{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 11/12/2018\n",
    "\n",
    "RNA-seq\n",
    "\n",
    "Griffith lab https://github.com/griffithlab/rnaseq_tutorial/wiki\n",
    "\n",
    "nathalia"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Environment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME\n",
    "\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq\n",
    "\n",
    "[nathalia@aws] $  ls\n",
    "tools\n",
    "\n",
    "[nathalia@aws] $  cd tools/\n",
    "\n",
    "[nathalia@aws] $  ls\n",
    "FastQC                                 kallisto_linux-v0.44.0.tar.gz\n",
    "bam-readcount                          picard.jar\n",
    "fastqc_v0.11.8.zip                     regtools\n",
    "flexbar-3.4.0-linux                    release_0.11.0.tar.gz\n",
    "flexbar-3.4.0-linux.tar.gz             samtools-1.9\n",
    "gffcompare-0.10.6.Linux_x86_64         samtools-1.9.tar\n",
    "gffcompare-0.10.6.Linux_x86_64.tar.gz  samtools-1.9.tar.bz2\n",
    "hisat2-2.1.0                           stringtie-1.3.4d.Linux_x86_64\n",
    "hisat2-2.1.0-Linux_x86_64.zip          stringtie-1.3.4d.Linux_x86_64.tar.gz\n",
    "htseq-release_0.11.0                   tophat-2.1.1.Linux_x86_64\n",
    "kallisto_linux-v0.44.0                 tophat-2.1.1.Linux_x86_64.tar.gz\n",
    "\n",
    "[nathalia@aws] $  env\n",
    "LD_LIBRARY_PATH=/home/ubuntu/lib:/home/ubuntu/workspace/rnaseq/tools/flexbar-3.4.0-linux:\n",
    "LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:\n",
    "SSH_CONNECTION=143.48.10.183 58108 172.31.39.185 22\n",
    "LESSCLOSE=/usr/bin/lesspipe %s %s\n",
    "LANG=C.UTF-8\n",
    "RNA_HOME=/home/ubuntu/workspace/rnaseq\n",
    "RNA_REF_FASTA=/home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92.fa\n",
    "RNA_DATA_DIR=/home/ubuntu/workspace/rnaseq/data\n",
    "RNA_ALIGN_DIR=/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "XDG_SESSION_ID=5\n",
    "USER=ubuntu\n",
    "WORKSPACE=/workspace\n",
    "DATA=/home/ubuntu/workspace/data\n",
    "PWD=/home/ubuntu/workspace/rnaseq/tools\n",
    "HOME=/home/ubuntu\n",
    "SSH_CLIENT=143.48.10.183 58108 22\n",
    "XDG_DATA_DIRS=/usr/local/share:/usr/share:/var/lib/snapd/desktop\n",
    "RNA_REF_GTF=/home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92.gtf\n",
    "SSH_TTY=/dev/pts/0\n",
    "MAIL=/var/mail/ubuntu\n",
    "TERM=xterm-256color\n",
    "SHELL=/bin/bash\n",
    "PERL5LIB=/home/ubuntu/FunSeq2_DC/:\n",
    "SHLVL=1\n",
    "LOGNAME=ubuntu\n",
    "RNA_REF_INDEX=/home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92\n",
    "XDG_RUNTIME_DIR=/run/user/1000\n",
    "RNA_REFS_DIR=/home/ubuntu/workspace/rnaseq/refs\n",
    "RNA_EXT_DATA_DIR=/home/ubuntu/CourseData/RNA_data\n",
    "PATH=/home/ubuntu/.local/bin:/home/ubuntu/bin:/home/ubuntu/bin:/home/ubuntu/bin/edirect:/home/ubuntu/bin/sratoolkit.2.9.2-ubuntu64/bin:/home/ubuntu/workspace/rnaseq/tools/samtools-1.9:/home/ubuntu/workspace/rnaseq/tools/bam-readcount/bin:/home/ubuntu/workspace/rnaseq/tools/hisat2-2.1.0:/home/ubuntu/workspace/rnaseq/tools/stringtie-1.3.4d.Linux_x86_64:/home/ubuntu/workspace/rnaseq/tools/gffcompare-0.10.6.Linux_x86_64:/home/ubuntu/workspace/rnaseq/tools/htseq-release_0.11.0/scripts:/home/ubuntu/workspace/rnaseq/tools/tophat-2.1.1.Linux_x86_64:/home/ubuntu/workspace/rnaseq/tools/kallisto_linux-v0.44.0:/home/ubuntu/workspace/rnaseq/tools/FastQC:/home/ubuntu/workspace/rnaseq/tools/flexbar-3.4.0-linux:/home/ubuntu/workspace/rnaseq/tools/regtools/build:/home/ubuntu/bin/bedtools2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin\n",
    "VAT_CONFIG_FILE=/home/ubuntu/.vatrc\n",
    "RNA_DATA_TRIM_DIR=/home/ubuntu/workspace/rnaseq/data/trimmed\n",
    "LESSOPEN=| /usr/bin/lesspipe %s\n",
    "_=/usr/bin/env\n",
    "OLDPWD=/home/ubuntu/workspace/rnaseq\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Installation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd ..\n",
    "[nathalia@aws] $  ls\n",
    "tools\n",
    "[nathalia@aws] $  mkdir student_tools\n",
    "[nathalia@aws] $  cd student_tools/\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/student_tools\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  wget https://github.com/samtools/samtools/releases/download/1.9/samtools-1.9.tar.bz2\n",
    "--2018-11-12 22:32:06--  https://github.com/samtools/samtools/releases/download/1.9/samtools-1.9.tar.bz2\n",
    "Resolving github.com (github.com)... 192.30.255.113, 192.30.255.112\n",
    "Connecting to github.com (github.com)|192.30.255.113|:443... connected.\n",
    "HTTP request sent, awaiting response... 302 Found\n",
    "Location: https://github-production-release-asset-2e65be.s3.amazonaws.com/3666841/fe586164-8a73-11e8-84ad-bb90bbd3b7c0?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20181112%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20181112T223211Z&X-Amz-Expires=300&X-Amz-Signature=5df7de0074c9bdd49f05c97992d57bec1e1678dbffc9a101366552df696b4ec0&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3Dsamtools-1.9.tar.bz2&response-content-type=application%2Foctet-stream [following]\n",
    "--2018-11-12 22:32:11--  https://github-production-release-asset-2e65be.s3.amazonaws.com/3666841/fe586164-8a73-11e8-84ad-bb90bbd3b7c0?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20181112%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20181112T223211Z&X-Amz-Expires=300&X-Amz-Signature=5df7de0074c9bdd49f05c97992d57bec1e1678dbffc9a101366552df696b4ec0&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3Dsamtools-1.9.tar.bz2&response-content-type=application%2Foctet-stream\n",
    "Resolving github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)... 52.216.128.243\n",
    "Connecting to github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)|52.216.128.243|:443... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 4440405 (4.2M) [application/octet-stream]\n",
    "Saving to: ‘samtools-1.9.tar.bz2’\n",
    "\n",
    "samtools-1.9.tar.bz2      100%[==================================>]   4.23M  6.54MB/s    in 0.6s    \n",
    "\n",
    "2018-11-12 22:32:12 (6.54 MB/s) - ‘samtools-1.9.tar.bz2’ saved [4440405/4440405]\n",
    "\n",
    "[nathalia@aws] $  bunzip2 samtools-1.9.tar.bz2\n",
    "\n",
    "[nathalia@aws] $  \n",
    "[nathalia@aws] $  tar -xvf samtools-1.9.tar\n",
    "samtools-1.9/\n",
    "samtools-1.9/AUTHORS\n",
    "samtools-1.9/ChangeLog.old\n",
    "samtools-1.9/INSTALL\n",
    "samtools-1.9/LICENSE\n",
    "samtools-1.9/Makefile\n",
    "samtools-1.9/Makefile.mingw\n",
    "samtools-1.9/NEWS\n",
    "samtools-1.9/README\n",
    "samtools-1.9/bam.c\n",
    "samtools-1.9/bam.h\n",
    "samtools-1.9/bam2bcf.c\n",
    "samtools-1.9/bam2bcf.h\n",
    "samtools-1.9/bam2bcf_indel.c\n",
    "samtools-1.9/bam2depth.c\n",
    "samtools-1.9/bam_addrprg.c\n",
    "samtools-1.9/bam_aux.c\n",
    "samtools-1.9/bam_cat.c\n",
    "samtools-1.9/bam_color.c\n",
    "samtools-1.9/bam_endian.h\n",
    "samtools-1.9/bam_flags.c\n",
    "samtools-1.9/bam_import.c\n",
    "samtools-1.9/bam_index.c\n",
    "samtools-1.9/bam_lpileup.c\n",
    "samtools-1.9/bam_lpileup.h\n",
    "samtools-1.9/bam_markdup.c\n",
    "samtools-1.9/bam_mate.c\n",
    "samtools-1.9/bam_md.c\n",
    "samtools-1.9/bam_plbuf.c\n",
    "samtools-1.9/bam_plbuf.h\n",
    "samtools-1.9/bam_plcmd.c\n",
    "samtools-1.9/bam_quickcheck.c\n",
    "samtools-1.9/bam_reheader.c\n",
    "samtools-1.9/bam_rmdup.c\n",
    "samtools-1.9/bam_rmdupse.c\n",
    "samtools-1.9/bam_sort.c\n",
    "samtools-1.9/bam_split.c\n",
    "samtools-1.9/bam_stat.c\n",
    "samtools-1.9/bam_tview.c\n",
    "samtools-1.9/bam_tview.h\n",
    "samtools-1.9/bam_tview_curses.c\n",
    "samtools-1.9/bam_tview_html.c\n",
    "samtools-1.9/bamshuf.c\n",
    "samtools-1.9/bamtk.c\n",
    "samtools-1.9/bedcov.c\n",
    "samtools-1.9/bedidx.c\n",
    "samtools-1.9/bedidx.h\n",
    "samtools-1.9/config.mk.in\n",
    "samtools-1.9/configure.ac\n",
    "samtools-1.9/cut_target.c\n",
    "samtools-1.9/dict.c\n",
    "samtools-1.9/examples/\n",
    "samtools-1.9/examples/00README.txt\n",
    "samtools-1.9/examples/ex1.fa\n",
    "samtools-1.9/examples/ex1.sam.gz\n",
    "samtools-1.9/examples/toy.fa\n",
    "samtools-1.9/examples/toy.sam\n",
    "samtools-1.9/faidx.c\n",
    "samtools-1.9/install-sh\n",
    "samtools-1.9/lz4/\n",
    "samtools-1.9/lz4/LICENSE\n",
    "samtools-1.9/lz4/lz4.c\n",
    "samtools-1.9/lz4/lz4.h\n",
    "samtools-1.9/m4/\n",
    "samtools-1.9/m4/ax_with_curses.m4\n",
    "samtools-1.9/m4/ax_with_htslib.m4\n",
    "samtools-1.9/misc/\n",
    "samtools-1.9/misc/HmmGlocal.java\n",
    "samtools-1.9/misc/ace2sam.c\n",
    "samtools-1.9/misc/blast2sam.pl\n",
    "samtools-1.9/misc/bowtie2sam.pl\n",
    "samtools-1.9/misc/export2sam.pl\n",
    "samtools-1.9/misc/interpolate_sam.pl\n",
    "samtools-1.9/misc/maq2sam.c\n",
    "samtools-1.9/misc/md5fa.c\n",
    "samtools-1.9/misc/md5sum-lite.c\n",
    "samtools-1.9/misc/novo2sam.pl\n",
    "samtools-1.9/misc/plot-bamstats\n",
    "samtools-1.9/misc/psl2sam.pl\n",
    "samtools-1.9/misc/r2plot.lua\n",
    "samtools-1.9/misc/sam2vcf.pl\n",
    "samtools-1.9/misc/samtools.pl\n",
    "samtools-1.9/misc/samtools_tab_completion\n",
    "samtools-1.9/misc/seq_cache_populate.pl\n",
    "samtools-1.9/misc/soap2sam.pl\n",
    "samtools-1.9/misc/varfilter.py\n",
    "samtools-1.9/misc/vcfutils.lua\n",
    "samtools-1.9/misc/wgsim.1\n",
    "samtools-1.9/misc/wgsim.c\n",
    "samtools-1.9/misc/wgsim_eval.pl\n",
    "samtools-1.9/misc/zoom2sam.pl\n",
    "samtools-1.9/padding.c\n",
    "samtools-1.9/phase.c\n",
    "samtools-1.9/sam.c\n",
    "samtools-1.9/sam.h\n",
    "samtools-1.9/sam_header.c\n",
    "samtools-1.9/sam_header.h\n",
    "samtools-1.9/sam_opts.c\n",
    "samtools-1.9/sam_opts.h\n",
    "samtools-1.9/sam_utils.c\n",
    "samtools-1.9/sam_view.c\n",
    "samtools-1.9/sample.c\n",
    "samtools-1.9/sample.h\n",
    "samtools-1.9/samtools.1\n",
    "samtools-1.9/samtools.h\n",
    "samtools-1.9/stats.c\n",
    "samtools-1.9/stats_isize.c\n",
    "samtools-1.9/stats_isize.h\n",
    "samtools-1.9/test/\n",
    "samtools-1.9/test/addrprg/\n",
    "samtools-1.9/test/addrprg/1_fixup.sam\n",
    "samtools-1.9/test/addrprg/1_fixup.sam.expected\n",
    "samtools-1.9/test/addrprg/1_fixup.sam.expected.err\n",
    "samtools-1.9/test/addrprg/2_fixup_orphan.sam\n",
    "samtools-1.9/test/addrprg/2_fixup_orphan.sam.expected\n",
    "samtools-1.9/test/addrprg/2_fixup_orphan.sam.expected.err\n",
    "samtools-1.9/test/addrprg/3_fixup.sam.expected\n",
    "samtools-1.9/test/addrprg/3_fixup.sam.expected.err\n",
    "samtools-1.9/test/addrprg/4_fixup_norg.sam\n",
    "samtools-1.9/test/addrprg/4_fixup_norg.sam.expected\n",
    "samtools-1.9/test/addrprg/4_fixup_norg.sam.expected.err\n",
    "samtools-1.9/test/bam2fq/\n",
    "samtools-1.9/test/bam2fq/1.1.fq.expected\n",
    "samtools-1.9/test/bam2fq/1.2.fq.expected\n",
    "samtools-1.9/test/bam2fq/1.stdout.expected\n",
    "samtools-1.9/test/bam2fq/10.1.fq.expected\n",
    "samtools-1.9/test/bam2fq/10.2.fq.expected\n",
    "samtools-1.9/test/bam2fq/10.s.fq.expected\n",
    "samtools-1.9/test/bam2fq/2.1.fq.expected\n",
    "samtools-1.9/test/bam2fq/2.2.fq.expected\n",
    "samtools-1.9/test/bam2fq/2.s.fq.expected\n",
    "samtools-1.9/test/bam2fq/2.stdout.expected\n",
    "samtools-1.9/test/bam2fq/3.1.fq.expected\n",
    "samtools-1.9/test/bam2fq/3.2.fq.expected\n",
    "samtools-1.9/test/bam2fq/3.s.fq.expected\n",
    "samtools-1.9/test/bam2fq/3.stdout.expected\n",
    "samtools-1.9/test/bam2fq/4.1.fq.expected\n",
    "samtools-1.9/test/bam2fq/4.2.fq.expected\n",
    "samtools-1.9/test/bam2fq/4.s.fq.expected\n",
    "samtools-1.9/test/bam2fq/4.stdout.expected\n",
    "samtools-1.9/test/bam2fq/5.1.fq.expected\n",
    "samtools-1.9/test/bam2fq/5.2.fq.expected\n",
    "samtools-1.9/test/bam2fq/5.s.fq.expected\n",
    "samtools-1.9/test/bam2fq/6.1.fq.expected\n",
    "samtools-1.9/test/bam2fq/6.2.fq.expected\n",
    "samtools-1.9/test/bam2fq/6.s.fq.expected\n",
    "samtools-1.9/test/bam2fq/7.1.fq.expected\n",
    "samtools-1.9/test/bam2fq/7.2.fq.expected\n",
    "samtools-1.9/test/bam2fq/7.s.fq.expected\n",
    "samtools-1.9/test/bam2fq/8.1.fq.expected\n",
    "samtools-1.9/test/bam2fq/8.2.fq.expected\n",
    "samtools-1.9/test/bam2fq/8.i.fq.expected\n",
    "samtools-1.9/test/bam2fq/8.s.fq.expected\n",
    "samtools-1.9/test/bam2fq/9.1.fq.expected\n",
    "samtools-1.9/test/bam2fq/9.2.fq.expected\n",
    "samtools-1.9/test/bam2fq/bc.fq.expected\n",
    "samtools-1.9/test/bam2fq/bc10.fq.expected\n",
    "samtools-1.9/test/bam2fq/bc_split.fq.expected\n",
    "samtools-1.9/test/bedcov/\n",
    "samtools-1.9/test/bedcov/bedcov.bam\n",
    "samtools-1.9/test/bedcov/bedcov.bam.bai\n",
    "samtools-1.9/test/bedcov/bedcov.bed\n",
    "samtools-1.9/test/bedcov/bedcov.expected\n",
    "samtools-1.9/test/bedcov/bedcov_j.expected\n",
    "samtools-1.9/test/collate/\n",
    "samtools-1.9/test/collate/1_fast_collate.sam.expected\n",
    "samtools-1.9/test/collate/2_fast_collate_with_tmp_used.sam.expected\n",
    "samtools-1.9/test/collate/collate.expected.sam\n",
    "samtools-1.9/test/collate/fast_collate.sam\n",
    "samtools-1.9/test/dat/\n",
    "samtools-1.9/test/dat/bam2fq.001.sam\n",
    "samtools-1.9/test/dat/bam2fq.002.sam\n",
    "samtools-1.9/test/dat/bam2fq.003.sam\n",
    "samtools-1.9/test/dat/bam2fq.004.sam\n",
    "samtools-1.9/test/dat/bam2fq.005.sam\n",
    "samtools-1.9/test/dat/bam2fq.010.sam\n",
    "samtools-1.9/test/dat/bam2fq.703.sam\n",
    "samtools-1.9/test/dat/cat.hdr\n",
    "samtools-1.9/test/dat/cram_md5/\n",
    "samtools-1.9/test/dat/cram_md5/08c04d512d4797d9ba2a156c1daba468\n",
    "samtools-1.9/test/dat/cram_md5/7c35feac7036c1cdef3bee0cc4b21437\n",
    "samtools-1.9/test/dat/cram_md5/89b96de7acc5918c2858ef92710be7d4\n",
    "samtools-1.9/test/dat/cram_md5/8a3d8deb9edd4ce3346b6061fcee6d0b\n",
    "samtools-1.9/test/dat/depad.001.fa\n",
    "samtools-1.9/test/dat/depad.001.fa.fai\n",
    "samtools-1.9/test/dat/depad.001p.sam\n",
    "samtools-1.9/test/dat/depad.001u.sam\n",
    "samtools-1.9/test/dat/dict.fa\n",
    "samtools-1.9/test/dat/dict.out\n",
    "samtools-1.9/test/dat/empty.expected\n",
    "samtools-1.9/test/dat/large_chrom.out\n",
    "samtools-1.9/test/dat/large_chrom.sam\n",
    "samtools-1.9/test/dat/large_chrom_bai_index.err\n",
    "samtools-1.9/test/dat/mpileup.1.sam\n",
    "samtools-1.9/test/dat/mpileup.2.sam\n",
    "samtools-1.9/test/dat/mpileup.3.sam\n",
    "samtools-1.9/test/dat/mpileup.err.1\n",
    "samtools-1.9/test/dat/mpileup.out.1\n",
    "samtools-1.9/test/dat/mpileup.out.2\n",
    "samtools-1.9/test/dat/mpileup.out.3\n",
    "samtools-1.9/test/dat/mpileup.out.4\n",
    "samtools-1.9/test/dat/mpileup.ref.fa\n",
    "samtools-1.9/test/dat/mpileup.ref.fa.fai\n",
    "samtools-1.9/test/dat/test_input_1_a.bam\n",
    "samtools-1.9/test/dat/test_input_1_a.bam.bai.expected\n",
    "samtools-1.9/test/dat/test_input_1_a.cram\n",
    "samtools-1.9/test/dat/test_input_1_a.sam\n",
    "samtools-1.9/test/dat/test_input_1_a_regex.sam\n",
    "samtools-1.9/test/dat/test_input_1_b.bam\n",
    "samtools-1.9/test/dat/test_input_1_b.sam\n",
    "samtools-1.9/test/dat/test_input_1_b_regex.sam\n",
    "samtools-1.9/test/dat/test_input_1_c.bam\n",
    "samtools-1.9/test/dat/test_input_1_c.sam\n",
    "samtools-1.9/test/dat/test_input_1_d.sam\n",
    "samtools-1.9/test/dat/view.001.01.bed\n",
    "samtools-1.9/test/dat/view.001.02.bed\n",
    "samtools-1.9/test/dat/view.001.fa\n",
    "samtools-1.9/test/dat/view.001.fa.fai\n",
    "samtools-1.9/test/dat/view.001.sam\n",
    "samtools-1.9/test/dat/view.002.01.bed\n",
    "samtools-1.9/test/dat/view.002.02.bed\n",
    "samtools-1.9/test/dat/view.002.fa\n",
    "samtools-1.9/test/dat/view.002.fa.fai\n",
    "samtools-1.9/test/dat/view.002.sam\n",
    "samtools-1.9/test/dat/view.003.expected.sam\n",
    "samtools-1.9/test/dat/view.003.sam\n",
    "samtools-1.9/test/fixmate/\n",
    "samtools-1.9/test/fixmate/1_coord_sort.sam\n",
    "samtools-1.9/test/fixmate/1_coord_sort.sam.expected\n",
    "samtools-1.9/test/fixmate/1_coord_sort.sam.expected.err\n",
    "samtools-1.9/test/fixmate/2_isize_overflow.sam\n",
    "samtools-1.9/test/fixmate/2_isize_overflow.sam.expected\n",
    "samtools-1.9/test/fixmate/3_reverse_read_pp_lt.sam\n",
    "samtools-1.9/test/fixmate/3_reverse_read_pp_lt.sam.expected\n",
    "samtools-1.9/test/fixmate/4_reverse_read_pp_equal.sam\n",
    "samtools-1.9/test/fixmate/4_reverse_read_pp_equal.sam.expected\n",
    "samtools-1.9/test/fixmate/5_ct.sam\n",
    "samtools-1.9/test/fixmate/5_ct.sam.expected\n",
    "samtools-1.9/test/fixmate/6_ct_replace.sam\n",
    "samtools-1.9/test/fixmate/6_ct_replace.sam.expected\n",
    "samtools-1.9/test/fixmate/7_two_read_mapped.sam\n",
    "samtools-1.9/test/fixmate/7_two_read_mapped.sam.expected\n",
    "samtools-1.9/test/idxstats/\n",
    "samtools-1.9/test/idxstats/test_input_1_a.bam.expected\n",
    "samtools-1.9/test/idxstats/test_input_1_a.bam.expected.err\n",
    "samtools-1.9/test/markdup/\n",
    "samtools-1.9/test/markdup/1_name_sort.expected.sam\n",
    "samtools-1.9/test/markdup/1_name_sort.expected.sam.err\n",
    "samtools-1.9/test/markdup/1_name_sort.sam\n",
    "samtools-1.9/test/markdup/2_bad_order.expected.sam\n",
    "samtools-1.9/test/markdup/2_bad_order.expected.sam.err\n",
    "samtools-1.9/test/markdup/2_bad_order.sam\n",
    "samtools-1.9/test/markdup/3_missing_mc.expected.sam\n",
    "samtools-1.9/test/markdup/3_missing_mc.expected.sam.err\n",
    "samtools-1.9/test/markdup/3_missing_mc.sam\n",
    "samtools-1.9/test/markdup/4_missing_ms.expected.sam\n",
    "samtools-1.9/test/markdup/4_missing_ms.expected.sam.err\n",
    "samtools-1.9/test/markdup/4_missing_ms.sam\n",
    "samtools-1.9/test/markdup/5_markdup.expected.sam\n",
    "samtools-1.9/test/markdup/5_markdup.sam\n",
    "samtools-1.9/test/markdup/6_remove_dups.expected.sam\n",
    "samtools-1.9/test/markdup/6_remove_dups.sam\n",
    "samtools-1.9/test/markdup/7_mark_supp_dup.expected.sam\n",
    "samtools-1.9/test/markdup/7_mark_supp_dup.sam\n",
    "samtools-1.9/test/merge/\n",
    "samtools-1.9/test/merge/2.merge.expected.sam\n",
    "samtools-1.9/test/merge/3.merge.expected.err\n",
    "samtools-1.9/test/merge/3.merge.expected.sam\n",
    "samtools-1.9/test/merge/4.merge.expected.sam\n",
    "samtools-1.9/test/merge/5.merge.expected.sam\n",
    "samtools-1.9/test/merge/6.merge.expected.sam\n",
    "samtools-1.9/test/merge/7.merge.expected.sam\n",
    "samtools-1.9/test/merge/tag.pg.merge.expected.sam\n",
    "samtools-1.9/test/merge/tag.pg.n.merge.expected.sam\n",
    "samtools-1.9/test/merge/test_3.fofn\n",
    "samtools-1.9/test/merge/test_bam_translate.c\n",
    "samtools-1.9/test/merge/test_header.sam\n",
    "samtools-1.9/test/merge/test_rtrans_build.c\n",
    "samtools-1.9/test/merge/test_trans_tbl_init.c\n",
    "samtools-1.9/test/mpileup/\n",
    "samtools-1.9/test/mpileup/1read.bam\n",
    "samtools-1.9/test/mpileup/1read.sam\n",
    "samtools-1.9/test/mpileup/anomalous.sam\n",
    "samtools-1.9/test/mpileup/c1#ID.bam\n",
    "samtools-1.9/test/mpileup/c1#ID2.bam\n",
    "samtools-1.9/test/mpileup/c1#clip.bam\n",
    "samtools-1.9/test/mpileup/c1#pad1.bam\n",
    "samtools-1.9/test/mpileup/c1#pad2.bam\n",
    "samtools-1.9/test/mpileup/c1#pad3.bam\n",
    "samtools-1.9/test/mpileup/c1.fa\n",
    "samtools-1.9/test/mpileup/c1.fa.fai\n",
    "samtools-1.9/test/mpileup/ce#5b.bam\n",
    "samtools-1.9/test/mpileup/ce#5b.bam.bai\n",
    "samtools-1.9/test/mpileup/ce#5b.cram\n",
    "samtools-1.9/test/mpileup/ce#5b.cram.crai\n",
    "samtools-1.9/test/mpileup/ce#5b.sam\n",
    "samtools-1.9/test/mpileup/ce#large_seq.bam\n",
    "samtools-1.9/test/mpileup/ce#unmap.bam\n",
    "samtools-1.9/test/mpileup/ce#unmap1.bam\n",
    "samtools-1.9/test/mpileup/ce#unmap2.bam\n",
    "samtools-1.9/test/mpileup/ce.fa\n",
    "samtools-1.9/test/mpileup/ce.fa.fai\n",
    "samtools-1.9/test/mpileup/deep.sam\n",
    "samtools-1.9/test/mpileup/depth.reg\n",
    "samtools-1.9/test/mpileup/expected/\n",
    "samtools-1.9/test/mpileup/expected/1.out.f3-6.gz\n",
    "samtools-1.9/test/mpileup/expected/10.out\n",
    "samtools-1.9/test/mpileup/expected/11.out\n",
    "samtools-1.9/test/mpileup/expected/12.out\n",
    "samtools-1.9/test/mpileup/expected/13.out\n",
    "samtools-1.9/test/mpileup/expected/14.out\n",
    "samtools-1.9/test/mpileup/expected/15.out\n",
    "samtools-1.9/test/mpileup/expected/16.out\n",
    "samtools-1.9/test/mpileup/expected/17.out\n",
    "samtools-1.9/test/mpileup/expected/18.out\n",
    "samtools-1.9/test/mpileup/expected/19.out\n",
    "samtools-1.9/test/mpileup/expected/2.out\n",
    "samtools-1.9/test/mpileup/expected/20.out\n",
    "samtools-1.9/test/mpileup/expected/21.out\n",
    "samtools-1.9/test/mpileup/expected/22.out\n",
    "samtools-1.9/test/mpileup/expected/23.out\n",
    "samtools-1.9/test/mpileup/expected/24.out\n",
    "samtools-1.9/test/mpileup/expected/25.out\n",
    "samtools-1.9/test/mpileup/expected/26.out\n",
    "samtools-1.9/test/mpileup/expected/27.out\n",
    "samtools-1.9/test/mpileup/expected/28.out\n",
    "samtools-1.9/test/mpileup/expected/29.out\n",
    "samtools-1.9/test/mpileup/expected/3.out\n",
    "samtools-1.9/test/mpileup/expected/30.out\n",
    "samtools-1.9/test/mpileup/expected/31.out\n",
    "samtools-1.9/test/mpileup/expected/32.out\n",
    "samtools-1.9/test/mpileup/expected/33.out\n",
    "samtools-1.9/test/mpileup/expected/34.out\n",
    "samtools-1.9/test/mpileup/expected/35.out\n",
    "samtools-1.9/test/mpileup/expected/37.out\n",
    "samtools-1.9/test/mpileup/expected/38.out\n",
    "samtools-1.9/test/mpileup/expected/39.out\n",
    "samtools-1.9/test/mpileup/expected/4.out\n",
    "samtools-1.9/test/mpileup/expected/40.out\n",
    "samtools-1.9/test/mpileup/expected/41.out\n",
    "samtools-1.9/test/mpileup/expected/42.out\n",
    "samtools-1.9/test/mpileup/expected/43.out\n",
    "samtools-1.9/test/mpileup/expected/44.out\n",
    "samtools-1.9/test/mpileup/expected/45.out\n",
    "samtools-1.9/test/mpileup/expected/46.out\n",
    "samtools-1.9/test/mpileup/expected/47.out\n",
    "samtools-1.9/test/mpileup/expected/48.out\n",
    "samtools-1.9/test/mpileup/expected/49.out\n",
    "samtools-1.9/test/mpileup/expected/5.out\n",
    "samtools-1.9/test/mpileup/expected/50.out\n",
    "samtools-1.9/test/mpileup/expected/51.out\n",
    "samtools-1.9/test/mpileup/expected/52.out\n",
    "samtools-1.9/test/mpileup/expected/53.out\n",
    "samtools-1.9/test/mpileup/expected/54.out\n",
    "samtools-1.9/test/mpileup/expected/55.out\n",
    "samtools-1.9/test/mpileup/expected/56.out\n",
    "samtools-1.9/test/mpileup/expected/57.out\n",
    "samtools-1.9/test/mpileup/expected/58.out\n",
    "samtools-1.9/test/mpileup/expected/59.out\n",
    "samtools-1.9/test/mpileup/expected/6.out\n",
    "samtools-1.9/test/mpileup/expected/60.out\n",
    "samtools-1.9/test/mpileup/expected/61.out\n",
    "samtools-1.9/test/mpileup/expected/62.out\n",
    "samtools-1.9/test/mpileup/expected/63.out\n",
    "samtools-1.9/test/mpileup/expected/64.out\n",
    "samtools-1.9/test/mpileup/expected/65.out\n",
    "samtools-1.9/test/mpileup/expected/66.out\n",
    "samtools-1.9/test/mpileup/expected/67.out\n",
    "samtools-1.9/test/mpileup/expected/68.out\n",
    "samtools-1.9/test/mpileup/expected/69.out\n",
    "samtools-1.9/test/mpileup/expected/7.out\n",
    "samtools-1.9/test/mpileup/expected/70.out\n",
    "samtools-1.9/test/mpileup/expected/71.out\n",
    "samtools-1.9/test/mpileup/expected/72.out\n",
    "samtools-1.9/test/mpileup/expected/73.out\n",
    "samtools-1.9/test/mpileup/expected/74.out\n",
    "samtools-1.9/test/mpileup/expected/75.out\n",
    "samtools-1.9/test/mpileup/expected/76.out\n",
    "samtools-1.9/test/mpileup/expected/77.out\n",
    "samtools-1.9/test/mpileup/expected/8.out\n",
    "samtools-1.9/test/mpileup/expected/9.out\n",
    "samtools-1.9/test/mpileup/expected/a0.out\n",
    "samtools-1.9/test/mpileup/expected/a1.out\n",
    "samtools-1.9/test/mpileup/expected/a10.out\n",
    "samtools-1.9/test/mpileup/expected/a11.out\n",
    "samtools-1.9/test/mpileup/expected/a13.out\n",
    "samtools-1.9/test/mpileup/expected/a14.out\n",
    "samtools-1.9/test/mpileup/expected/a15.out\n",
    "samtools-1.9/test/mpileup/expected/a16.out\n",
    "samtools-1.9/test/mpileup/expected/a17.out\n",
    "samtools-1.9/test/mpileup/expected/a2.out\n",
    "samtools-1.9/test/mpileup/expected/a3.out\n",
    "samtools-1.9/test/mpileup/expected/a4.out\n",
    "samtools-1.9/test/mpileup/expected/a5.out\n",
    "samtools-1.9/test/mpileup/expected/a6.out\n",
    "samtools-1.9/test/mpileup/expected/a8.out\n",
    "samtools-1.9/test/mpileup/expected/a9.out\n",
    "samtools-1.9/test/mpileup/expected/d1_1.out\n",
    "samtools-1.9/test/mpileup/expected/d1_12.out\n",
    "samtools-1.9/test/mpileup/expected/d1_2.out\n",
    "samtools-1.9/test/mpileup/expected/d2_12r.out\n",
    "samtools-1.9/test/mpileup/expected/d3_12.out\n",
    "samtools-1.9/test/mpileup/expected/d3_12r1a.out\n",
    "samtools-1.9/test/mpileup/expected/d3_12r2a.out\n",
    "samtools-1.9/test/mpileup/expected/d3_3.out\n",
    "samtools-1.9/test/mpileup/expected/d4_12.out\n",
    "samtools-1.9/test/mpileup/expected/d4_12b.out\n",
    "samtools-1.9/test/mpileup/expected/d4_12r.out\n",
    "samtools-1.9/test/mpileup/expected/d5_all1.out\n",
    "samtools-1.9/test/mpileup/expected/d5_all2.out\n",
    "samtools-1.9/test/mpileup/expected/d5_all3.out\n",
    "samtools-1.9/test/mpileup/expected/d5_b3.out\n",
    "samtools-1.9/test/mpileup/expected/d5_b3a.out\n",
    "samtools-1.9/test/mpileup/expected/d5_b3aa.out\n",
    "samtools-1.9/test/mpileup/expected/d5_blank.out\n",
    "samtools-1.9/test/mpileup/expected/d5_xx1.out\n",
    "samtools-1.9/test/mpileup/expected/d5_xx2.out\n",
    "samtools-1.9/test/mpileup/expected/d5_xx3.out\n",
    "samtools-1.9/test/mpileup/expected/d5_xy1.out\n",
    "samtools-1.9/test/mpileup/expected/m5_all1.out\n",
    "samtools-1.9/test/mpileup/expected/m5_all2.out\n",
    "samtools-1.9/test/mpileup/expected/m5_all3.out\n",
    "samtools-1.9/test/mpileup/expected/m5_b3.out\n",
    "samtools-1.9/test/mpileup/expected/m5_b3a.out\n",
    "samtools-1.9/test/mpileup/expected/m5_b3aa.out\n",
    "samtools-1.9/test/mpileup/expected/m5_blank.out\n",
    "samtools-1.9/test/mpileup/expected/m5_xx1.out\n",
    "samtools-1.9/test/mpileup/expected/m5_xx2.out\n",
    "samtools-1.9/test/mpileup/expected/m5_xx3.out\n",
    "samtools-1.9/test/mpileup/expected/m5_xy1.out\n",
    "samtools-1.9/test/mpileup/indels.sam\n",
    "samtools-1.9/test/mpileup/md5/\n",
    "samtools-1.9/test/mpileup/md5/02099b2aad2cfe707759353724fb40ae\n",
    "samtools-1.9/test/mpileup/md5/0d4dfc45807d6a11439e858605cb364d\n",
    "samtools-1.9/test/mpileup/md5/351f64d4f4f9ddd45b35336ad97aa6de\n",
    "samtools-1.9/test/mpileup/md5/6ec590eb82c361a61e6baabce20109bc\n",
    "samtools-1.9/test/mpileup/md5/6f60cdf4c739651ae72d53e2e94825c1\n",
    "samtools-1.9/test/mpileup/md5/73d180d5cb3860a2713927b6812eab0e\n",
    "samtools-1.9/test/mpileup/mpileup-E.bam\n",
    "samtools-1.9/test/mpileup/mpileup.1.bam\n",
    "samtools-1.9/test/mpileup/mpileup.1.bam.bai\n",
    "samtools-1.9/test/mpileup/mpileup.2.bam\n",
    "samtools-1.9/test/mpileup/mpileup.3.bam\n",
    "samtools-1.9/test/mpileup/mpileup.ref.fa\n",
    "samtools-1.9/test/mpileup/mpileup.ref.fa.fai\n",
    "samtools-1.9/test/mpileup/mpileup.reg\n",
    "samtools-1.9/test/mpileup/overlap50.sam\n",
    "samtools-1.9/test/mpileup/overlapIllumina.bam\n",
    "samtools-1.9/test/mpileup/regions\n",
    "samtools-1.9/test/mpileup/regression.sh\n",
    "samtools-1.9/test/mpileup/xx#depth1.sam\n",
    "samtools-1.9/test/mpileup/xx#depth2.sam\n",
    "samtools-1.9/test/mpileup/xx#depth3.sam\n",
    "samtools-1.9/test/mpileup/xx#minimal.bam\n",
    "samtools-1.9/test/mpileup/xx#triplet.bam\n",
    "samtools-1.9/test/mpileup/xx.bed\n",
    "samtools-1.9/test/mpileup/xx.bed2\n",
    "samtools-1.9/test/mpileup/xx.bed3\n",
    "samtools-1.9/test/mpileup/xx.fa\n",
    "samtools-1.9/test/mpileup/xx.fa.fai\n",
    "samtools-1.9/test/quickcheck/\n",
    "samtools-1.9/test/quickcheck/1.quickcheck.badeof.bam\n",
    "samtools-1.9/test/quickcheck/2.quickcheck.badheader.bam\n",
    "samtools-1.9/test/quickcheck/3.quickcheck.ok.bam\n",
    "samtools-1.9/test/quickcheck/4.quickcheck.ok.bam\n",
    "samtools-1.9/test/quickcheck/5.quickcheck.scramble30.truncated.cram\n",
    "samtools-1.9/test/quickcheck/6.quickcheck.cram21.ok.cram\n",
    "samtools-1.9/test/quickcheck/7.quickcheck.cram30.ok.cram\n",
    "samtools-1.9/test/quickcheck/8.quickcheck.cram21.truncated.cram\n",
    "samtools-1.9/test/quickcheck/9.quickcheck.cram30.truncated.cram\n",
    "samtools-1.9/test/quickcheck/all.expected\n",
    "samtools-1.9/test/reheader/\n",
    "samtools-1.9/test/reheader/1_view1.sam.expected\n",
    "samtools-1.9/test/reheader/1_view1.sam.expected.err\n",
    "samtools-1.9/test/reheader/2_view1.sam.expected\n",
    "samtools-1.9/test/reheader/2_view1.sam.expected.err\n",
    "samtools-1.9/test/reheader/3_view1.sam.expected\n",
    "samtools-1.9/test/reheader/3_view1.sam.expected.err\n",
    "samtools-1.9/test/reheader/hdr.sam\n",
    "samtools-1.9/test/sort/\n",
    "samtools-1.9/test/sort/name.sort.expected.sam\n",
    "samtools-1.9/test/sort/pos.sort.expected.sam\n",
    "samtools-1.9/test/sort/tag.as.sort.expected.sam\n",
    "samtools-1.9/test/sort/tag.fi.sort.expected.sam\n",
    "samtools-1.9/test/sort/tag.rg.n.sort.expected.sam\n",
    "samtools-1.9/test/sort/tag.rg.sort.expected.sam\n",
    "samtools-1.9/test/split/\n",
    "samtools-1.9/test/split/test_count_rg.c\n",
    "samtools-1.9/test/split/test_expand_format_string.c\n",
    "samtools-1.9/test/split/test_filter_header_rg.c\n",
    "samtools-1.9/test/split/test_parse_args.c\n",
    "samtools-1.9/test/stat/\n",
    "samtools-1.9/test/stat/1.stats.expected\n",
    "samtools-1.9/test/stat/10.stats.expected\n",
    "samtools-1.9/test/stat/10_map_cigar.sam\n",
    "samtools-1.9/test/stat/10_map_cigar.sam_s1_a_1.expected.bamstat\n",
    "samtools-1.9/test/stat/10_map_cigar.sam_s1_b_1.expected.bamstat\n",
    "samtools-1.9/test/stat/11.stats.expected\n",
    "samtools-1.9/test/stat/11.stats.g4.expected\n",
    "samtools-1.9/test/stat/11.stats.targets\n",
    "samtools-1.9/test/stat/11_target.bam\n",
    "samtools-1.9/test/stat/11_target.bam.bai\n",
    "samtools-1.9/test/stat/11_target.sam\n",
    "samtools-1.9/test/stat/12.2reads.nooverlap.expected\n",
    "samtools-1.9/test/stat/12.2reads.overlap.expected\n",
    "samtools-1.9/test/stat/12.3reads.nooverlap.expected\n",
    "samtools-1.9/test/stat/12.3reads.overlap.expected\n",
    "samtools-1.9/test/stat/12_2reads.bed\n",
    "samtools-1.9/test/stat/12_3reads.bed\n",
    "samtools-1.9/test/stat/12_overlaps.bam\n",
    "samtools-1.9/test/stat/1_map_cigar.sam\n",
    "samtools-1.9/test/stat/1_map_cigar.sam_s1_a_1.expected.bamstat\n",
    "samtools-1.9/test/stat/2.stats.expected\n",
    "samtools-1.9/test/stat/2_equal_cigar_full_seq.sam\n",
    "samtools-1.9/test/stat/3.stats.expected\n",
    "samtools-1.9/test/stat/3_map_cigar_equal_seq.sam\n",
    "samtools-1.9/test/stat/4.stats.expected\n",
    "samtools-1.9/test/stat/4_X_cigar_full_seq.sam\n",
    "samtools-1.9/test/stat/5.stats.expected\n",
    "samtools-1.9/test/stat/5_insert_cigar.sam\n",
    "samtools-1.9/test/stat/6.stats.expected\n",
    "samtools-1.9/test/stat/7.stats.expected\n",
    "samtools-1.9/test/stat/7_supp.sam\n",
    "samtools-1.9/test/stat/8.stats.expected\n",
    "samtools-1.9/test/stat/8_secondary.sam\n",
    "samtools-1.9/test/stat/9.stats.expected\n",
    "samtools-1.9/test/stat/test.fa\n",
    "samtools-1.9/test/stat/test.fa.fai\n",
    "samtools-1.9/test/test.c\n",
    "samtools-1.9/test/test.h\n",
    "samtools-1.9/test/test.pl\n",
    "samtools-1.9/test/tview/\n",
    "samtools-1.9/test/tview/test_get_rg_sample.c\n",
    "samtools-1.9/test/vcf-miniview.c\n",
    "samtools-1.9/tmp_file.c\n",
    "samtools-1.9/tmp_file.h\n",
    "samtools-1.9/version.sh\n",
    "samtools-1.9/win32/\n",
    "samtools-1.9/win32/libcurses.a\n",
    "samtools-1.9/win32/libz.a\n",
    "samtools-1.9/win32/xcurses.h\n",
    "samtools-1.9/win32/zconf.h\n",
    "samtools-1.9/win32/zlib.h\n",
    "samtools-1.9/config.h.in\n",
    "samtools-1.9/configure\n",
    "samtools-1.9/htslib-1.9/\n",
    "samtools-1.9/htslib-1.9/INSTALL\n",
    "samtools-1.9/htslib-1.9/LICENSE\n",
    "samtools-1.9/htslib-1.9/Makefile\n",
    "samtools-1.9/htslib-1.9/NEWS\n",
    "samtools-1.9/htslib-1.9/README\n",
    "samtools-1.9/htslib-1.9/bcf_sr_sort.c\n",
    "samtools-1.9/htslib-1.9/bcf_sr_sort.h\n",
    "samtools-1.9/htslib-1.9/bgzf.c\n",
    "samtools-1.9/htslib-1.9/bgzip.1\n",
    "samtools-1.9/htslib-1.9/bgzip.c\n",
    "samtools-1.9/htslib-1.9/config.mk.in\n",
    "samtools-1.9/htslib-1.9/configure.ac\n",
    "samtools-1.9/htslib-1.9/cram/\n",
    "samtools-1.9/htslib-1.9/cram/cram.h\n",
    "samtools-1.9/htslib-1.9/cram/cram_codecs.c\n",
    "samtools-1.9/htslib-1.9/cram/cram_codecs.h\n",
    "samtools-1.9/htslib-1.9/cram/cram_decode.c\n",
    "samtools-1.9/htslib-1.9/cram/cram_decode.h\n",
    "samtools-1.9/htslib-1.9/cram/cram_encode.c\n",
    "samtools-1.9/htslib-1.9/cram/cram_encode.h\n",
    "samtools-1.9/htslib-1.9/cram/cram_external.c\n",
    "samtools-1.9/htslib-1.9/cram/cram_index.c\n",
    "samtools-1.9/htslib-1.9/cram/cram_index.h\n",
    "samtools-1.9/htslib-1.9/cram/cram_io.c\n",
    "samtools-1.9/htslib-1.9/cram/cram_io.h\n",
    "samtools-1.9/htslib-1.9/cram/cram_samtools.c\n",
    "samtools-1.9/htslib-1.9/cram/cram_samtools.h\n",
    "samtools-1.9/htslib-1.9/cram/cram_stats.c\n",
    "samtools-1.9/htslib-1.9/cram/cram_stats.h\n",
    "samtools-1.9/htslib-1.9/cram/cram_structs.h\n",
    "samtools-1.9/htslib-1.9/cram/files.c\n",
    "samtools-1.9/htslib-1.9/cram/mFILE.c\n",
    "samtools-1.9/htslib-1.9/cram/mFILE.h\n",
    "samtools-1.9/htslib-1.9/cram/misc.h\n",
    "samtools-1.9/htslib-1.9/cram/open_trace_file.c\n",
    "samtools-1.9/htslib-1.9/cram/open_trace_file.h\n",
    "samtools-1.9/htslib-1.9/cram/os.h\n",
    "samtools-1.9/htslib-1.9/cram/pooled_alloc.c\n",
    "samtools-1.9/htslib-1.9/cram/pooled_alloc.h\n",
    "samtools-1.9/htslib-1.9/cram/rANS_byte.h\n",
    "samtools-1.9/htslib-1.9/cram/rANS_static.c\n",
    "samtools-1.9/htslib-1.9/cram/rANS_static.h\n",
    "samtools-1.9/htslib-1.9/cram/sam_header.c\n",
    "samtools-1.9/htslib-1.9/cram/sam_header.h\n",
    "samtools-1.9/htslib-1.9/cram/string_alloc.c\n",
    "samtools-1.9/htslib-1.9/cram/string_alloc.h\n",
    "samtools-1.9/htslib-1.9/errmod.c\n",
    "samtools-1.9/htslib-1.9/faidx.5\n",
    "samtools-1.9/htslib-1.9/faidx.c\n",
    "samtools-1.9/htslib-1.9/hfile.c\n",
    "samtools-1.9/htslib-1.9/hfile_gcs.c\n",
    "samtools-1.9/htslib-1.9/hfile_internal.h\n",
    "samtools-1.9/htslib-1.9/hfile_libcurl.c\n",
    "samtools-1.9/htslib-1.9/hfile_net.c\n",
    "samtools-1.9/htslib-1.9/hfile_s3.c\n",
    "samtools-1.9/htslib-1.9/hts.c\n",
    "samtools-1.9/htslib-1.9/hts_internal.h\n",
    "samtools-1.9/htslib-1.9/hts_os.c\n",
    "samtools-1.9/htslib-1.9/htsfile.1\n",
    "samtools-1.9/htslib-1.9/htsfile.c\n",
    "samtools-1.9/htslib-1.9/htslib.mk\n",
    "samtools-1.9/htslib-1.9/htslib.pc.in\n",
    "samtools-1.9/htslib-1.9/htslib/\n",
    "samtools-1.9/htslib-1.9/htslib/bgzf.h\n",
    "samtools-1.9/htslib-1.9/htslib/cram.h\n",
    "samtools-1.9/htslib-1.9/htslib/faidx.h\n",
    "samtools-1.9/htslib-1.9/htslib/hfile.h\n",
    "samtools-1.9/htslib-1.9/htslib/hts.h\n",
    "samtools-1.9/htslib-1.9/htslib/hts_defs.h\n",
    "samtools-1.9/htslib-1.9/htslib/hts_endian.h\n",
    "samtools-1.9/htslib-1.9/htslib/hts_log.h\n",
    "samtools-1.9/htslib-1.9/htslib/hts_os.h\n",
    "samtools-1.9/htslib-1.9/htslib/kbitset.h\n",
    "samtools-1.9/htslib-1.9/htslib/kfunc.h\n",
    "samtools-1.9/htslib-1.9/htslib/khash.h\n",
    "samtools-1.9/htslib-1.9/htslib/khash_str2int.h\n",
    "samtools-1.9/htslib-1.9/htslib/klist.h\n",
    "samtools-1.9/htslib-1.9/htslib/knetfile.h\n",
    "samtools-1.9/htslib-1.9/htslib/kseq.h\n",
    "samtools-1.9/htslib-1.9/htslib/ksort.h\n",
    "samtools-1.9/htslib-1.9/htslib/kstring.h\n",
    "samtools-1.9/htslib-1.9/htslib/regidx.h\n",
    "samtools-1.9/htslib-1.9/htslib/sam.h\n",
    "samtools-1.9/htslib-1.9/htslib/synced_bcf_reader.h\n",
    "samtools-1.9/htslib-1.9/htslib/tbx.h\n",
    "samtools-1.9/htslib-1.9/htslib/thread_pool.h\n",
    "samtools-1.9/htslib-1.9/htslib/vcf.h\n",
    "samtools-1.9/htslib-1.9/htslib/vcf_sweep.h\n",
    "samtools-1.9/htslib-1.9/htslib/vcfutils.h\n",
    "samtools-1.9/htslib-1.9/htslib_vars.mk\n",
    "samtools-1.9/htslib-1.9/kfunc.c\n",
    "samtools-1.9/htslib-1.9/knetfile.c\n",
    "samtools-1.9/htslib-1.9/kstring.c\n",
    "samtools-1.9/htslib-1.9/m4/\n",
    "samtools-1.9/htslib-1.9/m4/hts_prog_cc_warnings.m4\n",
    "samtools-1.9/htslib-1.9/md5.c\n",
    "samtools-1.9/htslib-1.9/multipart.c\n",
    "samtools-1.9/htslib-1.9/os/\n",
    "samtools-1.9/htslib-1.9/os/lzma_stub.h\n",
    "samtools-1.9/htslib-1.9/os/rand.c\n",
    "samtools-1.9/htslib-1.9/plugin.c\n",
    "samtools-1.9/htslib-1.9/probaln.c\n",
    "samtools-1.9/htslib-1.9/realn.c\n",
    "samtools-1.9/htslib-1.9/regidx.c\n",
    "samtools-1.9/htslib-1.9/sam.5\n",
    "samtools-1.9/htslib-1.9/sam.c\n",
    "samtools-1.9/htslib-1.9/synced_bcf_reader.c\n",
    "samtools-1.9/htslib-1.9/tabix.1\n",
    "samtools-1.9/htslib-1.9/tabix.c\n",
    "samtools-1.9/htslib-1.9/tbx.c\n",
    "samtools-1.9/htslib-1.9/test/\n",
    "samtools-1.9/htslib-1.9/test/auxf#values.sam\n",
    "samtools-1.9/htslib-1.9/test/auxf#values_java.cram\n",
    "samtools-1.9/htslib-1.9/test/auxf.fa\n",
    "samtools-1.9/htslib-1.9/test/auxf.fa.fai\n",
    "samtools-1.9/htslib-1.9/test/bgziptest.txt\n",
    "samtools-1.9/htslib-1.9/test/bgziptest.txt.gz\n",
    "samtools-1.9/htslib-1.9/test/bgziptest.txt.gz.gzi\n",
    "samtools-1.9/htslib-1.9/test/c1#bounds.sam\n",
    "samtools-1.9/htslib-1.9/test/c1#clip.sam\n",
    "samtools-1.9/htslib-1.9/test/c1#noseq.sam\n",
    "samtools-1.9/htslib-1.9/test/c1#pad1.sam\n",
    "samtools-1.9/htslib-1.9/test/c1#pad2.sam\n",
    "samtools-1.9/htslib-1.9/test/c1#pad3.sam\n",
    "samtools-1.9/htslib-1.9/test/c1#unknown.sam\n",
    "samtools-1.9/htslib-1.9/test/c1.fa\n",
    "samtools-1.9/htslib-1.9/test/c1.fa.fai\n",
    "samtools-1.9/htslib-1.9/test/c2#pad.sam\n",
    "samtools-1.9/htslib-1.9/test/c2.fa\n",
    "samtools-1.9/htslib-1.9/test/c2.fa.fai\n",
    "samtools-1.9/htslib-1.9/test/ce#1.sam\n",
    "samtools-1.9/htslib-1.9/test/ce#1000.sam\n",
    "samtools-1.9/htslib-1.9/test/ce#2.sam\n",
    "samtools-1.9/htslib-1.9/test/ce#5.sam\n",
    "samtools-1.9/htslib-1.9/test/ce#5b.sam\n",
    "samtools-1.9/htslib-1.9/test/ce#5b_java.cram\n",
    "samtools-1.9/htslib-1.9/test/ce#large_seq.sam\n",
    "samtools-1.9/htslib-1.9/test/ce#supp.sam\n",
    "samtools-1.9/htslib-1.9/test/ce#tag_depadded.sam\n",
    "samtools-1.9/htslib-1.9/test/ce#tag_padded.sam\n",
    "samtools-1.9/htslib-1.9/test/ce#unmap.sam\n",
    "samtools-1.9/htslib-1.9/test/ce#unmap1.sam\n",
    "samtools-1.9/htslib-1.9/test/ce#unmap2.sam\n",
    "samtools-1.9/htslib-1.9/test/ce.fa\n",
    "samtools-1.9/htslib-1.9/test/ce.fa.fai\n",
    "samtools-1.9/htslib-1.9/test/compare_sam.pl\n",
    "samtools-1.9/htslib-1.9/test/cross_validate.sh\n",
    "samtools-1.9/htslib-1.9/test/faidx.fa\n",
    "samtools-1.9/htslib-1.9/test/fastqs.fq\n",
    "samtools-1.9/htslib-1.9/test/fastqs_README.txt\n",
    "samtools-1.9/htslib-1.9/test/fieldarith.c\n",
    "samtools-1.9/htslib-1.9/test/fieldarith.sam\n",
    "samtools-1.9/htslib-1.9/test/formatcols.vcf\n",
    "samtools-1.9/htslib-1.9/test/formatmissing-out.vcf\n",
    "samtools-1.9/htslib-1.9/test/formatmissing.vcf\n",
    "samtools-1.9/htslib-1.9/test/hfile.c\n",
    "samtools-1.9/htslib-1.9/test/hts_endian.c\n",
    "samtools-1.9/htslib-1.9/test/md#1.sam\n",
    "samtools-1.9/htslib-1.9/test/md.fa\n",
    "samtools-1.9/htslib-1.9/test/md.fa.fai\n",
    "samtools-1.9/htslib-1.9/test/noroundtrip-out.vcf\n",
    "samtools-1.9/htslib-1.9/test/noroundtrip.vcf\n",
    "samtools-1.9/htslib-1.9/test/range.bam\n",
    "samtools-1.9/htslib-1.9/test/range.bam.bai\n",
    "samtools-1.9/htslib-1.9/test/range.cram\n",
    "samtools-1.9/htslib-1.9/test/range.cram.crai\n",
    "samtools-1.9/htslib-1.9/test/range.out\n",
    "samtools-1.9/htslib-1.9/test/realn01.fa\n",
    "samtools-1.9/htslib-1.9/test/realn01.fa.fai\n",
    "samtools-1.9/htslib-1.9/test/realn01.sam\n",
    "samtools-1.9/htslib-1.9/test/realn01_exp-a.sam\n",
    "samtools-1.9/htslib-1.9/test/realn01_exp-e.sam\n",
    "samtools-1.9/htslib-1.9/test/realn01_exp.sam\n",
    "samtools-1.9/htslib-1.9/test/realn02-r.sam\n",
    "samtools-1.9/htslib-1.9/test/realn02.fa\n",
    "samtools-1.9/htslib-1.9/test/realn02.fa.fai\n",
    "samtools-1.9/htslib-1.9/test/realn02.sam\n",
    "samtools-1.9/htslib-1.9/test/realn02_exp-a.sam\n",
    "samtools-1.9/htslib-1.9/test/realn02_exp-e.sam\n",
    "samtools-1.9/htslib-1.9/test/realn02_exp.sam\n",
    "samtools-1.9/htslib-1.9/test/sam.c\n",
    "samtools-1.9/htslib-1.9/test/tabix/\n",
    "samtools-1.9/htslib-1.9/test/tabix/bed_file.Y.100200.out\n",
    "samtools-1.9/htslib-1.9/test/tabix/bed_file.bed\n",
    "samtools-1.9/htslib-1.9/test/tabix/gff_file.X.2934832.2935190.out\n",
    "samtools-1.9/htslib-1.9/test/tabix/gff_file.gff\n",
    "samtools-1.9/htslib-1.9/test/tabix/large_chr.20.1.2147483647.out\n",
    "samtools-1.9/htslib-1.9/test/tabix/large_chr.vcf\n",
    "samtools-1.9/htslib-1.9/test/tabix/tabix.tst\n",
    "samtools-1.9/htslib-1.9/test/tabix/test-tabix.sh\n",
    "samtools-1.9/htslib-1.9/test/tabix/vcf_file.1.3000151.out\n",
    "samtools-1.9/htslib-1.9/test/tabix/vcf_file.2.3199812.out\n",
    "samtools-1.9/htslib-1.9/test/tabix/vcf_file.vcf\n",
    "samtools-1.9/htslib-1.9/test/test-bcf-sr.c\n",
    "samtools-1.9/htslib-1.9/test/test-bcf-sr.pl\n",
    "samtools-1.9/htslib-1.9/test/test-bcf-translate.c\n",
    "samtools-1.9/htslib-1.9/test/test-bcf-translate.out\n",
    "samtools-1.9/htslib-1.9/test/test-logging.pl\n",
    "samtools-1.9/htslib-1.9/test/test-regidx.c\n",
    "samtools-1.9/htslib-1.9/test/test-vcf-api.c\n",
    "samtools-1.9/htslib-1.9/test/test-vcf-api.out\n",
    "samtools-1.9/htslib-1.9/test/test-vcf-hdr-in.vcf\n",
    "samtools-1.9/htslib-1.9/test/test-vcf-hdr.out\n",
    "samtools-1.9/htslib-1.9/test/test-vcf-sweep.c\n",
    "samtools-1.9/htslib-1.9/test/test-vcf-sweep.out\n",
    "samtools-1.9/htslib-1.9/test/test.pl\n",
    "samtools-1.9/htslib-1.9/test/test_bgzf.c\n",
    "samtools-1.9/htslib-1.9/test/test_realn.c\n",
    "samtools-1.9/htslib-1.9/test/test_view.c\n",
    "samtools-1.9/htslib-1.9/test/thrash_threads1.c\n",
    "samtools-1.9/htslib-1.9/test/thrash_threads2.c\n",
    "samtools-1.9/htslib-1.9/test/thrash_threads3.c\n",
    "samtools-1.9/htslib-1.9/test/thrash_threads4.c\n",
    "samtools-1.9/htslib-1.9/test/thrash_threads5.c\n",
    "samtools-1.9/htslib-1.9/test/thrash_threads6.c\n",
    "samtools-1.9/htslib-1.9/test/thread_pool.md\n",
    "samtools-1.9/htslib-1.9/test/xx#blank.sam\n",
    "samtools-1.9/htslib-1.9/test/xx#large_aux.sam\n",
    "samtools-1.9/htslib-1.9/test/xx#large_aux2.sam\n",
    "samtools-1.9/htslib-1.9/test/xx#large_aux_java.cram\n",
    "samtools-1.9/htslib-1.9/test/xx#minimal.sam\n",
    "samtools-1.9/htslib-1.9/test/xx#pair.sam\n",
    "samtools-1.9/htslib-1.9/test/xx#repeated.sam\n",
    "samtools-1.9/htslib-1.9/test/xx#rg.sam\n",
    "samtools-1.9/htslib-1.9/test/xx#tlen.sam\n",
    "samtools-1.9/htslib-1.9/test/xx#tlen2.sam\n",
    "samtools-1.9/htslib-1.9/test/xx#triplet.sam\n",
    "samtools-1.9/htslib-1.9/test/xx#unsorted.sam\n",
    "samtools-1.9/htslib-1.9/test/xx.fa\n",
    "samtools-1.9/htslib-1.9/test/xx.fa.fai\n",
    "samtools-1.9/htslib-1.9/textutils.c\n",
    "samtools-1.9/htslib-1.9/textutils_internal.h\n",
    "samtools-1.9/htslib-1.9/thread_pool.c\n",
    "samtools-1.9/htslib-1.9/thread_pool_internal.h\n",
    "samtools-1.9/htslib-1.9/vcf.5\n",
    "samtools-1.9/htslib-1.9/vcf.c\n",
    "samtools-1.9/htslib-1.9/vcf_sweep.c\n",
    "samtools-1.9/htslib-1.9/vcfutils.c\n",
    "samtools-1.9/htslib-1.9/version.sh\n",
    "samtools-1.9/htslib-1.9/config.h.in\n",
    "samtools-1.9/htslib-1.9/configure\n",
    "[nathalia@aws] $  cd samtools-1.9\n",
    "[nathalia@aws] $  make\n",
    "Makefile:110: config.mk: No such file or directory\n",
    "config.mk:34: htslib-1.9/htslib_static.mk: No such file or directory\n",
    "cd htslib-1.9 && make htslib.pc.tmp\n",
    "make[1]: Entering directory '/workspace/rnaseq/student_tools/samtools-1.9/htslib-1.9'\n",
    "sed -e '/^static_libs=/s/@static_LIBS@/-lz -lm -lbz2 -llzma/;s#@[^-][^@]*@##g' htslib.pc.in > htslib.pc.tmp\n",
    "make[1]: Leaving directory '/workspace/rnaseq/student_tools/samtools-1.9/htslib-1.9'\n",
    "cd htslib-1.9 && make htslib_static.mk\n",
    "make[1]: Entering directory '/workspace/rnaseq/student_tools/samtools-1.9/htslib-1.9'\n",
    "sed -n '/^static_libs=/s/[^=]*=/HTSLIB_static_LIBS = /p;/^static_ldflags=/s/[^=]*=/HTSLIB_static_LDFLAGS = /p' htslib.pc.tmp > htslib_static.mk\n",
    "make[1]: Leaving directory '/workspace/rnaseq/student_tools/samtools-1.9/htslib-1.9'\n",
    "echo '/* Basic config.h generated by Makefile */' > config.h\n",
    "echo '#define HAVE_CURSES' >> config.h\n",
    "echo '#define HAVE_CURSES_H' >> config.h\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_index.o bam_index.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_plcmd.o bam_plcmd.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o sam_view.o sam_view.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_cat.o bam_cat.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_md.o bam_md.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_reheader.o bam_reheader.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_sort.o bam_sort.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bedidx.o bedidx.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_rmdup.o bam_rmdup.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_rmdupse.o bam_rmdupse.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_mate.o bam_mate.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_stat.o bam_stat.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_color.o bam_color.c\n",
    "echo '#define SAMTOOLS_VERSION \"1.9\"' > version.h\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bamtk.o bamtk.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam2bcf.o bam2bcf.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam2bcf_indel.o bam2bcf_indel.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o sample.o sample.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o cut_target.o cut_target.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o phase.o phase.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam2depth.o bam2depth.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o padding.o padding.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bedcov.o bedcov.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bamshuf.o bamshuf.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o faidx.o faidx.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o dict.o dict.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o stats.o stats.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o stats_isize.o stats_isize.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_flags.o bam_flags.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_split.o bam_split.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_tview.o bam_tview.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_tview_curses.o bam_tview_curses.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_tview_html.o bam_tview_html.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_lpileup.o bam_lpileup.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_quickcheck.o bam_quickcheck.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_addrprg.o bam_addrprg.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_markdup.o bam_markdup.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o tmp_file.o tmp_file.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o lz4/lz4.o lz4/lz4.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_aux.o bam_aux.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam.o bam.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_import.o bam_import.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o sam.o sam.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o sam_header.o sam_header.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o bam_plbuf.o bam_plbuf.c\n",
    "ar -csru libbam.a bam_aux.o bam.o bam_import.o sam.o sam_header.o bam_plbuf.o\n",
    "ar: `u' modifier ignored since `D' is the default (see `U')\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o sam_opts.o sam_opts.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o sam_utils.o sam_utils.c\n",
    "ar -rcs libst.a sam_opts.o sam_utils.o\n",
    "cd htslib-1.9 && make config.h\n",
    "make[1]: Entering directory '/workspace/rnaseq/student_tools/samtools-1.9/htslib-1.9'\n",
    "echo '/* Default config.h generated by Makefile */' > config.h\n",
    "echo '#define HAVE_LIBBZ2 1' >> config.h\n",
    "echo '#define HAVE_LIBLZMA 1' >> config.h\n",
    "echo '#define HAVE_LZMA_H 1' >> config.h\n",
    "echo '#define HAVE_FSEEKO 1' >> config.h\n",
    "echo '#define HAVE_DRAND48 1' >> config.h\n",
    "make[1]: Leaving directory '/workspace/rnaseq/student_tools/samtools-1.9/htslib-1.9'\n",
    "cd htslib-1.9 && make lib-static\n",
    "make[1]: Entering directory '/workspace/rnaseq/student_tools/samtools-1.9/htslib-1.9'\n",
    "gcc -g -Wall -O2 -I.  -c -o kfunc.o kfunc.c\n",
    "gcc -g -Wall -O2 -I.  -c -o knetfile.o knetfile.c\n",
    "gcc -g -Wall -O2 -I.  -c -o kstring.o kstring.c\n",
    "gcc -g -Wall -O2 -I.  -c -o bcf_sr_sort.o bcf_sr_sort.c\n",
    "gcc -g -Wall -O2 -I.  -c -o bgzf.o bgzf.c\n",
    "gcc -g -Wall -O2 -I.  -c -o errmod.o errmod.c\n",
    "gcc -g -Wall -O2 -I.  -c -o faidx.o faidx.c\n",
    "gcc -g -Wall -O2 -I.  -c -o hfile.o hfile.c\n",
    "gcc -g -Wall -O2 -I.  -c -o hfile_net.o hfile_net.c\n",
    "echo '#define HTS_VERSION \"1.9\"' > version.h\n",
    "gcc -g -Wall -O2 -I.  -c -o hts.o hts.c\n",
    "gcc -g -Wall -O2 -I.  -c -o hts_os.o hts_os.c\n",
    "gcc -g -Wall -O2 -I.  -c -o md5.o md5.c\n",
    "gcc -g -Wall -O2 -I.  -c -o multipart.o multipart.c\n",
    "gcc -g -Wall -O2 -I.  -c -o probaln.o probaln.c\n",
    "gcc -g -Wall -O2 -I.  -c -o realn.o realn.c\n",
    "gcc -g -Wall -O2 -I.  -c -o regidx.o regidx.c\n",
    "gcc -g -Wall -O2 -I.  -c -o sam.o sam.c\n",
    "gcc -g -Wall -O2 -I.  -c -o synced_bcf_reader.o synced_bcf_reader.c\n",
    "gcc -g -Wall -O2 -I.  -c -o vcf_sweep.o vcf_sweep.c\n",
    "gcc -g -Wall -O2 -I.  -c -o tbx.o tbx.c\n",
    "gcc -g -Wall -O2 -I.  -c -o textutils.o textutils.c\n",
    "gcc -g -Wall -O2 -I.  -c -o thread_pool.o thread_pool.c\n",
    "gcc -g -Wall -O2 -I.  -c -o vcf.o vcf.c\n",
    "gcc -g -Wall -O2 -I.  -c -o vcfutils.o vcfutils.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/cram_codecs.o cram/cram_codecs.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/cram_decode.o cram/cram_decode.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/cram_encode.o cram/cram_encode.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/cram_external.o cram/cram_external.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/cram_index.o cram/cram_index.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/cram_io.o cram/cram_io.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/cram_samtools.o cram/cram_samtools.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/cram_stats.o cram/cram_stats.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/files.o cram/files.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/mFILE.o cram/mFILE.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/open_trace_file.o cram/open_trace_file.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/pooled_alloc.o cram/pooled_alloc.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/rANS_static.o cram/rANS_static.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/sam_header.o cram/sam_header.c\n",
    "gcc -g -Wall -O2 -I.  -c -o cram/string_alloc.o cram/string_alloc.c\n",
    "ar -rc libhts.a kfunc.o knetfile.o kstring.o bcf_sr_sort.o bgzf.o errmod.o faidx.o hfile.o hfile_net.o hts.o hts_os.o md5.o multipart.o probaln.o realn.o regidx.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o textutils.o thread_pool.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/files.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/rANS_static.o cram/sam_header.o cram/string_alloc.o\n",
    "ranlib libhts.a\n",
    "make[1]: Leaving directory '/workspace/rnaseq/student_tools/samtools-1.9/htslib-1.9'\n",
    "gcc  -L./lz4  -o samtools bam_index.o bam_plcmd.o sam_view.o bam_cat.o bam_md.o bam_reheader.o bam_sort.o bedidx.o bam_rmdup.o bam_rmdupse.o bam_mate.o bam_stat.o bam_color.o bamtk.o bam2bcf.o bam2bcf_indel.o sample.o cut_target.o phase.o bam2depth.o padding.o bedcov.o bamshuf.o faidx.o dict.o stats.o stats_isize.o bam_flags.o bam_split.o bam_tview.o bam_tview_curses.o bam_tview_html.o bam_lpileup.o bam_quickcheck.o bam_addrprg.o bam_markdup.o tmp_file.o ./lz4/lz4.o libbam.a libst.a htslib-1.9/libhts.a -lz -lm -lbz2 -llzma -lcurses -lm -lz  -lpthread\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o misc/ace2sam.o misc/ace2sam.c\n",
    "gcc  -o misc/ace2sam misc/ace2sam.o -lz \n",
    "gcc -g -Wall -O2 -DMAQ_LONGREADS -I. -Ihtslib-1.9 -I./lz4  -c -o misc/maq2sam-long.o misc/maq2sam.c\n",
    "gcc  -o misc/maq2sam-long misc/maq2sam-long.o -lz \n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o misc/maq2sam-short.o misc/maq2sam.c\n",
    "gcc  -o misc/maq2sam-short misc/maq2sam-short.o -lz \n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o misc/md5fa.o misc/md5fa.c\n",
    "gcc  -L./lz4  -o misc/md5fa misc/md5fa.o htslib-1.9/libhts.a -lz -lm -lbz2 -llzma -lz \n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o misc/md5sum-lite.o misc/md5sum-lite.c\n",
    "gcc  -L./lz4  -o misc/md5sum-lite misc/md5sum-lite.o htslib-1.9/libhts.a -lz -lm -lbz2 -llzma -lz \n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o misc/wgsim.o misc/wgsim.c\n",
    "gcc  -L./lz4  -o misc/wgsim misc/wgsim.o -lm htslib-1.9/libhts.a -lz -lm -lbz2 -llzma -lz \n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o test/merge/test_bam_translate.o test/merge/test_bam_translate.c\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o test/test.o test/test.c\n",
    "gcc  -L./lz4  -o test/merge/test_bam_translate test/merge/test_bam_translate.o test/test.o libst.a htslib-1.9/libhts.a -lz -lm -lbz2 -llzma -lz  -lpthread\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o test/merge/test_rtrans_build.o test/merge/test_rtrans_build.c\n",
    "gcc  -L./lz4  -o test/merge/test_rtrans_build test/merge/test_rtrans_build.o libst.a htslib-1.9/libhts.a -lz -lm -lbz2 -llzma -lz  -lpthread\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o test/merge/test_trans_tbl_init.o test/merge/test_trans_tbl_init.c\n",
    "gcc  -L./lz4  -o test/merge/test_trans_tbl_init test/merge/test_trans_tbl_init.o libst.a htslib-1.9/libhts.a -lz -lm -lbz2 -llzma -lz  -lpthread\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o test/split/test_count_rg.o test/split/test_count_rg.c\n",
    "gcc  -L./lz4  -o test/split/test_count_rg test/split/test_count_rg.o test/test.o libst.a htslib-1.9/libhts.a -lz -lm -lbz2 -llzma -lz  -lpthread\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o test/split/test_expand_format_string.o test/split/test_expand_format_string.c\n",
    "gcc  -L./lz4  -o test/split/test_expand_format_string test/split/test_expand_format_string.o test/test.o libst.a htslib-1.9/libhts.a -lz -lm -lbz2 -llzma -lz  -lpthread\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o test/split/test_filter_header_rg.o test/split/test_filter_header_rg.c\n",
    "gcc  -L./lz4  -o test/split/test_filter_header_rg test/split/test_filter_header_rg.o test/test.o libst.a htslib-1.9/libhts.a -lz -lm -lbz2 -llzma -lz  -lpthread\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o test/split/test_parse_args.o test/split/test_parse_args.c\n",
    "gcc  -L./lz4  -o test/split/test_parse_args test/split/test_parse_args.o test/test.o libst.a htslib-1.9/libhts.a -lz -lm -lbz2 -llzma -lz  -lpthread\n",
    "gcc -g -Wall -O2 -I. -Ihtslib-1.9 -I./lz4  -c -o test/vcf-miniview.o test/vcf-miniview.c\n",
    "gcc  -L./lz4  -o test/vcf-miniview test/vcf-miniview.o htslib-1.9/libhts.a -lz -lm -lbz2 -llzma -lz  -lpthread\n",
    "\n",
    "[nathalia@aws] $  ./samtools \n",
    "\n",
    "Program: samtools (Tools for alignments in the SAM format)\n",
    "Version: 1.9 (using htslib 1.9)\n",
    "\n",
    "Usage:   samtools <command> [options]\n",
    "\n",
    "Commands:\n",
    "  -- Indexing\n",
    "     dict           create a sequence dictionary file\n",
    "     faidx          index/extract FASTA\n",
    "     fqidx          index/extract FASTQ\n",
    "     index          index alignment\n",
    "\n",
    "  -- Editing\n",
    "     calmd          recalculate MD/NM tags and '=' bases\n",
    "     fixmate        fix mate information\n",
    "     reheader       replace BAM header\n",
    "     targetcut      cut fosmid regions (for fosmid pool only)\n",
    "     addreplacerg   adds or replaces RG tags\n",
    "     markdup        mark duplicates\n",
    "\n",
    "  -- File operations\n",
    "     collate        shuffle and group alignments by name\n",
    "     cat            concatenate BAMs\n",
    "     merge          merge sorted alignments\n",
    "     mpileup        multi-way pileup\n",
    "     sort           sort alignment file\n",
    "     split          splits a file by read group\n",
    "     quickcheck     quickly check if SAM/BAM/CRAM file appears intact\n",
    "     fastq          converts a BAM to a FASTQ\n",
    "     fasta          converts a BAM to a FASTA\n",
    "\n",
    "  -- Statistics\n",
    "     bedcov         read depth per BED region\n",
    "     depth          compute the depth\n",
    "     flagstat       simple stats\n",
    "     idxstats       BAM index stats\n",
    "     phase          phase heterozygotes\n",
    "     stats          generate stats (former bamcheck)\n",
    "\n",
    "  -- Viewing\n",
    "     flags          explain BAM flags\n",
    "     tview          text alignment viewer\n",
    "     view           SAM<->BAM<->CRAM conversion\n",
    "     depad          convert padded BAM to unpadded BAM\n",
    "\n",
    "\n",
    "[nathalia@aws] $  cd ..\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 20M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20M Jul 18 09:19 samtools-1.9.tar\n",
    "drwxrwxr-x 9 ubuntu ubuntu 4.0K Nov 12 22:32 samtools-1.9\n",
    "\n",
    "[nathalia@aws] $  export SAMTOOLS_ROOT=$RNA_HOME/student_tools/samtools-1.9\n",
    "\n",
    "[nathalia@aws] $  echo $SAMTOOLS_ROOT\n",
    "/home/ubuntu/workspace/rnaseq/student_tools/samtools-1.9\n",
    "\n",
    "```\n",
    "\n",
    "Will not copy all the stdout of all instalation... Follow https://github.com/griffithlab/rnaseq_tutorial/wiki/Installation which is excellent! It also changes the directory to the `$RNA_HOME/student_tools/` which I totally forgot when copying and pasting the stuff hahaha"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Practice 1\n",
    "\n",
    "Bedtools"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  wget https://github.com/arq5x/bedtools2/releases/download/v2.27.1/bedtools-2.27.1.tar.gz\n",
    "--2018-11-12 22:55:17--  https://github.com/arq5x/bedtools2/releases/download/v2.27.1/bedtools-2.27.1.tar.gz\n",
    "Resolving github.com (github.com)... 192.30.255.112, 192.30.255.113\n",
    "Connecting to github.com (github.com)|192.30.255.112|:443... connected.\n",
    "HTTP request sent, awaiting response... 302 Found\n",
    "Location: https://github-production-release-asset-2e65be.s3.amazonaws.com/15059334/d4d69050-e0b8-11e7-970f-946a0a2e64d3?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20181112%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20181112T225517Z&X-Amz-Expires=300&X-Amz-Signature=c8931cf386cb58fb7238c7379a049bc93e5033d33373a0fa93d7c557e5eb0761&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3Dbedtools-2.27.1.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
    "--2018-11-12 22:55:17--  https://github-production-release-asset-2e65be.s3.amazonaws.com/15059334/d4d69050-e0b8-11e7-970f-946a0a2e64d3?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20181112%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20181112T225517Z&X-Amz-Expires=300&X-Amz-Signature=c8931cf386cb58fb7238c7379a049bc93e5033d33373a0fa93d7c557e5eb0761&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3Dbedtools-2.27.1.tar.gz&response-content-type=application%2Foctet-stream\n",
    "Resolving github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)... 52.216.163.147\n",
    "Connecting to github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)|52.216.163.147|:443... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 20014704 (19M) [application/octet-stream]\n",
    "Saving to: ‘bedtools-2.27.1.tar.gz’\n",
    "\n",
    "bedtools-2.27.1.tar.gz    100%[==================================>]  19.09M  13.2MB/s    in 1.5s    \n",
    "\n",
    "2018-11-12 22:55:19 (13.2 MB/s) - ‘bedtools-2.27.1.tar.gz’ saved [20014704/20014704]\n",
    "\n",
    "[nathalia@aws] $  tar -xzvf b\n",
    "bam-readcount/          bedtools-2.27.1.tar.gz  \n",
    "[nathalia@aws] $  tar -xzvf b\n",
    "bam-readcount/          bedtools-2.27.1.tar.gz  \n",
    "[nathalia@aws] $  tar -xzvf bedtools-2.27.1.tar.gz \n",
    "bedtools2/\n",
    "bedtools2/obj/\n",
    "bedtools2/tutorial/\n",
    "bedtools2/LICENSE\n",
    "bedtools2/test/\n",
    "bedtools2/bin/\n",
    "bedtools2/Makefile\n",
    "bedtools2/docs/\n",
    "bedtools2/README.md\n",
    "bedtools2/genomes/\n",
    "bedtools2/.gitignore\n",
    "bedtools2/scripts/\n",
    "bedtools2/data/\n",
    "bedtools2/RELEASE_HISTORY\n",
    "bedtools2/.travis.yml\n",
    "bedtools2/src/\n",
    "bedtools2/src/spacingFile/\n",
    "bedtools2/src/sortBed/\n",
    "bedtools2/src/split/\n",
    "bedtools2/src/bedToBam/\n",
    "bedtools2/src/complementFile/\n",
    "bedtools2/src/expand/\n",
    "bedtools2/src/fisher/\n",
    "bedtools2/src/linksBed/\n",
    "bedtools2/src/pairToPair/\n",
    "bedtools2/src/bedtools.cpp\n",
    "bedtools2/src/mapFile/\n",
    "bedtools2/src/slopBed/\n",
    "bedtools2/src/annotateBed/\n",
    "bedtools2/src/randomBed/\n",
    "bedtools2/src/getOverlap/\n",
    "bedtools2/src/reldist/\n",
    "bedtools2/src/nucBed/\n",
    "bedtools2/src/bedToIgv/\n",
    "bedtools2/src/regressTest/\n",
    "bedtools2/src/maskFastaFromBed/\n",
    "bedtools2/src/genomeCoverageBed/\n",
    "bedtools2/src/utils/\n",
    "bedtools2/src/windowBed/\n",
    "bedtools2/src/jaccard/\n",
    "bedtools2/src/groupBy/\n",
    "bedtools2/src/bamToFastq/\n",
    "bedtools2/src/fastaFromBed/\n",
    "bedtools2/src/nekSandbox1/\n",
    "bedtools2/src/unionBedGraphs/\n",
    "bedtools2/src/bed12ToBed6/\n",
    "bedtools2/src/subtractFile/\n",
    "bedtools2/src/bedpeToBam/\n",
    "bedtools2/src/windowMaker/\n",
    "bedtools2/src/coverageFile/\n",
    "bedtools2/src/multiIntersectBed/\n",
    "bedtools2/src/intersectFile/\n",
    "bedtools2/src/closestFile/\n",
    "bedtools2/src/shuffleBed/\n",
    "bedtools2/src/flankBed/\n",
    "bedtools2/src/tagBam/\n",
    "bedtools2/src/multiBamCov/\n",
    "bedtools2/src/sampleFile/\n",
    "bedtools2/src/clusterBed/\n",
    "bedtools2/src/shiftBed/\n",
    "bedtools2/src/mergeFile/\n",
    "bedtools2/src/bamToBed/\n",
    "bedtools2/src/pairToBed/\n",
    "bedtools2/src/pairToBed/pairToBed.h\n",
    "bedtools2/src/pairToBed/Makefile\n",
    "bedtools2/src/pairToBed/pairToBedMain.cpp\n",
    "bedtools2/src/pairToBed/pairToBed.cpp\n",
    "bedtools2/src/bamToBed/bamToBed.cpp\n",
    "bedtools2/src/bamToBed/Makefile\n",
    "bedtools2/src/mergeFile/mergeFile.cpp\n",
    "bedtools2/src/mergeFile/Makefile\n",
    "bedtools2/src/mergeFile/mergeHelp.cpp\n",
    "bedtools2/src/mergeFile/mergeFile.h\n",
    "bedtools2/src/shiftBed/shiftBed.cpp\n",
    "bedtools2/src/shiftBed/Makefile\n",
    "bedtools2/src/shiftBed/shiftBedMain.cpp\n",
    "bedtools2/src/shiftBed/shiftBed.h\n",
    "bedtools2/src/clusterBed/clusterBed.h\n",
    "bedtools2/src/clusterBed/Makefile\n",
    "bedtools2/src/clusterBed/clusterBed.cpp\n",
    "bedtools2/src/clusterBed/clusterMain.cpp\n",
    "bedtools2/src/sampleFile/sampleFile.cpp\n",
    "bedtools2/src/sampleFile/Makefile\n",
    "bedtools2/src/sampleFile/sampleFile.h\n",
    "bedtools2/src/sampleFile/sampleHelp.cpp\n",
    "bedtools2/src/multiBamCov/Makefile\n",
    "bedtools2/src/multiBamCov/multiBamCov.cpp\n",
    "bedtools2/src/multiBamCov/multiBamCov.h\n",
    "bedtools2/src/multiBamCov/multiBamCovMain.cpp\n",
    "bedtools2/src/tagBam/tagBam.cpp\n",
    "bedtools2/src/tagBam/tagBamMain.cpp\n",
    "bedtools2/src/tagBam/Makefile\n",
    "bedtools2/src/tagBam/tagBam.h\n",
    "bedtools2/src/flankBed/flankBed.cpp\n",
    "bedtools2/src/flankBed/flankBedMain.cpp\n",
    "bedtools2/src/flankBed/Makefile\n",
    "bedtools2/src/flankBed/flankBed.h\n",
    "bedtools2/src/shuffleBed/shuffleBed.h\n",
    "bedtools2/src/shuffleBed/Makefile\n",
    "bedtools2/src/shuffleBed/shuffleBed.cpp\n",
    "bedtools2/src/shuffleBed/shuffleBedMain.cpp\n",
    "bedtools2/src/closestFile/Makefile\n",
    "bedtools2/src/closestFile/closestHelp.cpp\n",
    "bedtools2/src/closestFile/closestFile.h\n",
    "bedtools2/src/closestFile/closestFile.cpp\n",
    "bedtools2/src/intersectFile/Makefile\n",
    "bedtools2/src/intersectFile/intersectHelp.cpp\n",
    "bedtools2/src/intersectFile/intersectFile.cpp\n",
    "bedtools2/src/intersectFile/intersectFile.h\n",
    "bedtools2/src/multiIntersectBed/multiIntersectBed.cpp\n",
    "bedtools2/src/multiIntersectBed/Makefile\n",
    "bedtools2/src/multiIntersectBed/multiIntersectBed.h\n",
    "bedtools2/src/multiIntersectBed/multiIntersectBedMain.cpp\n",
    "bedtools2/src/coverageFile/Makefile\n",
    "bedtools2/src/coverageFile/coverageFile.cpp\n",
    "bedtools2/src/coverageFile/coverageFile.h\n",
    "bedtools2/src/coverageFile/coverageHelp.cpp\n",
    "bedtools2/src/windowMaker/Makefile\n",
    "bedtools2/src/windowMaker/windowMakerMain.cpp\n",
    "bedtools2/src/windowMaker/windowMaker.h\n",
    "bedtools2/src/windowMaker/windowMaker.cpp\n",
    "bedtools2/src/bedpeToBam/notes.txt\n",
    "bedtools2/src/bedpeToBam/bedpeToBam.cpp\n",
    "bedtools2/src/bedpeToBam/Makefile\n",
    "bedtools2/src/subtractFile/subtractHelp.cpp\n",
    "bedtools2/src/subtractFile/Makefile\n",
    "bedtools2/src/subtractFile/subtractFile.h\n",
    "bedtools2/src/subtractFile/subtractFile.cpp\n",
    "bedtools2/src/bed12ToBed6/Makefile\n",
    "bedtools2/src/bed12ToBed6/bed12ToBed6.cpp\n",
    "bedtools2/src/unionBedGraphs/unionBedGraphs.cpp\n",
    "bedtools2/src/unionBedGraphs/Makefile\n",
    "bedtools2/src/unionBedGraphs/unionBedGraphsMain.cpp\n",
    "bedtools2/src/unionBedGraphs/unionBedGraphs.h\n",
    "bedtools2/src/nekSandbox1/FormatGuess.h\n",
    "bedtools2/src/nekSandbox1/Makefile\n",
    "bedtools2/src/nekSandbox1/nekSandboxMain.cpp\n",
    "bedtools2/src/fastaFromBed/Makefile\n",
    "bedtools2/src/fastaFromBed/fastaFromBedMain.cpp\n",
    "bedtools2/src/fastaFromBed/fastaFromBed.cpp\n",
    "bedtools2/src/fastaFromBed/fastaFromBed.h\n",
    "bedtools2/src/bamToFastq/bamToFastq.h\n",
    "bedtools2/src/bamToFastq/Makefile\n",
    "bedtools2/src/bamToFastq/bamToFastq.cpp\n",
    "bedtools2/src/bamToFastq/bamToFastqMain.cpp\n",
    "bedtools2/src/groupBy/groupBy.cpp\n",
    "bedtools2/src/groupBy/Makefile\n",
    "bedtools2/src/groupBy/groupBy.h\n",
    "bedtools2/src/groupBy/groupByHelp.cpp\n",
    "bedtools2/src/jaccard/jaccard.cpp\n",
    "bedtools2/src/jaccard/Makefile\n",
    "bedtools2/src/jaccard/jaccardHelp.cpp\n",
    "bedtools2/src/jaccard/jaccard.h\n",
    "bedtools2/src/windowBed/windowBed.cpp\n",
    "bedtools2/src/windowBed/Makefile\n",
    "bedtools2/src/windowBed/windowBed.h\n",
    "bedtools2/src/windowBed/windowMain.cpp\n",
    "bedtools2/src/utils/tabFile/\n",
    "bedtools2/src/utils/NewChromsweep/\n",
    "bedtools2/src/utils/RecordOutputMgr/\n",
    "bedtools2/src/utils/bedFile/\n",
    "bedtools2/src/utils/Contexts/\n",
    "bedtools2/src/utils/BamTools/\n",
    "bedtools2/src/utils/BlockedIntervals/\n",
    "bedtools2/src/utils/general/\n",
    "bedtools2/src/utils/driver/\n",
    "bedtools2/src/utils/Point/\n",
    "bedtools2/src/utils/bedGraphFile/\n",
    "bedtools2/src/utils/sequenceUtilities/\n",
    "bedtools2/src/utils/Fasta/\n",
    "bedtools2/src/utils/gzstream/\n",
    "bedtools2/src/utils/fileType/\n",
    "bedtools2/src/utils/KeyListOps/\n",
    "bedtools2/src/utils/BamTools-Ancillary/\n",
    "bedtools2/src/utils/GenomeFile/\n",
    "bedtools2/src/utils/version/\n",
    "bedtools2/src/utils/ToolBase/\n",
    "bedtools2/src/utils/lineFileUtilities/\n",
    "bedtools2/src/utils/bedFilePE/\n",
    "bedtools2/src/utils/chromsweep/\n",
    "bedtools2/src/utils/FileRecordTools/\n",
    "bedtools2/src/utils/BinTree/\n",
    "bedtools2/src/utils/VectorOps/\n",
    "bedtools2/src/utils/stringUtilities/\n",
    "bedtools2/src/utils/stringUtilities/stringUtilities.h\n",
    "bedtools2/src/utils/VectorOps/Makefile\n",
    "bedtools2/src/utils/VectorOps/VectorOps.cpp\n",
    "bedtools2/src/utils/VectorOps/VectorOps.h\n",
    "bedtools2/src/utils/BinTree/BinTree.cpp\n",
    "bedtools2/src/utils/BinTree/Makefile\n",
    "bedtools2/src/utils/BinTree/BinTree.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/\n",
    "bedtools2/src/utils/FileRecordTools/FileRecordMergeMgr.h\n",
    "bedtools2/src/utils/FileRecordTools/FileRecordMgr.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Makefile\n",
    "bedtools2/src/utils/FileRecordTools/FileRecordMgr.h\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/\n",
    "bedtools2/src/utils/FileRecordTools/FileRecordMergeMgr.cpp\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/InputStreamMgr.h\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/BamFileReader.h\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.cpp\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/Makefile\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.cpp\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTransferBuffer.cpp\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTransferBuffer.h\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/InputStreamMgr.cpp\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/FileReader.cpp\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/BamFileReader.cpp\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/SingleLineDelimTextFileReader.h\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/FileReader.h\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/BufferedStreamMgr.h\n",
    "bedtools2/src/utils/FileRecordTools/FileReaders/Makefile~\n",
    "bedtools2/src/utils/FileRecordTools/Records/BedGraphInterval.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/RecordMgr.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/RecordList.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/Bed12Interval.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/GffRecord.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/Bed4Interval.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/BedGraphInterval.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/RecordList.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/BamRecord.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/RecordKeyVector.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/Makefile\n",
    "bedtools2/src/utils/FileRecordTools/Records/PlusFields.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/StrandQueue.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/Bed5Interval.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/Record.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/GffRecord.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/Bed4Interval.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/VcfRecord.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/Bed3Interval.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/RecordKeyList.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/VcfRecord.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/EmptyRecord.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/Bed3Interval.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/RecordKeyVector.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/Bed6Interval.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/BlockMgr.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/RecordMgr.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/Bed6Interval.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/GffPlusRecord.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/GffPlusRecord.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/BedPlusInterval.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/StrandQueue.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/Record.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/Bed5Interval.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/NoPosPlusRecord.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/Bed12Interval.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/BlockMgr.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/RecordKeyList.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/PlusFields.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/BamRecord.h\n",
    "bedtools2/src/utils/FileRecordTools/Records/NoPosPlusRecord.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/EmptyRecord.cpp\n",
    "bedtools2/src/utils/FileRecordTools/Records/BedPlusInterval.h\n",
    "bedtools2/src/utils/chromsweep/Makefile\n",
    "bedtools2/src/utils/chromsweep/chromsweep.cpp\n",
    "bedtools2/src/utils/chromsweep/chromsweep.h\n",
    "bedtools2/src/utils/bedFilePE/Makefile\n",
    "bedtools2/src/utils/bedFilePE/bedFilePE.cpp\n",
    "bedtools2/src/utils/bedFilePE/bedFilePE.h\n",
    "bedtools2/src/utils/lineFileUtilities/lineFileUtilities.h\n",
    "bedtools2/src/utils/ToolBase/Makefile\n",
    "bedtools2/src/utils/ToolBase/ToolBase.cpp\n",
    "bedtools2/src/utils/ToolBase/ToolBase.h\n",
    "bedtools2/src/utils/version/version.h\n",
    "bedtools2/src/utils/version/version.cpp\n",
    "bedtools2/src/utils/version/Makefile\n",
    "bedtools2/src/utils/version/version_release.txt\n",
    "bedtools2/src/utils/GenomeFile/NewGenomeFile.cpp\n",
    "bedtools2/src/utils/GenomeFile/NewGenomeFile.h\n",
    "bedtools2/src/utils/GenomeFile/Makefile\n",
    "bedtools2/src/utils/GenomeFile/GenomeFile.cpp\n",
    "bedtools2/src/utils/GenomeFile/GenomeFile.h\n",
    "bedtools2/src/utils/BamTools-Ancillary/BamAncillary.h\n",
    "bedtools2/src/utils/BamTools-Ancillary/Makefile\n",
    "bedtools2/src/utils/BamTools-Ancillary/BamAncillary.cpp\n",
    "bedtools2/src/utils/KeyListOps/KeyListOps.h\n",
    "bedtools2/src/utils/KeyListOps/KeyListOps.cpp\n",
    "bedtools2/src/utils/KeyListOps/KeyListOpsMethods.cpp\n",
    "bedtools2/src/utils/KeyListOps/Makefile\n",
    "bedtools2/src/utils/KeyListOps/KeyListOpsMethods.h\n",
    "bedtools2/src/utils/fileType/fileType.cpp\n",
    "bedtools2/src/utils/fileType/Makefile\n",
    "bedtools2/src/utils/fileType/FileRecordTypeChecker.h\n",
    "bedtools2/src/utils/fileType/FileRecordTypeChecker.cpp\n",
    "bedtools2/src/utils/fileType/fileType.h\n",
    "bedtools2/src/utils/gzstream/Makefile\n",
    "bedtools2/src/utils/gzstream/gzstream.C\n",
    "bedtools2/src/utils/gzstream/README\n",
    "bedtools2/src/utils/gzstream/COPYING.LIB\n",
    "bedtools2/src/utils/gzstream/version\n",
    "bedtools2/src/utils/gzstream/gzstream.h\n",
    "bedtools2/src/utils/Fasta/Fasta.h\n",
    "bedtools2/src/utils/Fasta/split.cpp\n",
    "bedtools2/src/utils/Fasta/Makefile\n",
    "bedtools2/src/utils/Fasta/Fasta.cpp\n",
    "bedtools2/src/utils/Fasta/LargeFileSupport.h\n",
    "bedtools2/src/utils/Fasta/split.h\n",
    "bedtools2/src/utils/sequenceUtilities/Makefile\n",
    "bedtools2/src/utils/sequenceUtilities/sequenceUtils.h\n",
    "bedtools2/src/utils/sequenceUtilities/sequenceUtils.cpp\n",
    "bedtools2/src/utils/bedGraphFile/bedGraphFile.h\n",
    "bedtools2/src/utils/bedGraphFile/Makefile\n",
    "bedtools2/src/utils/bedGraphFile/bedGraphFile.cpp\n",
    "bedtools2/src/utils/Point/Point.h\n",
    "bedtools2/src/utils/driver/BedtoolsDriver.h\n",
    "bedtools2/src/utils/driver/BedtoolsDriver.cpp\n",
    "bedtools2/src/utils/driver/Makefile\n",
    "bedtools2/src/utils/general/QuickString.cpp\n",
    "bedtools2/src/utils/general/ParseTools.cpp\n",
    "bedtools2/src/utils/general/CommonHelp.h\n",
    "bedtools2/src/utils/general/FreeList.h\n",
    "bedtools2/src/utils/general/CompressionTools.cpp\n",
    "bedtools2/src/utils/general/Makefile\n",
    "bedtools2/src/utils/general/BedtoolsTypes.h\n",
    "bedtools2/src/utils/general/ErrorMsg.h\n",
    "bedtools2/src/utils/general/PushBackStreamBuf.h\n",
    "bedtools2/src/utils/general/Tokenizer.h\n",
    "bedtools2/src/utils/general/ParseTools.h\n",
    "bedtools2/src/utils/general/QuickString.h\n",
    "bedtools2/src/utils/general/Tokenizer.cpp\n",
    "bedtools2/src/utils/general/ErrorMsg.cpp\n",
    "bedtools2/src/utils/general/CommonHelp.cpp\n",
    "bedtools2/src/utils/general/BTlist.h\n",
    "bedtools2/src/utils/general/CompressionTools.h\n",
    "bedtools2/src/utils/general/PushBackStreamBuf.cpp\n",
    "bedtools2/src/utils/general/InflateStreamBuf.h\n",
    "bedtools2/src/utils/BlockedIntervals/BlockedIntervals.cpp\n",
    "bedtools2/src/utils/BlockedIntervals/BlockedIntervals.h\n",
    "bedtools2/src/utils/BlockedIntervals/Makefile\n",
    "bedtools2/src/utils/BamTools/LICENSE\n",
    "bedtools2/src/utils/BamTools/Makefile\n",
    "bedtools2/src/utils/BamTools/.gitignore\n",
    "bedtools2/src/utils/BamTools/src/\n",
    "bedtools2/src/utils/BamTools/src/shared/\n",
    "bedtools2/src/utils/BamTools/src/api/\n",
    "bedtools2/src/utils/BamTools/src/api/SamProgramChain.h\n",
    "bedtools2/src/utils/BamTools/src/api/SamProgram.h\n",
    "bedtools2/src/utils/BamTools/src/api/BamWriter.h\n",
    "bedtools2/src/utils/BamTools/src/api/BamWriter.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/BamMultiReader.h\n",
    "bedtools2/src/utils/BamTools/src/api/SamProgram.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/BamIndex.h\n",
    "bedtools2/src/utils/BamTools/src/api/BamAlignment.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/BamConstants.h\n",
    "bedtools2/src/utils/BamTools/src/api/SamHeader.h\n",
    "bedtools2/src/utils/BamTools/src/api/api_global.h\n",
    "bedtools2/src/utils/BamTools/src/api/SamSequenceDictionary.h\n",
    "bedtools2/src/utils/BamTools/src/api/SamHeader.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/BamReader.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/SamSequence.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/algorithms/\n",
    "bedtools2/src/utils/BamTools/src/api/internal/\n",
    "bedtools2/src/utils/BamTools/src/api/SamReadGroupDictionary.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/SamReadGroupDictionary.h\n",
    "bedtools2/src/utils/BamTools/src/api/SamConstants.h\n",
    "bedtools2/src/utils/BamTools/src/api/BamAlgorithms.h\n",
    "bedtools2/src/utils/BamTools/src/api/BamAlignment.h\n",
    "bedtools2/src/utils/BamTools/src/api/BamReader.h\n",
    "bedtools2/src/utils/BamTools/src/api/SamReadGroup.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/SamSequence.h\n",
    "bedtools2/src/utils/BamTools/src/api/SamReadGroup.h\n",
    "bedtools2/src/utils/BamTools/src/api/SamSequenceDictionary.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/SamProgramChain.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/BamMultiReader.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/BamAux.h\n",
    "bedtools2/src/utils/BamTools/src/api/IBamIODevice.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/sam/\n",
    "bedtools2/src/utils/BamTools/src/api/internal/bam/\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/\n",
    "bedtools2/src/utils/BamTools/src/api/internal/utils/\n",
    "bedtools2/src/utils/BamTools/src/api/internal/index/\n",
    "bedtools2/src/utils/BamTools/src/api/internal/index/BamStandardIndex_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/index/BamIndexFactory_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/index/BamToolsIndex_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/index/BamToolsIndex_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/index/BamStandardIndex_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/index/BamIndexFactory_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/utils/BamException_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/utils/BamException_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/HostInfo_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/BgzfStream_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/TcpSocket_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/BamFile_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/BamDeviceFactory_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/TcpSocket_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/TcpSocketEngine_win_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/TcpSocketEngine_unix_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/BamFtp_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/StdStreamDevice_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/BamHttp_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/BamHttp_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/BamFile_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/ILocalIODevice_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/ByteArray_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/ILocalIODevice_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/BgzfStream_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/BamDeviceFactory_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/TcpSocketEngine_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/NetWin_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/BamFtp_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/ByteArray_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/TcpSocketEngine_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/BamPipe_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/HostInfo_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/RollingBuffer_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/HttpHeader_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/NetUnix_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/BamPipe_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/HostAddress_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/HostAddress_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/HttpHeader_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/io/RollingBuffer_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/bam/BamWriter_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/bam/BamWriter_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/bam/BamMultiMerger_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/bam/BamRandomAccessController_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/bam/BamReader_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/bam/BamHeader_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/bam/BamHeader_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/bam/BamRandomAccessController_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/bam/BamMultiReader_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/bam/BamMultiReader_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/bam/BamReader_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/sam/SamFormatPrinter_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/sam/SamHeaderValidator_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/sam/SamHeaderValidator_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/sam/SamHeaderVersion_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/internal/sam/SamFormatPrinter_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/sam/SamFormatParser_p.cpp\n",
    "bedtools2/src/utils/BamTools/src/api/internal/sam/SamFormatParser_p.h\n",
    "bedtools2/src/utils/BamTools/src/api/algorithms/Sort.h\n",
    "bedtools2/src/utils/BamTools/src/shared/bamtools_global.h\n",
    "bedtools2/src/utils/Contexts/ContextBase.h\n",
    "bedtools2/src/utils/Contexts/ContextClosest.h\n",
    "bedtools2/src/utils/Contexts/ContextIntersect.h\n",
    "bedtools2/src/utils/Contexts/ContextIntersect.cpp\n",
    "bedtools2/src/utils/Contexts/ContextFisher.cpp\n",
    "bedtools2/src/utils/Contexts/ContextMerge.h\n",
    "bedtools2/src/utils/Contexts/ContextGroupBy.h\n",
    "bedtools2/src/utils/Contexts/ContextComplement.h\n",
    "bedtools2/src/utils/Contexts/Makefile\n",
    "bedtools2/src/utils/Contexts/ContextSubtract.h\n",
    "bedtools2/src/utils/Contexts/ContextCoverage.cpp\n",
    "bedtools2/src/utils/Contexts/ContextBase.cpp\n",
    "bedtools2/src/utils/Contexts/ContextFisher.h\n",
    "bedtools2/src/utils/Contexts/ContextSpacing.cpp\n",
    "bedtools2/src/utils/Contexts/ContextMap.h\n",
    "bedtools2/src/utils/Contexts/ContextSubtract.cpp\n",
    "bedtools2/src/utils/Contexts/ContextJaccard.h\n",
    "bedtools2/src/utils/Contexts/ContextComplement.cpp\n",
    "bedtools2/src/utils/Contexts/ContextCoverage.h\n",
    "bedtools2/src/utils/Contexts/ContextSample.cpp\n",
    "bedtools2/src/utils/Contexts/ContextClosest.cpp\n",
    "bedtools2/src/utils/Contexts/ContextJaccard.cpp\n",
    "bedtools2/src/utils/Contexts/ContextSample.h\n",
    "bedtools2/src/utils/Contexts/ContextGroupBy.cpp\n",
    "bedtools2/src/utils/Contexts/ContextMap.cpp\n",
    "bedtools2/src/utils/Contexts/ContextSpacing.h\n",
    "bedtools2/src/utils/Contexts/ContextMerge.cpp\n",
    "bedtools2/src/utils/bedFile/Makefile\n",
    "bedtools2/src/utils/bedFile/bedFile.h\n",
    "bedtools2/src/utils/bedFile/bedFile.cpp\n",
    "bedtools2/src/utils/RecordOutputMgr/Makefile\n",
    "bedtools2/src/utils/RecordOutputMgr/RecordOutputMgr.h\n",
    "bedtools2/src/utils/RecordOutputMgr/RecordOutputMgr.cpp\n",
    "bedtools2/src/utils/NewChromsweep/Makefile\n",
    "bedtools2/src/utils/NewChromsweep/NewChromsweep.h\n",
    "bedtools2/src/utils/NewChromsweep/CloseSweep.h\n",
    "bedtools2/src/utils/NewChromsweep/NewChromsweep.cpp\n",
    "bedtools2/src/utils/NewChromsweep/CloseSweep.cpp\n",
    "bedtools2/src/utils/tabFile/Makefile\n",
    "bedtools2/src/utils/tabFile/tabFile.cpp\n",
    "bedtools2/src/utils/tabFile/tabFile.h\n",
    "bedtools2/src/genomeCoverageBed/genomeCoverageMain.cpp\n",
    "bedtools2/src/genomeCoverageBed/Makefile\n",
    "bedtools2/src/genomeCoverageBed/genomeCoverageBed.h\n",
    "bedtools2/src/genomeCoverageBed/genomeCoverageBed.cpp\n",
    "bedtools2/src/maskFastaFromBed/maskFastaFromBed.cpp\n",
    "bedtools2/src/maskFastaFromBed/Makefile\n",
    "bedtools2/src/maskFastaFromBed/maskFastaFromBed.h\n",
    "bedtools2/src/maskFastaFromBed/maskFastaFromBedMain.cpp\n",
    "bedtools2/src/regressTest/RegressTest.h\n",
    "bedtools2/src/regressTest/Makefile\n",
    "bedtools2/src/regressTest/RegressTest.cpp\n",
    "bedtools2/src/regressTest/regressTestMain.cpp\n",
    "bedtools2/src/bedToIgv/bedToIgv.cpp\n",
    "bedtools2/src/bedToIgv/Makefile\n",
    "bedtools2/src/nucBed/nucBed.h\n",
    "bedtools2/src/nucBed/nucBedMain.cpp\n",
    "bedtools2/src/nucBed/Makefile\n",
    "bedtools2/src/nucBed/LargeFileSupport.h\n",
    "bedtools2/src/nucBed/nucBed.cpp\n",
    "bedtools2/src/reldist/Makefile\n",
    "bedtools2/src/reldist/reldist.cpp\n",
    "bedtools2/src/reldist/reldistMain.cpp\n",
    "bedtools2/src/reldist/reldist.h\n",
    "bedtools2/src/getOverlap/Makefile\n",
    "bedtools2/src/getOverlap/getOverlap.cpp\n",
    "bedtools2/src/randomBed/randomBedMain.cpp\n",
    "bedtools2/src/randomBed/randomBed.cpp\n",
    "bedtools2/src/randomBed/Makefile\n",
    "bedtools2/src/randomBed/randomBed.h\n",
    "bedtools2/src/annotateBed/annotateBed.cpp\n",
    "bedtools2/src/annotateBed/annotateBed.h\n",
    "bedtools2/src/annotateBed/Makefile\n",
    "bedtools2/src/annotateBed/annotateMain.cpp\n",
    "bedtools2/src/slopBed/Makefile\n",
    "bedtools2/src/slopBed/slopBed.h\n",
    "bedtools2/src/slopBed/slopBed.cpp\n",
    "bedtools2/src/slopBed/slopBedMain.cpp\n",
    "bedtools2/src/mapFile/Makefile\n",
    "bedtools2/src/mapFile/mapHelp.cpp\n",
    "bedtools2/src/mapFile/mapFile.cpp\n",
    "bedtools2/src/mapFile/mapFile.h\n",
    "bedtools2/src/pairToPair/pairToPair.h\n",
    "bedtools2/src/pairToPair/Makefile\n",
    "bedtools2/src/pairToPair/pairToPairMain.cpp\n",
    "bedtools2/src/pairToPair/pairToPair.cpp\n",
    "bedtools2/src/linksBed/Makefile\n",
    "bedtools2/src/linksBed/linksBed.cpp\n",
    "bedtools2/src/linksBed/linksBed.h\n",
    "bedtools2/src/linksBed/linksMain.cpp\n",
    "bedtools2/src/fisher/Makefile\n",
    "bedtools2/src/fisher/fisher.h\n",
    "bedtools2/src/fisher/kfunc.h\n",
    "bedtools2/src/fisher/kfunc.cpp\n",
    "bedtools2/src/fisher/fisher.cpp\n",
    "bedtools2/src/fisher/fisherHelp.cpp\n",
    "bedtools2/src/expand/Makefile\n",
    "bedtools2/src/expand/expand.cpp\n",
    "bedtools2/src/complementFile/complementHelp.cpp\n",
    "bedtools2/src/complementFile/Makefile\n",
    "bedtools2/src/complementFile/complementFile.h\n",
    "bedtools2/src/complementFile/complementFile.cpp\n",
    "bedtools2/src/bedToBam/Makefile\n",
    "bedtools2/src/bedToBam/bedToBam.cpp\n",
    "bedtools2/src/split/Makefile\n",
    "bedtools2/src/split/splitBed.h\n",
    "bedtools2/src/split/splitBed.cpp\n",
    "bedtools2/src/split/splitBedMain.cpp\n",
    "bedtools2/src/sortBed/sortMain.cpp\n",
    "bedtools2/src/sortBed/Makefile\n",
    "bedtools2/src/sortBed/sortBed.cpp\n",
    "bedtools2/src/sortBed/sortBed.h\n",
    "bedtools2/src/spacingFile/spacingHelp.cpp\n",
    "bedtools2/src/spacingFile/spacingFile.h\n",
    "bedtools2/src/spacingFile/Makefile\n",
    "bedtools2/src/spacingFile/spacingFile.cpp\n",
    "bedtools2/data/gerp.chr1.bed.gz\n",
    "bedtools2/data/simpleRepeats.chr1.bed.gz\n",
    "bedtools2/data/aluY.chr1.bed.gz\n",
    "bedtools2/data/refseq.chr1.exons.bed.gz\n",
    "bedtools2/data/knownGene.hg18.chr21.short.bed\n",
    "bedtools2/data/knownGene.hg18.chr21.bed\n",
    "bedtools2/scripts/makeBashScripts.py\n",
    "bedtools2/scripts/vcfsort.sh\n",
    "bedtools2/genomes/human.hg38.genome\n",
    "bedtools2/genomes/mouse.mm10.genome\n",
    "bedtools2/genomes/mouse.mm8.genome\n",
    "bedtools2/genomes/human.hg18.genome\n",
    "bedtools2/genomes/human.hg19.genome\n",
    "bedtools2/genomes/mouse.mm9.genome\n",
    "bedtools2/docs/index.rst\n",
    "bedtools2/docs/Makefile\n",
    "bedtools2/docs/conf.py\n",
    "bedtools2/docs/_static/\n",
    "bedtools2/docs/content/\n",
    "bedtools2/docs/bedtools.swiss.png\n",
    "bedtools2/docs/build.sh\n",
    "bedtools2/docs/make.bat\n",
    "bedtools2/docs/bedtools.ai\n",
    "bedtools2/docs/templates/\n",
    "bedtools2/docs/themes/\n",
    "bedtools2/docs/bedtools.png\n",
    "bedtools2/docs/themes/rtd/\n",
    "bedtools2/docs/themes/rtd/theme.conf\n",
    "bedtools2/docs/themes/rtd/static/\n",
    "bedtools2/docs/themes/rtd/static/rtd.css_t\n",
    "bedtools2/docs/templates/page.html\n",
    "bedtools2/docs/templates/sidebar-intro.html\n",
    "bedtools2/docs/content/general-usage.rst\n",
    "bedtools2/docs/content/tools/\n",
    "bedtools2/docs/content/images/\n",
    "bedtools2/docs/content/quick-start.rst\n",
    "bedtools2/docs/content/example-usage.rst\n",
    "bedtools2/docs/content/overview.rst\n",
    "bedtools2/docs/content/advanced-usage.rst\n",
    "bedtools2/docs/content/history.rst\n",
    "bedtools2/docs/content/installation.rst\n",
    "bedtools2/docs/content/tips-and-tricks.rst\n",
    "bedtools2/docs/content/bedtools-suite.rst\n",
    "bedtools2/docs/content/faq.rst\n",
    "bedtools2/docs/content/related-tools.rst\n",
    "bedtools2/docs/content/images/memory-comparo.png\n",
    "bedtools2/docs/content/images/speed-comparo.png\n",
    "bedtools2/docs/content/images/fisher/\n",
    "bedtools2/docs/content/images/github-zip-button.png\n",
    "bedtools2/docs/content/images/galaxy-bedtools.png\n",
    "bedtools2/docs/content/images/bedtoolsMenu.png\n",
    "bedtools2/docs/content/images/tool-glyphs/\n",
    "bedtools2/docs/content/images/tool-glyphs/map-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/jaccard-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/merge-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/reldist-glyph.ai\n",
    "bedtools2/docs/content/images/tool-glyphs/getfasta-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/map-speed-comparo.png\n",
    "bedtools2/docs/content/images/tool-glyphs/reldist-plot.png\n",
    "bedtools2/docs/content/images/tool-glyphs/subtract-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/flank-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/jaccard-glyph.ai\n",
    "bedtools2/docs/content/images/tool-glyphs/map-glyph.svg\n",
    "bedtools2/docs/content/images/tool-glyphs/random-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/genomecov-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/cluster-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/maskfasta-glyph.svg\n",
    "bedtools2/docs/content/images/tool-glyphs/random-glyph.ai\n",
    "bedtools2/docs/content/images/tool-glyphs/complement-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/slop-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/window-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/shuffle-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/reldist-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/closest-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/maskfasta-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/intersect-glyph.png\n",
    "bedtools2/docs/content/images/tool-glyphs/barski_binding_site.png\n",
    "bedtools2/docs/content/images/fisher/res-genes.png\n",
    "bedtools2/docs/content/tools/expand.rst\n",
    "bedtools2/docs/content/tools/genomecov.rst\n",
    "bedtools2/docs/content/tools/makewindows.rst\n",
    "bedtools2/docs/content/tools/igv.rst\n",
    "bedtools2/docs/content/tools/overlap.rst\n",
    "bedtools2/docs/content/tools/nuc.rst\n",
    "bedtools2/docs/content/tools/fisher.rst\n",
    "bedtools2/docs/content/tools/cluster.rst\n",
    "bedtools2/docs/content/tools/pairtobed.rst\n",
    "bedtools2/docs/content/tools/shift.rst\n",
    "bedtools2/docs/content/tools/complement.rst\n",
    "bedtools2/docs/content/tools/bed12tobed6.rst\n",
    "bedtools2/docs/content/tools/map.rst\n",
    "bedtools2/docs/content/tools/tag.rst\n",
    "bedtools2/docs/content/tools/pairtopair.rst\n",
    "bedtools2/docs/content/tools/slop.rst\n",
    "bedtools2/docs/content/tools/sort.rst\n",
    "bedtools2/docs/content/tools/reldist.rst\n",
    "bedtools2/docs/content/tools/closest.rst\n",
    "bedtools2/docs/content/tools/bedtobam.rst\n",
    "bedtools2/docs/content/tools/maskfasta.rst\n",
    "bedtools2/docs/content/tools/bamtofastq.rst\n",
    "bedtools2/docs/content/tools/multicov.rst\n",
    "bedtools2/docs/content/tools/flank.rst\n",
    "bedtools2/docs/content/tools/random.rst\n",
    "bedtools2/docs/content/tools/merge.rst\n",
    "bedtools2/docs/content/tools/shuffle.rst\n",
    "bedtools2/docs/content/tools/window.rst\n",
    "bedtools2/docs/content/tools/bedpetobam.rst\n",
    "bedtools2/docs/content/tools/groupby.rst\n",
    "bedtools2/docs/content/tools/multiinter.rst\n",
    "bedtools2/docs/content/tools/bamtobed.rst\n",
    "bedtools2/docs/content/tools/getfasta.rst\n",
    "bedtools2/docs/content/tools/unionbedg.rst\n",
    "bedtools2/docs/content/tools/subtract.rst\n",
    "bedtools2/docs/content/tools/coverage.rst\n",
    "bedtools2/docs/content/tools/jaccard.rst\n",
    "bedtools2/docs/content/tools/links.rst\n",
    "bedtools2/docs/content/tools/annotate.rst\n",
    "bedtools2/docs/content/tools/intersect.rst\n",
    "bedtools2/docs/_static/labibi.css\n",
    "bedtools2/docs/_static/labibi.js\n",
    "bedtools2/test/split/\n",
    "bedtools2/test/cluster/\n",
    "bedtools2/test/expand/\n",
    "bedtools2/test/shuffle/\n",
    "bedtools2/test/fisher/\n",
    "bedtools2/test/complement/\n",
    "bedtools2/test/intersect/\n",
    "bedtools2/test/sort/\n",
    "bedtools2/test/subtract/\n",
    "bedtools2/test/merge/\n",
    "bedtools2/test/general/\n",
    "bedtools2/test/slop/\n",
    "bedtools2/test/reldist/\n",
    "bedtools2/test/shift/\n",
    "bedtools2/test/genomecov/\n",
    "bedtools2/test/jaccard/\n",
    "bedtools2/test/groupby/\n",
    "bedtools2/test/bamtofastq/\n",
    "bedtools2/test/bed12tobed6/\n",
    "bedtools2/test/getfasta/\n",
    "bedtools2/test/map/\n",
    "bedtools2/test/spacing/\n",
    "bedtools2/test/sample/\n",
    "bedtools2/test/flank/\n",
    "bedtools2/test/negativecontrol/\n",
    "bedtools2/test/multicov/\n",
    "bedtools2/test/closest/\n",
    "bedtools2/test/test.sh\n",
    "bedtools2/test/coverage/\n",
    "bedtools2/test/bamtobed/\n",
    "bedtools2/test/makewindows/\n",
    "bedtools2/test/makewindows/test-makewindows.sh\n",
    "bedtools2/test/makewindows/a.33bp.bed\n",
    "bedtools2/test/makewindows/a.20bp.bed\n",
    "bedtools2/test/makewindows/input.bed\n",
    "bedtools2/test/makewindows/a.19bp.bed\n",
    "bedtools2/test/bamtobed/two_blocks.sam\n",
    "bedtools2/test/bamtobed/three_blocks.sam\n",
    "bedtools2/test/bamtobed/test-bamtobed.sh\n",
    "bedtools2/test/bamtobed/two_blocks_w_D.sam\n",
    "bedtools2/test/bamtobed/one_block.sam\n",
    "bedtools2/test/coverage/two_blocks.sam\n",
    "bedtools2/test/coverage/chr_0-100.bed\n",
    "bedtools2/test/coverage/three_blocks.bam.bai\n",
    "bedtools2/test/coverage/sam-w-del.sam\n",
    "bedtools2/test/coverage/three_blocks.bam\n",
    "bedtools2/test/coverage/three_blocks_nomatch.bed\n",
    "bedtools2/test/coverage/sam-w-del.bam\n",
    "bedtools2/test/coverage/two_blocks.bam.bai\n",
    "bedtools2/test/coverage/three_blocks.sam\n",
    "bedtools2/test/coverage/one_block.bam.bai\n",
    "bedtools2/test/coverage/test-coverage.sh\n",
    "bedtools2/test/coverage/x.bed\n",
    "bedtools2/test/coverage/two_blocks.bam\n",
    "bedtools2/test/coverage/y.bed\n",
    "bedtools2/test/coverage/a.bed\n",
    "bedtools2/test/coverage/c.bed\n",
    "bedtools2/test/coverage/one_block.bam\n",
    "bedtools2/test/coverage/three_blocks_match.bam\n",
    "bedtools2/test/coverage/sam-w-del.bam.bai\n",
    "bedtools2/test/coverage/b.bed\n",
    "bedtools2/test/coverage/three_blocks_match.bed\n",
    "bedtools2/test/coverage/one_block.sam\n",
    "bedtools2/test/coverage/three_blocks_match_1bp.bed\n",
    "bedtools2/test/closest/d_d1F.bed\n",
    "bedtools2/test/closest/d_q1.bed\n",
    "bedtools2/test/closest/d_iu.bed\n",
    "bedtools2/test/closest/d_d1R.bed\n",
    "bedtools2/test/closest/d_q2.bed\n",
    "bedtools2/test/closest/tfbs.bed\n",
    "bedtools2/test/closest/null_b.bed\n",
    "bedtools2/test/closest/bug281_a.medium.bed\n",
    "bedtools2/test/closest/null_c.bed\n",
    "bedtools2/test/closest/null_a.bed\n",
    "bedtools2/test/closest/d_id.bed\n",
    "bedtools2/test/closest/a2.bed\n",
    "bedtools2/test/closest/test-closest.sh\n",
    "bedtools2/test/closest/kclosest/\n",
    "bedtools2/test/closest/close-a.bed\n",
    "bedtools2/test/closest/b.names.bed\n",
    "bedtools2/test/closest/close-b.bed\n",
    "bedtools2/test/closest/strand-test-d.bed\n",
    "bedtools2/test/closest/b-one-bp-closer.bed\n",
    "bedtools2/test/closest/islands.2.bed\n",
    "bedtools2/test/closest/bug157_b.bed\n",
    "bedtools2/test/closest/shores.bed\n",
    "bedtools2/test/closest/strand-test-a.bed\n",
    "bedtools2/test/closest/strand-test-c.bed\n",
    "bedtools2/test/closest/bug157_a.bed\n",
    "bedtools2/test/closest/strand-test-b.bed\n",
    "bedtools2/test/closest/bug281_b.bed\n",
    "bedtools2/test/closest/dmr.bed\n",
    "bedtools2/test/closest/a.bed\n",
    "bedtools2/test/closest/b.bed\n",
    "bedtools2/test/closest/a.names.bed\n",
    "bedtools2/test/closest/mdb3.bed\n",
    "bedtools2/test/closest/mdb2.bed\n",
    "bedtools2/test/closest/b2.bed\n",
    "bedtools2/test/closest/mdb1.bed\n",
    "bedtools2/test/closest/mq1.bed\n",
    "bedtools2/test/closest/d.bed\n",
    "bedtools2/test/closest/d4.bed\n",
    "bedtools2/test/closest/sortAndNaming/\n",
    "bedtools2/test/closest/islands.bed\n",
    "bedtools2/test/closest/bug281_b.flip.bed\n",
    "bedtools2/test/closest/bug281_a.flip.medium.bed\n",
    "bedtools2/test/closest/d_d2R.bed\n",
    "bedtools2/test/closest/d_d2F.bed\n",
    "bedtools2/test/closest/sortAndNaming/num_all.bed\n",
    "bedtools2/test/closest/sortAndNaming/db1_noChr.bed\n",
    "bedtools2/test/closest/sortAndNaming/alpha_all.bed\n",
    "bedtools2/test/closest/sortAndNaming/test-sort-and-naming.sh\n",
    "bedtools2/test/closest/sortAndNaming/q1_gls.bed\n",
    "bedtools2/test/closest/sortAndNaming/db1_num.bed\n",
    "bedtools2/test/closest/sortAndNaming/db2_numBackwards.bed\n",
    "bedtools2/test/closest/sortAndNaming/num_missing.bed\n",
    "bedtools2/test/closest/sortAndNaming/q1_num.bed\n",
    "bedtools2/test/closest/sortAndNaming/db2_num.bed\n",
    "bedtools2/test/closest/sortAndNaming/db1_leadingZero.txt\n",
    "bedtools2/test/closest/sortAndNaming/sq1.bed\n",
    "bedtools2/test/closest/sortAndNaming/db1_gls.bed\n",
    "bedtools2/test/closest/sortAndNaming/alpha_missing.bed\n",
    "bedtools2/test/closest/sortAndNaming/sdb1.bed\n",
    "bedtools2/test/closest/sortAndNaming/db3_numBackwards.bed\n",
    "bedtools2/test/closest/sortAndNaming/q1a_num.bed\n",
    "bedtools2/test/closest/kclosest/test-kclosest.sh\n",
    "bedtools2/test/closest/kclosest/bug471_b.bed\n",
    "bedtools2/test/closest/kclosest/bug471_a.bed\n",
    "bedtools2/test/closest/kclosest/d2.bed\n",
    "bedtools2/test/closest/kclosest/q2.bed\n",
    "bedtools2/test/closest/kclosest/d3.bed\n",
    "bedtools2/test/closest/kclosest/d1.bed\n",
    "bedtools2/test/closest/kclosest/q1.bed\n",
    "bedtools2/test/multicov/two_blocks.sam\n",
    "bedtools2/test/multicov/two_blocks.bam.bai\n",
    "bedtools2/test/multicov/one_block.bam.bai\n",
    "bedtools2/test/multicov/test-multicov.sh\n",
    "bedtools2/test/multicov/multicov.bed\n",
    "bedtools2/test/multicov/one_block.sam\n",
    "bedtools2/test/negativecontrol/test-negativecontrol.sh\n",
    "bedtools2/test/flank/test-flank.sh\n",
    "bedtools2/test/flank/a.bed\n",
    "bedtools2/test/flank/tiny.genome\n",
    "bedtools2/test/sample/test-sample.sh\n",
    "bedtools2/test/sample/human.hg19.genome\n",
    "bedtools2/test/spacing/a.bed\n",
    "bedtools2/test/spacing/test-spacing.sh\n",
    "bedtools2/test/map/values.bam\n",
    "bedtools2/test/map/.Rhistory\n",
    "bedtools2/test/map/a.vsorted.bed\n",
    "bedtools2/test/map/three_blocks_nomatch.bed\n",
    "bedtools2/test/map/b.vsorted.bed\n",
    "bedtools2/test/map/test.vcf\n",
    "bedtools2/test/map/values.bed\n",
    "bedtools2/test/map/test.gff\n",
    "bedtools2/test/map/values5.bed\n",
    "bedtools2/test/map/ivls.bed\n",
    "bedtools2/test/map/values4.bed\n",
    "bedtools2/test/map/t.bed\n",
    "bedtools2/test/map/ivls2.bed\n",
    "bedtools2/test/map/values3.bed\n",
    "bedtools2/test/map/.RData\n",
    "bedtools2/test/map/values2.bed\n",
    "bedtools2/test/map/d.bed\n",
    "bedtools2/test/map/genome\n",
    "bedtools2/test/map/three_blocks_match.bed\n",
    "bedtools2/test/map/test-map.sh\n",
    "bedtools2/test/map/bug262_a.bed\n",
    "bedtools2/test/map/bug262_b.bed\n",
    "bedtools2/test/map/fullFields.bam\n",
    "bedtools2/test/getfasta/test.iupac.bed\n",
    "bedtools2/test/getfasta/t.fa\n",
    "bedtools2/test/getfasta/test.iupac.fa\n",
    "bedtools2/test/getfasta/blocks.bed\n",
    "bedtools2/test/getfasta/t_fH.fa.fai\n",
    "bedtools2/test/getfasta/t.bed\n",
    "bedtools2/test/getfasta/t.fa.fai\n",
    "bedtools2/test/getfasta/test-getfasta.sh\n",
    "bedtools2/test/getfasta/t_fH.fa\n",
    "bedtools2/test/bed12tobed6/two_blocks.bed\n",
    "bedtools2/test/bed12tobed6/bed6.bed\n",
    "bedtools2/test/bed12tobed6/three_blocks.bed\n",
    "bedtools2/test/bed12tobed6/test-bed12tobed6.sh\n",
    "bedtools2/test/bed12tobed6/one_blocks.bed\n",
    "bedtools2/test/bamtofastq/test-bamtofastq.sh\n",
    "bedtools2/test/bamtofastq/golden.fq\n",
    "bedtools2/test/bamtofastq/golden.fq2\n",
    "bedtools2/test/bamtofastq/test.sam\n",
    "bedtools2/test/groupby/gdc.bam\n",
    "bedtools2/test/groupby/values3_case.header.bed\n",
    "bedtools2/test/groupby/test-groupby.sh\n",
    "bedtools2/test/groupby/values3.unmarked_header.bed\n",
    "bedtools2/test/groupby/a_vcfSVtest.vcf\n",
    "bedtools2/test/groupby/values3.7fields.header.bed\n",
    "bedtools2/test/groupby/values3.unmarked_header.bed.2\n",
    "bedtools2/test/groupby/values3.no_header.bed\n",
    "bedtools2/test/groupby/values3.header.bed\n",
    "bedtools2/test/groupby/noPosvalues.header.bed\n",
    "bedtools2/test/groupby/test.bed\n",
    "bedtools2/test/groupby/bug569_problem.txt\n",
    "bedtools2/test/jaccard/bMixedStrands.bed\n",
    "bedtools2/test/jaccard/a.bed\n",
    "bedtools2/test/jaccard/c.bed\n",
    "bedtools2/test/jaccard/three_blocks_match.bam\n",
    "bedtools2/test/jaccard/b.bed\n",
    "bedtools2/test/jaccard/e.bed\n",
    "bedtools2/test/jaccard/test-jaccard.sh\n",
    "bedtools2/test/jaccard/three_blocks_match.bed\n",
    "bedtools2/test/jaccard/aMixedStrands.bed\n",
    "bedtools2/test/jaccard/a.bam\n",
    "bedtools2/test/genomecov/two_blocks.sam\n",
    "bedtools2/test/genomecov/test-genomecov.sh\n",
    "bedtools2/test/genomecov/chip.sam\n",
    "bedtools2/test/genomecov/y.bam\n",
    "bedtools2/test/genomecov/sam-w-del.sam\n",
    "bedtools2/test/genomecov/three_blocks.bam\n",
    "bedtools2/test/genomecov/three_blocks_nomatch.bed\n",
    "bedtools2/test/genomecov/sam-w-del.bam\n",
    "bedtools2/test/genomecov/three_blocks.sam\n",
    "bedtools2/test/genomecov/two_blocks.bam\n",
    "bedtools2/test/genomecov/y.bed\n",
    "bedtools2/test/genomecov/chip.bam\n",
    "bedtools2/test/genomecov/pair-chip.bam\n",
    "bedtools2/test/genomecov/one_block.bam\n",
    "bedtools2/test/genomecov/genome.txt\n",
    "bedtools2/test/genomecov/three_blocks_match.bed\n",
    "bedtools2/test/genomecov/pair-chip.sam\n",
    "bedtools2/test/genomecov/one_block.sam\n",
    "bedtools2/test/genomecov/three_blocks_match_1bp.bed\n",
    "bedtools2/test/shift/test-shift.sh\n",
    "bedtools2/test/shift/genome.len\n",
    "bedtools2/test/shift/a.bed\n",
    "bedtools2/test/shift/b.bed\n",
    "bedtools2/test/shift/tiny.genome\n",
    "bedtools2/test/shift/huge.genome\n",
    "bedtools2/test/reldist/test-reldist.sh\n",
    "bedtools2/test/slop/a.bed\n",
    "bedtools2/test/slop/tiny.genome\n",
    "bedtools2/test/slop/test-slop.sh\n",
    "bedtools2/test/general/hg19.fa.fai\n",
    "bedtools2/test/general/empty.bed\n",
    "bedtools2/test/general/a.chromheader.bed\n",
    "bedtools2/test/general/test-general.sh\n",
    "bedtools2/test/general/non-empty.bed.gz\n",
    "bedtools2/test/general/o\n",
    "bedtools2/test/general/t.bed\n",
    "bedtools2/test/general/a.trackheader.bed\n",
    "bedtools2/test/merge/bamCol8Mean.txt\n",
    "bedtools2/test/merge/mergeBugTest.bed\n",
    "bedtools2/test/merge/expFormat.bed\n",
    "bedtools2/test/merge/bamCol9Mean.txt\n",
    "bedtools2/test/merge/unsorted.bed\n",
    "bedtools2/test/merge/bug254_d.bed\n",
    "bedtools2/test/merge/bug254_e.bed\n",
    "bedtools2/test/merge/bamCol6Collapse.txt\n",
    "bedtools2/test/merge/bamCol3Collapse.txt\n",
    "bedtools2/test/merge/vcfSVtest.vcf\n",
    "bedtools2/test/merge/bug254_c.bed\n",
    "bedtools2/test/merge/bamCol11Collapse.txt\n",
    "bedtools2/test/merge/a.full.bed\n",
    "bedtools2/test/merge/a.bed\n",
    "bedtools2/test/merge/test-merge.sh\n",
    "bedtools2/test/merge/precisionTest.bed\n",
    "bedtools2/test/merge/b.bed\n",
    "bedtools2/test/merge/a.names.bed\n",
    "bedtools2/test/merge/testA.vcf\n",
    "bedtools2/test/merge/bamCol1Collapse.txt\n",
    "bedtools2/test/merge/precisionTest2.bed\n",
    "bedtools2/test/merge/bamCol10Collapse.txt\n",
    "bedtools2/test/merge/bamCol5Mean.txt\n",
    "bedtools2/test/merge/a.gff\n",
    "bedtools2/test/merge/bamCol7Collapse.txt\n",
    "bedtools2/test/merge/bamCol4Mean.txt\n",
    "bedtools2/test/merge/a.full.bam\n",
    "bedtools2/test/merge/mixedStrands.bed\n",
    "bedtools2/test/merge/fullFields.bam\n",
    "bedtools2/test/subtract/test-subtract.sh\n",
    "bedtools2/test/subtract/a.bed\n",
    "bedtools2/test/subtract/c.bed\n",
    "bedtools2/test/subtract/b.bed\n",
    "bedtools2/test/subtract/b2.bed\n",
    "bedtools2/test/subtract/d.bed\n",
    "bedtools2/test/sort/names.txt\n",
    "bedtools2/test/sort/a.bed\n",
    "bedtools2/test/sort/test-sort.sh\n",
    "bedtools2/test/sort/b.bed\n",
    "bedtools2/test/intersect/two_blocks.sam\n",
    "bedtools2/test/intersect/a_with_bothUnmapped.bam\n",
    "bedtools2/test/intersect/new_test-intersect.sh~\n",
    "bedtools2/test/intersect/gdc.gff\n",
    "bedtools2/test/intersect/bug223_sv1_b.vcf\n",
    "bedtools2/test/intersect/bed3.bed\n",
    "bedtools2/test/intersect/bug150_a.bed\n",
    "bedtools2/test/intersect/x.bam\n",
    "bedtools2/test/intersect/performanceTest.sh\n",
    "bedtools2/test/intersect/bed6.strand.bed\n",
    "bedtools2/test/intersect/queryUnmappedMateMappedCoordsInvalid.bam\n",
    "bedtools2/test/intersect/bug223_sv1_a.vcf\n",
    "bedtools2/test/intersect/bug150_b.bed\n",
    "bedtools2/test/intersect/headerOnly.vcf\n",
    "bedtools2/test/intersect/gdc_one.bed\n",
    "bedtools2/test/intersect/bed6.strand2.bed\n",
    "bedtools2/test/intersect/null_b.bed\n",
    "bedtools2/test/intersect/a_withLargeHeader.bed\n",
    "bedtools2/test/intersect/human.hg19.vSort.genome\n",
    "bedtools2/test/intersect/gdc.bam\n",
    "bedtools2/test/intersect/bed4.bed\n",
    "bedtools2/test/intersect/chromOrderB.bed\n",
    "bedtools2/test/intersect/bedplus.bed\n",
    "bedtools2/test/intersect/bed5.bed\n",
    "bedtools2/test/intersect/oneRecordNoNewline.bed\n",
    "bedtools2/test/intersect/aWithHeaderVsB.txt\n",
    "bedtools2/test/intersect/null_c.bed\n",
    "bedtools2/test/intersect/j1.bed\n",
    "bedtools2/test/intersect/b.issue311.vcf\n",
    "bedtools2/test/intersect/null_a.bed\n",
    "bedtools2/test/intersect/oneUnmapped.bam\n",
    "bedtools2/test/intersect/chromOrderA.bed\n",
    "bedtools2/test/intersect/bed6.bed\n",
    "bedtools2/test/intersect/new_test-intersect.sh\n",
    "bedtools2/test/intersect/chromsOutOfOrder.bed\n",
    "bedtools2/test/intersect/jim.bed\n",
    "bedtools2/test/intersect/three_blocks_nomatch.bed\n",
    "bedtools2/test/intersect/b_vcfSVtest.vcf\n",
    "bedtools2/test/intersect/bug223_f.vcf\n",
    "bedtools2/test/intersect/a.issue311.bed\n",
    "bedtools2/test/intersect/three_blocks.sam\n",
    "bedtools2/test/intersect/gdc_exp\n",
    "bedtools2/test/intersect/bug44_b.bed\n",
    "bedtools2/test/intersect/bug223_d.vcf\n",
    "bedtools2/test/intersect/bug223_e.vcf\n",
    "bedtools2/test/intersect/a_testZeroLen.bed\n",
    "bedtools2/test/intersect/a_bgzipped.bed.gz\n",
    "bedtools2/test/intersect/x.bed\n",
    "bedtools2/test/intersect/a_withLargeHeader_bgzipped.bed.gz\n",
    "bedtools2/test/intersect/dosLineChar_a.bed\n",
    "bedtools2/test/intersect/bug44_a.vcf.gz\n",
    "bedtools2/test/intersect/y.bed\n",
    "bedtools2/test/intersect/test-intersect.sh\n",
    "bedtools2/test/intersect/a_vcfSVtest.vcf\n",
    "bedtools2/test/intersect/blocks.bed12\n",
    "bedtools2/test/intersect/recordsOutOfOrder.bed\n",
    "bedtools2/test/intersect/a.bed\n",
    "bedtools2/test/intersect/jim.vcf\n",
    "bedtools2/test/intersect/c.bed\n",
    "bedtools2/test/intersect/three_blocks_match.bam\n",
    "bedtools2/test/intersect/dosLineCharWithExtraTab_b.bed\n",
    "bedtools2/test/intersect/b.bed\n",
    "bedtools2/test/intersect/two_blocks_partial.bam.bai\n",
    "bedtools2/test/intersect/foo\n",
    "bedtools2/test/intersect/mapped_and_unmapped.sam\n",
    "bedtools2/test/intersect/two_blocks_partial.bam\n",
    "bedtools2/test/intersect/e.bed\n",
    "bedtools2/test/intersect/a_withLargeHeader_gzipped.bed.gz\n",
    "bedtools2/test/intersect/svlen.vcf\n",
    "bedtools2/test/intersect/d.bed\n",
    "bedtools2/test/intersect/splitBug155_a.bed\n",
    "bedtools2/test/intersect/multi_intersect/\n",
    "bedtools2/test/intersect/nonamecheck_a.bed\n",
    "bedtools2/test/intersect/sortAndNaming/\n",
    "bedtools2/test/intersect/a_gzipped.bed.gz\n",
    "bedtools2/test/intersect/splitBug155_b.bed\n",
    "bedtools2/test/intersect/bug167_strandSweep.bed\n",
    "bedtools2/test/intersect/nonamecheck_b.bed\n",
    "bedtools2/test/intersect/b.issue311.gff\n",
    "bedtools2/test/intersect/two_blocks_partial.bed\n",
    "bedtools2/test/intersect/three_blocks_match.bed\n",
    "bedtools2/test/intersect/human.hg19.genome\n",
    "bedtools2/test/intersect/bug187_a.bed\n",
    "bedtools2/test/intersect/b_testZeroLen.bed\n",
    "bedtools2/test/intersect/one_block.sam\n",
    "bedtools2/test/intersect/bug187_b.bed\n",
    "bedtools2/test/intersect/three_blocks_match_1bp.bed\n",
    "bedtools2/test/intersect/aVSb.bam\n",
    "bedtools2/test/intersect/a.bam\n",
    "bedtools2/test/intersect/sortAndNaming/num_all.bed\n",
    "bedtools2/test/intersect/sortAndNaming/diffs.txt\n",
    "bedtools2/test/intersect/sortAndNaming/db1_noChr.bed\n",
    "bedtools2/test/intersect/sortAndNaming/alpha_all.bed\n",
    "bedtools2/test/intersect/sortAndNaming/test-sort-and-naming.sh\n",
    "bedtools2/test/intersect/sortAndNaming/exp1\n",
    "bedtools2/test/intersect/sortAndNaming/q1_gls.bed\n",
    "bedtools2/test/intersect/sortAndNaming/db1_num.bed\n",
    "bedtools2/test/intersect/sortAndNaming/db2_numBackwards.bed\n",
    "bedtools2/test/intersect/sortAndNaming/num_missing.bed\n",
    "bedtools2/test/intersect/sortAndNaming/q1_num.bed\n",
    "bedtools2/test/intersect/sortAndNaming/db2_num.bed\n",
    "bedtools2/test/intersect/sortAndNaming/db1_leadingZero.txt\n",
    "bedtools2/test/intersect/sortAndNaming/sq1.bed\n",
    "bedtools2/test/intersect/sortAndNaming/db1_gls.bed\n",
    "bedtools2/test/intersect/sortAndNaming/alpha_missing.bed\n",
    "bedtools2/test/intersect/sortAndNaming/bigTests/\n",
    "bedtools2/test/intersect/sortAndNaming/sdb1.bed\n",
    "bedtools2/test/intersect/sortAndNaming/db3_numBackwards.bed\n",
    "bedtools2/test/intersect/sortAndNaming/q1a_num.bed\n",
    "bedtools2/test/intersect/sortAndNaming/bigTests/curr5v5.txt\n",
    "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_500Kvs500K.txt\n",
    "bedtools2/test/intersect/sortAndNaming/bigTests/q500K.bed\n",
    "bedtools2/test/intersect/sortAndNaming/bigTests/dbchr1.bed\n",
    "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_chr1.txt\n",
    "bedtools2/test/intersect/sortAndNaming/bigTests/diffs.txt\n",
    "bedtools2/test/intersect/sortAndNaming/bigTests/curr500Kvs500K.txt\n",
    "bedtools2/test/intersect/sortAndNaming/bigTests/qchr1.bed\n",
    "bedtools2/test/intersect/sortAndNaming/bigTests/2.21_5v5.txt\n",
    "bedtools2/test/intersect/sortAndNaming/bigTests/currChr1.txt\n",
    "bedtools2/test/intersect/sortAndNaming/bigTests/db500K.bed\n",
    "bedtools2/test/intersect/multi_intersect/query3.bed\n",
    "bedtools2/test/intersect/multi_intersect/query2.bed\n",
    "bedtools2/test/intersect/multi_intersect/d8.bed\n",
    "bedtools2/test/intersect/multi_intersect/d9.bed\n",
    "bedtools2/test/intersect/multi_intersect/query.bed\n",
    "bedtools2/test/intersect/multi_intersect/g.bed\n",
    "bedtools2/test/intersect/multi_intersect/test-multi_intersect.sh\n",
    "bedtools2/test/intersect/multi_intersect/d4.bed\n",
    "bedtools2/test/intersect/multi_intersect/d5.bed\n",
    "bedtools2/test/intersect/multi_intersect/d7.bed\n",
    "bedtools2/test/intersect/multi_intersect/d6.bed\n",
    "bedtools2/test/intersect/multi_intersect/d2.bed\n",
    "bedtools2/test/intersect/multi_intersect/d3.bed\n",
    "bedtools2/test/intersect/multi_intersect/d1.bed\n",
    "bedtools2/test/complement/test-complement.sh\n",
    "bedtools2/test/fisher/test-fisher.sh\n",
    "bedtools2/test/fisher/sim.py\n",
    "bedtools2/test/fisher/plot.py\n",
    "bedtools2/test/fisher/cmp.sh\n",
    "bedtools2/test/fisher/t.60.genome\n",
    "bedtools2/test/fisher/get.sh\n",
    "bedtools2/test/fisher/README.md\n",
    "bedtools2/test/fisher/a.bed\n",
    "bedtools2/test/fisher/b.bed\n",
    "bedtools2/test/fisher/t.500.genome\n",
    "bedtools2/test/fisher/a_merge.bed\n",
    "bedtools2/test/fisher/shuf.sh\n",
    "bedtools2/test/shuffle/incl.sizedif.bed\n",
    "bedtools2/test/shuffle/test-shuffle.sh\n",
    "bedtools2/test/shuffle/excl.bed\n",
    "bedtools2/test/shuffle/simrep.bed\n",
    "bedtools2/test/shuffle/incl.bed\n",
    "bedtools2/test/expand/expand.txt\n",
    "bedtools2/test/expand/test-expand.sh\n",
    "bedtools2/test/cluster/in.bed\n",
    "bedtools2/test/cluster/test-cluster.sh\n",
    "bedtools2/test/split/test-split.sh\n",
    "bedtools2/test/split/randData.bed\n",
    "bedtools2/test/split/exp\n",
    "bedtools2/tutorial/answers.md\n",
    "bedtools2/tutorial/template.class.html\n",
    "bedtools2/tutorial/answers.html\n",
    "bedtools2/tutorial/bootstrap.css\n",
    "bedtools2/tutorial/Makefile\n",
    "bedtools2/tutorial/bedtools.html\n",
    "bedtools2/tutorial/bedtools.md\n",
    "bedtools2/tutorial/template.html\n",
    "[nathalia@aws] $  cd bedtools2/\n",
    "[nathalia@aws] $  make\n",
    "$CXXFLAGS is [-Wall -O2 -D_FILE_OFFSET_BITS=64 -fPIC -Isrc/utils/bedFile -Isrc/utils/BinTree -Isrc/utils/version -Isrc/utils/bedGraphFile -Isrc/utils/chromsweep -Isrc/utils/Contexts -Isrc/utils/FileRecordTools -Isrc/utils/FileRecordTools/FileReaders -Isrc/utils/FileRecordTools/Records -Isrc/utils/general -Isrc/utils/gzstream -Isrc/utils/fileType -Isrc/utils/gzstream/ -Isrc/utils/lineFileUtilities -Isrc/utils/KeyListOps -Isrc/utils/NewChromsweep -Isrc/utils/sequenceUtilities -Isrc/utils/tabFile -Isrc/utils/BamTools -Isrc/utils/BamTools/include -Isrc/utils/BamTools/src -Isrc/utils/BamTools-Ancillary -Isrc/utils/BlockedIntervals -Isrc/utils/Fasta -Isrc/utils/VectorOps -Isrc/utils/GenomeFile -Isrc/utils/RecordOutputMgr -Isrc/utils/ToolBase -Isrc/utils/driver  -std=c++11]\n",
    "Building BEDTools:\n",
    "=========================================================\n",
    "DETECTED_VERSION = v2.27.1\n",
    "CURRENT_VERSION  = \n",
    "Updating version file.\n",
    " * Creating BamTools API\n",
    "- Building in src/utils/FileRecordTools\n",
    "  * compiling FileRecordMgr.cpp\n",
    "  * compiling FileRecordMergeMgr.cpp\n",
    "- Building in FileReaders\n",
    "  * compiling FileReader.cpp\n",
    "  * compiling SingleLineDelimTextFileReader.cpp\n",
    "  * compiling BamFileReader.cpp\n",
    "  * compiling BufferedStreamMgr.cpp\n",
    "  * compiling InputStreamMgr.cpp\n",
    "- Building in Records\n",
    "  * compiling Record.cpp\n",
    "  * compiling EmptyRecord.cpp\n",
    "  * compiling Bed3Interval.cpp\n",
    "  * compiling Bed4Interval.cpp\n",
    "  * compiling BedGraphInterval.cpp\n",
    "  * compiling Bed5Interval.cpp\n",
    "  * compiling Bed6Interval.cpp\n",
    "  * compiling PlusFields.cpp\n",
    "  * compiling BedPlusInterval.cpp\n",
    "  * compiling Bed12Interval.cpp\n",
    "  * compiling BamRecord.cpp\n",
    "  * compiling GffRecord.cpp\n",
    "  * compiling GffPlusRecord.cpp\n",
    "  * compiling VcfRecord.cpp\n",
    "  * compiling NoPosPlusRecord.cpp\n",
    "  * compiling BlockMgr.cpp\n",
    "  * compiling StrandQueue.cpp\n",
    "  * compiling RecordMgr.cpp\n",
    "  * compiling RecordList.cpp\n",
    "  * compiling RecordKeyList.cpp\n",
    "  * compiling RecordKeyVector.cpp\n",
    "- Building in src/utils/FileRecordTools/FileReaders\n",
    "make[1]: Nothing to be done for 'all'.\n",
    "- Building in src/utils/FileRecordTools/Records\n",
    "make[1]: Nothing to be done for 'all'.\n",
    "- Building in src/utils/bedFile\n",
    "  * compiling bedFile.cpp\n",
    "- Building in src/utils/BinTree\n",
    "  * compiling BinTree.cpp\n",
    "- Building in src/utils/version\n",
    "  * compiling version.cpp\n",
    "- Building in src/utils/bedGraphFile\n",
    "  * compiling bedGraphFile.cpp\n",
    "- Building in src/utils/chromsweep\n",
    "  * compiling chromsweep.cpp\n",
    "- Building in src/utils/Contexts\n",
    "  * compiling ContextBase.cpp\n",
    "  * compiling ContextIntersect.cpp\n",
    "  * compiling ContextFisher.cpp\n",
    "  * compiling ContextMap.cpp\n",
    "  * compiling ContextSample.cpp\n",
    "  * compiling ContextSpacing.cpp\n",
    "  * compiling ContextMerge.cpp\n",
    "  * compiling ContextJaccard.cpp\n",
    "  * compiling ContextClosest.cpp\n",
    "  * compiling ContextSubtract.cpp\n",
    "  * compiling ContextCoverage.cpp\n",
    "  * compiling ContextComplement.cpp\n",
    "  * compiling ContextGroupBy.cpp\n",
    "- Building in src/utils/general\n",
    "  * compiling ParseTools.cpp\n",
    "  * compiling PushBackStreamBuf.cpp\n",
    "  * compiling CompressionTools.cpp\n",
    "  * compiling Tokenizer.cpp\n",
    "  * compiling CommonHelp.cpp\n",
    "- Building in src/utils/gzstream\n",
    "g++ -Wall -O2 -D_FILE_OFFSET_BITS=64 -fPIC -Isrc/utils/bedFile -Isrc/utils/BinTree -Isrc/utils/version -Isrc/utils/bedGraphFile -Isrc/utils/chromsweep -Isrc/utils/Contexts -Isrc/utils/FileRecordTools -Isrc/utils/FileRecordTools/FileReaders -Isrc/utils/FileRecordTools/Records -Isrc/utils/general -Isrc/utils/gzstream -Isrc/utils/fileType -Isrc/utils/gzstream/ -Isrc/utils/lineFileUtilities -Isrc/utils/KeyListOps -Isrc/utils/NewChromsweep -Isrc/utils/sequenceUtilities -Isrc/utils/tabFile -Isrc/utils/BamTools -Isrc/utils/BamTools/include -Isrc/utils/BamTools/src -Isrc/utils/BamTools-Ancillary -Isrc/utils/BlockedIntervals -Isrc/utils/Fasta -Isrc/utils/VectorOps -Isrc/utils/GenomeFile -Isrc/utils/RecordOutputMgr -Isrc/utils/ToolBase -Isrc/utils/driver  -std=c++11  -c -o ../../../obj//gzstream.o gzstream.C -I.\n",
    "- Building in src/utils/fileType\n",
    "  * compiling fileType.cpp\n",
    "  * compiling FileRecordTypeChecker.cpp\n",
    "FileRecordTypeChecker.cpp: In member function ‘bool FileRecordTypeChecker::isTextDelimtedFormat(const char*, size_t)’:\n",
    "FileRecordTypeChecker.cpp:365:19: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]\n",
    "   for (int j=0; j < len; j++) {\n",
    "                 ~~^~~~~\n",
    "- Building in src/utils/bedFilePE\n",
    "  * compiling bedFilePE.cpp\n",
    "- Building in src/utils/KeyListOps\n",
    "  * compiling KeyListOps.cpp\n",
    "  * compiling KeyListOpsMethods.cpp\n",
    "- Building in src/utils/NewChromsweep\n",
    "  * compiling NewChromsweep.cpp\n",
    "  * compiling CloseSweep.cpp\n",
    "- Building in src/utils/sequenceUtilities\n",
    "  * compiling sequenceUtils.cpp\n",
    "- Building in src/utils/tabFile\n",
    "  * compiling tabFile.cpp\n",
    "- Building in src/utils/BamTools\n",
    " * compiling BamAlignment.cpp\n",
    " * compiling BamMultiReader.cpp\n",
    " * compiling BamReader.cpp\n",
    " * compiling BamWriter.cpp\n",
    " * compiling SamHeader.cpp\n",
    " * compiling SamProgram.cpp\n",
    " * compiling SamProgramChain.cpp\n",
    " * compiling SamReadGroup.cpp\n",
    " * compiling SamReadGroupDictionary.cpp\n",
    " * compiling SamSequence.cpp\n",
    " * compiling SamSequenceDictionary.cpp\n",
    " * compiling BamHeader_p.cpp\n",
    " * compiling BamMultiReader_p.cpp\n",
    " * compiling BamRandomAccessController_p.cpp\n",
    " * compiling BamReader_p.cpp\n",
    " * compiling BamWriter_p.cpp\n",
    " * compiling BamIndexFactory_p.cpp\n",
    " * compiling BamStandardIndex_p.cpp\n",
    "src/api/internal/index/BamStandardIndex_p.cpp: In member function ‘void BamTools::Internal::BamStandardIndex::WriteLinearOffsets(const int&, BamTools::Internal::BaiLinearOffsetVector&)’:\n",
    "src/api/internal/index/BamStandardIndex_p.cpp:958:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]\n",
    "     if ( numBytesWritten != (sizeof(offsetCount) + linearOffsets.size()*sizeof(uint64_t)) )\n",
    "          ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
    " * compiling BamToolsIndex_p.cpp\n",
    " * compiling BamDeviceFactory_p.cpp\n",
    " * compiling BamFile_p.cpp\n",
    " * compiling BamFtp_p.cpp\n",
    " * compiling BamHttp_p.cpp\n",
    "src/api/internal/io/BamHttp_p.cpp: In member function ‘bool BamTools::Internal::BamHttp::SendRequest(size_t)’:\n",
    "src/api/internal/io/BamHttp_p.cpp:396:63: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]\n",
    "     return ( WriteToSocket(requestHeader.c_str(), headerSize) == headerSize );\n",
    "              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~\n",
    " * compiling BamPipe_p.cpp\n",
    " * compiling BgzfStream_p.cpp\n",
    " * compiling ByteArray_p.cpp\n",
    " * compiling HostAddress_p.cpp\n",
    "src/api/internal/io/HostAddress_p.cpp: In member function ‘std::__cxx11::string BamTools::Internal::HostAddress::GetIPString() const’:\n",
    "src/api/internal/io/HostAddress_p.cpp:325:13: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]\n",
    "             if ( i != 0 )\n",
    "             ^~\n",
    "src/api/internal/io/HostAddress_p.cpp:327:17: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’\n",
    "                 ss << hex << ( (uint16_t(m_ip6Address[2*i]) << 8) |\n",
    "                 ^~\n",
    " * compiling HostInfo_p.cpp\n",
    " * compiling HttpHeader_p.cpp\n",
    " * compiling ILocalIODevice_p.cpp\n",
    " * compiling RollingBuffer_p.cpp\n",
    " * compiling TcpSocketEngine_p.cpp\n",
    " * compiling TcpSocketEngine_unix_p.cpp\n",
    " * compiling TcpSocket_p.cpp\n",
    " * compiling SamFormatParser_p.cpp\n",
    " * compiling SamFormatPrinter_p.cpp\n",
    " * compiling SamHeaderValidator_p.cpp\n",
    " * compiling BamException_p.cpp\n",
    " * linking lib/libbamtools.a\n",
    "ar cr lib/libbamtools.a src/api/BamAlignment.o src/api/BamMultiReader.o src/api/BamReader.o src/api/BamWriter.o src/api/SamHeader.o src/api/SamProgram.o src/api/SamProgramChain.o src/api/SamReadGroup.o src/api/SamReadGroupDictionary.o src/api/SamSequence.o src/api/SamSequenceDictionary.o src/api/internal/bam/BamHeader_p.o src/api/internal/bam/BamMultiReader_p.o src/api/internal/bam/BamRandomAccessController_p.o src/api/internal/bam/BamReader_p.o src/api/internal/bam/BamWriter_p.o src/api/internal/index/BamIndexFactory_p.o src/api/internal/index/BamStandardIndex_p.o src/api/internal/index/BamToolsIndex_p.o src/api/internal/io/BamDeviceFactory_p.o src/api/internal/io/BamFile_p.o src/api/internal/io/BamFtp_p.o src/api/internal/io/BamHttp_p.o src/api/internal/io/BamPipe_p.o src/api/internal/io/BgzfStream_p.o src/api/internal/io/ByteArray_p.o src/api/internal/io/HostAddress_p.o src/api/internal/io/HostInfo_p.o src/api/internal/io/HttpHeader_p.o src/api/internal/io/ILocalIODevice_p.o src/api/internal/io/RollingBuffer_p.o src/api/internal/io/TcpSocketEngine_p.o src/api/internal/io/TcpSocketEngine_unix_p.o src/api/internal/io/TcpSocket_p.o src/api/internal/sam/SamFormatParser_p.o src/api/internal/sam/SamFormatPrinter_p.o src/api/internal/sam/SamHeaderValidator_p.o src/api/internal/utils/BamException_p.o\n",
    "ranlib lib/libbamtools.a\n",
    "- Building in src/utils/BamTools-Ancillary\n",
    "  * compiling BamAncillary.cpp\n",
    "  * compiling BamAncillary.cpp\n",
    "- Building in src/utils/BlockedIntervals\n",
    "  * compiling BlockedIntervals.cpp\n",
    "- Building in src/utils/Fasta\n",
    "  * compiling Fasta.cpp\n",
    "Fasta.cpp: In member function ‘std::__cxx11::string FastaReference::getSequence(std::__cxx11::string’:\n",
    "Fasta.cpp:290:14: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’, declared with attribute warn_unused_result [-Wunused-result]\n",
    "         fread(seq, sizeof(char), seqlen, file);\n",
    "         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
    "Fasta.cpp: In member function ‘std::__cxx11::string FastaReference::getSubSequence(std::__cxx11::string, int, int)’:\n",
    "Fasta.cpp:332:14: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’, declared with attribute warn_unused_result [-Wunused-result]\n",
    "         fread(seq, sizeof(char), (off_t) seqlen, file);\n",
    "         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
    "  * compiling split.cpp\n",
    "- Building in src/utils/VectorOps\n",
    "  * compiling VectorOps.cpp\n",
    "- Building in src/utils/GenomeFile\n",
    "  * compiling GenomeFile.cpp\n",
    "GenomeFile.cpp: In member function ‘void GenomeFile::loadGenomeFileIntoMap()’:\n",
    "GenomeFile.cpp:60:26: warning: variable ‘c2’ set but not used [-Wunused-but-set-variable]\n",
    "                     long c2;\n",
    "                          ^~\n",
    "  * compiling NewGenomeFile.cpp\n",
    "- Building in src/utils/RecordOutputMgr\n",
    "  * compiling RecordOutputMgr.cpp\n",
    "- Building in src/utils/ToolBase\n",
    "  * compiling ToolBase.cpp\n",
    "- Building in src/utils/driver\n",
    "  * compiling BedtoolsDriver.cpp\n",
    "- Building in src/annotateBed\n",
    "  * compiling annotateMain.cpp\n",
    "annotateMain.cpp: In function ‘int annotate_main(int, char**)’:\n",
    "annotateMain.cpp:39:10: warning: variable ‘haveTitles’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveTitles     = false;\n",
    "          ^~~~~~~~~~\n",
    "  * compiling annotateBed.cpp\n",
    "- Building in src/bamToBed\n",
    "  * compiling bamToBed.cpp\n",
    "bamToBed.cpp: In function ‘int bamtobed_main(int, char**)’:\n",
    "bamToBed.cpp:88:10: warning: variable ‘useAlignmentScore’ set but not used [-Wunused-but-set-variabl]\n",
    "     bool useAlignmentScore = false;\n",
    "          ^~~~~~~~~~~~~~~~~\n",
    "- Building in src/bamToFastq\n",
    "  * compiling bamToFastqMain.cpp\n",
    "bamToFastqMain.cpp: In function ‘int bamtofastq_main(int, char**)’:\n",
    "bamToFastqMain.cpp:38:10: warning: variable ‘haveFastq2’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveFastq2    = false;\n",
    "          ^~~~~~~~~~\n",
    "  * compiling bamToFastq.cpp\n",
    "- Building in src/bedToBam\n",
    "  * compiling bedToBam.cpp\n",
    "bedToBam.cpp: In function ‘int bedtobam_main(int, char**)’:\n",
    "bedToBam.cpp:60:10: warning: variable ‘haveMapQual’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveMapQual     = false;\n",
    "          ^~~~~~~~~~~\n",
    "- Building in src/bedpeToBam\n",
    "  * compiling bedpeToBam.cpp\n",
    "bedpeToBam.cpp: In function ‘int bedpetobam_main(int, char**)’:\n",
    "bedpeToBam.cpp:61:10: warning: variable ‘haveMapQual’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveMapQual     = false;\n",
    "          ^~~~~~~~~~~\n",
    "- Building in src/bedToIgv\n",
    "  * compiling bedToIgv.cpp\n",
    "- Building in src/bed12ToBed6\n",
    "  * compiling bed12ToBed6.cpp\n",
    "- Building in src/closestFile\n",
    "  * compiling closestHelp.cpp\n",
    "  * compiling closestFile.cpp\n",
    "- Building in src/clusterBed\n",
    "  * compiling clusterMain.cpp\n",
    "clusterMain.cpp: In function ‘int cluster_main(int, char**)’:\n",
    "clusterMain.cpp:38:10: warning: variable ‘haveMaxDistance’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveMaxDistance = false;\n",
    "          ^~~~~~~~~~~~~~~\n",
    "  * compiling clusterBed.cpp\n",
    "- Building in src/complementFile\n",
    "  * compiling complementHelp.cpp\n",
    "  * compiling complementFile.cpp\n",
    "- Building in src/coverageFile\n",
    "  * compiling coverageHelp.cpp\n",
    "  * compiling coverageFile.cpp\n",
    "coverageFile.cpp: In member function ‘virtual void CoverageFile::giveFinalReport(RecordOutputMgr*)’:\n",
    "coverageFile.cpp:100:11: warning: ignoring return value of ‘int asprintf(char**, const char*, ...)’, declared with attribute warn_unused_result [-Wunused-result]\n",
    "   asprintf(&depthPctString, \"%0.7f\", depthPct);\n",
    "   ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
    "coverageFile.cpp: In member function ‘void CoverageFile::doMean(RecordOutputMgr*, RecordKeyVector&)’:\n",
    "coverageFile.cpp:181:10: warning: ignoring return value of ‘int asprintf(char**, const char*, ...)’, declared with attribute warn_unused_result [-Wunused-result]\n",
    "  asprintf(&meanString, \"%0.7f\", mean);\n",
    "  ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
    "coverageFile.cpp: In member function ‘void CoverageFile::doHist(RecordOutputMgr*, RecordKeyVector&)’:\n",
    "coverageFile.cpp:210:11: warning: ignoring return value of ‘int asprintf(char**, const char*, ...)’, declared with attribute warn_unused_result [-Wunused-result]\n",
    "   asprintf(&coveredFractionString, \"%0.7f\", coveredFraction);\n",
    "   ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
    "coverageFile.cpp: In member function ‘void CoverageFile::doDefault(RecordOutputMgr*, RecordKeyVector&)’:\n",
    "coverageFile.cpp:230:10: warning: ignoring return value of ‘int asprintf(char**, const char*, ...)’, declared with attribute warn_unused_result [-Wunused-result]\n",
    "  asprintf(&coveredFractionString, \"%0.7f\", coveredFraction);\n",
    "  ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
    "- Building in src/expand\n",
    "  * compiling expand.cpp\n",
    "- Building in src/fastaFromBed\n",
    "  * compiling fastaFromBedMain.cpp\n",
    "  * compiling fastaFromBed.cpp\n",
    "- Building in src/flankBed\n",
    "  * compiling flankBedMain.cpp\n",
    "  * compiling flankBed.cpp\n",
    "- Building in src/genomeCoverageBed\n",
    "  * compiling genomeCoverageMain.cpp\n",
    "  * compiling genomeCoverageBed.cpp\n",
    "genomeCoverageBed.cpp: In member function ‘void BedGenomeCoverage::CoverageBam(std::__cxx11::string)’:\n",
    "genomeCoverageBed.cpp:322:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]\n",
    "                 if(end<_fragmentSize) { //sometimes fragmentSize is bigger :(\n",
    "                    ~~~^~~~~~~~~~~~~~\n",
    "- Building in src/getOverlap\n",
    "  * compiling getOverlap.cpp\n",
    "getOverlap.cpp: In function ‘int getoverlap_main(int, char**)’:\n",
    "getOverlap.cpp:44:10: warning: variable ‘haveColumns’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveColumns = false;\n",
    "          ^~~~~~~~~~~\n",
    "- Building in src/groupBy\n",
    "  * compiling groupBy.cpp\n",
    "  * compiling groupByHelp.cpp\n",
    "- Building in src/intersectFile\n",
    "  * compiling intersectHelp.cpp\n",
    "  * compiling intersectFile.cpp\n",
    "intersectFile.cpp: In member function ‘virtual void IntersectFile::processHits(RecordOutputMgr*, RecordKeyVector&)’:\n",
    "intersectFile.cpp:70:33: warning: variable ‘hitListIter’ set but not used [-Wunused-but-set-variable]\n",
    "  RecordKeyVector::iterator_type hitListIter = hits.begin();\n",
    "                                 ^~~~~~~~~~~\n",
    "- Building in src/fisher\n",
    "  * compiling fisherHelp.cpp\n",
    "  * compiling fisher.cpp\n",
    "  * compiling kfunc.cpp\n",
    "- Building in src/jaccard\n",
    "  * compiling jaccardHelp.cpp\n",
    "  * compiling jaccard.cpp\n",
    "- Building in src/linksBed\n",
    "  * compiling linksMain.cpp\n",
    "  * compiling linksBed.cpp\n",
    "- Building in src/maskFastaFromBed\n",
    "  * compiling maskFastaFromBedMain.cpp\n",
    "  * compiling maskFastaFromBed.cpp\n",
    "- Building in src/mapFile\n",
    "  * compiling mapHelp.cpp\n",
    "  * compiling mapFile.cpp\n",
    "- Building in src/mergeFile\n",
    "  * compiling mergeHelp.cpp\n",
    "  * compiling mergeFile.cpp\n",
    "- Building in src/multiBamCov\n",
    "  * compiling multiBamCovMain.cpp\n",
    "multiBamCovMain.cpp: In function ‘int multibamcov_main(int, char**)’:\n",
    "multiBamCovMain.cpp:38:10: warning: variable ‘haveBed’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveBed           = false;\n",
    "          ^~~~~~~\n",
    "multiBamCovMain.cpp:39:10: warning: variable ‘haveBams’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveBams          = false;\n",
    "          ^~~~~~~~\n",
    "multiBamCovMain.cpp:47:10: warning: variable ‘haveFraction’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveFraction       = false;\n",
    "          ^~~~~~~~~~~~\n",
    "  * compiling multiBamCov.cpp\n",
    "- Building in src/multiIntersectBed\n",
    "  * compiling multiIntersectBedMain.cpp\n",
    "multiIntersectBedMain.cpp: In function ‘int multiintersect_main(int, char**)’:\n",
    "multiIntersectBedMain.cpp:45:10: warning: variable ‘haveFiles’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveFiles         = false;\n",
    "          ^~~~~~~~~\n",
    "multiIntersectBedMain.cpp:47:10: warning: variable ‘haveGenome’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveGenome        = false;\n",
    "          ^~~~~~~~~~\n",
    "multiIntersectBedMain.cpp:48:10: warning: variable ‘haveFiller’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveFiller        = true;\n",
    "          ^~~~~~~~~~\n",
    "  * compiling multiIntersectBed.cpp\n",
    "- Building in src/nucBed\n",
    "  * compiling nucBedMain.cpp\n",
    "  * compiling nucBed.cpp\n",
    "- Building in src/pairToBed\n",
    "  * compiling pairToBedMain.cpp\n",
    "pairToBedMain.cpp: In function ‘int pairtobed_main(int, char**)’:\n",
    "pairToBedMain.cpp:43:10: warning: variable ‘haveFraction’ set but not used [-Wunused-but-set-variabl]\n",
    "     bool haveFraction       = false;\n",
    "          ^~~~~~~~~~~~\n",
    "  * compiling pairToBed.cpp\n",
    "pairToBed.cpp: In member function ‘void BedIntersectPE::FindSpanningOverlaps(const BEDPE&, std::vector<BED>&, const string&)’:\n",
    "pairToBed.cpp:257:12: warning: variable ‘spanLength’ set but not used [-Wunused-but-set-variable]\n",
    "     CHRPOS spanLength = 0;\n",
    "            ^~~~~~~~~~\n",
    "pairToBed.cpp: In member function ‘bool BedIntersectPE::FindOneOrMoreSpanningOverlaps(const BEDPE&, const string&)’:\n",
    "pairToBed.cpp:310:9: warning: variable ‘spanLength’ set but not used [-Wunused-but-set-variable]\n",
    "     int spanLength = 0;\n",
    "         ^~~~~~~~~~\n",
    "- Building in src/pairToPair\n",
    "  * compiling pairToPairMain.cpp\n",
    "pairToPairMain.cpp: In function ‘int pairtopair_main(int, char**)’:\n",
    "pairToPairMain.cpp:44:10: warning: variable ‘haveFraction’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveFraction = false;\n",
    "          ^~~~~~~~~~~~\n",
    "  * compiling pairToPair.cpp\n",
    "pairToPair.cpp: In member function ‘void PairToPair::FindOverlaps(const BEDPE&)’:\n",
    "pairToPair.cpp:112:14: warning: variable ‘found1’ set but not used [-Wunused-but-set-variable]\n",
    "         bool found1 = false;\n",
    "              ^~~~~~\n",
    "pairToPair.cpp:113:14: warning: variable ‘found2’ set but not used [-Wunused-but-set-variable]\n",
    "         bool found2 = false;\n",
    "              ^~~~~~\n",
    "- Building in src/randomBed\n",
    "  * compiling randomBedMain.cpp\n",
    "  * compiling randomBed.cpp\n",
    "- Building in src/regressTest\n",
    "compiling RegressTest.cpp\n",
    "RegressTest.cpp: In member function ‘bool RegressTest::executeAndCompareCorrectness(const fileListType&)’:\n",
    "RegressTest.cpp:431:8: warning: ignoring return value of ‘int system(const char*)’, declared with attribute warn_unused_result [-Wunused-result]\n",
    "  system(diffCmd.c_str());\n",
    "  ~~~~~~^~~~~~~~~~~~~~~~~\n",
    "RegressTest.cpp: In member function ‘bool RegressTest::startMemoryProfile(bool)’:\n",
    "RegressTest.cpp:573:7: warning: ignoring return value of ‘char* fgets(char*, int, FILE*)’, declared with attribute warn_unused_result [-Wunused-result]\n",
    "  fgets(sLine, 4192, fp);\n",
    "  ~~~~~^~~~~~~~~~~~~~~~~\n",
    "RegressTest.cpp: In member function ‘bool RegressTest::calcMemoryStats()’:\n",
    "RegressTest.cpp:622:8: warning: ignoring return value of ‘char* fgets(char*, int, FILE*)’, declared with attribute warn_unused_result [-Wunused-result]\n",
    "   fgets(sLine, 4192, fp);\n",
    "   ~~~~~^~~~~~~~~~~~~~~~~\n",
    "compiling regressTestMain.cpp\n",
    "- Building in src/reldist\n",
    "  * compiling reldistMain.cpp\n",
    "  * compiling reldist.cpp\n",
    "- Building in src/sampleFile\n",
    "  * compiling sampleHelp.cpp\n",
    "  * compiling sampleFile.cpp\n",
    "- Building in src/shiftBed\n",
    "  * compiling shiftBedMain.cpp\n",
    "  * compiling shiftBed.cpp\n",
    "- Building in src/shuffleBed\n",
    "  * compiling shuffleBedMain.cpp\n",
    "  * compiling shuffleBed.cpp\n",
    "- Building in src/slopBed\n",
    "  * compiling slopBedMain.cpp\n",
    "  * compiling slopBed.cpp\n",
    "- Building in src/sortBed\n",
    "  * compiling sortMain.cpp\n",
    "  * compiling sortBed.cpp\n",
    "- Building in src/spacingFile\n",
    "  * compiling spacingHelp.cpp\n",
    "  * compiling spacingFile.cpp\n",
    "- Building in src/split\n",
    "  * compiling splitBed.cpp\n",
    "  * compiling splitBedMain.cpp\n",
    "- Building in src/subtractFile\n",
    "  * compiling subtractHelp.cpp\n",
    "  * compiling subtractFile.cpp\n",
    "- Building in src/tagBam\n",
    "  * compiling tagBamMain.cpp\n",
    "  * compiling tagBam.cpp\n",
    "- Building in src/unionBedGraphs\n",
    "  * compiling unionBedGraphsMain.cpp\n",
    "unionBedGraphsMain.cpp: In function ‘int unionbedgraphs_main(int, char**)’:\n",
    "unionBedGraphsMain.cpp:46:10: warning: variable ‘haveFiles’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveFiles         = false;\n",
    "          ^~~~~~~~~\n",
    "unionBedGraphsMain.cpp:48:10: warning: variable ‘haveGenome’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveGenome        = false;\n",
    "          ^~~~~~~~~~\n",
    "unionBedGraphsMain.cpp:49:10: warning: variable ‘haveFiller’ set but not used [-Wunused-but-set-variable]\n",
    "     bool haveFiller        = true;\n",
    "          ^~~~~~~~~~\n",
    "  * compiling unionBedGraphs.cpp\n",
    "- Building in src/windowBed\n",
    "  * compiling windowMain.cpp\n",
    "  * compiling windowBed.cpp\n",
    "- Building in src/windowMaker\n",
    "  * compiling windowMakerMain.cpp\n",
    "  * compiling windowMaker.cpp\n",
    "- Building main bedtools binary.\n",
    "done.\n",
    "- Creating executables for old CLI.\n",
    "done.\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  PATH=$RNA_HOME/student_tools/samtools-1.9:$RNA_HOME/student_tools/bam-readcount/bin:$RNA_HOME/student_tools/hisat2-2.1.0:$RNA_HOME/student_tools/stringtie-1.3.4d.Linux_x86_64:$RNA_HOME/student_tools/gffcompare-0.10.6.Linux_x86_64:$RNA_HOME/student_tools/htseq-release_0.11.0/scripts:$RNA_HOME/student_tools/tophat-2.1.1.Linux_x86_64:$RNA_HOME/student_tools/kallisto_linux-v0.44.0:$RNA_HOME/student_tools/FastQC:$RNA_HOME/student_tools/flexbar-3.4.0-linux:$RNA_HOME/student_tools/regtools/build:/home/ubuntu/bin/bedtools2/bin:$PATH\n",
    "[nathalia@aws] $  \n",
    "[nathalia@aws] $  export LD_LIBRARY_PATH=$RNA_HOME/student_tools/flexbar-3.4.0-linux:$LD_LIBRARY_PATH[nathalia@aws] $  echo $PATH\n",
    "/home/ubuntu/workspace/rnaseq/student_tools/samtools-1.9:/home/ubuntu/workspace/rnaseq/student_tools/bam-readcount/bin:/home/ubuntu/workspace/rnaseq/student_tools/hisat2-2.1.0:/home/ubuntu/workspace/rnaseq/student_tools/stringtie-1.3.4d.Linux_x86_64:/home/ubuntu/workspace/rnaseq/student_tools/gffcompare-0.10.6.Linux_x86_64:/home/ubuntu/workspace/rnaseq/student_tools/htseq-release_0.11.0/scripts:/home/ubuntu/workspace/rnaseq/student_tools/tophat-2.1.1.Linux_x86_64:/home/ubuntu/workspace/rnaseq/student_tools/kallisto_linux-v0.44.0:/home/ubuntu/workspace/rnaseq/student_tools/FastQC:/home/ubuntu/workspace/rnaseq/student_tools/flexbar-3.4.0-linux:/home/ubuntu/workspace/rnaseq/student_tools/regtools/build:/home/ubuntu/bin/bedtools2/bin:/home/ubuntu/.local/bin:/home/ubuntu/bin:/home/ubuntu/bin:/home/ubuntu/bin/edirect:/home/ubuntu/bin/sratoolkit.2.9.2-ubuntu64/bin:/home/ubuntu/workspace/rnaseq/tools/samtools-1.9:/home/ubuntu/workspace/rnaseq/tools/bam-readcount/bin:/home/ubuntu/workspace/rnaseq/tools/hisat2-2.1.0:/home/ubuntu/workspace/rnaseq/tools/stringtie-1.3.4d.Linux_x86_64:/home/ubuntu/workspace/rnaseq/tools/gffcompare-0.10.6.Linux_x86_64:/home/ubuntu/workspace/rnaseq/tools/htseq-release_0.11.0/scripts:/home/ubuntu/workspace/rnaseq/tools/tophat-2.1.1.Linux_x86_64:/home/ubuntu/workspace/rnaseq/tools/kallisto_linux-v0.44.0:/home/ubuntu/workspace/rnaseq/tools/FastQC:/home/ubuntu/workspace/rnaseq/tools/flexbar-3.4.0-linux:/home/ubuntu/workspace/rnaseq/tools/regtools/build:/home/ubuntu/bin/bedtools2/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Reference genome"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq\n",
    "[nathalia@aws] $  echo $RNA_REFS_DIR\n",
    "/home/ubuntu/workspace/rnaseq/refs\n",
    "[nathalia@aws] $  mkdir -p $RNA_REFS_DIR\n",
    "[nathalia@aws] $  ls\n",
    "refs  student_tools  tools\n",
    "[nathalia@aws] $  \n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_REFS_DIR\n",
    "[nathalia@aws] $  wget http://genomedata.org/rnaseq-tutorial/fasta/GRCh38/chr22_with_ERCC92.fa\n",
    "--2018-11-13 00:02:36--  http://genomedata.org/rnaseq-tutorial/fasta/GRCh38/chr22_with_ERCC92.fa\n",
    "Resolving genomedata.org (genomedata.org)... 3.8.108.6\n",
    "Connecting to genomedata.org (genomedata.org)|3.8.108.6|:80... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 51751056 (49M)\n",
    "Saving to: ‘chr22_with_ERCC92.fa’\n",
    "\n",
    "chr22_with_ERCC92.fa      100%[==================================>]  49.35M  9.09MB/s    in 5.4s    \n",
    "\n",
    "2018-11-13 00:02:41 (9.09 MB/s) - ‘chr22_with_ERCC92.fa’ saved [51751056/51751056]\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 50M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 50M Oct 23 16:48 chr22_with_ERCC92.fa\n",
    "\n",
    "\n",
    "[nathalia@aws] $  head chr22_with_ERCC92.fa \n",
    ">22 dna_sm:chromosome chromosome:GRCh38:22:1:50818468:1 REF\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
    "[nathalia@aws] $  tail chr22_with_ERCC92.fa \n",
    "GGACGGGCACGCTCATATCAGGCTATATTTGGTCCGGGTTATTATCGTCG\n",
    "CGGTTACCGTAATACTTCAGATCAGTTAAGTAGGGCCATATGCCTCGGGA\n",
    "ATAAGCTGACGGTGACAAGGTTTCCCCCTAATCGAGACGCTGCAATAACA\n",
    "CAGGGGCATACAGTAACCAGGCAAGAGTTCAATCGCTTAGTTTCGTGGCG\n",
    "GGATTTGAGGAAAACTGCGACTGTTCTTTAACCAAACATCCGTGCGATTC\n",
    "GTGCCACTCGTAGACGGCATCTCACAGTCACTGAAGGCTATTAAAGAGTT\n",
    "AGCACCCACCATTGGATGAAGCCCAGGATAAGTGACCCCCCCGGACCTTG\n",
    "GAGTTTCATGCTAATCAAAGAAGAGCTAATCCGACGTAAAGTTGCGGCGT\n",
    "TGATTACGCAGGATTGCGACCAAAGAACGAGAAAAAAAAAAAAAAAAAAA\n",
    "AAAAA\n",
    "[nathalia@aws] $  wc -l chr22_with_ERCC92.fa \n",
    "848761 chr22_with_ERCC92.fa\n",
    "\n",
    "[nathalia@aws] $  cat chr22_with_ERCC92.fa | grep -v \">\" | perl -ne 'chomp $_; $bases{$_}++ for split //; if (eof){print \"$_ $bases{$_}\\n\" for sort keys %bases}'\n",
    "A 4455938\n",
    "C 4406493\n",
    "G 4411768\n",
    "N 10710000\n",
    "T 4445994\n",
    "Y 1\n",
    "a 5950524\n",
    "c 4772185\n",
    "g 4853055\n",
    "n 948691\n",
    "t 5946575\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Practice 2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_REFS_DIR\n",
    "\n",
    "[nathalia@aws] $  ls\n",
    "chr22_with_ERCC92.1.ht2  chr22_with_ERCC92.5.ht2  chr22_with_ERCC92.fa   splicesites.tsv\n",
    "chr22_with_ERCC92.2.ht2  chr22_with_ERCC92.6.ht2  chr22_with_ERCC92.gtf\n",
    "chr22_with_ERCC92.3.ht2  chr22_with_ERCC92.7.ht2  exons.tsv\n",
    "chr22_with_ERCC92.4.ht2  chr22_with_ERCC92.8.ht2  illumina_multiplex.fa\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "dict_chr22={}\n",
    "\n",
    "with open('chr22_with_ERCC92.fa', 'r') as chr22:\n",
    "    for line in chr22:\n",
    "        line=line.rstrip()\n",
    "        if line.startswith('>22'):\n",
    "                header=line\n",
    "                if header not in dict_chr22:\n",
    "                    dict_chr22[header]=''\n",
    "        else:\n",
    "            seq=line\n",
    "            dict_chr22[header]+=seq\n",
    "            \n",
    "count=0\n",
    "for gene in dict_chr22:\n",
    "    repet_elem=dict_chr22[gene].count(\"N\")\n",
    "    whole_seq=len(dict_chr22[gene])\n",
    "    a=repet_elem/whole_seq\n",
    "    for i in re.findall(r\"[AG]AATT[CT]\",sequence):\n",
    "        count+=1\n",
    "\n",
    "print('Number of EcoRI restriction sites found in chr22: {}'.format(count))\n",
    "\n",
    "[nathalia@aws] $  ./practice2.py &\n",
    "[2] 20101\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Annotations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_REFS_DIR\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/refs\n",
    "[nathalia@aws] $  lhtr\n",
    "total 50M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 50M Oct 23 16:48 chr22_with_ERCC92.fa\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  wget http://genomedata.org/rnaseq-tutorial/annotations/GRCh38/chr22_with_ERCC92.gtf--2018-11-13 00:14:06--  http://genomedata.org/rnaseq-tutorial/annotations/GRCh38/chr22_with_ERCC92.gtf\n",
    "Resolving genomedata.org (genomedata.org)... 3.8.108.6\n",
    "Connecting to genomedata.org (genomedata.org)|3.8.108.6|:80... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 30712117 (29M)\n",
    "Saving to: ‘chr22_with_ERCC92.gtf’\n",
    "\n",
    "chr22_with_ERCC92.gtf     100%[==================================>]  29.29M  8.20MB/s    in 3.6s    \n",
    "\n",
    "2018-11-13 00:14:10 (8.20 MB/s) - ‘chr22_with_ERCC92.gtf’ saved [30712117/30712117]\n",
    "\n",
    "[nathalia@aws] $  echo $RNA_REF_GTF\n",
    "/home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92.gtf\n",
    "\n",
    "[nathalia@aws] $  less -p start_codon -S $RNA_REF_GTF\n",
    "\n",
    "# -p highlights matches in the file\n",
    "\n",
    "[nathalia@aws] $  grep ENST00000342247 $RNA_REF_GTF | less -p \"exon\" -S\n",
    "\n",
    "# shows all the entries of this particular gene\n",
    "\n",
    "[nathalia@aws] $  perl -ne 'if ($_ =~ /(gene_id\\s\\\"ENSG\\w+\\\")/){print \"$1\\n\"}' $RNA_REF_GTF | sort | uniq | wc -l\n",
    "1318\n",
    "\n",
    "[nathalia@aws] $  cat $RNA_REF_GTF | cut -f 3 | grep -c gene\n",
    "1318\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Index"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME\n",
    "\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq\n",
    "\n",
    "[nathalia@aws] $  hisat2_extract_splice_sites.py $RNA_REF_GTF > $RNA_REFS_DIR/splicesites.tsv\n",
    "[nathalia@aws] $  hisat2_extract_exons.py $RNA_REF_GTF > $RNA_REFS_DIR/exons.tsv\n",
    "[nathalia@aws] $  hisat2-build -p 8 --ss $RNA_REFS_DIR/splicesites.tsv --exon $RNA_REFS_DIR/exons.tsv $RNA_REF_FASTA $RNA_REF_INDEX\n",
    "Settings:\n",
    "  Output files: \"/home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92.*.ht2\"\n",
    "  Line rate: 7 (line is 128 bytes)\n",
    "  Lines per side: 1 (side is 128 bytes)\n",
    "  Offset rate: 4 (one in 16)\n",
    "  FTable chars: 10\n",
    "  Strings: unpacked\n",
    "  Local offset rate: 3 (one in 8)\n",
    "  Local fTable chars: 6\n",
    "  Local sequence length: 57344\n",
    "  Local sequence overlap between two consecutive indexes: 1024\n",
    "  Endianness: little\n",
    "  Actual local endianness: little\n",
    "  Sanity checking: disabled\n",
    "  Assertions: disabled\n",
    "  Random seed: 0\n",
    "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
    "Input files DNA, FASTA:\n",
    "  /home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92.fa\n",
    "Reading reference sizes\n",
    "  Time reading reference sizes: 00:00:00\n",
    "Calculating joined length\n",
    "Writing header\n",
    "Reserving space for joined string\n",
    "Joining reference sequences\n",
    "  Time to join reference sequences: 00:00:00\n",
    "  Time to read SNPs and splice sites: 00:00:00\n",
    "Generation 0 (39250236 -> 39250236 nodes, 0 ranks)\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "Generation 1 (39257595 -> 39257595 nodes, 0 ranks)\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "Generation 2 (39272316 -> 39272316 nodes, 0 ranks)\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "Generation 3 (39301776 -> 39301776 nodes, 0 ranks)\n",
    "BUILT FROM_INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 1\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "RESIZE NODES: 1\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "12 5120817\n",
    "16 5313362\n",
    "48 5650456\n",
    "54 4705498\n",
    "16 4869620\n",
    "49 4609511\n",
    "16 5217917\n",
    "155 3873596\n",
    "FINISHED RECURSIVE SORTS: 1\n",
    "SORT NODES: 1\n",
    "MERGE, UPDATE RANK: 0\n",
    "Generation 4 (39360777 -> 39315196 nodes, 29466727 ranks)\n",
    "ALLOCATE FROM_TABLE: 1\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "38 4981795\n",
    "38 4993576\n",
    "38 4991933\n",
    "38 4991322\n",
    "37 4855436\n",
    "37 4862563\n",
    "37 4859282\n",
    "37 4779289\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 1\n",
    "TOTAL TIME: 2\n",
    "Generation 5 (39322184 -> 39317557 nodes, 34612896 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "38 4981868\n",
    "38 4995067\n",
    "38 4992416\n",
    "38 4991339\n",
    "37 4855384\n",
    "37 4862647\n",
    "37 4859641\n",
    "37 4779195\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 1\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 0\n",
    "TOTAL TIME: 1\n",
    "Generation 6 (39323256 -> 39321583 nodes, 37172451 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "38 4981954\n",
    "38 4997504\n",
    "38 4993158\n",
    "38 4991483\n",
    "37 4855378\n",
    "37 4862698\n",
    "37 4860218\n",
    "37 4779190\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 1\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 0\n",
    "TOTAL TIME: 1\n",
    "Generation 7 (39327438 -> 39325919 nodes, 38260854 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "38 4982133\n",
    "38 5000683\n",
    "38 4993620\n",
    "38 4991460\n",
    "37 4855378\n",
    "37 4862703\n",
    "37 4860753\n",
    "37 4779189\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 1\n",
    "TOTAL TIME: 1\n",
    "Generation 8 (39332047 -> 39330184 nodes, 39014973 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "38 4982517\n",
    "38 5003436\n",
    "38 4993876\n",
    "38 4991416\n",
    "37 4855378\n",
    "37 4862658\n",
    "37 4861714\n",
    "37 4779189\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 1\n",
    "TOTAL TIME: 1\n",
    "Generation 9 (39337374 -> 39335766 nodes, 39213952 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "38 4983285\n",
    "38 5006515\n",
    "38 4993894\n",
    "38 4991416\n",
    "37 4855378\n",
    "37 4862658\n",
    "37 4863431\n",
    "37 4779189\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 0\n",
    "TOTAL TIME: 0\n",
    "Generation 10 (39342142 -> 39339288 nodes, 39289153 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "38 4984120\n",
    "38 5008324\n",
    "38 4993894\n",
    "38 4991416\n",
    "37 4855378\n",
    "37 4862658\n",
    "37 4864309\n",
    "37 4779189\n",
    "FINISHED RECURSIVE SORTS: 1\n",
    "BUILD TABLE: 1\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 0\n",
    "TOTAL TIME: 1\n",
    "Generation 11 (39342851 -> 39339120 nodes, 39321913 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "38 4985142\n",
    "38 5008045\n",
    "38 4993894\n",
    "38 4991416\n",
    "37 4855378\n",
    "37 4862658\n",
    "37 4863398\n",
    "37 4779189\n",
    "FINISHED RECURSIVE SORTS: 1\n",
    "BUILD TABLE: 1\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 0\n",
    "TOTAL TIME: 1\n",
    "Generation 12 (39340396 -> 39335818 nodes, 39331595 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "38 4986389\n",
    "38 5004772\n",
    "38 4993894\n",
    "38 4991416\n",
    "37 4855378\n",
    "37 4862658\n",
    "37 4862122\n",
    "37 4779189\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 1\n",
    "TOTAL TIME: 1\n",
    "Generation 13 (39335818 -> 39335810 nodes, 39335810 ranks)\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "38 4986389\n",
    "38 5004764\n",
    "38 4993894\n",
    "38 4991416\n",
    "37 4855378\n",
    "37 4862658\n",
    "37 4862122\n",
    "37 4779189\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "Generating edges... \n",
    "NODE.TO -> GENOME POS: 0\n",
    "BUILD FROM_INDEX 0\n",
    "COUNTED NEW EDGES: 0\n",
    "MADE NEW EDGES: 1\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "27 1315740\n",
    "39 1309145\n",
    "39 1307884\n",
    "47 1302209\n",
    "31 1317027\n",
    "32 1308488\n",
    "40 1292062\n",
    "46 1278333\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "35 1161676\n",
    "26 1176848\n",
    "32 1150809\n",
    "31 1170097\n",
    "48 1141516\n",
    "68 1147212\n",
    "27 1145463\n",
    "34 1111689\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "40 1172640\n",
    "35 1170972\n",
    "40 1172745\n",
    "37 1166096\n",
    "29 1160233\n",
    "32 1166931\n",
    "44 1177265\n",
    "44 1104224\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "61 1314084\n",
    "43 1325913\n",
    "38 1307043\n",
    "32 1300337\n",
    "32 1335684\n",
    "30 1290259\n",
    "39 1289419\n",
    "26 1253171\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "SORTED NEW EDGES: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "38 4980736\n",
    "38 4980736\n",
    "38 4980736\n",
    "38 4980736\n",
    "38 4980736\n",
    "37 4849664\n",
    "37 4849664\n",
    "37 4732802\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "RE-SORTED NODES: 1\n",
    "PROCESS EDGES: 0\n",
    "REMOVE Y: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "38 4981325\n",
    "38 4982124\n",
    "38 4982127\n",
    "38 4982902\n",
    "38 4981513\n",
    "37 4850354\n",
    "37 4849918\n",
    "37 4732952\n",
    "FINISHED RECURSIVE SORTS: 1\n",
    "SORT, Make index: 1\n",
    "TOTAL: 3\n",
    "Allocating ftab, absorbFtab\n",
    "Entering GFM loop\n",
    "Exited GFM loop\n",
    "fchr[A]: 0\n",
    "fchr[C]: 10430888\n",
    "fchr[G]: 19636198\n",
    "fchr[T]: 28927304\n",
    "fchr[$]: 39343214\n",
    "Exiting GFM::buildToDisk()\n",
    "Returning from initFromVector\n",
    "Wrote 28409188 bytes to primary GFM file: /home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92.1.ht2\n",
    "Wrote 9833960 bytes to secondary GFM file: /home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92.2.ht2\n",
    "Re-opening _in1 and _in2 as input streams\n",
    "Returning from GFM constructor\n",
    "Returning from initFromVector\n",
    "Wrote 25762107 bytes to primary GFM file: /home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92.5.ht2\n",
    "Wrote 9998776 bytes to secondary GFM file: /home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92.6.ht2\n",
    "Re-opening _in5 and _in5 as input streams\n",
    "Returning from HierEbwt constructor\n",
    "Headers:\n",
    "    len: 39242532\n",
    "    gbwtLen: 39343215\n",
    "    nodes: 39335809\n",
    "    sz: 19621266\n",
    "    gbwtSz: 19671608\n",
    "    lineRate: 7\n",
    "    offRate: 4\n",
    "    offMask: 0xfffffff0\n",
    "    ftabChars: 10\n",
    "    eftabLen: 0\n",
    "    eftabSz: 0\n",
    "    ftabLen: 1048577\n",
    "    ftabSz: 4194308\n",
    "    offsLen: 2458489\n",
    "    offsSz: 9833956\n",
    "    lineSz: 128\n",
    "    sideSz: 128\n",
    "    sideGbwtSz: 104\n",
    "    sideGbwtLen: 208\n",
    "    numSides: 189151\n",
    "    numLines: 189151\n",
    "    gbwtTotLen: 24211328\n",
    "    gbwtTotSz: 24211328\n",
    "    reverse: 0\n",
    "    linearFM: No\n",
    "Total time for call to driver() for forward index: 00:00:57\n",
    "\n",
    "[nathalia@aws] $  lhtr refs/\n",
    "total 160M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  50M Oct 23 16:48 chr22_with_ERCC92.fa\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  30M Oct 23 16:49 chr22_with_ERCC92.gtf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 172K Nov 13 00:26 splicesites.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 148K Nov 13 00:26 exons.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 9.4M Nov 13 00:26 chr22_with_ERCC92.4.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.3K Nov 13 00:26 chr22_with_ERCC92.3.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  55K Nov 13 00:26 chr22_with_ERCC92.8.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 278K Nov 13 00:26 chr22_with_ERCC92.7.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 9.4M Nov 13 00:27 chr22_with_ERCC92.2.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  28M Nov 13 00:27 chr22_with_ERCC92.1.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 9.6M Nov 13 00:27 chr22_with_ERCC92.6.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25M Nov 13 00:27 chr22_with_ERCC92.5.ht2\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  echo $RNA_DATA_DIR\n",
    "/home/ubuntu/workspace/rnaseq/data\n",
    "[nathalia@aws] $  mkdir -p $RNA_DATA_DIR\n",
    "[nathalia@aws] $  cd $RNA_DATA_DIR\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/data\n",
    "[nathalia@aws] $  wget http://genomedata.org/rnaseq-tutorial/HBR_UHR_ERCC_ds_5pc.tar\n",
    "--2018-11-13 00:33:33--  http://genomedata.org/rnaseq-tutorial/HBR_UHR_ERCC_ds_5pc.tar\n",
    "Resolving genomedata.org (genomedata.org)... 3.8.108.6\n",
    "Connecting to genomedata.org (genomedata.org)|3.8.108.6|:80... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 116602880 (111M) [application/x-tar]\n",
    "Saving to: ‘HBR_UHR_ERCC_ds_5pc.tar’\n",
    "\n",
    "HBR_UHR_ERCC_ds_5pc.tar   100%[==================================>] 111.20M  10.5MB/s    in 11s     \n",
    "\n",
    "2018-11-13 00:33:44 (9.83 MB/s) - ‘HBR_UHR_ERCC_ds_5pc.tar’ saved [116602880/116602880]\n",
    "\n",
    "[nathalia@aws] $  tar -xvf HBR_UHR_ERCC_ds_5pc.tar\n",
    "HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "```\n",
    "\n",
    "\n",
    "- UHR + ERCC Spike-In Mix1, Replicate 1\n",
    "- UHR + ERCC Spike-In Mix1, Replicate 2\n",
    "- UHR + ERCC Spike-In Mix1, Replicate 3\n",
    "- HBR + ERCC Spike-In Mix2, Replicate 1\n",
    "- HBR + ERCC Spike-In Mix2, Replicate 2\n",
    "- HBR + ERCC Spike-In Mix2, Replicate 3\n",
    "\n",
    "\n",
    "```\n",
    "[nathalia@aws] $  zcat UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz | grep -P \"^\\@HWI\" | wc -l\n",
    "227392\n",
    "\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Practice 3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  mkdir practice3\n",
    "[nathalia@aws] $  cd practice3/\n",
    "\n",
    "[nathalia@aws] $  wget http://genomedata.org/rnaseq-tutorial/practical.tar\n",
    "--2018-11-13 00:48:17--  http://genomedata.org/rnaseq-tutorial/practical.tar\n",
    "Resolving genomedata.org (genomedata.org)... 3.8.108.6\n",
    "Connecting to genomedata.org (genomedata.org)|3.8.108.6|:80... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 363653120 (347M) [application/x-tar]\n",
    "Saving to: ‘practical.tar’\n",
    "\n",
    "practical.tar            100%[=================================>] 346.81M  9.25MB/s    in 45s     \n",
    "\n",
    "2018-11-13 00:49:02 (7.74 MB/s) - ‘practical.tar’ saved [363653120/363653120]\n",
    "\n",
    "[nathalia@aws] $  tar -xvf practical.tar \n",
    "hcc1395_normal_rep1_r1.fastq.gz\n",
    "hcc1395_normal_rep1_r2.fastq.gz\n",
    "hcc1395_normal_rep2_r1.fastq.gz\n",
    "hcc1395_normal_rep2_r2.fastq.gz\n",
    "hcc1395_normal_rep3_r1.fastq.gz\n",
    "hcc1395_normal_rep3_r2.fastq.gz\n",
    "hcc1395_tumor_rep1_r1.fastq.gz\n",
    "hcc1395_tumor_rep1_r2.fastq.gz\n",
    "hcc1395_tumor_rep2_r1.fastq.gz\n",
    "hcc1395_tumor_rep2_r2.fastq.gz\n",
    "hcc1395_tumor_rep3_r1.fastq.gz\n",
    "hcc1395_tumor_rep3_r2.fastq.gz\n",
    "\n",
    "[nathalia@aws] $  ls | grep -c \".gz\" # lazy\n",
    "12\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### **Questions**\n",
    "\n",
    "- How many data files were contained in the 'practical.tar' archive? What commonly used sequence data file format are they?\n",
    "\n",
    "12, fastq\n",
    "\n",
    "- In the first read of the hcc1395, normal, replicate 1, read 1 file, what was the physical location of the read on the flow cell (i.e. lane, tile, x, y)?\n",
    "\n",
    "@K00193:38:H3MYFBBXX:4:1101:10003:44458/1\n",
    "\n",
    "Lane 4, tile = 1101, x=10003, y=44458\n",
    "\n",
    "\n",
    "- In the first read of this same file, how many 'T' bases are there?\n",
    "\n",
    "```\n",
    "[nathalia@aws] $  zcat hcc1395_normal_rep1_r1.fastq.gz | head -n 2 | sed '1'd \n",
    "TTCCTTATGAAACAGGAAGAGTCCCTGGGCCCAGGCCTGGCCCACGGTTGTCAAGGCACATCATTGCCAGCAAGCTGAAGCATACCAGCAGCCACAACCTAGATCTCATTCCCAACCCAAAGTTCTGACTTCTGTACAAACTCGTTTCCAG\n",
    "\n",
    "[nathalia@aws] $  python\n",
    "Python 2.7.15rc1 (default, Apr 15 2018, 21:51:34) \n",
    "[GCC 7.3.0] on linux2\n",
    "Type \"help\", \"copyright\", \"credits\" or \"license\" for more information.\n",
    ">>> seq=\"TTCCTTATGAAACAGGAAGAGTCCCTGGGCCCAGGCCTGGCCCACGGTTGTCAAGGCACATCATTGCCAGCAAGCTGAAGCATACCAGCAGCCACAACCTAGATCTCATTCCCAACCCAAAGTTCTGACTTCTGTACAAACTCGTTTCCAG\"\n",
    ">>> seq.count('T')\n",
    "32\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data QC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  screen -S fastqc\n",
    "\n",
    "cd $RNA_HOME/data\n",
    "fastqc *.fastq.gz\n",
    "\n",
    "(...)\n",
    "\n",
    "[detached from 18496.fastqc]\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In Safari,\n",
    "\n",
    "http://ec2-34-220-220-193.us-west-2.compute.amazonaws.com/workspace/rnaseq/data/\n",
    "\n",
    "Click on Data/\n",
    "\n",
    "Click on the `.html` "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![fastqc](fastqc.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Over-represented sequence\n",
    "\n",
    "`CTTATGTGATAGATGCCTCTTTAAAATATCTAAGTGCTGGGGTTATGAGT`\n",
    "\n",
    "Blastn"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![over represented sequence](over_represented_sequence.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is a spike-in control sequence from Methanocaldococcus jannaschii (archaean)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  multiqc .\n",
    "[INFO   ]         multiqc : This is MultiQC v1.6\n",
    "[INFO   ]         multiqc : Template    : default\n",
    "[INFO   ]         multiqc : Searching '.'\n",
    "[INFO   ]          fastqc : Found 12 reports\n",
    "[INFO   ]         multiqc : Compressing plot data\n",
    "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
    "[INFO   ]         multiqc : Data        : multiqc_data\n",
    "[INFO   ]         multiqc : MultiQC complete\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://ec2-34-220-220-193.us-west-2.compute.amazonaws.com/workspace/rnaseq/data/multiqc_report.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![multiqc](multiqc.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Adapter Trim"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME\n",
    "\n",
    "[nathalia@aws] $  echo $RNA_DATA_TRIM_DIR\n",
    "/home/ubuntu/workspace/rnaseq/data/trimmed\n",
    "\n",
    "[nathalia@aws] $  mkdir -p $RNA_DATA_TRIM_DIR\n",
    "\n",
    "[nathalia@aws] $  echo $RNA_REFS_DIR\n",
    "/home/ubuntu/workspace/rnaseq/refs\n",
    "\n",
    "[nathalia@aws] $  mkdir -p $RNA_REFS_DIR\n",
    "\n",
    "[nathalia@aws] $  cd $RNA_REFS_DIR\n",
    "\n",
    "[nathalia@aws] $  wget http://genomedata.org/rnaseq-tutorial/illumina_multiplex.fa\n",
    "--2018-11-13 01:11:28--  http://genomedata.org/rnaseq-tutorial/illumina_multiplex.fa\n",
    "Resolving genomedata.org (genomedata.org)... 3.8.108.6\n",
    "Connecting to genomedata.org (genomedata.org)|3.8.108.6|:80... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 161\n",
    "Saving to: ‘illumina_multiplex.fa’\n",
    "\n",
    "illumina_multiplex.fa     100%[==================================>]     161  --.-KB/s    in 0s      \n",
    "\n",
    "2018-11-13 01:11:28 (36.5 MB/s) - ‘illumina_multiplex.fa’ saved [161/161]\n",
    "\n",
    "[nathalia@aws] $  cd $RNA_HOME\n",
    "\n",
    "[nathalia@aws] $  emacs flexbar.sh \n",
    "\n",
    "#!bin/bash\n",
    "\n",
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters $RNA_REFS_DIR/illumina_multiplex.fa --pre-trim-left 13 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads $RNA_DATA_DIR/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz --reads2 $RNA_DATA_DIR/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz --target $RNA_DATA_TRIM_DIR/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22\n",
    "\n",
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters $RNA_REFS_DIR/illumina_multiplex.fa --pre-trim-left 13 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads $RNA_DATA_DIR/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz --reads2 $RNA_DATA_DIR/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz --target $RNA_DATA_TRIM_DIR/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22\n",
    "\n",
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters $RNA_REFS_DIR/illumina_multiplex.fa --pre-trim-left 13 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads $RNA_DATA_DIR/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz --reads2 $RNA_DATA_DIR/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz --target $RNA_DATA_TRIM_DIR/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22\n",
    "\n",
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters $RNA_REFS_DIR/illumina_multiplex.fa --pre-trim-left 13 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads $RNA_DATA_DIR/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz --reads2 $RNA_DATA_DIR/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz --target $RNA_DATA_TRIM_DIR/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22\n",
    "\n",
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters $RNA_REFS_DIR/illumina_multiplex.fa --pre-trim-left 13 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads $RNA_DATA_DIR/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz --reads2 $RNA_DATA_DIR/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz --target $RNA_DATA_TRIM_DIR/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22\n",
    "\n",
    "\n",
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters $RNA_REFS_DIR/illumina_multiplex.fa --pre-trim-left 13 --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads $RNA_DATA_DIR/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz --reads2 $RNA_DATA_DIR/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz --target $RNA_DATA_TRIM_DIR/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "[nathalia@aws] $  screen -R fastqc \n",
    "\n",
    "[nathalia@aws] $  bash flexbar.sh\n",
    "\n",
    "[detached from 18496.fastqc]\n",
    "\n",
    "\n",
    "[nathalia@aws] $  cd $RNA_DATA_TRIM_DIR\n",
    "\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/data/trimmed\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 105M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.7K Nov 13 01:18 UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.log\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13M Nov 13 01:18 UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22_1.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13M Nov 13 01:18 UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22_2.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.7K Nov 13 01:18 UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.log\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 9.1M Nov 13 01:18 UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22_1.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 9.6M Nov 13 01:18 UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22_2.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.7K Nov 13 01:18 UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.log\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  10M Nov 13 01:18 UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22_1.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  11M Nov 13 01:18 UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22_2.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.7K Nov 13 01:18 HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.log\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 6.0M Nov 13 01:18 HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22_1.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 6.4M Nov 13 01:18 HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22_2.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.7K Nov 13 01:19 HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.log\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 7.3M Nov 13 01:19 HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22_1.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 7.7M Nov 13 01:19 HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22_2.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.7K Nov 13 01:19 HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.log\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 6.6M Nov 13 01:19 HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22_1.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 6.9M Nov 13 01:19 HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22_2.fastq.gz\n",
    "\n",
    "[nathalia@aws] $  screen -R fastqc \n",
    "\n",
    "[nathalia@aws] $ cd $RNA_DATA_TRIM_DIR\n",
    "\n",
    "[nathalia@aws] $ fastqc *.fastq.gz\n",
    "\n",
    "[nathalia@aws] $  multiqc .\n",
    "[INFO   ]         multiqc : This is MultiQC v1.6\n",
    "[INFO   ]         multiqc : Template    : default\n",
    "[INFO   ]         multiqc : Searching '.'\n",
    "[INFO   ]         flexbar : Found 6 logs\n",
    "[INFO   ]          fastqc : Found 12 reports\n",
    "[INFO   ]         multiqc : Compressing plot data\n",
    "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
    "[INFO   ]         multiqc : Data        : multiqc_data\n",
    "[INFO   ]         multiqc : MultiQC complete\n",
    "\n",
    "\n",
    "[detached from 18496.fastqc]\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://ec2-34-220-220-193.us-west-2.compute.amazonaws.com/workspace/rnaseq/data/trimmed/multiqc_report.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![fastqc after trim](fastqc_after_trim.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Practice 5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/practice3\n",
    "\n",
    "[nathalia@aws] $  emacs qc1.sh\n",
    "\n",
    "\n",
    "#!/bin/bash\n",
    "\n",
    "fastqc *.fastq.gz\n",
    "\n",
    "multiqc .\n",
    "\n",
    "mkdir trimmed\n",
    "\n",
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters $RNA_REFS_DIR/illumina_multiplex.fa  --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads hcc1395_normal_rep1_r1.fastq.gz --reads2 hcc1395_normal_rep1_r2.fastq.gz --target trimmed/hcc1395_normal_rep1\n",
    "\n",
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters $RNA_REFS_DIR/illumina_multiplex.fa  --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads hcc1395_normal_rep2_r1.fastq.gz --reads2 hcc1395_normal_rep2_r2.fastq.gz --target trimmed/hcc1395_normal_rep2\n",
    "\n",
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters $RNA_REFS_DIR/illumina_multiplex.fa --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads hcc1395_normal_rep3_r1.fastq.gz --reads2 hcc1395_normal_rep3_r2.fastq.gz --target trimmed/hcc1395_normal_rep3\n",
    "\n",
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters $RNA_REFS_DIR/illumina_multiplex.fa --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads hcc1395_tumor_rep1_r1.fastq.gz --reads2 hcc1395_tumor_rep1_r2.fastq.gz --target trimmed/hcc1395_tumor_rep1\n",
    "\n",
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters $RNA_REFS_DIR/illumina_multiplex.fa --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads hcc1395_tumor_rep2_r1.fastq.gz --reads2 hcc1395_tumor_rep2_r2.fastq.gz --target trimmed/hcc1395_tumor_rep2\n",
    "\n",
    "flexbar --adapter-min-overlap 7 --adapter-trim-end RIGHT --adapters $RNA_REFS_DIR/illumina_multiplex.fa  --max-uncalled 300 --min-read-length 25 --threads 8 --zip-output GZ --reads hcc1395_tumor_rep3_r1.fastq.gz --reads2 hcc1395_tumor_rep3_r2.fastq.gz --target trimmed/hcc1395_tumor_rep3\n",
    "\n",
    "fastqc trimmed/*.fastq.gz\n",
    "\n",
    "multiqc trimmed/\n",
    "\n",
    "[nathalia@aws] $  screen -R fastqc \n",
    "\n",
    "[nathalia@aws] $  cd /home/ubuntu/workspace/rnaseq/practice3\n",
    "\n",
    "[nathalia@aws] $  bash qc1.sh\n",
    "\n",
    "[detached from 18496.fastqc]\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### **Questions**\n",
    "\n",
    "- After trimming, what is the range of read lengths observed for hcc1395 normal replicate 1, read 1?  \n",
    "\n",
    "118-151 (It didn't match to the solution...)\n",
    "\n",
    "\n",
    "- Which sections of the FastQC report are most informative for observing the effect of trimming?  \n",
    "\n",
    "'Basic Statistics', 'Sequence Length Distribution', and 'Adapter Content'\n",
    "\n",
    "\n",
    "- In the 'Per base sequence content section', what pattern do you see? What could explain this pattern?  \n",
    "\n",
    "5' UTR bias"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
