{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 11/11/2018\n",
    "\n",
    "### Variant calling practice SEQTEC2018\n",
    "\n",
    "NATHALIA GRAF-GRACHET\n",
    "\n",
    "http://iobio.io/cshl_practical"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# SETUP"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd workspace\n",
    "[nathalia@aws] $  mkdir cshl-practical\n",
    "[nathalia@aws] $  cd cshl-practical/\n",
    "[nathalia@aws] $  mkdir reference\n",
    "[nathalia@aws] $  mkdir analysis\n",
    "[nathalia@aws] $  REF=\"/home/ubuntu/workspace/data/fasta/GRCh37/Homo_sapiens.GRCh37.75.dna_sm.primary_assembly\"\n",
    "[nathalia@aws] $  DATA=\"/home/ubuntu/workspace/data/utah\"\n",
    "[nathalia@aws] $  cd reference/\n",
    "[nathalia@aws] $  cp ~/workspace/data/utah/chr20_fragment.fa .\n",
    "\n",
    "[nathalia@aws] $  echo $REF\n",
    "/home/ubuntu/workspace/data/fasta/GRCh37/Homo_sapiens.GRCh37.75.dna_sm.primary_assembly\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prep the reference sequence\n",
    "\n",
    "create an index\n",
    "\n",
    "https://github.com/lh3/bwa.git"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  bwa\n",
    "\n",
    "Program: bwa (alignment via Burrows-Wheeler transformation)\n",
    "Version: 0.7.17-r1194-dirty\n",
    "Contact: Heng Li <lh3@sanger.ac.uk>\n",
    "\n",
    "Usage:   bwa <command> [options]\n",
    "\n",
    "Command: index         index sequences in the FASTA format\n",
    "         mem           BWA-MEM algorithm\n",
    "         fastmap       identify super-maximal exact matches\n",
    "         pemerge       merge overlapping paired ends (EXPERIMENTAL)\n",
    "         aln           gapped/ungapped alignment\n",
    "         samse         generate alignment (single ended)\n",
    "         sampe         generate alignment (paired ended)\n",
    "         bwasw         BWA-SW for long queries\n",
    "\n",
    "         shm           manage indices in shared memory\n",
    "         fa2pac        convert FASTA to PAC format\n",
    "         pac2bwt       generate BWT from PAC\n",
    "         pac2bwtgen    alternative algorithm for generating BWT\n",
    "         bwtupdate     update .bwt to the new format\n",
    "         bwt2sa        generate SA from BWT and Occ\n",
    "\n",
    "Note: To use BWA, you need to first index the genome with `bwa index'.\n",
    "      There are three alignment algorithms in BWA: `mem', `bwasw', and\n",
    "      `aln/samse/sampe'. If you are not sure which to use, try `bwa mem'\n",
    "      first. Please `man ./bwa.1' for the manual.\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bwa index\n",
    "\n",
    "Usage:   bwa index [options] <in.fasta>\n",
    "\n",
    "Options: -a STR    BWT construction algorithm: bwtsw, is or rb2 [auto]\n",
    "         -p STR    prefix of the index [same as fasta name]\n",
    "         -b INT    block size for the bwtsw algorithm (effective with -a bwtsw) [10000000]\n",
    "         -6        index files named as <in.fasta>.64.* instead of <in.fasta>.* \n",
    "\n",
    "Warning: `-a bwtsw' does not work for short genomes, while `-a is' and\n",
    "         `-a div' do not work not for long genomes.\n",
    "\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  bwa index -p chr20_fragment chr20_fragment.fa\n",
    "[bwa_index] Pack FASTA... 0.00 sec\n",
    "[bwa_index] Construct BWT for the packed sequence...\n",
    "[bwa_index] 0.02 seconds elapse.\n",
    "[bwa_index] Update BWT... 0.00 sec\n",
    "[bwa_index] Pack forward-only FASTA... 0.00 sec\n",
    "[bwa_index] Construct SA from BWT and Occ... 0.01 sec\n",
    "[main] Version: 0.7.17-r1194-dirty\n",
    "[main] CMD: bwa index -p chr20_fragment chr20_fragment.fa\n",
    "[main] Real time: 0.039 sec; CPU: 0.027 sec\n",
    "[nathalia@aws] $  lhtr\n",
    "total 300K\n",
    "-rwxr-xr-- 1 ubuntu ubuntu 100K Nov 11 15:13 chr20_fragment.fa\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  98K Nov 11 15:17 chr20_fragment.bwt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 11 15:17 chr20_fragment.pac\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   76 Nov 11 15:17 chr20_fragment.ann\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   11 Nov 11 15:17 chr20_fragment.amb\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  49K Nov 11 15:17 chr20_fragment.sa\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  ls -l $DATA\n",
    "total 68188\n",
    "-rwxr-xr-- 1 ubuntu ubuntu 10694300 Nov  8 18:51 NA12878_1_sample.fastq\n",
    "-rwxr-xr-- 1 ubuntu ubuntu 10693786 Nov  8 18:51 NA12878_2_sample.fastq\n",
    "-rwxr-xr-- 1 ubuntu ubuntu 10772596 Nov  8 18:51 NA12891_1_sample.fastq\n",
    "-rwxr-xr-- 1 ubuntu ubuntu 10772200 Nov  8 18:51 NA12891_2_sample.fastq\n",
    "-rwxr-xr-- 1 ubuntu ubuntu 11323228 Nov  8 18:51 NA12892_1_sample.fastq\n",
    "-rwxr-xr-- 1 ubuntu ubuntu 11322974 Nov  8 18:51 NA12892_2_sample.fastq\n",
    "-rwxr-xr-- 1 ubuntu ubuntu   101739 Nov  8 18:51 chr20_fragment.fa\n",
    "-rwxr-xr-- 1 ubuntu ubuntu  4024499 Nov  8 18:51 sample.bam\n",
    "```\n",
    "\n",
    "NA is the most sequenced person in history. It's a she, and she is dead. There's also her parents sequence here in this folder."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Align reads"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd ../analysis/\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bwa mem\n",
    "\n",
    "Usage: bwa mem [options] <idxbase> <in1.fq> [in2.fq]\n",
    "\n",
    "Algorithm options:\n",
    "\n",
    "       -t INT        number of threads [1]\n",
    "       -k INT        minimum seed length [19]\n",
    "       -w INT        band width for banded alignment [100]\n",
    "       -d INT        off-diagonal X-dropoff [100]\n",
    "       -r FLOAT      look for internal seeds inside a seed longer than {-k} * FLOAT [1.5]\n",
    "       -y INT        seed occurrence for the 3rd round seeding [20]\n",
    "       -c INT        skip seeds with more than INT occurrences [500]\n",
    "       -D FLOAT      drop chains shorter than FLOAT fraction of the longest overlapping chain [0.50]\n",
    "       -W INT        discard a chain if seeded bases shorter than INT [0]\n",
    "       -m INT        perform at most INT rounds of mate rescues for each read [50]\n",
    "       -S            skip mate rescue\n",
    "       -P            skip pairing; mate rescue performed unless -S also in use\n",
    "\n",
    "Scoring options:\n",
    "\n",
    "       -A INT        score for a sequence match, which scales options -TdBOELU unless overridden [1]\n",
    "       -B INT        penalty for a mismatch [4]\n",
    "       -O INT[,INT]  gap open penalties for deletions and insertions [6,6]\n",
    "       -E INT[,INT]  gap extension penalty; a gap of size k cost '{-O} + {-E}*k' [1,1]\n",
    "       -L INT[,INT]  penalty for 5'- and 3'-end clipping [5,5]\n",
    "       -U INT        penalty for an unpaired read pair [17]\n",
    "\n",
    "       -x STR        read type. Setting -x changes multiple parameters unless overridden [null]\n",
    "                     pacbio: -k17 -W40 -r10 -A1 -B1 -O1 -E1 -L0  (PacBio reads to ref)\n",
    "                     ont2d: -k14 -W20 -r10 -A1 -B1 -O1 -E1 -L0  (Oxford Nanopore 2D-reads to ref)\n",
    "                     intractg: -B9 -O16 -L5  (intra-species contigs to ref)\n",
    "\n",
    "Input/output options:\n",
    "\n",
    "       -p            smart pairing (ignoring in2.fq)\n",
    "       -R STR        read group header line such as '@RG\\tID:foo\\tSM:bar' [null]\n",
    "       -H STR/FILE   insert STR to header if it starts with @; or insert lines in FILE [null]\n",
    "       -o FILE       sam file to output results to [stdout]\n",
    "       -j            treat ALT contigs as part of the primary assembly (i.e. ignore <idxbase>.alt file)\n",
    "       -5            for split alignment, take the alignment with the smallest coordinate as primary\n",
    "       -q            don't modify mapQ of supplementary alignments\n",
    "       -K INT        process INT input bases in each batch regardless of nThreads (for reproducibility) []\n",
    "\n",
    "       -v INT        verbosity level: 1=error, 2=warning, 3=message, 4+=debugging [3]\n",
    "       -T INT        minimum score to output [30]\n",
    "       -h INT[,INT]  if there are <INT hits with score >80% of the max score, output all in XA [5,200]\n",
    "       -a            output all alignments for SE or unpaired PE\n",
    "       -C            append FASTA/FASTQ comment to SAM output\n",
    "       -V            output the reference FASTA header in the XR tag\n",
    "       -Y            use soft clipping for supplementary alignments\n",
    "       -M            mark shorter split hits as secondary\n",
    "\n",
    "       -I FLOAT[,FLOAT[,INT[,INT]]]\n",
    "                     specify the mean, standard deviation (10% of the mean if absent), max\n",
    "                     (4 sigma from the mean if absent) and min of the insert size distribution.\n",
    "                     FR orientation only. [inferred]\n",
    "\n",
    "Note: Please read the man page for detailed description of the command line and options.\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  lscpu\n",
    "Architecture:        x86_64\n",
    "CPU op-mode(s):      32-bit, 64-bit\n",
    "Byte Order:          Little Endian\n",
    "CPU(s):              8\n",
    "On-line CPU(s) list: 0-7\n",
    "Thread(s) per core:  2\n",
    "Core(s) per socket:  4\n",
    "Socket(s):           1\n",
    "NUMA node(s):        1\n",
    "Vendor ID:           GenuineIntel\n",
    "CPU family:          6\n",
    "Model:               85\n",
    "Model name:          Intel(R) Xeon(R) Platinum 8175M CPU @ 2.50GHz\n",
    "Stepping:            4\n",
    "CPU MHz:             3128.053\n",
    "BogoMIPS:            5000.00\n",
    "Hypervisor vendor:   KVM\n",
    "Virtualization type: full\n",
    "L1d cache:           32K\n",
    "L1i cache:           32K\n",
    "L2 cache:            1024K\n",
    "L3 cache:            33792K\n",
    "NUMA node0 CPU(s):   0-7\n",
    "Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves ida arat pku ospke\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  bwa mem -t 8 -R \"@RG\\tID:NABOB\\tPL:ILLUMINA\\tSM:NA12878\" ../reference/chr20_fragment /home/ubuntu/workspace/data/utah//NA12878_1_sample.fastq /home/ubuntu/workspace/data/utah//NA12878_2_sample.fastq | samblaster | samtools view -b -S - | samtools sort -o output.bam -\n",
    "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
    "samblaster: Version 0.1.24\n",
    "samblaster: Inputting from stdin\n",
    "samblaster: Outputting to stdout\n",
    "[M::process] read 93400 sequences (9433143 bp)...\n",
    "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 26, 4, 8)\n",
    "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
    "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
    "[M::mem_pestat] (25, 50, 75) percentile: (93, 134, 2946)\n",
    "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 8652)\n",
    "[M::mem_pestat] mean and std.dev: (1139.08, 1689.00)\n",
    "[M::mem_pestat] low and high boundaries for proper pairs: (1, 11505)\n",
    "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
    "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
    "[M::mem_process_seqs] Processed 93400 reads in 18.184 CPU sec, 2.277 real sec\n",
    "samblaster: Loaded 1 header sequence entries.\n",
    "[main] Version: 0.7.17-r1194-dirty\n",
    "[main] CMD: bwa mem -t 8 -R @RG\\tID:NABOB\\tPL:ILLUMINA\\tSM:NA12878 ../reference/chr20_fragment /home/ubuntu/workspace/data/utah//NA12878_1_sample.fastq /home/ubuntu/workspace/data/utah//NA12878_2_sample.fastq\n",
    "[main] Real time: 3.184 sec; CPU: 18.297 sec\n",
    "samblaster: Marked 669 of 46700 (1.43%) read ids as duplicates using 3400k memory in 0.047S CPU seconds and 3S wall time.\n",
    "\n",
    "[nathalia@aws] $  samtools index output.bam \n",
    "\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 6.7M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 6.7M Nov 11 15:28 output.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  288 Nov 11 15:30 output.bam.bai\n",
    "\n",
    "```\n",
    "\n",
    "Yas it works... delete and repeat for 3 files with their respective names."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### tangent..\n",
    "\n",
    "```\n",
    "[nathalia@aws] $  sudo apt install tree\n",
    "Reading package lists... Done\n",
    "Building dependency tree       \n",
    "Reading state information... Done\n",
    "The following NEW packages will be installed:\n",
    "  tree\n",
    "0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.\n",
    "Need to get 40.7 kB of archives.\n",
    "After this operation, 105 kB of additional disk space will be used.\n",
    "Get:1 http://us-west-2.ec2.archive.ubuntu.com/ubuntu bionic/universe amd64 tree amd64 1.7.0-5 [40.7 kB]\n",
    "Fetched 40.7 kB in 0s (0 B/s)   \n",
    "Selecting previously unselected package tree.\n",
    "(Reading database ... 148761 files and directories currently installed.)\n",
    "Preparing to unpack .../tree_1.7.0-5_amd64.deb ...\n",
    "Unpacking tree (1.7.0-5) ...\n",
    "Setting up tree (1.7.0-5) ...\n",
    "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n",
    "[nathalia@aws] $  tree\n",
    ".\n",
    "├── analysis\n",
    "│   ├── output.bam\n",
    "│   └── output.bam.bai\n",
    "└── reference\n",
    "    ├── chr20_fragment.amb\n",
    "    ├── chr20_fragment.ann\n",
    "    ├── chr20_fragment.bwt\n",
    "    ├── chr20_fragment.fa\n",
    "    ├── chr20_fragment.pac\n",
    "    └── chr20_fragment.sa\n",
    "\n",
    "2 directories, 8 files\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### and back to it..."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  screen -S screen1\n",
    "\n",
    "#crtl+a and crtl+d = detaches\n",
    "#crtl+d = terminates\n",
    "[detached from 2639.pts-0.ip-172-31-39-185]\n",
    "\n",
    "\n",
    "[nathalia@aws] $  screen -list\n",
    "There are screens on:\n",
    "\t3240.screen1\t(11/11/18 16:47:00)\t(Detached)\n",
    "1 Sockets in /run/screen/S-ubuntu.\n",
    "\n",
    "\n",
    "[nathalia@aws] $  screen -r screen1 \n",
    "[detached from 2674.screen1]\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  rm -f output.bam*\n",
    "\n",
    "[nathalia@aws] $  screen -S screen1\n",
    "#need to set variables again in each screen...\n",
    "\n",
    "[nathalia@aws] $  REF=\"/home/ubuntu/workspace/data/fasta/GRCh37/Homo_sapiens.GRCh37.75.dna_sm.primary_assembly\"\n",
    "[nathalia@aws] $  DATA=\"/home/ubuntu/workspace/data/utah/\"\n",
    "\n",
    "[nathalia@aws] $  bwa mem -t 8 -R \"@RG\\tID:NA12878\\tPL:ILLUMINA\\tSM:NA12878\" $REF $DATA/NA12878_1_sample.fastq  $DATA/NA12878_2_sample.fastq |samblaster |samtools view -b -S - | samtools sort -o NA12878.bam -\n",
    "samblaster: Version 0.1.24\n",
    "samblaster: Inputting from stdin\n",
    "samblaster: Outputting to stdout\n",
    "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
    "[M::process] read 93400 sequences (9433143 bp)...\n",
    "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (11, 44457, 0, 7)\n",
    "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
    "[M::mem_pestat] (25, 50, 75) percentile: (66, 147, 235)\n",
    "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 573)\n",
    "[M::mem_pestat] mean and std.dev: (135.73, 79.80)\n",
    "[M::mem_pestat] low and high boundaries for proper pairs: (1, 742)\n",
    "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
    "[M::mem_pestat] (25, 50, 75) percentile: (278, 323, 368)\n",
    "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (98, 548)\n",
    "[M::mem_pestat] mean and std.dev: (322.32, 71.71)\n",
    "[M::mem_pestat] low and high boundaries for proper pairs: (8, 638)\n",
    "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
    "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
    "[M::mem_pestat] skip orientation FF\n",
    "[M::mem_process_seqs] Processed 93400 reads in 14.201 CPU sec, 1.781 real sec\n",
    "samblaster: Loaded 84 header sequence entries.\n",
    "[main] Version: 0.7.17-r1194-dirty\n",
    "[main] CMD: bwa mem -tsamblaster: Marked 148 of 46700 (0.32%) read ids as duplicates 8 -R using 4196k memory in  @RG\\tID:NA12878\\tPL:ILLUMINA\\tSM:NA128780.059S /home/ubuntu/workspace/data/fasta/GRCh37/Homo_sapiens.GRCh37.75.dna_sm.primary_assembly CPU seconds and  /home/ubuntu/workspace/data/utah//NA12878_1_sample.fastq5S /home/ubuntu/workspace/data/utah//NA12878_2_sample.fastq wall time.\n",
    "\n",
    "[main] Real time: 5.412 sec; CPU: 16.895 sec\n",
    "\n",
    "#crtl+a+d\n",
    "\n",
    "[detached from 3240.screen1]\n",
    "\n",
    "\n",
    "[nathalia@aws] $  screen -r screen2\n",
    "\n",
    "[nathalia@aws] $  REF=\"/home/ubuntu/workspace/data/fasta/GRCh37/Homo_sapiens.GRCh37.75.dna_sm.primary_assembly\"\n",
    "[nathalia@aws] $  DATA=\"/home/ubuntu/workspace/data/utah/\"\n",
    "\n",
    "[nathalia@aws] $  bwa mem -t 8 -R \"@RG\\tID:NA12891\\tPL:ILLUMINA\\tSM:NA12891\" $REF $DATA/NA12891_1_sample.fastq $DATA/NA12891_2_sample.fastq | samblaster |samtools view -b -S - | samtools sort -o NA12891.bam -\n",
    "samblaster: Version 0.1.24\n",
    "samblaster: Inputting from stdin\n",
    "samblaster: Outputting to stdout\n",
    "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
    "[M::process] read 94086 sequences (9502240 bp)...\n",
    "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 44764, 0, 0)\n",
    "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
    "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
    "[M::mem_pestat] (25, 50, 75) percentile: (264, 302, 343)\n",
    "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (106, 501)\n",
    "[M::mem_pestat] mean and std.dev: (303.38, 61.30)\n",
    "[M::mem_pestat] low and high boundaries for proper pairs: (27, 580)\n",
    "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
    "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
    "[M::mem_process_seqs] Processed 94086 reads in 14.442 CPU sec, 1.812 real sec\n",
    "samblaster: Loaded 84 header sequence entries.\n",
    "[main] Version: 0.7.17-r1194-dirty\n",
    "samblaster: Marked 412 of 47043 (0.88%) read ids as duplicates[main] CMD: bwa mem -t 8 using 4228k memory in  -R @RG\\tID:NA12891\\tPL:ILLUMINA\\tSM:NA128910.055S /home/ubuntu/workspace/data/fasta/GRCh37/Homo_sapiens.GRCh37.75.dna_sm.primary_assembly CPU seconds and  /home/ubuntu/workspace/data/utah//NA12891_1_sample.fastq6S /home/ubuntu/workspace/data/utah//NA12891_2_sample.fastq wall time.\n",
    "\n",
    "[main] Real time: 5.670 sec; CPU: 17.151 sec\n",
    "\n",
    "\n",
    "[detached from 2736.screen2]\n",
    "\n",
    "\n",
    "[nathalia@aws] $  screen -r screen3 \n",
    "\n",
    "\n",
    "[nathalia@aws] $  REF=\"/home/ubuntu/workspace/data/fasta/GRCh37/Homo_sapiens.GRCh37.75.dna_sm.primary_assembly\"\n",
    "[nathalia@aws] $  DATA=\"/home/ubuntu/workspace/data/utah/\"\n",
    "\n",
    "[nathalia@aws] $  bwa mem -t 8 -R \"@RG\\tID:NA12892\\tPL:ILLUMINA\\tSM:NA12892\" $REF $DATA/NA12892_1_sample.fastq  $DATA/NA12892_2_sample.fastq | samblaster | samtools view -b -S - | samtools sort -o NA12892.bam -\n",
    "samblaster: Version 0.1.24\n",
    "samblaster: Inputting from stdin\n",
    "samblaster: Outputting to stdout\n",
    "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
    "[M::process] read 98894 sequences (9988033 bp)...\n",
    "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 47343, 0, 1)\n",
    "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
    "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
    "[M::mem_pestat] (25, 50, 75) percentile: (267, 306, 348)\n",
    "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (105, 510)\n",
    "[M::mem_pestat] mean and std.dev: (308.26, 62.21)\n",
    "[M::mem_pestat] low and high boundaries for proper pairs: (24, 591)\n",
    "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
    "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
    "[M::mem_process_seqs] Processed 98894 reads in 13.476 CPU sec, 1.691 real sec\n",
    "samblaster: Loaded 84 header sequence entries.\n",
    "[main] Version: 0.7.17-r1194-dirty\n",
    "samblaster: Marked 243 of 49447 (0.49%) read ids as duplicates[main] CMD: bwa mem using 4224k memory in  -t 80.071S -R CPU seconds and  @RG\\tID:NA12892\\tPL:ILLUMINA\\tSM:NA128926S /home/ubuntu/workspace/data/fasta/GRCh37/Homo_sapiens.GRCh37.75.dna_sm.primary_assembly wall time.\n",
    " /home/ubuntu/workspace/data/utah//NA12892_1_sample.fastq /home/ubuntu/workspace/data/utah//NA12892_2_sample.fastq\n",
    "[main] Real time: 5.585 sec; CPU: 16.179 sec\n",
    "\n",
    "[detached from 2749.screen3]\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  lhtr\n",
    "total 21M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 7.4M Nov 11 16:50 NA12878.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 6.5M Nov 11 16:53 NA12891.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 6.9M Nov 11 16:55 NA12892.bam\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "`\"@RG\\tID:NA12891\\tPL:ILLUMINA\\tSM:NA12891\"`  \n",
    "`RG` is read group; which is not necessary equal to your sample id; read group is where the sequences come from  \n",
    "`PL` is platform of sequence  \n",
    "`SM` is sample\n",
    "\n",
    "In the line, I typed the following:\n",
    "\n",
    "`bwa mem -t 8 -R \"@RG\\tID:NA12878\\tPL:ILLUMINA\\tSM:NA12878\" $REF $DATA/NA12878_1_sample.fastq $DATA/NA12878_2_sample.fastq |\n",
    "samblaster |\n",
    "samtools view -b -S - |\n",
    "samtools sort -o NA12878.bam -`\n",
    "\n",
    "the `$REF` and `$DATA` get expanded; that's why in the code here it's written all out.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  for file in *.bam; do samtools index $file; done\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 24M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  7.4M Nov 11 16:50 NA12878.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.5M Nov 11 16:53 NA12891.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.9M Nov 11 16:55 NA12892.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1007K Nov 11 16:56 NA12878.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  803K Nov 11 16:56 NA12891.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  1.1M Nov 11 16:56 NA12892.bam.bai\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alignment stats"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  for file in *.bam; do output=${file/bam/stats}; samtools flagstat $file > $output; done\n",
    "[nathalia@aws] $  lhtr\n",
    "total 24M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  7.4M Nov 11 16:50 NA12878.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.5M Nov 11 16:53 NA12891.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.9M Nov 11 16:55 NA12892.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1007K Nov 11 16:56 NA12878.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  803K Nov 11 16:56 NA12891.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  1.1M Nov 11 16:56 NA12892.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12878.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12891.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12892.stats\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  for file in *.stats; do cat $file; echo '';done\n",
    "93479 + 0 in total (QC-passed reads + QC-failed reads)\n",
    "0 + 0 secondary\n",
    "79 + 0 supplementary\n",
    "296 + 0 duplicates\n",
    "93150 + 0 mapped (99.65% : N/A)\n",
    "93400 + 0 paired in sequencing\n",
    "46700 + 0 read1\n",
    "46700 + 0 read2\n",
    "92358 + 0 properly paired (98.88% : N/A)\n",
    "92742 + 0 with itself and mate mapped\n",
    "329 + 0 singletons (0.35% : N/A)\n",
    "4 + 0 with mate mapped to a different chr\n",
    "1 + 0 with mate mapped to a different chr (mapQ>=5)\n",
    "\n",
    "94151 + 0 in total (QC-passed reads + QC-failed reads)\n",
    "0 + 0 secondary\n",
    "65 + 0 supplementary\n",
    "824 + 0 duplicates\n",
    "93970 + 0 mapped (99.81% : N/A)\n",
    "94086 + 0 paired in sequencing\n",
    "47043 + 0 read1\n",
    "47043 + 0 read2\n",
    "92728 + 0 properly paired (98.56% : N/A)\n",
    "93728 + 0 with itself and mate mapped\n",
    "177 + 0 singletons (0.19% : N/A)\n",
    "4 + 0 with mate mapped to a different chr\n",
    "2 + 0 with mate mapped to a different chr (mapQ>=5)\n",
    "\n",
    "98961 + 0 in total (QC-passed reads + QC-failed reads)\n",
    "0 + 0 secondary\n",
    "67 + 0 supplementary\n",
    "486 + 0 duplicates\n",
    "98795 + 0 mapped (99.83% : N/A)\n",
    "98894 + 0 paired in sequencing\n",
    "49447 + 0 read1\n",
    "49447 + 0 read2\n",
    "97956 + 0 properly paired (99.05% : N/A)\n",
    "98564 + 0 with itself and mate mapped\n",
    "164 + 0 singletons (0.17% : N/A)\n",
    "2 + 0 with mate mapped to a different chr\n",
    "1 + 0 with mate mapped to a different chr (mapQ>=5)\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Checking the BAM files\n",
    "\n",
    "http://bam.iobio.io\n",
    "\n",
    "Choose BAM URL > Add this URL: https://s3.amazonaws.com/iobio/1000gSV/NA19238.mapped.ILLUMINA.bwa.YRI.exome.20130415.bam > Hit Go."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![iobio.bam visualization](bam_iobio_visualiz.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Call variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  freebayes -h\n",
    "usage: freebayes [OPTION] ... [BAM FILE] ... \n",
    "\n",
    "Bayesian haplotype-based polymorphism discovery.\n",
    "\n",
    "citation: Erik Garrison, Gabor Marth\n",
    "          \"Haplotype-based variant detection from short-read sequencing\"\n",
    "          arXiv:1207.3907 (http://arxiv.org/abs/1207.3907)\n",
    "\n",
    "overview:\n",
    "\n",
    "    To call variants from aligned short-read sequencing data, supply BAM files and\n",
    "    a reference.  FreeBayes will provide VCF output on standard out describing SNPs,\n",
    "    indels, and complex variants in samples in the input alignments.\n",
    "\n",
    "    By default, FreeBayes will consider variants supported by at least 2\n",
    "    observations in a single sample (-C) and also by at least 20% of the reads from\n",
    "    a single sample (-F).  These settings are suitable to low to high depth\n",
    "    sequencing in haploid and diploid samples, but users working with polyploid or\n",
    "    pooled samples may wish to adjust them depending on the characteristics of\n",
    "    their sequencing data.\n",
    "\n",
    "    FreeBayes is capable of calling variant haplotypes shorter than a read length\n",
    "    where multiple polymorphisms segregate on the same read.  The maximum distance\n",
    "    between polymorphisms phased in this way is determined by the\n",
    "    --max-complex-gap, which defaults to 3bp.  In practice, this can comfortably be\n",
    "    set to half the read length.\n",
    "\n",
    "    Ploidy may be set to any level (-p), but by default all samples are assumed to\n",
    "    be diploid.  FreeBayes can model per-sample and per-region variation in\n",
    "    copy-number (-A) using a copy-number variation map.\n",
    "\n",
    "    FreeBayes can act as a frequency-based pooled caller and describe variants\n",
    "    and haplotypes in terms of observation frequency rather than called genotypes.\n",
    "    To do so, use --pooled-continuous and set input filters to a suitable level.\n",
    "    Allele observation counts will be described by AO and RO fields in the VCF output.\n",
    "\n",
    "\n",
    "examples:\n",
    "\n",
    "    # call variants assuming a diploid sample\n",
    "    freebayes -f ref.fa aln.bam >var.vcf\n",
    "\n",
    "    # call variants assuming a diploid sample, providing gVCF output\n",
    "    freebayes -f ref.fa --gvcf aln.bam >var.gvcf\n",
    "\n",
    "    # require at least 5 supporting observations to consider a variant\n",
    "    freebayes -f ref.fa -C 5 aln.bam >var.vcf\n",
    "\n",
    "    # use a different ploidy\n",
    "    freebayes -f ref.fa -p 4 aln.bam >var.vcf\n",
    "\n",
    "    # assume a pooled sample with a known number of genome copies\n",
    "    freebayes -f ref.fa -p 20 --pooled-discrete aln.bam >var.vcf\n",
    "\n",
    "    # generate frequency-based calls for all variants passing input thresholds\n",
    "    freebayes -f ref.fa -F 0.01 -C 1 --pooled-continuous aln.bam >var.vcf\n",
    "\n",
    "    # use an input VCF (bgzipped + tabix indexed) to force calls at particular alleles\n",
    "    freebayes -f ref.fa -@ in.vcf.gz aln.bam >var.vcf\n",
    "\n",
    "    # generate long haplotype calls over known variants\n",
    "    freebayes -f ref.fa --haplotype-basis-alleles in.vcf.gz \\ \n",
    "                        --haplotype-length 50 aln.bam\n",
    "\n",
    "    # naive variant calling: simply annotate observation counts of SNPs and indels\n",
    "    freebayes -f ref.fa --haplotype-length 0 --min-alternate-count 1 \\ \n",
    "        --min-alternate-fraction 0 --pooled-continuous --report-monomorphic >var.vcf\n",
    "\n",
    "\n",
    "parameters:\n",
    "\n",
    "   -h --help       Prints this help dialog.\n",
    "   --version       Prints the release number and the git commit id.\n",
    "\n",
    "input:\n",
    "\n",
    "   -b --bam FILE   Add FILE to the set of BAM files to be analyzed.\n",
    "   -L --bam-list FILE\n",
    "                   A file containing a list of BAM files to be analyzed.\n",
    "   -c --stdin      Read BAM input on stdin.\n",
    "   -f --fasta-reference FILE\n",
    "                   Use FILE as the reference sequence for analysis.\n",
    "                   An index file (FILE.fai) will be created if none exists.\n",
    "                   If neither --targets nor --region are specified, FreeBayes\n",
    "                   will analyze every position in this reference.\n",
    "   -t --targets FILE\n",
    "                   Limit analysis to targets listed in the BED-format FILE.\n",
    "   -r --region <chrom>:<start_position>-<end_position>\n",
    "                   Limit analysis to the specified region, 0-base coordinates,\n",
    "                   end_position not included (same as BED format).\n",
    "                   Either '-' or '..' maybe used as a separator.\n",
    "   -s --samples FILE\n",
    "                   Limit analysis to samples listed (one per line) in the FILE.\n",
    "                   By default FreeBayes will analyze all samples in its input\n",
    "                   BAM files.\n",
    "   --populations FILE\n",
    "                   Each line of FILE should list a sample and a population which\n",
    "                   it is part of.  The population-based bayesian inference model\n",
    "                   will then be partitioned on the basis of the populations.\n",
    "   -A --cnv-map FILE\n",
    "                   Read a copy number map from the BED file FILE, which has\n",
    "                   either a sample-level ploidy:\n",
    "                      sample name, copy number\n",
    "                   or a region-specific format:\n",
    "                      reference sequence, start, end, sample name, copy number\n",
    "                   ... for each region in each sample which does not have the\n",
    "                   default copy number as set by --ploidy.\n",
    "\n",
    "output:\n",
    "\n",
    "   -v --vcf FILE   Output VCF-format results to FILE. (default: stdout)\n",
    "   --gvcf\n",
    "                   Write gVCF output, which indicates coverage in uncalled regions.\n",
    "   --gvcf-chunk NUM\n",
    "                   When writing gVCF output emit a record for every NUM bases.\n",
    "   -@ --variant-input VCF\n",
    "                   Use variants reported in VCF file as input to the algorithm.\n",
    "                   Variants in this file will included in the output even if\n",
    "                   there is not enough support in the data to pass input filters.\n",
    "   -l --only-use-input-alleles\n",
    "                   Only provide variant calls and genotype likelihoods for sites\n",
    "                   and alleles which are provided in the VCF input, and provide\n",
    "                   output in the VCF for all input alleles, not just those which\n",
    "                   have support in the data.\n",
    "   --haplotype-basis-alleles VCF\n",
    "                   When specified, only variant alleles provided in this input\n",
    "                   VCF will be used for the construction of complex or haplotype\n",
    "                   alleles.\n",
    "   --report-all-haplotype-alleles\n",
    "                   At sites where genotypes are made over haplotype alleles,\n",
    "                   provide information about all alleles in output, not only\n",
    "                   those which are called.\n",
    "   --report-monomorphic\n",
    "                   Report even loci which appear to be monomorphic, and report all\n",
    "                   considered alleles, even those which are not in called genotypes.\n",
    "                   Loci which do not have any potential alternates have '.' for ALT.\n",
    "   -P --pvar N     Report sites if the probability that there is a polymorphism\n",
    "                   at the site is greater than N.  default: 0.0.  Note that post-\n",
    "                   filtering is generally recommended over the use of this parameter.\n",
    "   --strict-vcf\n",
    "                   Generate strict VCF format (FORMAT/GQ will be an int)\n",
    "\n",
    "population model:\n",
    "\n",
    "   -T --theta N    The expected mutation rate or pairwise nucleotide diversity\n",
    "                   among the population under analysis.  This serves as the\n",
    "                   single parameter to the Ewens Sampling Formula prior model\n",
    "                   default: 0.001\n",
    "   -p --ploidy N   Sets the default ploidy for the analysis to N.  default: 2\n",
    "   -J --pooled-discrete\n",
    "                   Assume that samples result from pooled sequencing.\n",
    "                   Model pooled samples using discrete genotypes across pools.\n",
    "                   When using this flag, set --ploidy to the number of\n",
    "                   alleles in each sample or use the --cnv-map to define\n",
    "                   per-sample ploidy.\n",
    "   -K --pooled-continuous\n",
    "                   Output all alleles which pass input filters, regardles of\n",
    "                   genotyping outcome or model.\n",
    "\n",
    "reference allele:\n",
    "\n",
    "   -Z --use-reference-allele\n",
    "                   This flag includes the reference allele in the analysis as\n",
    "                   if it is another sample from the same population.\n",
    "   --reference-quality MQ,BQ\n",
    "                   Assign mapping quality of MQ to the reference allele at each\n",
    "                   site and base quality of BQ.  default: 100,60\n",
    "\n",
    "allele scope:\n",
    "\n",
    "   -n --use-best-n-alleles N\n",
    "                   Evaluate only the best N SNP alleles, ranked by sum of\n",
    "                   supporting quality scores.  (Set to 0 to use all; default: all)\n",
    "   -E --max-complex-gap N\n",
    "      --haplotype-length N\n",
    "                   Allow haplotype calls with contiguous embedded matches of up\n",
    "                   to this length. Set N=-1 to disable clumping. (default: 3)\n",
    "   --min-repeat-size N\n",
    "                   When assembling observations across repeats, require the total repeat\n",
    "                   length at least this many bp.  (default: 5)\n",
    "   --min-repeat-entropy N\n",
    "                   To detect interrupted repeats, build across sequence until it has\n",
    "                   entropy > N bits per bp. Set to 0 to turn off. (default: 1)\n",
    "   --no-partial-observations\n",
    "                   Exclude observations which do not fully span the dynamically-determined\n",
    "                   detection window.  (default, use all observations, dividing partial\n",
    "                   support across matching haplotypes when generating haplotypes.)\n",
    "\n",
    "  These flags are meant for testing.\n",
    "  They are not meant for filtering the output.\n",
    "  They actually filter the input to the algorithm by throwing away alignments.\n",
    "  This hurts performance by hiding information from the Bayesian model.\n",
    "  Do not use them unless you can validate that they improve results!\n",
    "\n",
    "   -I --throw-away-snp-obs     Remove SNP observations from input.\n",
    "   -i --throw-away-indels-obs  Remove indel observations from input.\n",
    "   -X --throw-away-mnp-obs     Remove MNP observations from input.\n",
    "   -u --throw-away-complex-obs Remove complex allele observations from input.\n",
    "\n",
    "  If you need to break apart haplotype calls to obtain one class of alleles,\n",
    "  run the call with default parameters, then normalize and subset the VCF:\n",
    "    freebayes ... | vcfallelicprimitives -kg >calls.vcf\n",
    "  For example, this would retain only biallelic SNPs.\n",
    "    <calls.vcf vcfsnps | vcfbiallelic >biallelic_snp_calls.vcf\n",
    "\n",
    "indel realignment:\n",
    "\n",
    "   -O --dont-left-align-indels\n",
    "                   Turn off left-alignment of indels, which is enabled by default.\n",
    "\n",
    "input filters:\n",
    "\n",
    "   -4 --use-duplicate-reads\n",
    "                   Include duplicate-marked alignments in the analysis.\n",
    "                   default: exclude duplicates marked as such in alignments\n",
    "   -m --min-mapping-quality Q\n",
    "                   Exclude alignments from analysis if they have a mapping\n",
    "                   quality less than Q.  default: 1\n",
    "   -q --min-base-quality Q\n",
    "                   Exclude alleles from analysis if their supporting base\n",
    "                   quality is less than Q.  default: 0\n",
    "   -R --min-supporting-allele-qsum Q\n",
    "                   Consider any allele in which the sum of qualities of supporting\n",
    "                   observations is at least Q.  default: 0\n",
    "   -Y --min-supporting-mapping-qsum Q\n",
    "                   Consider any allele in which and the sum of mapping qualities of\n",
    "                   supporting reads is at least Q.  default: 0\n",
    "   -Q --mismatch-base-quality-threshold Q\n",
    "                   Count mismatches toward --read-mismatch-limit if the base\n",
    "                   quality of the mismatch is >= Q.  default: 10\n",
    "   -U --read-mismatch-limit N\n",
    "                   Exclude reads with more than N mismatches where each mismatch\n",
    "                   has base quality >= mismatch-base-quality-threshold.\n",
    "                   default: ~unbounded\n",
    "   -z --read-max-mismatch-fraction N\n",
    "                   Exclude reads with more than N [0,1] fraction of mismatches where\n",
    "                   each mismatch has base quality >= mismatch-base-quality-threshold\n",
    "                   default: 1.0\n",
    "   -$ --read-snp-limit N\n",
    "                   Exclude reads with more than N base mismatches, ignoring gaps\n",
    "                   with quality >= mismatch-base-quality-threshold.\n",
    "                   default: ~unbounded\n",
    "   -e --read-indel-limit N\n",
    "                   Exclude reads with more than N separate gaps.\n",
    "                   default: ~unbounded\n",
    "   -0 --standard-filters  Use stringent input base and mapping quality filters\n",
    "                   Equivalent to -m 30 -q 20 -R 0 -S 0\n",
    "   -F --min-alternate-fraction N\n",
    "                   Require at least this fraction of observations supporting\n",
    "                   an alternate allele within a single individual in the\n",
    "                   in order to evaluate the position.  default: 0.05\n",
    "   -C --min-alternate-count N\n",
    "                   Require at least this count of observations supporting\n",
    "                   an alternate allele within a single individual in order\n",
    "                   to evaluate the position.  default: 2\n",
    "   -3 --min-alternate-qsum N\n",
    "                   Require at least this sum of quality of observations supporting\n",
    "                   an alternate allele within a single individual in order\n",
    "                   to evaluate the position.  default: 0\n",
    "   -G --min-alternate-total N\n",
    "                   Require at least this count of observations supporting\n",
    "                   an alternate allele within the total population in order\n",
    "                   to use the allele in analysis.  default: 1\n",
    "   --min-coverage N\n",
    "                   Require at least this coverage to process a site. default: 0\n",
    "   --max-coverage N\n",
    "                   Do not process sites with greater than this coverage. default: no limit\n",
    "\n",
    "population priors:\n",
    "\n",
    "   -k --no-population-priors\n",
    "                   Equivalent to --pooled-discrete --hwe-priors-off and removal of\n",
    "                   Ewens Sampling Formula component of priors.\n",
    "\n",
    "mappability priors:\n",
    "\n",
    "   -w --hwe-priors-off\n",
    "                   Disable estimation of the probability of the combination\n",
    "                   arising under HWE given the allele frequency as estimated\n",
    "                   by observation frequency.\n",
    "   -V --binomial-obs-priors-off\n",
    "                   Disable incorporation of prior expectations about observations.\n",
    "                   Uses read placement probability, strand balance probability,\n",
    "                   and read position (5'-3') probability.\n",
    "   -a --allele-balance-priors-off\n",
    "                   Disable use of aggregate probability of observation balance between alleles\n",
    "                   as a component of the priors.\n",
    "\n",
    "genotype likelihoods:\n",
    "\n",
    "   --observation-bias FILE\n",
    "                   Read length-dependent allele observation biases from FILE.\n",
    "                   The format is [length] [alignment efficiency relative to reference]\n",
    "                   where the efficiency is 1 if there is no relative observation bias.\n",
    "   --base-quality-cap Q\n",
    "                   Limit estimated observation quality by capping base quality at Q.\n",
    "   --prob-contamination F\n",
    "                   An estimate of contamination to use for all samples.  default: 10e-9\n",
    "   --legacy-gls    Use legacy (polybayes equivalent) genotype likelihood calculations\n",
    "   --contamination-estimates FILE\n",
    "                   A file containing per-sample estimates of contamination, such as\n",
    "                   those generated by VerifyBamID.  The format should be:\n",
    "                       sample p(read=R|genotype=AR) p(read=A|genotype=AA)\n",
    "                   Sample '*' can be used to set default contamination estimates.\n",
    "\n",
    "algorithmic features:\n",
    "\n",
    "   --report-genotype-likelihood-max\n",
    "                   Report genotypes using the maximum-likelihood estimate provided\n",
    "                   from genotype likelihoods.\n",
    "   -B --genotyping-max-iterations N\n",
    "                   Iterate no more than N times during genotyping step. default: 1000.\n",
    "   --genotyping-max-banddepth N\n",
    "                   Integrate no deeper than the Nth best genotype by likelihood when\n",
    "                   genotyping. default: 6.\n",
    "   -W --posterior-integration-limits N,M\n",
    "                   Integrate all genotype combinations in our posterior space\n",
    "                   which include no more than N samples with their Mth best\n",
    "                   data likelihood. default: 1,3.\n",
    "   -N --exclude-unobserved-genotypes\n",
    "                   Skip sample genotypings for which the sample has no supporting reads.\n",
    "   -S --genotype-variant-threshold N\n",
    "                   Limit posterior integration to samples where the second-best\n",
    "                   genotype likelihood is no more than log(N) from the highest\n",
    "                   genotype likelihood for the sample.  default: ~unbounded\n",
    "   -j --use-mapping-quality\n",
    "                   Use mapping quality of alleles when calculating data likelihoods.\n",
    "   -H --harmonic-indel-quality\n",
    "                   Use a weighted sum of base qualities around an indel, scaled by the\n",
    "                   distance from the indel.  By default use a minimum BQ in flanking sequence.\n",
    "   -D --read-dependence-factor N\n",
    "                   Incorporate non-independence of reads by scaling successive\n",
    "                   observations by this factor during data likelihood\n",
    "                   calculations.  default: 0.9\n",
    "   -= --genotype-qualities\n",
    "                   Calculate the marginal probability of genotypes and report as GQ in\n",
    "                   each sample field in the VCF output.\n",
    "\n",
    "debugging:\n",
    "\n",
    "   -d --debug      Print debugging output.\n",
    "   -dd             Print more verbose debugging output (requires \"make DEBUG\")\n",
    "\n",
    "\n",
    "author:   Erik Garrison <erik.garrison@bc.edu>, Marth Lab, Boston College, 2010-2014\n",
    "version:  v1.2.0-2-g29c4002\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Poor man multithreading**\n",
    "```\n",
    "[nathalia@aws] $  screen -r screen1\n",
    "\n",
    "[nathalia@aws] $  freebayes -f $REF.fa -b NA12878.bam -b NA12891.bam -b NA12892.bam -r 17 -v chr17.vcf\n",
    "\n",
    "# took approximately 2-4 mins\n",
    "\n",
    "[detached from 3240.screen1]\n",
    "\n",
    "[nathalia@aws] $  screen -r screen2\n",
    "\n",
    "[nathalia@aws] $  freebayes -f $REF.fa -b NA12878.bam -b NA12891.bam -b NA12892.bam -r 20 -v chr20.vcf\n",
    "\n",
    "[detached from 2736.screen2]\n",
    "\n",
    "[nathalia@aws] $  screen -r screen3\n",
    "\n",
    "[nathalia@aws] $  freebayes -f $REF.fa -b NA12878.bam -b NA12891.bam -b NA12892.bam -r 21 -v chr21.vcf\n",
    "\n",
    "[detached from 2749.screen3]\n",
    "\n",
    "[nathalia@aws] $  screen -r screen4\n",
    "\n",
    "[nathalia@aws] $  REF=\"/home/ubuntu/workspace/data/fasta/GRCh37/Homo_sapiens.GRCh37.75.dna_sm.primary_assembly\"\n",
    "[nathalia@aws] $  DATA=\"/home/ubuntu/workspace/data/utah/\"\n",
    "[nathalia@aws] $  freebayes -f $REF.fa -b NA12878.bam -b NA12891.bam -b NA12892.bam -r 22 -v chr22.vcf\n",
    "\n",
    "[detached from 3495.screen4]\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Could have done with `nohup`\n",
    "\n",
    "```\n",
    "$ nohup freebayes -f $REF.fa -b NA12878.bam -b NA12891.bam -b NA12892.bam -r 17 -v chr17.vcf & \n",
    "$ nohup freebayes -f $REF.fa -b NA12878.bam -b NA12891.bam -b NA12892.bam -r 20 -v chr20.vcf &\n",
    "$ nohup freebayes -f $REF.fa -b NA12878.bam -b NA12891.bam -b NA12892.bam -r 21 -v chr21.vcf &\n",
    "$ nodup freebayes -f $REF.fa -b NA12878.bam -b NA12891.bam -b NA12892.bam -r 22 -v chr22.vcf &\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  lhtr\n",
    "total 25M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  7.4M Nov 11 16:50 NA12878.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.5M Nov 11 16:53 NA12891.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.9M Nov 11 16:55 NA12892.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1007K Nov 11 16:56 NA12878.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  803K Nov 11 16:56 NA12891.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  1.1M Nov 11 16:56 NA12892.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12878.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12891.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12892.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  900K Nov 11 17:57 chr17.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   81K Nov 11 17:57 chr20.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  196K Nov 11 17:58 chr21.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  216K Nov 11 17:58 chr22.vcf\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  bcftools concat -o all.vcf chr17.vcf chr20.vcf chr21.vcf chr22.vcf\n",
    "[nathalia@aws] $  bgzip all.vcf\n",
    "[nathalia@aws] $  bcftools index all.vcf.gz\n",
    "[nathalia@aws] $  lhtr\n",
    "total 26M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  7.4M Nov 11 16:50 NA12878.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.5M Nov 11 16:53 NA12891.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.9M Nov 11 16:55 NA12892.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1007K Nov 11 16:56 NA12878.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  803K Nov 11 16:56 NA12891.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  1.1M Nov 11 16:56 NA12892.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12878.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12891.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12892.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  900K Nov 11 17:57 chr17.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   81K Nov 11 17:57 chr20.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  196K Nov 11 17:58 chr21.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  216K Nov 11 17:58 chr22.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  347K Nov 11 18:04 all.vcf.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   352 Nov 11 18:04 all.vcf.gz.csi\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Check VCF files\n",
    "\n",
    "https://vcf.iobio.io/?species=Human\n",
    "\n",
    "Hit Choose VCF URL > Check that Human and GRCh37 are selected > Hit Load"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![iobio.bam visualization](vcf_iobio_visualiz.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "#local machine\n",
    "\n",
    "[nathalia@cshl] $  cd Desktop/nathalia_seqtec18/11-11-2018/\n",
    "\n",
    "[nathalia@cshl] $  ls\n",
    "AaronsClass-3-30-17.pdf\t\t\tbam_iobio_visualiz.png\n",
    "CSHL-Seq-2017-Var-discovery.pdf\t\tvariant_calling_practice_andrew.ipynb\n",
    "aaron_class_2017.pdf\t\t\tvcf_iobio_visualiz.png\n",
    "\n",
    "[nathalia@cshl] $  scp -i ~/cshl/cshl_2018_student.pem ubuntu@52.26.253.73:/home/ubuntu/workspace/cshl-practical/analysis/all.vcf.gz .\n",
    "all.vcf.gz                                    100%  346KB 720.1KB/s   00:00    \n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Trying to load in the website...\n",
    "\n",
    "Message: `must select 2 files, both a .vcf.gz and .vcf.gz.tbi file`\n",
    "\n",
    "I don't have any `.tbi` file...\n",
    "```\n",
    "[nathalia@aws] $  lhtr\n",
    "total 26M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  7.4M Nov 11 16:50 NA12878.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.5M Nov 11 16:53 NA12891.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.9M Nov 11 16:55 NA12892.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1007K Nov 11 16:56 NA12878.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  803K Nov 11 16:56 NA12891.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  1.1M Nov 11 16:56 NA12892.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12878.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12891.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12892.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  900K Nov 11 17:57 chr17.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   81K Nov 11 17:57 chr20.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  196K Nov 11 17:58 chr21.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  216K Nov 11 17:58 chr22.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  347K Nov 11 18:04 all.vcf.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   352 Nov 11 18:04 all.vcf.gz.csi\n",
    "\n",
    "\n",
    "#testing with `.csi`\n",
    "\n",
    "[nathalia@cshl] $  scp -i ~/cshl/cshl_2018_student.pem ubuntu@52.26.253.73:/home/ubuntu/workspace/cshl-practical/analysis/all.vcf.gz.csi .\n",
    "all.vcf.gz.csi                                100%  352     4.3KB/s   00:00    \n",
    "\n",
    "\n",
    "# https://www.biostars.org/p/59492/\n",
    "\n",
    "[nathalia@aws] $  tabix -p vcf all.vcf.gz\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 26M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  7.4M Nov 11 16:50 NA12878.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.5M Nov 11 16:53 NA12891.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.9M Nov 11 16:55 NA12892.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1007K Nov 11 16:56 NA12878.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  803K Nov 11 16:56 NA12891.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  1.1M Nov 11 16:56 NA12892.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12878.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12891.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12892.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  900K Nov 11 17:57 chr17.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   81K Nov 11 17:57 chr20.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  196K Nov 11 17:58 chr21.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  216K Nov 11 17:58 chr22.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  347K Nov 11 18:04 all.vcf.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   352 Nov 11 18:04 all.vcf.gz.csi\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  116K Nov 11 18:40 NA12878.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   60K Nov 11 18:44 test.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   557 Nov 11 19:13 all.vcf.gz.tbi\n",
    "\n",
    "#now I have the .csi\n",
    "\n",
    "[nathalia@cshl] $  scp -i ~/cshl/cshl_2018_student.pem ubuntu@52.26.253.73:/home/ubuntu/workspace/cshl-practical/analysis/all.vcf.gz.tbi .\n",
    "all.vcf.gz.tbi                                100%  557     6.8KB/s   00:00    \n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Went to the website: https://vcf.iobio.io/?species=Human&build=GRCh37 > Make sure Human and GRCh37 are selected > Click on Select All Samples > Hit Load"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![Visualization with the files we used](practice_vcf_visualization.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The number of variants I had (252) didn't match with the girls around me. So I got the link to my public AWS and put it there.\n",
    "\n",
    "http://ec2-52-26-253-73.us-west-2.compute.amazonaws.com/workspace/cshl-practical/analysis/\n",
    "\n",
    "Right-click on `all.vcf.gz`\n",
    "\n",
    "Paste this URL onto to the iobio, and now I have all 766 variants.\n",
    "\n",
    "Seems like we still need the `.tbi` file, because the girls tried to upload their URL without the `.tbi` and it didn't load.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![now works](practice2_vcf_visialization.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Call variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  bcftools concat -o all.vcf chr17.vcf chr20.vcf chr21.vcf chr22.vcf\n",
    "# the above step has t be done with chr from the same person\n",
    "\n",
    "[nathalia@aws] $  bgzip all.vcf\n",
    "# never use gzip bc then cannot index\n",
    "\n",
    "[nathalia@aws] $  bcftools index all.vcf.gz\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 26M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  7.4M Nov 11 16:50 NA12878.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.5M Nov 11 16:53 NA12891.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.9M Nov 11 16:55 NA12892.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1007K Nov 11 16:56 NA12878.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  803K Nov 11 16:56 NA12891.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  1.1M Nov 11 16:56 NA12892.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12878.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12891.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12892.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  900K Nov 11 17:57 chr17.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   81K Nov 11 17:57 chr20.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  196K Nov 11 17:58 chr21.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  216K Nov 11 17:58 chr22.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  347K Nov 11 18:04 all.vcf.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   352 Nov 11 18:04 all.vcf.gz.csi\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Manipulate VCF files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  bcftools view all.vcf.gz 17 | less -S\n",
    "\n",
    "[nathalia@aws] $  bcftools view -H all.vcf.gz 17:17585028-17585100 | less -S\n",
    "\n",
    "[nathalia@aws] $  bcftools view -i 'TYPE=\"snp\"' all.vcf.gz | less -S\n",
    "\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bcftools view all.vcf.gz 22:39636944 | grep -v ^## | less -S\n",
    "\n",
    "#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  NA12892 NA12878 NA12891\n",
    "22      39636944        .       ACACTC  AC,ACTCTC,ACACAC        1703.26 .       AB=0.632075,0.141509,\n",
    "\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bcftools view all.vcf.gz 22:39636944 | bcftools norm -m - | grep -v ^## | less -S\n",
    "\n",
    "#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  NA12892 NA12878 NA12891\n",
    "22      39636944        .       ACACTC  AC      1703.26 .       AB=0.632075;ABP=19.071;AC=3;AF=0.5;AN\n",
    "22      39636944        .       ACACTC  ACTCTC  1703.26 .       AB=0.141509;ABP=121.335;AC=2;AF=0.333\n",
    "22      39636944        .       ACACTC  ACACAC  1703.26 .       AB=0.207547;ABP=81.7569;AC=1;AF=0.166\n",
    "\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bcftools view all.vcf.gz 22:39636944 | bcftools norm -f $REF.fa -m - | grep -v ^## | less -S\n",
    "\n",
    "#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  NA12892 NA12878 NA12891\n",
    "22      39636944        .       ACACT   A       1703.26 .       AB=0.632075;ABP=19.071;AC=3;AF=0.5;AN\n",
    "22      39636946        .       A       T       1703.26 .       AB=0.141509;ABP=121.335;AC=2;AF=0.333\n",
    "22      39636948        .       T       A       1703.26 .       AB=0.207547;ABP=81.7569;AC=1;AF=0.166\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Contents of vcf file explained: http://www.internationalgenome.org/wiki/Analysis/vcf4.0/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  zcat all.vcf.gz | grep -v ^## | head | column -t | less -S\n",
    "#grep canot take compressed file\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Subsetting on samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  bcftools view -s NA12878 all.vcf.gz | less -S\n",
    "\n",
    "[nathalia@aws] $  bcftools view -s NA12878 all.vcf.gz | wc -l \n",
    "2632\n",
    "[nathalia@aws] $  bcftools view all.vcf.gz | wc -l\n",
    "2632\n",
    "\n",
    "# These numbers are supposed to be the same because the files have the same number of lines; it doesn't represent the number of polymorphisms... this command only deleted the columns that match to the sample we selected\n",
    "\n",
    "[nathalia@aws] $  bcftools view -s NA12878 all.vcf.gz | grep \"0/0\" | less -S\n",
    "\n",
    "# not polymorphic\n",
    "\n",
    "[nathalia@aws] $  bcftools view -s NA12878 all.vcf.gz | bcftools view -e 'GT==\"RR\"' -o NA12878.vcf -\n",
    "\n",
    "# all sites that show polymorphism; 'GT==\"RR\"' Genotype == reference reference\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 26M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  7.4M Nov 11 16:50 NA12878.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.5M Nov 11 16:53 NA12891.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.9M Nov 11 16:55 NA12892.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1007K Nov 11 16:56 NA12878.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  803K Nov 11 16:56 NA12891.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  1.1M Nov 11 16:56 NA12892.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12878.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12891.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12892.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  900K Nov 11 17:57 chr17.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   81K Nov 11 17:57 chr20.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  196K Nov 11 17:58 chr21.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  216K Nov 11 17:58 chr22.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  347K Nov 11 18:04 all.vcf.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   352 Nov 11 18:04 all.vcf.gz.csi\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  116K Nov 11 18:40 NA12878.vcf\n",
    "\n",
    "[nathalia@aws] $  less NA12878.vcf | grep -v \"^#\" | wc -l\n",
    "218\n",
    "\n",
    "[nathalia@aws] $  bcftools view -s NA12878 -e 'GT==\"RR\"' -o test.vcf all.vcf.gz\n",
    "\n",
    "[nathalia@aws] $  less test.vcf | grep -v \"^#\" | wc -l\n",
    "101\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note: `screen` and `tmux` "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Joint calling"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://useast.ensembl.org/index.html\n",
    "\n",
    "Uner favourite genomes, select Human.\n",
    "Under other assemblies, select the 37 reference\n",
    "Click BioMart\n",
    "\n",
    "Select Human databases; Select Structural, then under Gene select Chromosome, gene start, gene end, gene name\n",
    "\n",
    "```\n",
    "Chromosome/scaffold name\tGene start (bp)\tGene end (bp)\tGene name\n",
    "HG991_PATCH\t66119285\t66465398\tSLC25A26\n",
    "13\t23551994\t23552136\tAL157931.1\n",
    "13\t23708313\t23708703\tHMGA1P6\n",
    "13\t23726725\t23726825\tRNY3P4\n",
    "13\t23743974\t23744736\tLINC00362\n",
    "13\t23791571\t23791673\tRNU6-58P\n",
    "13\t23817659\t23821323\tTATDN2P3\n",
    "13\t93708910\t93710179\tLINC00363\n",
    "13\t23841645\t23843289\tSDAD1P4\n",
    "13\t23947562\t23948200\tRPLP1P13\n",
    "\n",
    "```\n",
    "\n",
    "Download to local machine as tsv, then scp into AWS\n",
    "\n",
    "```\n",
    "[nathalia@cshl] $  scp -i /Users/seqtec/cshl/cshl_2018_student.pem mart_export-2.txt ubuntu@52.26.253.73:/home/ubuntu/workspace/cshl-practical/analysis/\n",
    "mart_export-2.txt                             100% 1955KB   1.9MB/s   00:01    \n",
    "\n",
    "\n",
    "[nathalia@aws] $  emacs mart_export-2.txt \n",
    "\n",
    "#delete two header lines, saved\n",
    "\n",
    "[nathalia@aws] $  mv mart_export-2.txt genes.bed\n",
    "\n",
    "[nathalia@aws] $  bedtools intersect -a genes.bed -b all.vcf.gz | cut -f 4 | less\n",
    "\n",
    "# if want the column with gene names, so -a should be the bed file\n",
    "\n",
    "[nathalia@aws] $  bedtools intersect -a genes.bed -b all.vcf.gz | cut -f 4 | uniq | less\n",
    "\n",
    "[nathalia@aws] $  bedtools intersect -a genes.bed -b NA12878.vcf | cut -f 4| uniq | less\n",
    "\n",
    "# these are the genes this particular sample has variants in\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  bcftools view -s NA12878 all.vcf.gz | bcftools view -e 'GT==\"RR\"' -O z -o NA12878.vcf.gz -\n",
    "\n",
    "[nathalia@aws] $  bcftools index NA12878.vcf.gz \n",
    "\n",
    "[nathalia@aws] $  tabix -p vcf NA12878.vcf.gz\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 30M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  7.4M Nov 11 16:50 NA12878.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.5M Nov 11 16:53 NA12891.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  6.9M Nov 11 16:55 NA12892.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1007K Nov 11 16:56 NA12878.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  803K Nov 11 16:56 NA12891.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  1.1M Nov 11 16:56 NA12892.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12878.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12891.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   412 Nov 11 16:57 NA12892.stats\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  900K Nov 11 17:57 chr17.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   81K Nov 11 17:57 chr20.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  196K Nov 11 17:58 chr21.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  216K Nov 11 17:58 chr22.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  347K Nov 11 18:04 all.vcf.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   352 Nov 11 18:04 all.vcf.gz.csi\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  116K Nov 11 18:40 NA12878.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   60K Nov 11 18:44 test.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   557 Nov 11 19:27 all.vcf.gz.tbi\n",
    "-rw-r--r-- 1 ubuntu ubuntu  2.0M Nov 11 20:51 mart_export-2.txt~\n",
    "-rw-r--r-- 1 ubuntu ubuntu  2.0M Nov 11 20:51 genes.bed\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   32K Nov 11 21:02 NA12878.vcf.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   316 Nov 11 21:02 NA12878.vcf.gz.csi\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   503 Nov 11 21:04 NA12878.vcf.gz.tbi\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Loaded the files into https://vcf.iobio.io/?species=Human&build=GRCh37&vcf=http%3A%2F%2Fec2-52-26-253-73.us-west-2.compute.amazonaws.com%2Fworkspace%2Fcshl-practical%2Fanalysis%2FNA12878.vcf.gz&tbi="
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![join_calling](joint_calling_visualization.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
