{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 11/13/2018\n",
    "\n",
    "RNA-seq continued\n",
    "\n",
    "Griffith lab https://github.com/griffithlab/rnaseq_tutorial/wiki/Alignment\n",
    "\n",
    "nathalia."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alignment with hisat2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  echo $RNA_ALIGN_DIR\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "[nathalia@aws] $  mkdir -p $RNA_ALIGN_DIR\n",
    "[nathalia@aws] $  cd $RNA_ALIGN_DIR\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  emacs alignment.sh \n",
    "#!/bin/bash\n",
    "\n",
    "hisat2 -p 8 --rg-id=UHR_Rep1 --rg SM:UHR --rg LB:UHR_Rep1_ERCC-Mix1 --rg PL:ILLUMINA --rg PU:CXX1234-ACTGAC.1 -x $RNA_REF_INDEX --dta --rna-strandness RF -1 $RNA_DATA_DIR/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz -2 $RNA_DATA_DIR/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz -S ./UHR_Rep1.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=UHR_Rep2 --rg SM:UHR --rg LB:UHR_Rep2_ERCC-Mix1 --rg PL:ILLUMINA --rg PU:CXX1234-TGACAC.1 -x $RNA_REF_INDEX --dta --rna-strandness RF -1 $RNA_DATA_DIR/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz -2 $RNA_DATA_DIR/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz -S ./UHR_Rep2.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=UHR_Rep3 --rg SM:UHR --rg LB:UHR_Rep3_ERCC-Mix1 --rg PL:ILLUMINA --rg PU:CXX1234-CTGACA.1 -x $RNA_REF_INDEX --dta --rna-strandness RF -1 $RNA_DATA_DIR/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz -2 $RNA_DATA_DIR/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz -S ./UHR_Rep3.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=HBR_Rep1 --rg SM:HBR --rg LB:HBR_Rep1_ERCC-Mix2 --rg PL:ILLUMINA --rg PU:CXX1234-TGACAC.1 -x $RNA_REF_INDEX --dta --rna-strandness RF -1 $RNA_DATA_DIR/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz -2 $RNA_DATA_DIR/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz -S ./HBR_Rep1.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=HBR_Rep2 --rg SM:HBR --rg LB:HBR_Rep2_ERCC-Mix2 --rg PL:ILLUMINA --rg PU:CXX1234-GACACT.1 -x $RNA_REF_INDEX --dta --rna-strandness RF -1 $RNA_DATA_DIR/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz -2 $RNA_DATA_DIR/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz -S ./HBR_Rep2.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=HBR_Rep3 --rg SM:HBR --rg LB:HBR_Rep3_ERCC-Mix2 --rg PL:ILLUMINA --rg PU:CXX1234-ACACTG.1 -x $RNA_REF_INDEX --dta --rna-strandness RF -1 $RNA_DATA_DIR/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz -2 $RNA_DATA_DIR/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz -S ./HBR_Rep3.sam\n",
    "\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bash alignment.sh \n",
    "227392 reads; of these:\n",
    "  227392 (100.00%) were paired; of these:\n",
    "    1167 (0.51%) aligned concordantly 0 times\n",
    "    222479 (97.84%) aligned concordantly exactly 1 time\n",
    "    3746 (1.65%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1167 pairs aligned concordantly 0 times; of these:\n",
    "      534 (45.76%) aligned discordantly 1 time\n",
    "    ----\n",
    "    633 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      1266 mates make up the pairs; of these:\n",
    "        515 (40.68%) aligned 0 times\n",
    "        648 (51.18%) aligned exactly 1 time\n",
    "        103 (8.14%) aligned >1 times\n",
    "99.89% overall alignment rate\n",
    "162373 reads; of these:\n",
    "  162373 (100.00%) were paired; of these:\n",
    "    971 (0.60%) aligned concordantly 0 times\n",
    "    158727 (97.75%) aligned concordantly exactly 1 time\n",
    "    2675 (1.65%) aligned concordantly >1 times\n",
    "    ----\n",
    "    971 pairs aligned concordantly 0 times; of these:\n",
    "      586 (60.35%) aligned discordantly 1 time\n",
    "    ----\n",
    "    385 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      770 mates make up the pairs; of these:\n",
    "        324 (42.08%) aligned 0 times\n",
    "        406 (52.73%) aligned exactly 1 time\n",
    "        40 (5.19%) aligned >1 times\n",
    "99.90% overall alignment rate\n",
    "185442 reads; of these:\n",
    "  185442 (100.00%) were paired; of these:\n",
    "    1162 (0.63%) aligned concordantly 0 times\n",
    "    181295 (97.76%) aligned concordantly exactly 1 time\n",
    "    2985 (1.61%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1162 pairs aligned concordantly 0 times; of these:\n",
    "      718 (61.79%) aligned discordantly 1 time\n",
    "    ----\n",
    "    444 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      888 mates make up the pairs; of these:\n",
    "        381 (42.91%) aligned 0 times\n",
    "        452 (50.90%) aligned exactly 1 time\n",
    "        55 (6.19%) aligned >1 times\n",
    "99.90% overall alignment rate\n",
    "118571 reads; of these:\n",
    "  118571 (100.00%) were paired; of these:\n",
    "    486 (0.41%) aligned concordantly 0 times\n",
    "    117363 (98.98%) aligned concordantly exactly 1 time\n",
    "    722 (0.61%) aligned concordantly >1 times\n",
    "    ----\n",
    "    486 pairs aligned concordantly 0 times; of these:\n",
    "      331 (68.11%) aligned discordantly 1 time\n",
    "    ----\n",
    "    155 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      310 mates make up the pairs; of these:\n",
    "        188 (60.65%) aligned 0 times\n",
    "        112 (36.13%) aligned exactly 1 time\n",
    "        10 (3.23%) aligned >1 times\n",
    "99.92% overall alignment rate\n",
    "144826 reads; of these:\n",
    "  144826 (100.00%) were paired; of these:\n",
    "    622 (0.43%) aligned concordantly 0 times\n",
    "    143265 (98.92%) aligned concordantly exactly 1 time\n",
    "    939 (0.65%) aligned concordantly >1 times\n",
    "    ----\n",
    "    622 pairs aligned concordantly 0 times; of these:\n",
    "      454 (72.99%) aligned discordantly 1 time\n",
    "    ----\n",
    "    168 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      336 mates make up the pairs; of these:\n",
    "        215 (63.99%) aligned 0 times\n",
    "        111 (33.04%) aligned exactly 1 time\n",
    "        10 (2.98%) aligned >1 times\n",
    "99.93% overall alignment rate\n",
    "129786 reads; of these:\n",
    "  129786 (100.00%) were paired; of these:\n",
    "    518 (0.40%) aligned concordantly 0 times\n",
    "    128459 (98.98%) aligned concordantly exactly 1 time\n",
    "    809 (0.62%) aligned concordantly >1 times\n",
    "    ----\n",
    "    518 pairs aligned concordantly 0 times; of these:\n",
    "      353 (68.15%) aligned discordantly 1 time\n",
    "    ----\n",
    "    165 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      330 mates make up the pairs; of these:\n",
    "        206 (62.42%) aligned 0 times\n",
    "        116 (35.15%) aligned exactly 1 time\n",
    "        8 (2.42%) aligned >1 times\n",
    "99.92% overall alignment rate\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sam to bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  emacs sam_bam.sh\n",
    "\n",
    "#!/bin/bash\n",
    "\n",
    "samtools sort -@ 8 -o UHR_Rep1.bam UHR_Rep1.sam\n",
    "samtools sort -@ 8 -o UHR_Rep2.bam UHR_Rep2.sam \n",
    "samtools sort -@ 8 -o UHR_Rep3.bam UHR_Rep3.sam \n",
    "samtools sort -@ 8 -o HBR_Rep1.bam HBR_Rep1.sam\n",
    "samtools sort -@ 8 -o HBR_Rep2.bam HBR_Rep2.sam\n",
    "samtools sort -@ 8 -o HBR_Rep3.bam HBR_Rep3.sam\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bash sam_bam.sh \n",
    "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
    "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
    "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
    "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
    "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
    "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 843M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.0K Nov 13 14:24 alignment.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 166M Nov 13 14:24 UHR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 119M Nov 13 14:24 UHR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 136M Nov 13 14:24 UHR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  86M Nov 13 14:25 HBR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105M Nov 13 14:25 HBR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  94M Nov 13 14:25 HBR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  303 Nov 13 14:27 sam_bam.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  34M Nov 13 14:28 UHR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25M Nov 13 14:28 UHR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  28M Nov 13 14:28 UHR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  17M Nov 13 14:28 HBR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20M Nov 13 14:28 HBR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  18M Nov 13 14:28 HBR_Rep3.bam\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge hisat2 bam files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME/alignments/hisat2\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  java -Xmx2g -jar $RNA_HOME/tools/picard.jar MergeSamFiles OUTPUT=UHR.bam INPUT=UHR_Rep1.bam INPUT=UHR_Rep2.bam INPUT=UHR_Rep3.bam\n",
    "INFO\t2018-11-13 14:29:56\tMergeSamFiles\t\n",
    "\n",
    "********** NOTE: Picard's command line syntax is changing.\n",
    "**********\n",
    "********** For more information, please see:\n",
    "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
    "**********\n",
    "********** The command line looks like this in the new syntax:\n",
    "**********\n",
    "**********    MergeSamFiles -OUTPUT UHR.bam -INPUT UHR_Rep1.bam -INPUT UHR_Rep2.bam -INPUT UHR_Rep3.bam\n",
    "**********\n",
    "\n",
    "\n",
    "14:30:02.450 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/workspace/rnaseq/tools/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
    "[Tue Nov 13 14:30:02 UTC 2018] MergeSamFiles INPUT=[UHR_Rep1.bam, UHR_Rep2.bam, UHR_Rep3.bam] OUTPUT=UHR.bam    SORT_ORDER=coordinate ASSUME_SORTED=false MERGE_SEQUENCE_DICTIONARIES=false USE_THREADING=false VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
    "[Tue Nov 13 14:30:02 UTC 2018] Executing as ubuntu@ip-172-31-39-185 on Linux 4.15.0-1025-aws amd64; OpenJDK 64-Bit Server VM 10.0.2+13-Ubuntu-1ubuntu0.18.04.3; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.15-SNAPSHOT\n",
    "INFO\t2018-11-13 14:30:02\tMergeSamFiles\tInput files are in same order as output so sorting to temp directory is not needed.\n",
    "INFO\t2018-11-13 14:30:13\tMergeSamFiles\tProcessed     1,000,000 records.  Elapsed time: 00:00:10s.  Time for last 1,000,000:   10s.  Last read position: ERCC-00130:135\n",
    "INFO\t2018-11-13 14:30:14\tMergeSamFiles\tFinished reading inputs.\n",
    "[Tue Nov 13 14:30:14 UTC 2018] picard.sam.MergeSamFiles done. Elapsed time: 0.21 minutes.\n",
    "Runtime.totalMemory()=515899392\n",
    "\n",
    "[nathalia@aws] $  java -Xmx2g -jar $RNA_HOME/tools/picard.jar MergeSamFiles OUTPUT=HBR.bam INPUT=HBR_Rep1.bam INPUT=HBR_Rep2.bam INPUT=HBR_Rep3.bam\n",
    "INFO\t2018-11-13 14:30:37\tMergeSamFiles\t\n",
    "\n",
    "********** NOTE: Picard's command line syntax is changing.\n",
    "**********\n",
    "********** For more information, please see:\n",
    "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
    "**********\n",
    "********** The command line looks like this in the new syntax:\n",
    "**********\n",
    "**********    MergeSamFiles -OUTPUT HBR.bam -INPUT HBR_Rep1.bam -INPUT HBR_Rep2.bam -INPUT HBR_Rep3.bam\n",
    "**********\n",
    "\n",
    "\n",
    "14:30:37.524 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/workspace/rnaseq/tools/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
    "[Tue Nov 13 14:30:37 UTC 2018] MergeSamFiles INPUT=[HBR_Rep1.bam, HBR_Rep2.bam, HBR_Rep3.bam] OUTPUT=HBR.bam    SORT_ORDER=coordinate ASSUME_SORTED=false MERGE_SEQUENCE_DICTIONARIES=false USE_THREADING=false VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
    "[Tue Nov 13 14:30:37 UTC 2018] Executing as ubuntu@ip-172-31-39-185 on Linux 4.15.0-1025-aws amd64; OpenJDK 64-Bit Server VM 10.0.2+13-Ubuntu-1ubuntu0.18.04.3; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.15-SNAPSHOT\n",
    "INFO\t2018-11-13 14:30:37\tMergeSamFiles\tInput files are in same order as output so sorting to temp directory is not needed.\n",
    "INFO\t2018-11-13 14:30:45\tMergeSamFiles\tFinished reading inputs.\n",
    "[Tue Nov 13 14:30:45 UTC 2018] picard.sam.MergeSamFiles done. Elapsed time: 0.14 minutes.\n",
    "Runtime.totalMemory()=515899392\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 981M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.0K Nov 13 14:24 alignment.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 166M Nov 13 14:24 UHR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 119M Nov 13 14:24 UHR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 136M Nov 13 14:24 UHR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  86M Nov 13 14:25 HBR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105M Nov 13 14:25 HBR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  94M Nov 13 14:25 HBR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  303 Nov 13 14:27 sam_bam.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  34M Nov 13 14:28 UHR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25M Nov 13 14:28 UHR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  28M Nov 13 14:28 UHR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  17M Nov 13 14:28 HBR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20M Nov 13 14:28 HBR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  18M Nov 13 14:28 HBR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  85M Nov 13 14:30 UHR.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53M Nov 13 14:30 HBR.bam\n",
    "\n",
    "[nathalia@aws] $  ls -l *.bam | wc -l\n",
    "8\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Practice 6"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/practice3\n",
    "[nathalia@aws] $  cd data/\n",
    "[nathalia@aws] $  ln -s ../trimmed/*.fastq.gz .\n",
    "[nathalia@aws] $  ls\n",
    "hcc1395_normal_rep1_1.fastq.gz  hcc1395_normal_rep3_1.fastq.gz  hcc1395_tumor_rep2_1.fastq.gz\n",
    "hcc1395_normal_rep1_2.fastq.gz  hcc1395_normal_rep3_2.fastq.gz  hcc1395_tumor_rep2_2.fastq.gz\n",
    "hcc1395_normal_rep2_1.fastq.gz  hcc1395_tumor_rep1_1.fastq.gz   hcc1395_tumor_rep3_1.fastq.gz\n",
    "hcc1395_normal_rep2_2.fastq.gz  hcc1395_tumor_rep1_2.fastq.gz   hcc1395_tumor_rep3_2.fastq.gz\n",
    "\n",
    "[nathalia@aws] $  cp ../../alignments/hisat2/alignment.sh .\n",
    "\n",
    "#!/bin/bash\n",
    "\n",
    "hisat2 -p 8 --rg-id=hcc1395_normal_rep1 --rg SM:hcc1395_normal --rg LB:hcc1395_normal_rep1 --rg PL:ILLUMINA  -x $RNA_REF_INDEX --dta --rna-strandness RF -1 hcc1395_normal_rep1_1.fastq.gz -2 hcc1395_normal_rep1_2.fastq.gz -S ./hcc1395_normal_rep1.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=hcc1395_normal_rep2 --rg SM:hcc1395_normal --rg LB:hcc1395_normal_rep2 --rg PL:ILLUMINA  -x $RNA_REF_INDEX --dta --rna-strandness RF -1 hcc1395_normal_rep2_1.fastq.gz -2 hcc1395_normal_rep2_2.fastq.gz -S ./hcc1395_normal_rep2.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=hcc1395_normal_rep3 --rg SM:hcc1395_normal --rg LB:hcc1395_normal_rep3 --rg PL:ILLUMINA  -x $RNA_REF_INDEX --dta --rna-strandness RF -1 hcc1395_normal_rep3_1.fastq.gz -2 hcc1395_normal_rep3_2.fastq.gz -S ./hcc1395_normal_rep3.sam\n",
    "\n",
    "##\n",
    "\n",
    "hisat2 -p 8 --rg-id=hcc1395_tumor_rep1 --rg SM:hcc1395_tumor --rg LB:hcc1395_tumor_rep1 --rg PL:ILLUMINA  -x $RNA_REF_INDEX --dta --rna-strandness RF -1 hcc1395_tumor_rep1_1.fastq.gz -2 hcc1395_tumor_rep1_2.fastq.gz -S ./hcc1395_tumor_rep1.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=hcc1395_tumor_rep2 --rg SM:hcc1395_tumor --rg LB:hcc1395_tumor_rep2 --rg PL:ILLUMINA  -x $RNA_REF_INDEX --dta --rna-strandness RF -1 hcc1395_tumor_rep2_1.fastq.gz -2 hcc1395_tumor_rep2_2.fastq.gz -S ./hcc1395_tumor_rep2.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=hcc1395_tumor_rep3 --rg SM:hcc1395_tumor --rg LB:hcc1395_tumor_rep3 --rg PL:ILLUMINA  -x $RNA_REF_INDEX --dta --rna-strandness RF -1 hcc1395_tumor_rep3_1.fastq.gz -2 hcc1395_tumor_rep3_2.fastq.gz -S ./hcc1395_tumor_rep3.sam\n",
    "\n",
    "\n",
    "samtools sort -@ 8 -o hcc1395_normal_rep1.bam hcc1395_normal_rep1.sam\n",
    "samtools sort -@ 8 -o hcc1395_normal_rep2.bam hcc1395_normal_rep2.sam \n",
    "samtools sort -@ 8 -o hcc1395_normal_rep3.bam hcc1395_normal_rep3.sam \n",
    "samtools sort -@ 8 -o hcc1395_tumor_rep1.bam hcc1395_tumor_rep1.sam\n",
    "samtools sort -@ 8 -o hcc1395_tumor_rep2.bam hcc1395_tumor_rep2.sam\n",
    "samtools sort -@ 8 -o hcc1395_tumor_rep3.bam hcc1395_tumor_rep3.sam\n",
    "\n",
    "\n",
    "java -Xmx2g -jar $RNA_HOME/tools/picard.jar MergeSamFiles OUTPUT=hcc_normal.bam INPUT=hcc1395_normal_rep1.bam INPUT=hcc1395_normal_rep2.bam INPUT=hcc1395_normal_rep3.bam\n",
    "\n",
    "java -Xmx2g -jar $RNA_HOME/tools/picard.jar MergeSamFiles OUTPUT=hcc_tumor.bam INPUT=hcc1395_tumor_rep1.bam INPUT=hcc1395_tumor_rep2.bam INPUT=hcc1395_tumor_rep3.bam\n",
    "\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 2.6G\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep3_2.fastq.gz -> ../trimmed/hcc1395_tumor_rep3_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep3_1.fastq.gz -> ../trimmed/hcc1395_tumor_rep3_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep2_2.fastq.gz -> ../trimmed/hcc1395_tumor_rep2_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep2_1.fastq.gz -> ../trimmed/hcc1395_tumor_rep2_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep1_2.fastq.gz -> ../trimmed/hcc1395_tumor_rep1_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep1_1.fastq.gz -> ../trimmed/hcc1395_tumor_rep1_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep3_2.fastq.gz -> ../trimmed/hcc1395_normal_rep3_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep3_1.fastq.gz -> ../trimmed/hcc1395_normal_rep3_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep2_2.fastq.gz -> ../trimmed/hcc1395_normal_rep2_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep2_1.fastq.gz -> ../trimmed/hcc1395_normal_rep2_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep1_2.fastq.gz -> ../trimmed/hcc1395_normal_rep1_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep1_1.fastq.gz -> ../trimmed/hcc1395_normal_rep1_1.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.3K Nov 13 15:15 alignment.sh~\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.4K Nov 13 15:22 alignment.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 314M Nov 13 15:22 hcc1395_normal_rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 315M Nov 13 15:23 hcc1395_normal_rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 315M Nov 13 15:23 hcc1395_normal_rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 369M Nov 13 15:26 hcc1395_tumor_rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 368M Nov 13 15:26 hcc1395_tumor_rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 368M Nov 13 15:27 hcc1395_tumor_rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  49M Nov 13 15:27 hcc1395_normal_rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  47M Nov 13 15:27 hcc1395_normal_rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  47M Nov 13 15:27 hcc1395_normal_rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  55M Nov 13 15:27 hcc1395_tumor_rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53M Nov 13 15:27 hcc1395_tumor_rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53M Nov 13 15:27 hcc1395_tumor_rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 141M Nov 13 15:28 hcc_normal.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 159M Nov 13 15:28 hcc_tumor.bam\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ?\n",
    "```\n",
    "[nathalia@aws] $  zcat hcc1395_normal_rep1_1.fastq.gz | head -n 5\n",
    "@K00193:38:H3MYFBBXX:4:1101:10003:44458/1\n",
    "TTCCTTATGAAACAGGAAGAGTCCCTGGGCCCAGGCCTGGCCCACGGTTGTCAAGGCACATCATTGCCAGCAAGCTGAAGCATACCAGCAGCCACAACCTAGATCTCATTCCCAACCCAAAGTTCTGACTTCTGTACAAACTCGTTTCCAG\n",
    "+\n",
    "AAFFFKKKKKKKKKKKKKKKKKKKKKKKKFKKFKKKKF<AAKKKKKKKKKKKKKKKKFKKKFKKKKKKKKKKKFKAFKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKFKKKKKKKKKKKKFKFFKKKKKKKKKKKKFKKKK\n",
    "@K00193:38:H3MYFBBXX:4:1101:10003:46427/1\n",
    "[nathalia@aws] $  zcat /home/ubuntu/workspace/rnaseq/data/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz | head -n 5\n",
    "@HWI-ST718_146963544:6:1213:8996:10047/1\n",
    "CTTTTTTATTTTTGTCTGACTGGGTTGATTCAAAGGTCTGGTCTTTGAGCTCTTAAATTAGTTCTTCTATTTGGCCTAGTCTGTTGCTAAGGCTGCCAAC\n",
    "+\n",
    "CCCFFFFFHHHHGJHIIJHIHIIIFHIJJJJIJJGIBBFGEGGHIIHGGIJJIIHGGHIIIFGCGHHIIHIHHEEE?DFEFFFEEDCEEDDDDDDDBCDD\n",
    "@HWI-ST718_146963544:5:2303:11793:37095/1\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alignment visualization"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  echo $RNA_ALIGN_DIR\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "[nathalia@aws] $  cd $RNA_ALIGN_DIR\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "\n",
    "[nathalia@aws] $  find *.bam -exec echo samtools index {} \\; | sh\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 981M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.0K Nov 13 14:24 alignment.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 166M Nov 13 14:24 UHR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 119M Nov 13 14:24 UHR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 136M Nov 13 14:24 UHR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  86M Nov 13 14:25 HBR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105M Nov 13 14:25 HBR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  94M Nov 13 14:25 HBR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  303 Nov 13 14:27 sam_bam.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  34M Nov 13 14:28 UHR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25M Nov 13 14:28 UHR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  28M Nov 13 14:28 UHR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  17M Nov 13 14:28 HBR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20M Nov 13 14:28 HBR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  18M Nov 13 14:28 HBR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  85M Nov 13 14:30 UHR.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53M Nov 13 14:30 HBR.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  47K Nov 13 15:34 HBR.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  39K Nov 13 15:34 HBR_Rep1.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  40K Nov 13 15:34 HBR_Rep2.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  40K Nov 13 15:34 HBR_Rep3.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  51K Nov 13 15:34 UHR.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  42K Nov 13 15:34 UHR_Rep1.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  44K Nov 13 15:34 UHR_Rep2.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  43K Nov 13 15:34 UHR_Rep3.bam.bai\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In IGV, change the reference to hg38, and go to chr22, then\n",
    "\n",
    "`File` > `Load from URL` \n",
    "\n",
    "http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/alignments/hisat2/HBR.bam\n",
    "\n",
    "http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/alignments/hisat2/UHR.bam\n",
    "\n",
    "In the genomic region box, find `EIF3L`, hit Go\n",
    "\n",
    "Right-click, click on `Show splice-juction track`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![igv1](alignment_igv.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![variant](variant.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BAM read count"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME\n",
    "[nathalia@aws] $  mkdir bam_readcount\n",
    "[nathalia@aws] $  cd bam_readcount\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/bam_readcount\n",
    "\n",
    "[nathalia@aws] $  echo $RNA_REF_FASTA\n",
    "/home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92.fa\n",
    "\n",
    "[nathalia@aws] $  samtools faidx $RNA_REF_FASTA\n",
    "\n",
    "[nathalia@aws] $  ls /home/ubuntu/workspace/rnaseq/refs/\n",
    "chr22_with_ERCC92.1.ht2  chr22_with_ERCC92.6.ht2   chr22_with_ERCC92.gtf  splicesites.tsv\n",
    "chr22_with_ERCC92.2.ht2  chr22_with_ERCC92.7.ht2   exons.tsv              \n",
    "chr22_with_ERCC92.3.ht2  chr22_with_ERCC92.8.ht2   illumina_multiplex.fa\n",
    "chr22_with_ERCC92.4.ht2  chr22_with_ERCC92.fa      \n",
    "chr22_with_ERCC92.5.ht2  chr22_with_ERCC92.fa.fai  \n",
    "\n",
    "# produced the .fai file\n",
    "\n",
    "[nathalia@aws] $  samtools mpileup -f $RNA_REF_FASTA -r 22:18918457-18918467 $RNA_ALIGN_DIR/UHR.bam $RNA_ALIGN_DIR/HBR.bam \n",
    "[mpileup] 2 samples in 2 input files\n",
    "22\t18918457\tA\t7\t..,.,.,\t3EJJDD?\t32\t>.$.,.,,.,,..,,....,....,,,,.,.,.\tICD@mAJCIJDDJJDDDDJEEHJJIJEGEIDI\n",
    "22\t18918458\tG\t7\t..,.,.,\t8EIIEDC\t31\t>.$,.,,.,,..,,....,....,,,,.,.,.\tIB@k:JBJJDDJJDDDDJDEGHJGJFGEIDJ\n",
    "22\t18918459\tC\t7\t..,.,.,\t8DJJDH?\t30\t>,$.,,.,,..,,....,....,,,,.,.,.\tI@mAIDIIDDIJBDDDJDDFHJIJFIHJDJ\n",
    "22\t18918460\tT\t7\t..,.,.,\t4DIIAEB\t29\t>.,,.,,..,,....,....,,,,.,.,.\tImFIDGJDDIICDDCJDEFGIHJFIHJDJ\n",
    "22\t18918461\tC\t7\t..,.,.,\t4DJIDH@\t29\t>.,,.,,..,,....,....,,,,.,.,.\tImHJDJJDDIJDDDDJDDFHJFJFIEJDJ\n",
    "22\t18918462\tC\t8\t..,.,.,^].\t:DJJFE?@\t29\t>.,,.,,..,,....,....,,,,.,.,.\tImHJDIJDDIJDDDCIDDFHIGJHIDIDJ\n",
    "22\t18918463\tA\t8\t..,.,.,.\t>DIJBH5@\t29\t>.,,.,,..,,....,....,,,,.,.,.\tIkDIaHJDDEICDDCGDDFFHFHHGBGFJ\n",
    "22\t18918464\tC\t8\t..,.,.,.\t?DJJFG@@\t29\t>.,,.,t..,,....,....,,,,.,.,.\tIkDJgJJDDJJCDDCJDDEFHHJHGHHFJ\n",
    "22\t18918465\tG\t9\t..,.,.,.^].\t?DJJHG?FF\t30\t>.,a.,,..,,....,....,,,,.,A,.^].\tIkBIgJJDDJJBDDDHDDDFJBJJIFIFJC\n",
    "22\t18918466\tT\t9\t..,.,.,..\t2BJEHF5DF\t30\t>.,,.,,..,,....,....,,,,.,.,..\tIkBFeJJBDIIABBDJDDDFI?JJ@HJHHB\n",
    "22\t18918467\tC\t9\t..,.,.,..\t8DJHHG=DF\t30\t>.$,,.,,..,,....,....,,,,.,.,..\tIdDGgJJDDJJBDDDJDDDDHBJJEHJHIC\n",
    "```\n",
    "\n",
    "Each line consists of chromosome, 1-based coordinate, reference base, the number of reads covering the site, read bases and base qualities. At the read base column, a dot stands for a match to the reference base on the forward strand, a comma for a match on the reverse strand, ACGTN for a mismatch on the forward strand and acgtn for a mismatch on the reverse strand. A pattern \\+[0-9]+[ACGTNacgtn]+ indicates there is an insertion between this reference position and the next reference position.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  echo \"22 38483683 38483683\" > snvs.bed\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bam-readcount -l snvs.bed -f $RNA_REF_FASTA $RNA_ALIGN_DIR/UHR.bam 2>/dev/null\n",
    "22\t38483683\tG\t326\t=:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\tA:163:60.00:37.15:0.00:94:69:0.53:0.01:38.37:96:0.56:99.98:0.54\tC:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\tG:163:60.00:37.06:0.74:84:79:0.45:0.00:1.89:89:0.53:99.85:0.54\tT:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\tN:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\n",
    "\n",
    "#326 = depth\n",
    "#A/G 50/50\n",
    "\n",
    "#IGV chr22:38483683\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bam-readcount -l snvs.bed -f $RNA_REF_FASTA $RNA_ALIGN_DIR/HBR.bam 2>/dev/null\n",
    "22\t38483683\tG\t206\t=:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\tA:75:60.00:38.41:0.00:44:31:0.52:0.01:38.99:44:0.55:99.99:0.50\tC:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\tG:131:60.00:37.31:0.00:76:55:0.50:0.00:1.22:77:0.56:99.95:0.54\tT:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\tN:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\n",
    "\n",
    "#Here, is not exactly 50/50\n",
    "\n",
    "[nathalia@aws] $  bam-readcount -l snvs.bed -f $RNA_REF_FASTA $RNA_ALIGN_DIR/UHR.bam 2>/dev/null 1>UHR_bam-readcounts.txt\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bam-readcount -l snvs.bed -f $RNA_REF_FASTA $RNA_ALIGN_DIR/HBR.bam 2>/dev/null 1>HBR_bam-readcounts.txt\n",
    "\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 12K\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  21 Nov 13 15:53 snvs.bed\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 365 Nov 13 15:54 UHR_bam-readcounts.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 364 Nov 13 15:54 HBR_bam-readcounts.txt\n",
    "\n",
    "\n",
    "[nathalia@aws] $  cat UHR_bam-readcounts.txt | perl -ne '@data=split(\"\\t\", $_); @Adata=split(\":\", $data[5]); @Cdata=split(\":\", $data[6]); @Gdata=split(\":\", $data[7]); @Tdata=split(\":\", $data[8]); print \"UHR Counts\\t$data[0]\\t$data[1]\\tA: $Adata[1]\\tC: $Cdata[1]\\tT: $Tdata[1]\\tG: $Gdata[1]\\n\";'\n",
    "UHR Counts\t22\t38483683\tA: 163\tC: 0\tT: 0\tG: 163\n",
    "\n",
    "# uses perl to split the last columns, so we can easily see the counts of nt bases\n",
    "\n",
    "[nathalia@aws] $  cat HBR_bam-readcounts.txt | perl -ne '@data=split(\"\\t\", $_); @Adata=split(\":\", $data[5]); @Cdata=split(\":\", $data[6]); @Gdata=split(\":\", $data[7]); @Tdata=split(\":\", $data[8]); print \"HBR Counts\\t$data[0]\\t$data[1]\\tA: $Adata[1]\\tC: $Cdata[1]\\tT: $Tdata[1]\\tG: $Gdata[1]\\n\";'\n",
    "HBR Counts\t22\t38483683\tA: 75\tC: 0\tT: 0\tG: 131\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Practice 6"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME/practice3/data\n",
    "\n",
    "[nathalia@aws] $  find *.bam -exec samtools index {} \\; | sh\n",
    "```\n",
    "\n",
    "Loading into IGV\n",
    "\n",
    "http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/practice3/data/hcc_normal.bam\n",
    "\n",
    "\n",
    "http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/practice3/data/hcc_tumor.bam\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![practice](alignment_practice_igv.png) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![splices](Sashimi_.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alignment QC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_ALIGN_DIR\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "\n",
    "[nathalia@aws] $  samtools view UHR.bam | head\n",
    "HWI-ST718_146963544:5:1311:6035:51578\t419\t22\t10649269\t1\t100M\t=\t10649312\t143\tTGGGTCTATTTCTTTAAAACAAAGGGAGGGGAGTCTCTCATTTACATTAGTTTTTTTCATAGCCTTTTGGACTTTGCAATTTCTATGTTTTGGAACCTAT\t@@CFDDFFHHHHHJJJJJIJJJJJJJFHIJJBGFHIIJJJJJJJJJJJJJIIJHJIHFFFFFFFEEEEEDDDDDDDDEDDDDDEEEECDEDDDDDDDDDD\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:1311:6035:51578\t339\t22\t10649312\t1\t100M\t=\t10649269\t-143\tACATTAGTTTTTTTCATAGCCTTTTGGACTTTGCAATTTCTATGTTTTGGAACCTATTTCTTACAGTTTTTCTATGCTAAACTCTGTCCTGGTCAGTTCC\tCDDDDDDDDEEEEEEEFFFFHHHHEJHHIJJIIIGJJJIJJJJGHEJIJHGBGGHFIIIIIIGJJJJJJJJJJGJJJJJJIJJJJJJHHHHHFFFFFCCC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:6:1310:5357:21076\t355\t22\t10834840\t1\t100M\t=\t10834935\t195\tTGAATAGTGCATTCAAGTAATCTGTAGAACAAAGTTTGTATTTTATATTTTGGTGGGAGGGAGAAACCAGTTAATTTTCCCCTCTTAACTTCAGAAAGCA\tCCCFFFFEHHHHHJJJJGIJJJJJJJJJJJJJJJHIJJHIJJJJIJIJJJJIIGHIJIJJJIJJJJJJHHHHHFFDFFFFEEEDEDDEDDEDDDDDDDDC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:-\tNH:i:2\n",
    "HWI-ST718_146963544:6:1310:5357:21076\t403\t22\t10834935\t1\t100M\t=\t10834840\t-195\tAAGCATACTTGTTCAAATGTTTATAGATCATTTGTATTTTTCTATACTTTAGAAAAAAATAGTTCTATATTCTCTATTTTAGGGTATTAACTCTCAAGAG\tEEDEEDCDDDEEEEEEEFFFFFFFHHHHHHJJJJJJJJJJJJJIJHJJIIJJJJJJJJJJJJJJJIJIJJIJJJJJIJIJJJJJJJJHHHHHFFFFFCCC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:-\tNH:i:2\n",
    "HWI-ST718_146963544:5:2216:3503:44168\t419\t22\t10835740\t1\t100M\t=\t10835762\t122\tGCTACTTCTTCTTTCATGGGTCTTTTTGTTGGGTTTGCATCCTGGTTCTTCCATTGTTGATCTTGCATAGGAATATTTTTGTAATTCACATTTTTTATTA\tBBCDFFFFHHHHHJJEHIJJAFHJJJJJJJJJJHJJJJJJJJJJJJJJJJHIHIIJHIJHIJJJJJJJJJJIJHHHHHHHFBDFDFEEEEEEFEDDDDDF\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2216:3503:44168\t339\t22\t10835762\t1\t100M\t=\t10835740\t-122\tTTTTTGTTGGGTTTGCATCCTGGTTCTTCCATTGTTGATCTTGCATAGGAATATTTTTGTAATTCACATTTTTTATTAATATGCTGCCTGCTTTTCTTTC\tDDDDDDDDDDDDDDDDDDDDEDFFDFHHHHGIJJJJJJIJJJJIJIIJJJJJJJJJJJJJJJJIHGJJJJJJJJIJIJJJJJIJIIJHHHHHFFFFFCCC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2208:13984:35958\t163\t22\t10836084\t1\t100M\t=\t10836228\t244\tCACAAAGAAGCAAACTGTGTCCTGAACAGTTTTGCAAACTCTGTAGTTGGTAGTATTCTTTTCACACGTCTTTCTTGTCGTTCTTTTTAACAACAGTCAT\tCCCFFFDFHHHHHJJIIHGIJJJJIJIIJHGHIIIIIJJIIIGGIIIIIICFGIHIJEIHIIDGIIJJHGIJJIGJIHHHEFBBDEEECCACDDDDA:AD\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:-6\tYT:Z:CP\tRG:Z:UHR_Rep1\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2208:13984:35958\t83\t22\t10836228\t1\t100M\t=\t10836084\t-244\tGTGACATAGTATAGTATCCTTAGTATCCTAACACACAGTAGAGTACCTAAAGTCAGAACTTTTAAGAGAGATACCTGTAAATTTGGACATTACATCAGAT\tEDFCDDDEEHFHEHHIGIJJIHEHHIJIHHGIIJGIIIJJIHIHGFJHGIICGIGIJIIHIFJJJJJJIHHFBJJIJIJJJJJIJIJGHHHHFFFFF@CC\tAS:i:-6\tZS:i:-6\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:86G13\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep1\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:6:1103:13485:78315\t419\t22\t10837443\t1\t100M\t=\t10837541\t198\tCTAACTGATAAGAGAAAAAGTATAACACAACCTCCAAAATTAAAAATCACTTCAGAGGATTTCCAATACTTGTGTGGAGGGGTGAGCTCCTAACAAACTG\t@C@FFFFDHDHHBHIGIJJGCFHIIIIIIJGGIGGIJJJJJGIIJIJIJJIIIGIGHCGGIJJIDHIIIGGIIF?CCFFFB@>?BAC:ACCCDDCCBBD@\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:6:1103:13485:78315\t339\t22\t10837541\t1\t100M\t=\t10837443\t-198\tTGATCTTCTCACAAATAACCATTTGTAAACTCTGCACATAATATAGATAACATCTATCTGAGGGTTGTGGAGATTGAATAAAAGCAGGCAAGCTTTGGAG\tDCACC@CCCFFFFDDE?=FHHHAGGIHIHGIGGGIHGGHHIJJGJIGIHDGCIIHBGGEJJJJIJJIJJIIJJJIGIIIJJIJJJIIHHHHGFFFFF@C@\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://broadinstitute.github.io/picard/explain-flags.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  samtools view -f 3 -F 268 UHR.bam | head\n",
    "HWI-ST718_146963544:5:2208:13984:35958\t163\t22\t10836084\t1\t100M\t=\t10836228\t244\tCACAAAGAAGCAAACTGTGTCCTGAACAGTTTTGCAAACTCTGTAGTTGGTAGTATTCTTTTCACACGTCTTTCTTGTCGTTCTTTTTAACAACAGTCAT\tCCCFFFDFHHHHHJJIIHGIJJJJIJIIJHGHIIIIIJJIIIGGIIIIIICFGIHIJEIHIIDGIIJJHGIJJIGJIHHHEFBBDEEECCACDDDDA:AD\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:-6\tYT:Z:CP\tRG:Z:UHR_Rep1\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2208:13984:35958\t83\t22\t10836228\t1\t100M\t=\t10836084\t-244\tGTGACATAGTATAGTATCCTTAGTATCCTAACACACAGTAGAGTACCTAAAGTCAGAACTTTTAAGAGAGATACCTGTAAATTTGGACATTACATCAGAT\tEDFCDDDEEHFHEHHIGIJJIHEHHIJIHHGIIJGIIIJJIHIHGFJHGIICGIGIJIIHIFJJJJJJIHHFBJJIJIJJJJJIJIJGHHHHFFFFF@CC\tAS:i:-6\tZS:i:-6\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:86G13\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep1\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:1116:3649:92911\t163\t22\t10837841\t1\t100M\t=\t10837925\t184\tATATACCAACTCTAAGATAACCCACATGTTGGAAATATCACATGAGGTCTATTGTAACTAGCGAGGTAGAGAAAAGTTTACTTATAATGAATTAAAAGAT\tCCCFFFFFHHHHHIIHHIJJHHIIDHIJJIJJGIJJIHIJJGHIHGIFHIGHIIJJIJJJIJJFHIHIHHFHHHFFEFFEEEEEEEEEDDDCDDDDDDDD\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:1116:3649:92911\t83\t22\t10837925\t1\t100M\t=\t10837841\t-184\tTAATGAATTAAAAGATAGGAAGTACCAGCCGTGAAATAAAACAAAATCAGATGGCAATTCTAGAGCTGAAAAATATATCAGAATTAAAAATAGGCTCTAT\tEFDCEEEEDEEDEFFEFFFHHHHHHJJJJJJGHJJJIJJJIJJJIIIJJIIJJJJIIIJJJJJJJJIHJJJJJIJJJJJJJJJIHJJHHHHHFFFFFCCC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:6:2107:2342:27323\t163\t22\t10838055\t1\t100M\t=\t10838144\t189\tAAGTAAATGAACTTGAATATAGATTGATAGAAAATCTGAAGAGAGTTATGAAAGATTGGGGTGAAAAAATAGAACCATAGAGATGTATGGGGGCAGTTTT\t?B@DDFDFHHHHHJJJJJJJJJIJJIJJIIIJJJJJJJIIJGIEHHIIIJGIIJJJJJJJJFHIJJJJHGHHFFFFFDEEEECDDDDFEDDDDDBBCDDD\tAS:i:-6\tZS:i:-6\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:16T83\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep3\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:6:2107:2342:27323\t83\t22\t10838144\t1\t100M\t=\t10838055\t-189\tGGGGCAGTTTTGAAAGGTCTAATAGGCACAGTATAAGAGAAAGAGCCAGAAAAAATATTTGACTAAATGATGACAGAAAACTTTTCAGATTTGGTAAAAG\tDDEEEEEEFEDFFFHHHHGHJJIHEJJJJJJJJJJJJJJIJJJIGIGJJJJJJJJJJJIHHGJIJJJJJJJIIJJJJIHIJJJJJJJHHHHHFFFFFCCC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:-6\tYT:Z:CP\tRG:Z:UHR_Rep3\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2312:11656:62706\t163\t22\t10838443\t1\t100M\t=\t10838707\t364\tCCATAGAGGTGGAACAAAATCTTTAAAGTTCTGAAAGGAAGAAAAAAATCTGTCAACCTAGAATTTTTTATCCAGTGAAAATATTCTTTAAGCCTTTTGA\t@@7ADDDD?DDF;CBC@:;<ECHIAAFCCDFCBBCEFE;;DFGIF?DAFE>CFEE>@FFEE@EFF<CEDCBAAA>AB@@ABBB;>>BDBBBBBB@ABB><\tAS:i:-5\tZS:i:-5\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:92T7\tYS:i:-4\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2308:15338:36631\t99\t22\t10838665\t1\t100M\t=\t10838833\t268\tCTATAAATGCACCTATGTAGTTGAAAAGTTATGTATTTTGCAATGTGTCTTGATAGAATGTTGACTACAATTCAATGGCTAAAACAAAATTATCAAAGTC\tCCCFFFFFGHHHHJJJJIJJIJIJJJJJGHGJIIIIJJJJJJJJJIIIIIIJHIIIGJJJJJJJJJJJJGIJJJJHIIJJJJJJJHHHHHFFFFFFEEEC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:-\tNH:i:2\n",
    "HWI-ST718_146963544:5:2312:11656:62706\t83\t22\t10838707\t1\t96M4S\t=\t10838443\t-364\tATGTGTCTTGATAGAATGTTGACTACAATTCAATGGCTAAAACAAAATTATCAAAGTCAAAATTGCATGTCTTTTCTTGAATGATAGGTATGTATCCGGT\t>?CA?@;CCCADECDCBC?>FEHHFIHGGECFHIIGHHHF@BB*EFDCIIHDBF?GHGGGGGHD>?C???CIHF9<FA,HHGGGAFGBDGHHFDDFFCCC\tAS:i:-4\tZS:i:-4\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:96\tYS:i:-5\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2308:15338:36631\t147\t22\t10838833\t1\t100M\t=\t10838665\t-268\tTAGACCCCTCTTTCAAATTAAAGATAACAAAAGCCCAATTGAAAGGAAAACAAAATACATCAATGCTAACTAAATACAAATAATTAGCCTTCACTAATTC\tDDDDDDFFDHHCGHHHJIJJJJJJJJJJIIJJJJIJJIIIJJJJJJJJJJJJJIIIFJIIIJJJJJIIIJJJJJIHJJJJJIJJJJJHHGHHEFFFFCCC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:-\tNH:i:2\n",
    "\n",
    "# -f features to include\n",
    "# -F features to exclude\n",
    "# -f 3 include read paired, read mapped in proper pair\n",
    "# -F read unmapped, mate unmapped, not primary alignment\n",
    "\n",
    "[nathalia@aws] $  samtools view -f 1024 UHR.bam | head\n",
    "\n",
    "# -f 1024 include read is PCR or optical duplicate\n",
    "# nothing\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_ALIGN_DIR\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "\n",
    "\n",
    "[nathalia@aws] $  samtools flagstat UHR.bam\n",
    "1174925 + 0 in total (QC-passed reads + QC-failed reads)\n",
    "24511 + 0 secondary\n",
    "0 + 0 supplementary\n",
    "0 + 0 duplicates\n",
    "1173705 + 0 mapped (99.90% : N/A)\n",
    "1150414 + 0 paired in sequencing\n",
    "575207 + 0 read1\n",
    "575207 + 0 read2\n",
    "1143814 + 0 properly paired (99.43% : N/A)\n",
    "1148584 + 0 with itself and mate mapped\n",
    "610 + 0 singletons (0.05% : N/A)\n",
    "6 + 0 with mate mapped to a different chr\n",
    "6 + 0 with mate mapped to a different chr (mapQ>=5)\n",
    "\n",
    "\n",
    "[nathalia@aws] $  samtools flagstat HBR.bam\n",
    "793931 + 0 in total (QC-passed reads + QC-failed reads)\n",
    "7565 + 0 secondary\n",
    "0 + 0 supplementary\n",
    "0 + 0 duplicates\n",
    "793322 + 0 mapped (99.92% : N/A)\n",
    "786366 + 0 paired in sequencing\n",
    "393183 + 0 read1\n",
    "393183 + 0 read2\n",
    "783114 + 0 properly paired (99.59% : N/A)\n",
    "785492 + 0 with itself and mate mapped\n",
    "265 + 0 singletons (0.03% : N/A)\n",
    "0 + 0 with mate mapped to a different chr\n",
    "0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  ls *.bam\n",
    "HBR.bam  HBR_Rep1.bam  HBR_Rep2.bam  HBR_Rep3.bam  UHR.bam  UHR_Rep1.bam  UHR_Rep2.bam  UHR_Rep3.bam\n",
    "[nathalia@aws] $  fastqc *.bam\n",
    "Started analysis of HBR.bam\n",
    "Approx 5% complete for HBR.bam\n",
    "Approx 10% complete for HBR.bam\n",
    "Approx 15% complete for HBR.bam\n",
    "Approx 20% complete for HBR.bam\n",
    "Approx 25% complete for HBR.bam\n",
    "Approx 30% complete for HBR.bam\n",
    "Approx 35% complete for HBR.bam\n",
    "Approx 40% complete for HBR.bam\n",
    "Approx 45% complete for HBR.bam\n",
    "Approx 50% complete for HBR.bam\n",
    "Approx 55% complete for HBR.bam\n",
    "Approx 60% complete for HBR.bam\n",
    "Approx 65% complete for HBR.bam\n",
    "Approx 70% complete for HBR.bam\n",
    "Approx 75% complete for HBR.bam\n",
    "Approx 80% complete for HBR.bam\n",
    "Approx 85% complete for HBR.bam\n",
    "Approx 90% complete for HBR.bam\n",
    "Approx 95% complete for HBR.bam\n",
    "Approx 100% complete for HBR.bam\n",
    "Analysis complete for HBR.bam\n",
    "Started analysis of HBR_Rep1.bam\n",
    "Approx 5% complete for HBR_Rep1.bam\n",
    "Approx 10% complete for HBR_Rep1.bam\n",
    "Approx 15% complete for HBR_Rep1.bam\n",
    "Approx 20% complete for HBR_Rep1.bam\n",
    "Approx 25% complete for HBR_Rep1.bam\n",
    "Approx 30% complete for HBR_Rep1.bam\n",
    "Approx 35% complete for HBR_Rep1.bam\n",
    "Approx 40% complete for HBR_Rep1.bam\n",
    "Approx 45% complete for HBR_Rep1.bam\n",
    "Approx 50% complete for HBR_Rep1.bam\n",
    "Approx 55% complete for HBR_Rep1.bam\n",
    "Approx 60% complete for HBR_Rep1.bam\n",
    "Approx 65% complete for HBR_Rep1.bam\n",
    "Approx 70% complete for HBR_Rep1.bam\n",
    "Approx 75% complete for HBR_Rep1.bam\n",
    "Approx 80% complete for HBR_Rep1.bam\n",
    "Approx 85% complete for HBR_Rep1.bam\n",
    "Approx 90% complete for HBR_Rep1.bam\n",
    "Approx 95% complete for HBR_Rep1.bam\n",
    "Approx 100% complete for HBR_Rep1.bam\n",
    "Analysis complete for HBR_Rep1.bam\n",
    "Started analysis of HBR_Rep2.bam\n",
    "Approx 5% complete for HBR_Rep2.bam\n",
    "Approx 10% complete for HBR_Rep2.bam\n",
    "Approx 15% complete for HBR_Rep2.bam\n",
    "Approx 20% complete for HBR_Rep2.bam\n",
    "Approx 25% complete for HBR_Rep2.bam\n",
    "Approx 30% complete for HBR_Rep2.bam\n",
    "Approx 35% complete for HBR_Rep2.bam\n",
    "Approx 40% complete for HBR_Rep2.bam\n",
    "Approx 45% complete for HBR_Rep2.bam\n",
    "Approx 50% complete for HBR_Rep2.bam\n",
    "Approx 55% complete for HBR_Rep2.bam\n",
    "Approx 60% complete for HBR_Rep2.bam\n",
    "Approx 65% complete for HBR_Rep2.bam\n",
    "Approx 70% complete for HBR_Rep2.bam\n",
    "Approx 75% complete for HBR_Rep2.bam\n",
    "Approx 80% complete for HBR_Rep2.bam\n",
    "Approx 85% complete for HBR_Rep2.bam\n",
    "Approx 90% complete for HBR_Rep2.bam\n",
    "Approx 95% complete for HBR_Rep2.bam\n",
    "Approx 100% complete for HBR_Rep2.bam\n",
    "Analysis complete for HBR_Rep2.bam\n",
    "Started analysis of HBR_Rep3.bam\n",
    "Approx 5% complete for HBR_Rep3.bam\n",
    "Approx 10% complete for HBR_Rep3.bam\n",
    "Approx 15% complete for HBR_Rep3.bam\n",
    "Approx 20% complete for HBR_Rep3.bam\n",
    "Approx 25% complete for HBR_Rep3.bam\n",
    "Approx 30% complete for HBR_Rep3.bam\n",
    "Approx 35% complete for HBR_Rep3.bam\n",
    "Approx 40% complete for HBR_Rep3.bam\n",
    "Approx 45% complete for HBR_Rep3.bam\n",
    "Approx 50% complete for HBR_Rep3.bam\n",
    "Approx 55% complete for HBR_Rep3.bam\n",
    "Approx 60% complete for HBR_Rep3.bam\n",
    "Approx 65% complete for HBR_Rep3.bam\n",
    "Approx 70% complete for HBR_Rep3.bam\n",
    "Approx 75% complete for HBR_Rep3.bam\n",
    "Approx 80% complete for HBR_Rep3.bam\n",
    "Approx 85% complete for HBR_Rep3.bam\n",
    "Approx 90% complete for HBR_Rep3.bam\n",
    "Approx 95% complete for HBR_Rep3.bam\n",
    "Approx 100% complete for HBR_Rep3.bam\n",
    "Analysis complete for HBR_Rep3.bam\n",
    "Started analysis of UHR.bam\n",
    "Approx 5% complete for UHR.bam\n",
    "Approx 10% complete for UHR.bam\n",
    "Approx 15% complete for UHR.bam\n",
    "Approx 20% complete for UHR.bam\n",
    "Approx 25% complete for UHR.bam\n",
    "Approx 30% complete for UHR.bam\n",
    "Approx 35% complete for UHR.bam\n",
    "Approx 40% complete for UHR.bam\n",
    "Approx 45% complete for UHR.bam\n",
    "Approx 50% complete for UHR.bam\n",
    "Approx 55% complete for UHR.bam\n",
    "Approx 60% complete for UHR.bam\n",
    "Approx 65% complete for UHR.bam\n",
    "Approx 70% complete for UHR.bam\n",
    "Approx 75% complete for UHR.bam\n",
    "Approx 80% complete for UHR.bam\n",
    "Approx 85% complete for UHR.bam\n",
    "Approx 90% complete for UHR.bam\n",
    "Approx 95% complete for UHR.bam\n",
    "Analysis complete for UHR.bam\n",
    "Started analysis of UHR_Rep1.bam\n",
    "Approx 5% complete for UHR_Rep1.bam\n",
    "Approx 10% complete for UHR_Rep1.bam\n",
    "Approx 15% complete for UHR_Rep1.bam\n",
    "Approx 20% complete for UHR_Rep1.bam\n",
    "Approx 25% complete for UHR_Rep1.bam\n",
    "Approx 30% complete for UHR_Rep1.bam\n",
    "Approx 35% complete for UHR_Rep1.bam\n",
    "Approx 40% complete for UHR_Rep1.bam\n",
    "Approx 45% complete for UHR_Rep1.bam\n",
    "Approx 50% complete for UHR_Rep1.bam\n",
    "Approx 55% complete for UHR_Rep1.bam\n",
    "Approx 60% complete for UHR_Rep1.bam\n",
    "Approx 65% complete for UHR_Rep1.bam\n",
    "Approx 70% complete for UHR_Rep1.bam\n",
    "Approx 75% complete for UHR_Rep1.bam\n",
    "Approx 80% complete for UHR_Rep1.bam\n",
    "Approx 85% complete for UHR_Rep1.bam\n",
    "Approx 90% complete for UHR_Rep1.bam\n",
    "Approx 95% complete for UHR_Rep1.bam\n",
    "Approx 100% complete for UHR_Rep1.bam\n",
    "Analysis complete for UHR_Rep1.bam\n",
    "Started analysis of UHR_Rep2.bam\n",
    "Approx 5% complete for UHR_Rep2.bam\n",
    "Approx 10% complete for UHR_Rep2.bam\n",
    "Approx 15% complete for UHR_Rep2.bam\n",
    "Approx 20% complete for UHR_Rep2.bam\n",
    "Approx 25% complete for UHR_Rep2.bam\n",
    "Approx 30% complete for UHR_Rep2.bam\n",
    "Approx 35% complete for UHR_Rep2.bam\n",
    "Approx 40% complete for UHR_Rep2.bam\n",
    "Approx 45% complete for UHR_Rep2.bam\n",
    "Approx 50% complete for UHR_Rep2.bam\n",
    "Approx 55% complete for UHR_Rep2.bam\n",
    "Approx 60% complete for UHR_Rep2.bam\n",
    "Approx 65% complete for UHR_Rep2.bam\n",
    "Approx 70% complete for UHR_Rep2.bam\n",
    "Approx 75% complete for UHR_Rep2.bam\n",
    "Approx 80% complete for UHR_Rep2.bam\n",
    "Approx 85% complete for UHR_Rep2.bam\n",
    "Approx 90% complete for UHR_Rep2.bam\n",
    "Approx 95% complete for UHR_Rep2.bam\n",
    "Approx 100% complete for UHR_Rep2.bam\n",
    "Analysis complete for UHR_Rep2.bam\n",
    "Started analysis of UHR_Rep3.bam\n",
    "Approx 5% complete for UHR_Rep3.bam\n",
    "Approx 10% complete for UHR_Rep3.bam\n",
    "Approx 15% complete for UHR_Rep3.bam\n",
    "Approx 20% complete for UHR_Rep3.bam\n",
    "Approx 25% complete for UHR_Rep3.bam\n",
    "Approx 30% complete for UHR_Rep3.bam\n",
    "Approx 35% complete for UHR_Rep3.bam\n",
    "Approx 40% complete for UHR_Rep3.bam\n",
    "Approx 45% complete for UHR_Rep3.bam\n",
    "Approx 50% complete for UHR_Rep3.bam\n",
    "Approx 55% complete for UHR_Rep3.bam\n",
    "Approx 60% complete for UHR_Rep3.bam\n",
    "Approx 65% complete for UHR_Rep3.bam\n",
    "Approx 70% complete for UHR_Rep3.bam\n",
    "Approx 75% complete for UHR_Rep3.bam\n",
    "Approx 80% complete for UHR_Rep3.bam\n",
    "Approx 85% complete for UHR_Rep3.bam\n",
    "Approx 90% complete for UHR_Rep3.bam\n",
    "Approx 95% complete for UHR_Rep3.bam\n",
    "Approx 100% complete for UHR_Rep3.bam\n",
    "Analysis complete for UHR_Rep3.bam\n",
    "[nathalia@aws] $  lhtr\n",
    "total 990M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.0K Nov 13 14:24 alignment.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 166M Nov 13 14:24 UHR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 119M Nov 13 14:24 UHR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 136M Nov 13 14:24 UHR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  86M Nov 13 14:25 HBR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105M Nov 13 14:25 HBR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  94M Nov 13 14:25 HBR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  303 Nov 13 14:27 sam_bam.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  34M Nov 13 14:28 UHR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25M Nov 13 14:28 UHR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  28M Nov 13 14:28 UHR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  17M Nov 13 14:28 HBR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20M Nov 13 14:28 HBR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  18M Nov 13 14:28 HBR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  85M Nov 13 14:30 UHR.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53M Nov 13 14:30 HBR.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  47K Nov 13 15:34 HBR.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  39K Nov 13 15:34 HBR_Rep1.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  40K Nov 13 15:34 HBR_Rep2.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  40K Nov 13 15:34 HBR_Rep3.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  51K Nov 13 15:34 UHR.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  42K Nov 13 15:34 UHR_Rep1.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  44K Nov 13 15:34 UHR_Rep2.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  43K Nov 13 15:34 UHR_Rep3.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 447K Nov 13 17:29 HBR_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 638K Nov 13 17:29 HBR_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 454K Nov 13 17:29 HBR_Rep1_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 645K Nov 13 17:29 HBR_Rep1_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 452K Nov 13 17:29 HBR_Rep2_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 643K Nov 13 17:29 HBR_Rep2_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 453K Nov 13 17:29 HBR_Rep3_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 643K Nov 13 17:29 HBR_Rep3_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 462K Nov 13 17:29 UHR_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 646K Nov 13 17:29 UHR_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 469K Nov 13 17:29 UHR_Rep1_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 649K Nov 13 17:29 UHR_Rep1_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 472K Nov 13 17:29 UHR_Rep2_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 651K Nov 13 17:29 UHR_Rep2_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 470K Nov 13 17:29 UHR_Rep3_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 653K Nov 13 17:29 UHR_Rep3_fastqc.html\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/alignments/hisat2/HBR_fastqc.html\n",
    "\n",
    "http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/alignments/hisat2/UHR_fastqc.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME\n",
    "\n",
    "[nathalia@aws] $  wget http://genomedata.org/rnaseq-tutorial/RSeQC.zip\n",
    "--2018-11-13 17:21:07--  http://genomedata.org/rnaseq-tutorial/RSeQC.zip\n",
    "Resolving genomedata.org (genomedata.org)... 3.8.108.6\n",
    "Connecting to genomedata.org (genomedata.org)|3.8.108.6|:80... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 442199597 (422M) [application/zip]\n",
    "Saving to: RSeQC.zip\n",
    "\n",
    "RSeQC.zip                  100%[========================================>] 421.71M  2.05MB/s    in 6m 39s  \n",
    "\n",
    "2018-11-13 17:27:52 (1.06 MB/s) - RSeQC.zip saved [442199597/442199597]\n",
    "\n",
    "[nathalia@aws] $  unzip RSeQC.zip\n",
    "Archive:  RSeQC.zip\n",
    "   creating: RSeQC/\n",
    "  inflating: RSeQC/Pairend_nonStrandSpecific_36mer_Human_hg19.bam  \n",
    "  inflating: RSeQC/hg19_UCSC_knownGene.bed.gz  \n",
    "  inflating: RSeQC/Pairend_nonStrandSpecific_36mer_Human_hg19.bam.bai  \n",
    "\n",
    "\n",
    "[nathalia@aws] $  cd RSeQC/\n",
    "\n",
    "[nathalia@aws] $  gunzip hg19_UCSC_knownGene.bed.gz\n",
    "\n",
    "[nathalia@aws] $  emacs rseqc.sh\n",
    "\n",
    "\n",
    "bam_stat.py -i Pairend_nonStrandSpecific_36mer_Human_hg19.bam\n",
    "clipping_profile.py -i Pairend_nonStrandSpecific_36mer_Human_hg19.bam -o tutorial -s \"PE\"\n",
    "geneBody_coverage.py -r hg19_UCSC_knownGene.bed -i Pairend_nonStrandSpecific_36mer_Human_hg19.bam -o tutorial\n",
    "infer_experiment.py -r hg19_UCSC_knownGene.bed -i Pairend_nonStrandSpecific_36mer_Human_hg19.bam\n",
    "inner_distance.py -r hg19_UCSC_knownGene.bed -i Pairend_nonStrandSpecific_36mer_Human_hg19.bam -o tutorial\n",
    "junction_annotation.py -r hg19_UCSC_knownGene.bed -i Pairend_nonStrandSpecific_36mer_Human_hg19.bam -o tutorial\n",
    "junction_saturation.py -r hg19_UCSC_knownGene.bed -i Pairend_nonStrandSpecific_36mer_Human_hg19.bam -o tutorial\n",
    "read_distribution.py -r hg19_UCSC_knownGene.bed -i Pairend_nonStrandSpecific_36mer_Human_hg19.bam\n",
    "read_duplication.py -i Pairend_nonStrandSpecific_36mer_Human_hg19.bam -o tutorial\n",
    "read_GC.py -i Pairend_nonStrandSpecific_36mer_Human_hg19.bam -o tutorial\n",
    "read_NVC.py -i Pairend_nonStrandSpecific_36mer_Human_hg19.bam -o tutorial\n",
    "read_quality.py -i Pairend_nonStrandSpecific_36mer_Human_hg19.bam -o tutorial\n",
    "ls *.pdf\n",
    "\n",
    "```\n",
    "\n",
    "This is taking a long time... like almost 1 h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Expression with stringtie"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME\n",
    "\n",
    "[nathalia@aws] $  mkdir -p expression/stringtie/ref_only/\n",
    "\n",
    "[nathalia@aws] $  cd expression/stringtie/ref_only/\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  emacs stringtie.sh\n",
    "stringtie -p 8 -G $RNA_REF_GTF -e -B -o HBR_Rep1/transcripts.gtf -A HBR_Rep1/gene_abundances.tsv $RNA_ALIGN_DIR/HBR_Rep1.bam\n",
    "stringtie -p 8 -G $RNA_REF_GTF -e -B -o HBR_Rep2/transcripts.gtf -A HBR_Rep2/gene_abundances.tsv $RNA_ALIGN_DIR/HBR_Rep2.bam\n",
    "stringtie -p 8 -G $RNA_REF_GTF -e -B -o HBR_Rep3/transcripts.gtf -A HBR_Rep3/gene_abundances.tsv $RNA_ALIGN_DIR/HBR_Rep3.bam\n",
    "\n",
    "stringtie -p 8 -G $RNA_REF_GTF -e -B -o UHR_Rep1/transcripts.gtf -A UHR_Rep1/gene_abundances.tsv $RNA_ALIGN_DIR/UHR_Rep1.bam\n",
    "stringtie -p 8 -G $RNA_REF_GTF -e -B -o UHR_Rep2/transcripts.gtf -A UHR_Rep2/gene_abundances.tsv $RNA_ALIGN_DIR/UHR_Rep2.bam\n",
    "stringtie -p 8 -G $RNA_REF_GTF -e -B -o UHR_Rep3/transcripts.gtf -A UHR_Rep3/gene_abundances.tsv $RNA_ALIGN_DIR/UHR_Rep3.bam\n",
    "[nathalia@aws] $  bash stringtie.sh \n",
    "\n",
    "\n",
    "# reference-only mode because of -G and -e options\n",
    "\n",
    "\n",
    "[nathalia@aws] $  ls -R\n",
    ".:\n",
    "HBR_Rep1  HBR_Rep2  HBR_Rep3  UHR_Rep1  UHR_Rep2  UHR_Rep3  stringtie.sh\n",
    "\n",
    "./HBR_Rep1:\n",
    "e2t.ctab     gene_abundances.tsv  i_data.ctab  transcripts.gtf\n",
    "e_data.ctab  i2t.ctab             t_data.ctab\n",
    "\n",
    "./HBR_Rep2:\n",
    "e2t.ctab     gene_abundances.tsv  i_data.ctab  transcripts.gtf\n",
    "e_data.ctab  i2t.ctab             t_data.ctab\n",
    "\n",
    "./HBR_Rep3:\n",
    "e2t.ctab     gene_abundances.tsv  i_data.ctab  transcripts.gtf\n",
    "e_data.ctab  i2t.ctab             t_data.ctab\n",
    "\n",
    "./UHR_Rep1:\n",
    "e2t.ctab     gene_abundances.tsv  i_data.ctab  transcripts.gtf\n",
    "e_data.ctab  i2t.ctab             t_data.ctab\n",
    "\n",
    "./UHR_Rep2:\n",
    "e2t.ctab     gene_abundances.tsv  i_data.ctab  transcripts.gtf\n",
    "e_data.ctab  i2t.ctab             t_data.ctab\n",
    "\n",
    "./UHR_Rep3:\n",
    "e2t.ctab     gene_abundances.tsv  i_data.ctab  transcripts.gtf\n",
    "e_data.ctab  i2t.ctab             t_data.ctab\n",
    "\n",
    "[nathalia@aws] $  less -S UHR_Rep1/transcripts.gtf\n",
    "\n",
    "[nathalia@aws] $  awk '{if ($3==\"transcript\") print}' UHR_Rep1/transcripts.gtf | cut -f 1,4,9 | less\n",
    "\n",
    "[nathalia@aws] $  less -S UHR_Rep1/t_data.ctab\n",
    "\n",
    "# transcript level expression\n",
    "\n",
    "[nathalia@aws] $  less -S UHR_Rep1/gene_abundances.tsv\n",
    "\n",
    "# gene level expression\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  wget https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/stringtie_expression_matrix.pl\n",
    "--2018-11-13 19:34:08--  https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/stringtie_expression_matrix.pl\n",
    "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.52.133\n",
    "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.52.133|:443... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 6704 (6.5K) [text/plain]\n",
    "Saving to: stringtie_expression_matrix.pl\n",
    "\n",
    "stringtie_expression_matr 100%[=====================================>]   6.55K  --.-KB/s    in 0s      \n",
    "\n",
    "2018-11-13 19:34:09 (139 MB/s) - stringtie_expression_matrix.pl saved [6704/6704]\n",
    "\n",
    "[nathalia@aws] $  chmod +x stringtie_expression_matrix.pl\n",
    "\n",
    "[nathalia@aws] $  emacs expr_matrix.sh\n",
    "\n",
    "#!/bin/bash\n",
    "\n",
    "./stringtie_expression_matrix.pl --expression_metric=TPM --result_dirs='HBR_Rep1,HBR_Rep2,HBR_Rep3,UHR_Rep1,UHR_Rep2,UHR_Rep3' --transcript_matrix_file=transcript_tpm_all_samples.tsv --gene_matrix_file=gene_tpm_all_samples.tsv\n",
    "\n",
    "./stringtie_expression_matrix.pl --expression_metric=FPKM --result_dirs='HBR_Rep1,HBR_Rep2,HBR_Rep3,UHR_Rep1,UHR_Rep2,UHR_Rep3' --transcript_matrix_file=transcript_fpkm_all_samples.tsv --gene_matrix_file=gene_fpkm_all_samples.tsv\n",
    "\n",
    "./stringtie_expression_matrix.pl --expression_metric=Coverage --result_dirs='HBR_Rep1,HBR_Rep2,HBR_Rep3,UHR_Rep1,UHR_Rep2,UHR_Rep3' --transcript_matrix_file=transcript_coverage_all_samples.tsv --gene_matrix_file=gene_coverage_all_samples.tsv\n",
    "\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bash expr_matrix.sh \n",
    "\n",
    "\n",
    "Processing data for the following 6 samples:\n",
    "HBR_Rep1 HBR_Rep2 HBR_Rep3 UHR_Rep1 UHR_Rep2 UHR_Rep3\n",
    "\n",
    "Gathered TPM expression values for 4564 unique transcripts\n",
    "\n",
    "Gathered TPM expression values for 1410 unique genes\n",
    "\n",
    "Printed transcript TPM expression matrix to transcript_tpm_all_samples.tsv\n",
    "\n",
    "Printed gene TPM expression matrix to gene_tpm_all_samples.tsv\n",
    "\n",
    "\n",
    "\n",
    "Processing data for the following 6 samples:\n",
    "HBR_Rep1 HBR_Rep2 HBR_Rep3 UHR_Rep1 UHR_Rep2 UHR_Rep3\n",
    "\n",
    "Gathered FPKM expression values for 4564 unique transcripts\n",
    "\n",
    "Gathered FPKM expression values for 1410 unique genes\n",
    "\n",
    "Printed transcript FPKM expression matrix to transcript_fpkm_all_samples.tsv\n",
    "\n",
    "Printed gene FPKM expression matrix to gene_fpkm_all_samples.tsv\n",
    "\n",
    "\n",
    "\n",
    "Processing data for the following 6 samples:\n",
    "HBR_Rep1 HBR_Rep2 HBR_Rep3 UHR_Rep1 UHR_Rep2 UHR_Rep3\n",
    "\n",
    "Gathered Coverage expression values for 4564 unique transcripts\n",
    "\n",
    "Gathered Coverage expression values for 1410 unique genes\n",
    "\n",
    "Printed transcript Coverage expression matrix to transcript_coverage_all_samples.tsv\n",
    "\n",
    "Printed gene Coverage expression matrix to gene_coverage_all_samples.tsv\n",
    "\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 1.2M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  751 Nov 13 19:26 stringtie.sh\n",
    "drwxr-xr-x 2 ubuntu ubuntu 4.0K Nov 13 19:26 HBR_Rep1\n",
    "drwxr-xr-x 2 ubuntu ubuntu 4.0K Nov 13 19:26 HBR_Rep2\n",
    "drwxr-xr-x 2 ubuntu ubuntu 4.0K Nov 13 19:26 HBR_Rep3\n",
    "drwxr-xr-x 2 ubuntu ubuntu 4.0K Nov 13 19:26 UHR_Rep1\n",
    "drwxr-xr-x 2 ubuntu ubuntu 4.0K Nov 13 19:26 UHR_Rep2\n",
    "drwxr-xr-x 2 ubuntu ubuntu 4.0K Nov 13 19:26 UHR_Rep3\n",
    "-rwxrwxr-x 1 ubuntu ubuntu 6.6K Nov 13 19:34 stringtie_expression_matrix.pl\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  714 Nov 13 19:35 expr_matrix.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 307K Nov 13 19:35 transcript_tpm_all_samples.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  89K Nov 13 19:35 gene_tpm_all_samples.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 306K Nov 13 19:35 transcript_fpkm_all_samples.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  89K Nov 13 19:35 gene_fpkm_all_samples.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 236K Nov 13 19:35 transcript_coverage_all_samples.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  84K Nov 13 19:35 gene_coverage_all_samples.tsv\n",
    "\n",
    "[nathalia@aws] $  head transcript_tpm_all_samples.tsv gene_tpm_all_samples.tsv\n",
    "==> transcript_tpm_all_samples.tsv <==\n",
    "Transcript_ID\tHBR_Rep1\tHBR_Rep2\tHBR_Rep3\tUHR_Rep1\tUHR_Rep2\tUHR_Rep3\n",
    "ENST00000006251\t0.000000\t0.000000\t0.000000\t0.000000\t0.000000\t0.000000\n",
    "ENST00000008876\t131.426895\t163.158142\t136.729172\t0.000000\t0.000000\t0.000000\n",
    "ENST00000043402\t405.076324\t419.098175\t426.167725\t48.028503\t31.323360\t35.597595\n",
    "ENST00000086933\t0.0\t0.0\t0.0\t0.0\t0.0\t0.0\n",
    "ENST00000155674\t0.000430\t0.000000\t0.000000\t3.234303\t173.356018\t0.000000\n",
    "ENST00000159647\t158.713745\t69.338867\t3.367408\t9.448453\t0.000000\t0.000000\n",
    "ENST00000207636\t11.119016\t8.582876\t17.949173\t62.407532\t0.000000\t75.185722\n",
    "ENST00000215659\t0.000000\t122.947205\t62.079456\t11.538836\t5.100820\t1.432057\n",
    "ENST00000215727\t0.000000\t8.245975\t0.000000\t306.032196\t326.813080\t314.283936\n",
    "\n",
    "==> gene_tpm_all_samples.tsv <==\n",
    "Gene_ID\tHBR_Rep1\tHBR_Rep2\tHBR_Rep3\tUHR_Rep1\tUHR_Rep2\tUHR_Rep3\n",
    "ENSG00000008735\t1378.358276\t1502.894653\t1497.128784\t18.058046\t27.210226\t9.181973\n",
    "ENSG00000015475\t183.293930\t171.647629\t254.215347\t420.351257\t262.014618\t307.026245\n",
    "ENSG00000025708\t91.459953\t53.572464\t133.441071\t164.339783\t67.974785\t100.541763\n",
    "ENSG00000025770\t543.730225\t481.488129\t503.951569\t788.373291\t596.408813\t792.156189\n",
    "ENSG00000040608\t424.655518\t436.600983\t429.485779\t48.799942\t49.200336\t35.597595\n",
    "ENSG00000054611\t119.961060\t137.551483\t143.356598\t114.563286\t105.671730\t128.256882\n",
    "ENSG00000056487\t59.768002\t32.060265\t9.015366\t14.082806\t24.280190\t49.806057\n",
    "ENSG00000063515\t0.0\t0.0\t0.0\t0.0\t0.0\t0.0\n",
    "ENSG00000069998\t134.154922\t134.716873\t184.259293\t283.136200\t333.809479\t511.102905\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Expression with HTSEQ-COUNT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME/\n",
    "[nathalia@aws] $  mkdir -p expression/htseq_counts\n",
    "[nathalia@aws] $  cd expression/htseq_counts\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/expression/htseq_counts\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  emacs htseq.sh\n",
    "[nathalia@aws] $  cat htseq.sh \n",
    "#!/bin/bash\n",
    "\n",
    "htseq-count --format bam --order pos --mode intersection-strict --stranded reverse --minaqual 1 --type exon --idattr gene_id $RNA_ALIGN_DIR/UHR_Rep1.bam $RNA_REF_GTF > UHR_Rep1_gene.tsv\n",
    "htseq-count --format bam --order pos --mode intersection-strict --stranded reverse --minaqual 1 --type exon --idattr gene_id $RNA_ALIGN_DIR/UHR_Rep2.bam $RNA_REF_GTF > UHR_Rep2_gene.tsv\n",
    "htseq-count --format bam --order pos --mode intersection-strict --stranded reverse --minaqual 1 --type exon --idattr gene_id $RNA_ALIGN_DIR/UHR_Rep3.bam $RNA_REF_GTF > UHR_Rep3_gene.tsv\n",
    "\n",
    "htseq-count --format bam --order pos --mode intersection-strict --stranded reverse --minaqual 1 --type exon --idattr gene_id $RNA_ALIGN_DIR/HBR_Rep1.bam $RNA_REF_GTF > HBR_Rep1_gene.tsv\n",
    "htseq-count --format bam --order pos --mode intersection-strict --stranded reverse --minaqual 1 --type exon --idattr gene_id $RNA_ALIGN_DIR/HBR_Rep2.bam $RNA_REF_GTF > HBR_Rep2_gene.tsv\n",
    "htseq-count --format bam --order pos --mode intersection-strict --stranded reverse --minaqual 1 --type exon --idattr gene_id $RNA_ALIGN_DIR/HBR_Rep3.bam $RNA_REF_GTF > HBR_Rep3_gene.tsv\n",
    "[nathalia@aws] $  bash htseq.sh \n",
    "56295 GFF lines processed.\n",
    "100000 SAM alignment record pairs processed.\n",
    "200000 SAM alignment record pairs processed.\n",
    "Warning: Mate records missing for 794 records; first such record: <SAM_Alignment object: Paired-end read 'HWI-ST718_146963544:5:2104:10412:46400' aligned to 22:[22901080,22901180)/->.\n",
    "Warning: Mate pairing was ambiguous for 253 records; mate key for first such record: ('HWI-ST718_146963544:5:1113:19935:62480', 'first', '22', 17149631, '22', 17149650, 119).\n",
    "232476 SAM alignment pairs processed.\n",
    "56295 GFF lines processed.\n",
    "100000 SAM alignment record pairs processed.\n",
    "Warning: Mate records missing for 1130 records; first such record: <SAM_Alignment object: Paired-end read 'HWI-ST718_146963544:6:2208:13869:19680' aligned to 22:[22906676,22906771)/->.\n",
    "Warning: Mate pairing was ambiguous for 137 records; mate key for first such record: ('HWI-ST718_146963544:5:2313:1955:9570', 'first', '22', 17148472, '22', 17149650, 297).\n",
    "166624 SAM alignment pairs processed.\n",
    "56295 GFF lines processed.\n",
    "100000 SAM alignment record pairs processed.\n",
    "Warning: Mate records missing for 1245 records; first such record: <SAM_Alignment object: Paired-end read 'HWI-ST718_146963544:6:1210:6504:30099' aligned to 22:[36304121,36305098)/->.\n",
    "Warning: Mate pairing was ambiguous for 142 records; mate key for first such record: ('HWI-ST718_146963544:6:2201:10770:21087', 'first', '22', 17141096, '22', 17141134, 138).\n",
    "189948 SAM alignment pairs processed.\n",
    "56295 GFF lines processed.\n",
    "100000 SAM alignment record pairs processed.\n",
    "Warning: Mate records missing for 714 records; first such record: <SAM_Alignment object: Paired-end read 'HWI-ST718_146963544:8:2110:6678:83017' aligned to ERCC-00095:[168,268)/->.\n",
    "Warning: Mate pairing was ambiguous for 90 records; mate key for first such record: ('HWI-ST718_146963544:7:1213:16585:58791', 'first', '22', 17188425, '22', 17188443, 1576).\n",
    "120056 SAM alignment pairs processed.\n",
    "56295 GFF lines processed.\n",
    "100000 SAM alignment record pairs processed.\n",
    "Warning: Mate records missing for 1048 records; first such record: <SAM_Alignment object: Paired-end read 'HWI-ST718_146963544:7:1207:18904:36296' aligned to 22:[46085024,46085124)/->.\n",
    "Warning: Mate pairing was ambiguous for 112 records; mate key for first such record: ('HWI-ST718_146963544:8:1103:9833:24924', 'first', '22', 17592538, '22', 17594612, 382).\n",
    "146764 SAM alignment pairs processed.\n",
    "56295 GFF lines processed.\n",
    "100000 SAM alignment record pairs processed.\n",
    "Warning: Mate records missing for 755 records; first such record: <SAM_Alignment object: Paired-end read 'HWI-ST718_146963544:8:1213:2059:11831' aligned to 22:[50287688,50287785)/+>.\n",
    "Warning: Mate pairing was ambiguous for 92 records; mate key for first such record: ('HWI-ST718_146963544:7:2204:19299:75541', 'first', '22', 17592545, '22', 17594612, 375).\n",
    "131404 SAM alignment pairs processed.\n",
    "\n",
    "[nathalia@aws] $  join UHR_Rep1_gene.tsv UHR_Rep2_gene.tsv | join - UHR_Rep3_gene.tsv | join - HBR_Rep1_gene.tsv | join - HBR_Rep2_gene.tsv | join - HBR_Rep3_gene.tsv > gene_read_counts_table_all.tsv\n",
    "\n",
    "[nathalia@aws] $  echo \"GeneID UHR_Rep1 UHR_Rep2 UHR_Rep3 HBR_Rep1 HBR_Rep2 HBR_Rep3\" > header.txt\n",
    "\n",
    "[nathalia@aws] $  cat header.txt gene_read_counts_table_all.tsv | grep -v \"__\" | perl -ne 'chomp $_; $_ =~ s/\\s+/\\t/g; print \"$_\\n\"' > gene_read_counts_table_all_final.tsv\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 264K\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.2K Nov 13 19:39 htseq.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  26K Nov 13 19:39 UHR_Rep1_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:40 UHR_Rep2_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:40 UHR_Rep3_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:40 HBR_Rep1_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:40 HBR_Rep2_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:41 HBR_Rep3_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  42K Nov 13 19:42 gene_read_counts_table_all.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   61 Nov 13 19:42 header.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  42K Nov 13 19:42 gene_read_counts_table_all_final.tsv\n",
    "\n",
    "[nathalia@aws] $  head gene_read_counts_table_all.tsv gene_read_counts_table_all_final.tsv \n",
    "==> gene_read_counts_table_all.tsv <==\n",
    "ENSG00000008735 13 17 8 399 533 467\n",
    "ENSG00000015475 103 65 100 40 43 48\n",
    "ENSG00000025708 46 16 26 13 11 19\n",
    "ENSG00000025770 189 118 154 73 78 71\n",
    "ENSG00000040608 18 8 11 69 92 81\n",
    "ENSG00000054611 70 47 66 36 43 40\n",
    "ENSG00000056487 12 13 7 11 10 4\n",
    "ENSG00000063515 0 0 0 0 0 0\n",
    "ENSG00000069998 123 76 137 25 40 33\n",
    "ENSG00000070010 228 163 171 43 65 54\n",
    "\n",
    "==> gene_read_counts_table_all_final.tsv <==\n",
    "GeneID\tUHR_Rep1\tUHR_Rep2\tUHR_Rep3\tHBR_Rep1\tHBR_Rep2\tHBR_Rep3\n",
    "ENSG00000008735\t13\t17\t8\t399\t533\t467\n",
    "ENSG00000015475\t103\t65\t100\t40\t43\t48\n",
    "ENSG00000025708\t46\t16\t26\t13\t11\t19\n",
    "ENSG00000025770\t189\t118\t154\t73\t78\t71\n",
    "ENSG00000040608\t18\t8\t11\t69\t92\t81\n",
    "ENSG00000054611\t70\t47\t66\t36\t43\t40\n",
    "ENSG00000056487\t12\t13\t7\t11\t10\t4\n",
    "ENSG00000063515\t0\t0\t0\t0\t0\t0\n",
    "ENSG00000069998\t123\t76\t137\t25\t40\t33\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Practice 8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/practice3/data\n",
    "\n",
    "[nathalia@aws] $  stringtie -p 8 -G $RNA_REF_GTF -e -B -o normal_rep1/transcripts.gtf -A normal_rep1/gene_abundances.tsv hcc1395_normal_rep1.bam\n",
    "[nathalia@aws] $  stringtie -p 8 -G $RNA_REF_GTF -e -B -o normal_rep2/transcripts.gtf -A normal_rep2/gene_abundances.tsv hcc1395_normal_rep2.bam\n",
    "[nathalia@aws] $  stringtie -p 8 -G $RNA_REF_GTF -e -B -o normal_rep3/transcripts.gtf -A normal_rep3/gene_abundances.tsv hcc1395_normal_rep3.bam\n",
    "\n",
    "[nathalia@aws] $  stringtie -p 8 -G $RNA_REF_GTF -e -B -o tumor_rep1/transcripts.gtf -A tumor_rep1/gene_abundances.tsv hcc1395_tumor_rep1.bam\n",
    "[nathalia@aws] $  stringtie -p 8 -G $RNA_REF_GTF -e -B -o tumor_rep2/transcripts.gtf -A tumor_rep2/gene_abundances.tsv hcc1395_tumor_rep2.bam\n",
    "[nathalia@aws] $  stringtie -p 8 -G $RNA_REF_GTF -e -B -o tumor_rep3/transcripts.gtf -A tumor_rep3/gene_abundances.tsv hcc1395_tumor_rep3.bam\n",
    "\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 2.6G\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep3_2.fastq.gz -> ../trimmed/hcc1395_tumor_rep3_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep3_1.fastq.gz -> ../trimmed/hcc1395_tumor_rep3_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep2_2.fastq.gz -> ../trimmed/hcc1395_tumor_rep2_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep2_1.fastq.gz -> ../trimmed/hcc1395_tumor_rep2_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep1_2.fastq.gz -> ../trimmed/hcc1395_tumor_rep1_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep1_1.fastq.gz -> ../trimmed/hcc1395_tumor_rep1_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep3_2.fastq.gz -> ../trimmed/hcc1395_normal_rep3_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep3_1.fastq.gz -> ../trimmed/hcc1395_normal_rep3_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep2_2.fastq.gz -> ../trimmed/hcc1395_normal_rep2_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep2_1.fastq.gz -> ../trimmed/hcc1395_normal_rep2_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep1_2.fastq.gz -> ../trimmed/hcc1395_normal_rep1_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep1_1.fastq.gz -> ../trimmed/hcc1395_normal_rep1_1.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.3K Nov 13 15:15 alignment.sh~\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.4K Nov 13 15:22 alignment.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 314M Nov 13 15:22 hcc1395_normal_rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 315M Nov 13 15:23 hcc1395_normal_rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 315M Nov 13 15:23 hcc1395_normal_rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 369M Nov 13 15:26 hcc1395_tumor_rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 368M Nov 13 15:26 hcc1395_tumor_rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 368M Nov 13 15:27 hcc1395_tumor_rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  49M Nov 13 15:27 hcc1395_normal_rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  47M Nov 13 15:27 hcc1395_normal_rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  47M Nov 13 15:27 hcc1395_normal_rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  55M Nov 13 15:27 hcc1395_tumor_rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53M Nov 13 15:27 hcc1395_tumor_rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53M Nov 13 15:27 hcc1395_tumor_rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 141M Nov 13 15:28 hcc_normal.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 159M Nov 13 15:28 hcc_tumor.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  50K Nov 13 16:01 hcc1395_normal_rep1.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  49K Nov 13 16:01 hcc1395_normal_rep2.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  50K Nov 13 16:01 hcc1395_normal_rep3.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53K Nov 13 16:01 hcc1395_tumor_rep1.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53K Nov 13 16:01 hcc1395_tumor_rep2.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53K Nov 13 16:01 hcc1395_tumor_rep3.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  80K Nov 13 16:01 hcc_normal.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  81K Nov 13 16:01 hcc_tumor.bam.bai\n",
    "drwxr-xr-x 2 ubuntu ubuntu 4.0K Nov 13 19:58 normal_rep1\n",
    "drwxr-xr-x 2 ubuntu ubuntu 4.0K Nov 13 19:58 normal_rep2\n",
    "drwxr-xr-x 2 ubuntu ubuntu 4.0K Nov 13 19:58 normal_rep3\n",
    "drwxr-xr-x 2 ubuntu ubuntu 4.0K Nov 13 19:59 tumor_rep1\n",
    "drwxr-xr-x 2 ubuntu ubuntu 4.0K Nov 13 19:59 tumor_rep2\n",
    "drwxr-xr-x 2 ubuntu ubuntu 4.0K Nov 13 19:59 tumor_rep3\n",
    "\n",
    "\n",
    "[nathalia@aws] $  wget https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/stringtie_expression_matrix.pl\n",
    "--2018-11-13 20:00:31--  https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/stringtie_expression_matrix.pl\n",
    "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.52.133\n",
    "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.52.133|:443... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 6704 (6.5K) [text/plain]\n",
    "Saving to: stringtie_expression_matrix.pl\n",
    "\n",
    "stringtie_expression_matr 100%[=====================================>]   6.55K  --.-KB/s    in 0s      \n",
    "\n",
    "2018-11-13 20:00:31 (154 MB/s) - stringtie_expression_matrix.pl saved [6704/6704]\n",
    "\n",
    "[nathalia@aws] $  chmod +x stringtie_expression_matrix.pl\n",
    "\n",
    "\n",
    "[nathalia@aws] $  ./stringtie_expression_matrix.pl --expression_metric=TPM --result_dirs='normal_rep1,normal_rep2,normal_rep3,tumor_rep1,tumor_rep2,tumor_rep3' --transcript_matrix_file=transcript_tpm_all_samples.tsv --gene_matrix_file=gene_tpm_all_samples.tsv\n",
    "\n",
    "\n",
    "Processing data for the following 6 samples:\n",
    "normal_rep1 normal_rep2 normal_rep3 tumor_rep1 tumor_rep2 tumor_rep3\n",
    "\n",
    "Gathered TPM expression values for 4564 unique transcripts\n",
    "\n",
    "Gathered TPM expression values for 1410 unique genes\n",
    "\n",
    "Printed transcript TPM expression matrix to transcript_tpm_all_samples.tsv\n",
    "\n",
    "Printed gene TPM expression matrix to gene_tpm_all_samples.tsv\n",
    "\n",
    " \n",
    "[nathalia@aws] $  ./stringtie_expression_matrix.pl --expression_metric=FPKM --result_dirs='normal_rep1,normal_rep2,normal_rep3,tumor_rep1,tumor_rep2,tumor_rep3' --transcript_matrix_file=transcript_fpkm_all_samples.tsv --gene_matrix_file=gene_fpkm_all_samples.tsv\n",
    "\n",
    "\n",
    "Processing data for the following 6 samples:\n",
    "normal_rep1 normal_rep2 normal_rep3 tumor_rep1 tumor_rep2 tumor_rep3\n",
    "\n",
    "Gathered FPKM expression values for 4564 unique transcripts\n",
    "\n",
    "Gathered FPKM expression values for 1410 unique genes\n",
    "\n",
    "Printed transcript FPKM expression matrix to transcript_fpkm_all_samples.tsv\n",
    "\n",
    "Printed gene FPKM expression matrix to gene_fpkm_all_samples.tsv\n",
    "\n",
    "\n",
    "[nathalia@aws] $  ./stringtie_expression_matrix.pl --expression_metric=Coverage --result_dirs='normal_rep1,normal_rep2,normal_rep3,tumor_rep1,tumor_rep2,tumor_rep3' --transcript_matrix_file=transcript_coverage_all_samples.tsv --gene_matrix_file=gene_coverage_all_samples.tsv\n",
    "\n",
    "\n",
    "Processing data for the following 6 samples:\n",
    "normal_rep1 normal_rep2 normal_rep3 tumor_rep1 tumor_rep2 tumor_rep3\n",
    "\n",
    "Gathered Coverage expression values for 4564 unique transcripts\n",
    "\n",
    "Gathered Coverage expression values for 1410 unique genes\n",
    "\n",
    "Printed transcript Coverage expression matrix to transcript_coverage_all_samples.tsv\n",
    "\n",
    "Printed gene Coverage expression matrix to gene_coverage_all_samples.tsv\n",
    "\n",
    "\n",
    "\n",
    "[nathalia@aws] $  head transcript_tpm_all_samples.tsv gene_tpm_all_samples.tsv\n",
    "==> transcript_tpm_all_samples.tsv <==\n",
    "Transcript_ID\tnormal_rep1\tnormal_rep2\tnormal_rep3\ttumor_rep1\ttumor_rep2\ttumor_rep3\n",
    "ENST00000006251\t1.170631\t35.025623\t40.675278\t0.000000\t0.000000\t24.092852\n",
    "ENST00000008876\t0.000000\t0.000000\t19.190367\t18.919088\t19.228687\t27.821047\n",
    "ENST00000043402\t0.000000\t8.559146\t9.781459\t134.203674\t136.061142\t216.743011\n",
    "ENST00000086933\t8.947122\t0.0\t0.0\t0.0\t0.0\t0.0\n",
    "ENST00000155674\t96.469200\t2.371792\t0.000000\t0.000000\t0.000000\t564.934631\n",
    "ENST00000159647\t0.000000\t0.000000\t7.599782\t37.676743\t44.098915\t99.557777\n",
    "ENST00000207636\t171.608627\t168.076859\t228.643600\t90.503662\t95.923088\t48.446419\n",
    "ENST00000215659\t100.119423\t729.792358\t93.610970\t103.260765\t70.396370\t89.942009\n",
    "ENST00000215727\t0.000000\t0.000000\t0.000000\t0.000000\t0.000000\t0.000000\n",
    "\n",
    "==> gene_tpm_all_samples.tsv <==\n",
    "Gene_ID\tnormal_rep1\tnormal_rep2\tnormal_rep3\ttumor_rep1\ttumor_rep2\ttumor_rep3\n",
    "ENSG00000008735\t15.771706\t36.318851\t19.190367\t190.482315\t236.229797\t180.245193\n",
    "ENSG00000015475\t3616.129639\t3656.241943\t3067.918701\t758.651855\t858.584595\t752.987915\n",
    "ENSG00000025708\t1700.266113\t1668.025024\t1598.638428\t239.103317\t212.403687\t244.582092\n",
    "ENSG00000025770\t2504.575684\t2601.244385\t2280.569092\t4700.067871\t5041.317871\t4747.436523\n",
    "ENSG00000040608\t0.000000\t8.559146\t9.781459\t168.003281\t198.923309\t216.743011\n",
    "ENSG00000054611\t2559.363037\t2205.872803\t2204.499268\t754.682068\t745.984375\t892.792908\n",
    "ENSG00000056487\t5.035357\t7.302259\t14.831677\t258.549164\t273.201660\t199.371246\n",
    "ENSG00000063515\t8.947122\t0.0\t0.0\t0.0\t0.0\t0.0\n",
    "ENSG00000069998\t2483.770752\t2558.758545\t1764.495728\t622.776123\t576.681458\t855.656067\n",
    "\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ERCC expression analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/expression/htseq_counts\n",
    "\n",
    "[nathalia@aws] $  wget http://genomedata.org/rnaseq-tutorial/ERCC_Controls_Analysis.txt\n",
    "--2018-11-13 20:10:58--  http://genomedata.org/rnaseq-tutorial/ERCC_Controls_Analysis.txt\n",
    "Resolving genomedata.org (genomedata.org)... 3.8.108.6\n",
    "Connecting to genomedata.org (genomedata.org)|3.8.108.6|:80... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 4086 (4.0K) [text/plain]\n",
    "Saving to: ERCC_Controls_Analysis.txt\n",
    "\n",
    "ERCC_Controls_Analysis.tx 100%[=====================================>]   3.99K  --.-KB/s    in 0s      \n",
    "\n",
    "2018-11-13 20:10:59 (498 MB/s) - ERCC_Controls_Analysis.txt saved [4086/4086]\n",
    "\n",
    "[nathalia@aws] $  wget https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/Tutorial_ERCC_expression.pl\n",
    "--2018-11-13 20:11:04--  https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/Tutorial_ERCC_expression.pl\n",
    "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.52.133\n",
    "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.52.133|:443... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 1790 (1.7K) [text/plain]\n",
    "Saving to: Tutorial_ERCC_expression.pl\n",
    "\n",
    "Tutorial_ERCC_expression. 100%[=====================================>]   1.75K  --.-KB/s    in 0s      \n",
    "\n",
    "2018-11-13 20:11:04 (626 MB/s) - Tutorial_ERCC_expression.pl saved [1790/1790]\n",
    "\n",
    "[nathalia@aws] $  chmod +x Tutorial_ERCC_expression.pl\n",
    "[nathalia@aws] $  wget https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/Tutorial_ERCC_expression.R\n",
    "--2018-11-13 20:11:12--  https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/Tutorial_ERCC_expression.R\n",
    "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.52.133\n",
    "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.52.133|:443... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 913 [text/plain]\n",
    "Saving to: Tutorial_ERCC_expression.R\n",
    "\n",
    "Tutorial_ERCC_expression. 100%[=====================================>]     913  --.-KB/s    in 0s      \n",
    "\n",
    "2018-11-13 20:11:12 (317 MB/s) - Tutorial_ERCC_expression.R saved [913/913]\n",
    "\n",
    "[nathalia@aws] $  chmod +x Tutorial_ERCC_expression.R\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  less ERCC_Controls_Analysis.txt \n",
    "\n",
    "Re-sort ID      ERCC ID subgroup        concentration in Mix 1 (attomoles/ul)   concentration in Mix 2 (\n",
    "1       ERCC-00130      A       30000   7500    4       2\n",
    "2       ERCC-00004      A       7500    1875    4       2\n",
    "3       ERCC-00136      A       1875    468.75  4       2\n",
    "4       ERCC-00108      A       937.5   234.375 4       2\n",
    "5       ERCC-00116      A       468.75  117.1875        4       2\n",
    "6       ERCC-00092      A       234.375 58.59375        4       2\n",
    "7       ERCC-00095      A       117.1875        29.296875       4       2\n",
    "8       ERCC-00131      A       117.1875        29.296875       4       2\n",
    "9       ERCC-00062      A       58.59375        14.6484375      4       2\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  ./Tutorial_ERCC_expression.pl \n",
    "[nathalia@aws] $  lhtr\n",
    "total 308K\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.0K Oct 23 16:49 ERCC_Controls_Analysis.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.2K Nov 13 19:39 htseq.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  26K Nov 13 19:39 UHR_Rep1_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:40 UHR_Rep2_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:40 UHR_Rep3_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:40 HBR_Rep1_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:40 HBR_Rep2_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:41 HBR_Rep3_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  42K Nov 13 19:42 gene_read_counts_table_all.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   61 Nov 13 19:42 header.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  42K Nov 13 19:42 gene_read_counts_table_all_final.tsv\n",
    "-rwxrwxr-x 1 ubuntu ubuntu 1.8K Nov 13 20:11 Tutorial_ERCC_expression.pl\n",
    "-rwxrwxr-x 1 ubuntu ubuntu  913 Nov 13 20:11 Tutorial_ERCC_expression.R\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20K Nov 13 20:12 ercc_read_counts.tsv\n",
    "[nathalia@aws] $  less ercc_read_counts.tsv \n",
    "\n",
    "[nathalia@aws] $  ./Tutorial_ERCC_expression.R ercc_read_counts.tsv\n",
    "null device \n",
    "          1 \n",
    "[nathalia@aws] $  lhtr\n",
    "total 324K\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.0K Oct 23 16:49 ERCC_Controls_Analysis.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.2K Nov 13 19:39 htseq.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  26K Nov 13 19:39 UHR_Rep1_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:40 UHR_Rep2_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:40 UHR_Rep3_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:40 HBR_Rep1_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:40 HBR_Rep2_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25K Nov 13 19:41 HBR_Rep3_gene.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  42K Nov 13 19:42 gene_read_counts_table_all.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   61 Nov 13 19:42 header.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  42K Nov 13 19:42 gene_read_counts_table_all_final.tsv\n",
    "-rwxrwxr-x 1 ubuntu ubuntu 1.8K Nov 13 20:11 Tutorial_ERCC_expression.pl\n",
    "-rwxrwxr-x 1 ubuntu ubuntu  913 Nov 13 20:11 Tutorial_ERCC_expression.R\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20K Nov 13 20:12 ercc_read_counts.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  16K Nov 13 20:13 Tutorial_ERCC_expression.pdf\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/expression/htseq_counts/Tutorial_ERCC_expression.pdf\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Differential Expression"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  mkdir -p $RNA_HOME/de/ballgown/ref_only/\n",
    "[nathalia@aws] $  cd $RNA_HOME/de/ballgown/ref_only/\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/de/ballgown/ref_only\n",
    "[nathalia@aws] $  printf \"\\\"ids\\\",\\\"type\\\",\\\"path\\\"\\n\\\"UHR_Rep1\\\",\\\"UHR\\\",\\\"$RNA_HOME/expression/stringtie/ref_only/UHR_Rep1\\\"\\n\\\"UHR_Rep2\\\",\\\"UHR\\\",\\\"$RNA_HOME/expression/stringtie/ref_only/UHR_Rep2\\\"\\n\\\"UHR_Rep3\\\",\\\"UHR\\\",\\\"$RNA_HOME/expression/stringtie/ref_only/UHR_Rep3\\\"\\n\\\"HBR_Rep1\\\",\\\"HBR\\\",\\\"$RNA_HOME/expression/stringtie/ref_only/HBR_Rep1\\\"\\n\\\"HBR_Rep2\\\",\\\"HBR\\\",\\\"$RNA_HOME/expression/stringtie/ref_only/HBR_Rep2\\\"\\n\\\"HBR_Rep3\\\",\\\"HBR\\\",\\\"$RNA_HOME/expression/stringtie/ref_only/HBR_Rep3\\\"\\n\" > UHR_vs_HBR.csv\n",
    "[nathalia@aws] $  cat UHR_vs_HBR.csv \n",
    "\"ids\",\"type\",\"path\"\n",
    "\"UHR_Rep1\",\"UHR\",\"/home/ubuntu/workspace/rnaseq/expression/stringtie/ref_only/UHR_Rep1\"\n",
    "\"UHR_Rep2\",\"UHR\",\"/home/ubuntu/workspace/rnaseq/expression/stringtie/ref_only/UHR_Rep2\"\n",
    "\"UHR_Rep3\",\"UHR\",\"/home/ubuntu/workspace/rnaseq/expression/stringtie/ref_only/UHR_Rep3\"\n",
    "\"HBR_Rep1\",\"HBR\",\"/home/ubuntu/workspace/rnaseq/expression/stringtie/ref_only/HBR_Rep1\"\n",
    "\"HBR_Rep2\",\"HBR\",\"/home/ubuntu/workspace/rnaseq/expression/stringtie/ref_only/HBR_Rep2\"\n",
    "\"HBR_Rep3\",\"HBR\",\"/home/ubuntu/workspace/rnaseq/expression/stringtie/ref_only/HBR_Rep3\"\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://github.com/griffithlab/rnaseq_tutorial/blob/master/scripts/Tutorial_Part1_ballgown.R\n",
    "\n",
    "```\n",
    "[nathalia@aws] $  R\n",
    "\n",
    "R version 3.4.4 (2018-03-15) -- \"Someone to Lean On\"\n",
    "Copyright (C) 2018 The R Foundation for Statistical Computing\n",
    "Platform: x86_64-pc-linux-gnu (64-bit)\n",
    "\n",
    "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
    "You are welcome to redistribute it under certain conditions.\n",
    "Type 'license()' or 'licence()' for distribution details.\n",
    "\n",
    "R is a collaborative project with many contributors.\n",
    "Type 'contributors()' for more information and\n",
    "'citation()' on how to cite R or R packages in publications.\n",
    "\n",
    "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
    "'help.start()' for an HTML browser interface to help.\n",
    "Type 'q()' to quit R.\n",
    "\n",
    "> library(ballgown)\n",
    "\n",
    "Attaching package: ballgown\n",
    "\n",
    "The following object is masked from package:base:\n",
    "\n",
    "    structure\n",
    "\n",
    "> library(genefilter)\n",
    "> library(dplyr)\n",
    "\n",
    "Attaching package: dplyr\n",
    "\n",
    "The following objects are masked from package:ballgown:\n",
    "\n",
    "    contains, expr, last\n",
    "\n",
    "The following objects are masked from package:stats:\n",
    "\n",
    "    filter, lag\n",
    "\n",
    "The following objects are masked from package:base:\n",
    "\n",
    "    intersect, setdiff, setequal, union\n",
    "\n",
    "> library(devtools)\n",
    "> pheno_data = read.csv(\"UHR_vs_HBR.csv\")\n",
    "> bg = ballgown(samples=as.vector(pheno_data$path), pData=pheno_data)\n",
    "Tue Nov 13 20:19:48 2018\n",
    "Tue Nov 13 20:19:48 2018: Reading linking tables\n",
    "Tue Nov 13 20:19:48 2018: Reading intron data files\n",
    "Tue Nov 13 20:19:48 2018: Merging intron data\n",
    "Tue Nov 13 20:19:48 2018: Reading exon data files\n",
    "Tue Nov 13 20:19:48 2018: Merging exon data\n",
    "Tue Nov 13 20:19:49 2018: Reading transcript data files\n",
    "Tue Nov 13 20:19:49 2018: Merging transcript data\n",
    "Wrapping up the results\n",
    "Tue Nov 13 20:19:49 2018\n",
    "> bg\n",
    "ballgown instance with 4564 transcripts and 6 samples\n",
    "> bg_table = texpr(bg, 'all')\n",
    "> bg_gene_names = unique(bg_table[, 9:10])\n",
    "> save(bg, file='bg.rda')\n",
    "> results_transcripts = stattest(bg, feature=\"transcript\", covariate=\"type\", getFC=TRUE, meas=\"FPKM\")\n",
    "> results_genes = stattest(bg, feature=\"gene\", covariate=\"type\", getFC=TRUE, meas=\"FPKM\")\n",
    ")\n",
    "results_genes = stattest(bg_filt, feature=\"gene\", covariate=\"type\", getFC=TRUE, meas=\"FPKM\")\n",
    "results_genes = merge(results_genes, bg_filt_gene_names, by.x=c(\"id\"), by.y=c(\"gene_id\"))\n",
    "\n",
    "# Output the filtered list of genes and transcripts and save to tab delimited files\n",
    "write.table(results_transcripts, \"UHR_vs_HBR_transcript_results_filtered.tsv\", sep=\"\\t\", quote=FALSE, row.names = FALSE)\n",
    "write.table(results_genes, \"UHR_vs_HBR_gene_results_filtered.tsv\", sep=\"\\t\", quote=FALSE, row.names = FALSE)\n",
    "\n",
    "# Identify the significant genes with p-value < 0.05\n",
    "sig_transcripts = subset(results_transcripts, results_transcripts$pval<0.05)\n",
    "sig_genes = subset(results_genes, results_genes$pval<0.05)\n",
    "\n",
    "# Output the signifant gene results to a pair of tab delimited files\n",
    "write.table(sig_transcripts, \"UHR_vs_HBR_transcript_results_sig.tsv\", sep=\"\\t\", quote=FALSE, row.names = FALSE)\n",
    "write.table(sig_genes, \"UHR_vs_HBR_gene_results_sig.tsv\", sep=\"\\t\", quote=FALSE, row.names = FALSE)> results_genes = merge(results_genes, bg_gene_names, by.x=c(\"id\"), by.y=c(\"gene_id\"))\n",
    "> \n",
    "> # Save a tab delimited file for both the transcript and gene results\n",
    "> write.table(results_transcripts, \"UHR_vs_HBR_transcript_results.tsv\", sep=\"\\t\", quote=FALSE, row.names = FALSE)\n",
    "> write.table(results_genes, \"UHR_vs_HBR_gene_results.tsv\", sep=\"\\t\", quote=FALSE, row.names = FALSE)\n",
    "> \n",
    "> # Filter low-abundance genes. Here we remove all transcripts with a variance across the samples of less than one\n",
    "> bg_filt = subset (bg,\"rowVars(texpr(bg)) > 1\", genomesubset=TRUE)\n",
    "> \n",
    "> # Load all attributes including gene name\n",
    "> bg_filt_table = texpr(bg_filt , 'all')\n",
    "> bg_filt_gene_names = unique(bg_filt_table[, 9:10])\n",
    "> \n",
    "> # Perform DE analysis now using the filtered data\n",
    "> results_transcripts = stattest(bg_filt, feature=\"transcript\", covariate=\"type\", getFC=TRUE, meas=\"FPKM\")\n",
    "> results_genes = stattest(bg_filt, feature=\"gene\", covariate=\"type\", getFC=TRUE, meas=\"FPKM\")\n",
    "> results_genes = merge(results_genes, bg_filt_gene_names, by.x=c(\"id\"), by.y=c(\"gene_id\"))\n",
    "> \n",
    "> # Output the filtered list of genes and transcripts and save to tab delimited files\n",
    "> write.table(results_transcripts, \"UHR_vs_HBR_transcript_results_filtered.tsv\", sep=\"\\t\", quote=FALSE, row.names = FALSE)\n",
    "> write.table(results_genes, \"UHR_vs_HBR_gene_results_filtered.tsv\", sep=\"\\t\", quote=FALSE, row.names = FALSE)\n",
    "> \n",
    "> # Identify the significant genes with p-value < 0.05\n",
    "> sig_transcripts = subset(results_transcripts, results_transcripts$pval<0.05)\n",
    "> sig_genes = subset(results_genes, results_genes$pval<0.05)\n",
    "> \n",
    "> # Output the signifant gene results to a pair of tab delimited files\n",
    "> write.table(sig_transcripts, \"UHR_vs_HBR_transcript_results_sig.tsv\", sep=\"\\t\", quote=FALSE, row.names = FALSE)\n",
    "> write.table(sig_genes, \"UHR_vs_HBR_gene_results_sig.tsv\", sep=\"\\t\", quote=FALSE, row.names = FALSE)\n",
    "> quit(save=yes)\n",
    "Error in quit(save = yes) : object 'yes' not found\n",
    "> quit(save=\"yes\")\n",
    "\n",
    "# don't see anything that saved this session...\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 2.7M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  548 Nov 13 20:17 UHR_vs_HBR.csv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.1M Nov 13 20:20 bg.rda\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 247K Nov 13 20:20 UHR_vs_HBR_transcript_results.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  93K Nov 13 20:20 UHR_vs_HBR_gene_results.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 206K Nov 13 20:20 UHR_vs_HBR_transcript_results_filtered.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  70K Nov 13 20:20 UHR_vs_HBR_gene_results_filtered.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  21K Nov 13 20:20 UHR_vs_HBR_transcript_results_sig.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  27K Nov 13 20:20 UHR_vs_HBR_gene_results_sig.tsv\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  head UHR_vs_HBR_gene_results.tsv \n",
    "id\tfeature\tfc\tpval\tqval\tgene_name\n",
    "ENSG00000008735\tgene\t0.0102531569800447\t9.877118335333e-05\t0.00225818270165137\tMAPK8IP2\n",
    "ENSG00000015475\tgene\t1.32306263166779\t0.0764387804544768\t0.187661500329249\tBID\n",
    "ENSG00000025708\tgene\t1.55239124915212\t0.327682469296716\t0.518830576386466\tTYMP\n",
    "ENSG00000025770\tgene\t1.42936894146574\t0.00323664975305338\t0.0175703843737183\tNCAPH2\n",
    "ENSG00000040608\tgene\t0.108828865221793\t0.000255802217979273\t0.00378934133074381\tRTN4R\n",
    "ENSG00000054611\tgene\t1.05231738283307\t0.459151907204536\t0.621205521512019\tTBC1D22A\n",
    "ENSG00000056487\tgene\t0.810808181134758\t0.699213838381129\t0.842952674879717\tPHF21B\n",
    "ENSG00000063515\tgene\t1\tNA\tNA\tGSC2\n",
    "ENSG00000069998\tgene\t2.05557550268768\t0.0719916109898758\t0.179773337157576\tCECR5\n",
    "[nathalia@aws] $  grep -v feature UHR_vs_HBR_gene_results.tsv | wc -l\n",
    "1410\n",
    "[nathalia@aws] $  grep -v feature UHR_vs_HBR_gene_results_filtered.tsv | wc -l\n",
    "843\n",
    "[nathalia@aws] $  grep -v feature UHR_vs_HBR_gene_results_sig.tsv | wc -l\n",
    "319\n",
    "[nathalia@aws] $  grep -v feature UHR_vs_HBR_gene_results_sig.tsv | sort -rnk 3 | head -n 20 #Higher abundance in UHR\n",
    "ENSG00000211677\tgene\t17783.6814624069\t0.00230622619246057\t0.0135123529175882\tIGLC2\n",
    "ENSG00000211679\tgene\t12875.7816804218\t2.67621524940864e-09\t2.25604945525149e-06\tIGLC3\n",
    "ENSG00000211666\tgene\t2706.54605161754\t5.92881495098396e-06\t0.000624748875459935\tIGLV2-14\n",
    "ENSG00000185686\tgene\t902.15788112243\t1.39903310603096e-07\t5.89692454192048e-05\tPRAME\n",
    "ENSG00000093009\tgene\t327.609915785601\t1.37755588625055e-06\t0.000290319903027303\tCDC45\n",
    "ENSG00000100196\tgene\t253.13063487409\t6.89706710010896e-06\t0.000646025285043539\tKDELR3\n",
    "ENSG00000271672\tgene\t194.226413239759\t0.0108699948128967\t0.0412766019246484\tDUXAP8\n",
    "ENSG00000280623\tgene\t148.039565857311\t0.00168433900440768\t0.0115438843960624\tPCAT14\n",
    "ENSG00000099937\tgene\t125.732790843451\t0.00988451659156242\t0.0385770716976256\tSERPIND1\n",
    "ENSG00000179750\tgene\t97.1715539921813\t1.19265099662869e-06\t0.000290319903027303\tAPOBEC3B\n",
    "ENSG00000211685\tgene\t87.2127202707906\t0.000194597761772775\t0.00321658653283233\tIGLC7\n",
    "ENSG00000075275\tgene\t79.6342032810874\t0.000346185545310185\t0.00444508678980326\tCELSR1\n",
    "ENSG00000075218\tgene\t68.53169076568\t0.00784325025466293\t0.0327319800231725\tGTSE1\n",
    "ENSG00000211660\tgene\t65.5200427367455\t5.45875004183127e-05\t0.00163353396365886\tIGLV2-23\n",
    "ENSG00000251357\tgene\t62.0201550972963\t0.000374673330653774\t0.00459821472890528\tAP000350.10\n",
    "ENSG00000233903\tgene\t45.119195597068\t8.81728826557238e-05\t0.00205379674066896\tZ83851.4\n",
    "ENSG00000100162\tgene\t44.7279435560213\t0.0292549159636819\t0.088078193419228\tCENPM\n",
    "ENSG00000133454\tgene\t43.6478978899329\t7.91387334353466e-06\t0.000667139522859972\tMYO18B\n",
    "ENSG00000128165\tgene\t40.9662834227171\t7.08636200452695e-05\t0.00192703328058588\tADM2\n",
    "ENSG00000183765\tgene\t40.8713940792141\t0.0323878429442334\t0.0944738809757395\tCHEK2\n",
    "[nathalia@aws] $  grep -v feature UHR_vs_HBR_gene_results_sig.tsv | sort -nk 3 | head -n 20 #Higher abundance in HBR\n",
    "ENSG00000261675\tgene\t0.004698954946935\t2.40212386415362e-06\t0.0004049980834963\tRP5-1119A7.17\n",
    "ENSG00000184113\tgene\t0.00564170307051958\t9.46454013689424e-05\t0.00209963350931627\tCLDN5\n",
    "ENSG00000186732\tgene\t0.00587743100221453\t0.000319011020402327\t0.00433752080966389\tMPPED1\n",
    "ENSG00000225465\tgene\t0.00838284867422027\t0.00541696803759573\t0.0249535740748262\tRFPL1S\n",
    "ENSG00000130540\tgene\t0.00851210627845688\t6.41585206511142e-05\t0.00180285443029631\tSULT4A1\n",
    "ENSG00000205704\tgene\t0.00863326654592059\t0.000523479708357866\t0.00556427659784253\tLINC00634\n",
    "ENSG00000182902\tgene\t0.00876481037194695\t0.000451185200520454\t0.0050713216538499\tSLC25A18\n",
    "ENSG00000166862\tgene\t0.00962041424045346\t0.000108832859428309\t0.00223770976824548\tCACNG2\n",
    "ENSG00000183785\tgene\t0.00971335981477572\t0.00562092369106626\t0.0256195073341291\tTUBA8\n",
    "ENSG00000100399\tgene\t0.0100939218003004\t0.000931313774164222\t0.00811780364397272\tCHADL\n",
    "ENSG00000008735\tgene\t0.0101540642196127\t7.93486177163016e-05\t0.00198707065689339\tMAPK8IP2\n",
    "ENSG00000283366\tgene\t0.0112054663682816\t0.000166727996999239\t0.00299046173341187\tAC000095.12\n",
    "ENSG00000273164\tgene\t0.0112312945235097\t0.000551281839530793\t0.00566744622834705\tAC000095.11\n",
    "ENSG00000172346\tgene\t0.0113768834332585\t0.0061157180991559\t0.0271697644729183\tCSDC2\n",
    "ENSG00000100346\tgene\t0.0114341583826291\t5.12271447427048e-05\t0.00163353396365886\tCACNA1I\n",
    "ENSG00000273289\tgene\t0.011896108037934\t0.000398871236137355\t0.00467011738977486\tRP6-109B7.5\n",
    "ENSG00000099960\tgene\t0.0118977385213888\t5.61951185600318e-05\t0.00163353396365886\tSLC7A4\n",
    "ENSG00000100276\tgene\t0.0143417567863054\t0.0011006325024624\t0.00889262985156803\tRASL10A\n",
    "ENSG00000237517\tgene\t0.0156412145513587\t0.00856499216676576\t0.0349593246151362\tDGCR5\n",
    "ENSG00000128285\tgene\t0.0195226793397065\t0.00121905970535852\t0.00934243028742936\tMCHR1\n",
    "\n",
    "[nathalia@aws] $  grep -v feature UHR_vs_HBR_gene_results_sig.tsv | cut -f 6 | sed 's/\\\"//g' > DE_genes.txt\n",
    "\n",
    "\n",
    "[nathalia@aws] $  head DE_genes.txt\n",
    "MAPK8IP2\n",
    "NCAPH2\n",
    "RTN4R\n",
    "UFD1L\n",
    "CLTCL1\n",
    "DGCR2\n",
    "PANX2\n",
    "GTSE1\n",
    "TTC38\n",
    "CELSR1\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ERCC DE Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/de/ballgown/ref_only\n",
    "[nathalia@aws] $  wget https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/Tutorial_ERCC_DE.R\n",
    "--2018-11-13 20:26:31--  https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/Tutorial_ERCC_DE.R\n",
    "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.52.133\n",
    "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.52.133|:443... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 1568 (1.5K) [text/plain]\n",
    "Saving to: Tutorial_ERCC_DE.R\n",
    "\n",
    "Tutorial_ERCC_DE.R        100%[=====================================>]   1.53K  --.-KB/s    in 0s      \n",
    "\n",
    "2018-11-13 20:26:31 (537 MB/s) - Tutorial_ERCC_DE.R saved [1568/1568]\n",
    "\n",
    "[nathalia@aws] $  chmod +x Tutorial_ERCC_DE.R\n",
    "[nathalia@aws] $  ./Tutorial_ERCC_DE.R $RNA_HOME/expression/htseq_counts/ERCC_Controls_Analysis.txt $RNA_HOME/de/ballgown/ref_only/UHR_vs_HBR_gene_results.tsv\n",
    "null device \n",
    "          1 \n",
    "[nathalia@aws] $  lhtr\n",
    "total 2.8M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  548 Nov 13 20:17 UHR_vs_HBR.csv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.1M Nov 13 20:20 bg.rda\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 247K Nov 13 20:20 UHR_vs_HBR_transcript_results.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  93K Nov 13 20:20 UHR_vs_HBR_gene_results.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 206K Nov 13 20:20 UHR_vs_HBR_transcript_results_filtered.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  70K Nov 13 20:20 UHR_vs_HBR_gene_results_filtered.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  21K Nov 13 20:20 UHR_vs_HBR_transcript_results_sig.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  27K Nov 13 20:20 UHR_vs_HBR_gene_results_sig.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.5K Nov 13 20:25 DE_genes.txt\n",
    "-rwxrwxr-x 1 ubuntu ubuntu 1.6K Nov 13 20:26 Tutorial_ERCC_DE.R\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 7.3K Nov 13 20:26 Tutorial_ERCC_DE.pdf\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<img src=\"http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/de/ballgown/ref_only/Tutorial_ERCC_DE.pdf\"/>"
      ],
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from IPython.display import Image\n",
    "from IPython.core.display import HTML \n",
    "Image(url= \"http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/de/ballgown/ref_only/Tutorial_ERCC_DE.pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# edgeR Analysis\n",
    "\n",
    "https://github.com/griffithlab/rnaseq_tutorial/blob/master/scripts/Tutorial_Part1_ballgown.R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME/\n",
    "[nathalia@aws] $  mkdir -p de/htseq_counts\n",
    "[nathalia@aws] $  cd de/htseq_counts\n",
    "\n",
    "[nathalia@aws] $  perl -ne 'if ($_ =~ /gene_id\\s\\\"(ENSG\\S+)\\\"\\;/) { $id = $1; $name = undef; if ($_ =~ /gene_name\\s\\\"(\\S+)\"\\;/) { $name = $1; }; }; if ($id && $name) {print \"$id\\t$name\\n\";} if ($_=~/gene_id\\s\\\"(ERCC\\S+)\\\"/){print \"$1\\t$1\\n\";}' $RNA_REF_GTF | sort | uniq > ENSG_ID2Name.txt\n",
    "[nathalia@aws] $  ls\n",
    "ENSG_ID2Name.txt\n",
    "\n",
    "[nathalia@aws] $  cut -f 2 ENSG_ID2Name.txt | sort | uniq -c | sort -r | head\n",
    "     13 Y_RNA\n",
    "     12 Metazoa_SRP\n",
    "      3 SCARNA17\n",
    "      2 U6\n",
    "      2 RNU12\n",
    "      2 RFPL3-AS1_1\n",
    "      2 LINC01422\n",
    "      2 ELFN2\n",
    "      2 DUXAP8\n",
    "      2 DGCR5\n",
    "      ```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  R\n",
    "\n",
    "R version 3.4.4 (2018-03-15) -- \"Someone to Lean On\"\n",
    "Copyright (C) 2018 The R Foundation for Statistical Computing\n",
    "Platform: x86_64-pc-linux-gnu (64-bit)\n",
    "\n",
    "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
    "You are welcome to redistribute it under certain conditions.\n",
    "Type 'license()' or 'licence()' for distribution details.\n",
    "\n",
    "R is a collaborative project with many contributors.\n",
    "Type 'contributors()' for more information and\n",
    "'citation()' on how to cite R or R packages in publications.\n",
    "\n",
    "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
    "'help.start()' for an HTML browser interface to help.\n",
    "Type 'q()' to quit R.\n",
    "\n",
    "> #Malachi Griffith, mgriffit[AT]wustl.edu\n",
    "> #Obi Griffith, obigriffith[AT]wustl.edu\n",
    "> #The McDonnell Genome Institute, Washington University School of Medicine\n",
    "> \n",
    "> #R tutorial for Informatics for RNA-sequence Analysis workshops\n",
    "> \n",
    "> #######################\n",
    "> # Loading Data into R #\n",
    "> #######################\n",
    "> \n",
    "> #Set working directory where output will go\n",
    "> working_dir = \"~/workspace/rnaseq/de/htseq_counts\"\n",
    "> setwd(working_dir)\n",
    "> \n",
    "> #Read in gene mapping\n",
    "> mapping=read.table(\"~/workspace/rnaseq/de/htseq_counts/ENSG_ID2Name.txt\", header=FALSE, stringsAsFactors=FALSE, row.names=1)\n",
    "> \n",
    "> # Read in count matrix\n",
    "> rawdata=read.table(\"~/workspace/rnaseq/expression/htseq_counts/gene_read_counts_table_all_final.tsv\", header=TRUE, stringsAsFactors=FALSE, row.names=1)\n",
    "> \n",
    "> # Check dimensions\n",
    "> dim(rawdata)\n",
    "[1] 1410    6\n",
    "> \n",
    "> # Require at least 25% of samples to have count > 25\n",
    "> quant <- apply(rawdata,1,quantile,0.75)\n",
    "ake class labels\n",
    "class <- factor( c( rep(\"UHR\",3), rep(\"HBR\",3) ))\n",
    "\n",
    "# Get common gene names\n",
    "genes=rownames(rawdata)\n",
    "gene_names=mapping[genes,1]\n",
    "\n",
    "\n",
    "# Make DGEList object\n",
    "y <- DGEList(counts=rawdata, genes=genes, group=class)\n",
    "nrow(y)\n",
    "\n",
    "# TMM Normalization\n",
    "y <- calcNormFactors(y)\n",
    "\n",
    "# Estimate dispersion\n",
    "y <- estimateCommonDisp(y, verbose=TRUE)\n",
    "y <- estimateTagwiseDisp(y)\n",
    "\n",
    "# Differential expression test\n",
    "et <- exactTest(y)\n",
    "\n",
    "# Print top genes\n",
    "topTags(et)\n",
    "\n",
    "# Print number of up/down significant genes at FDR = 0.05  significance level\n",
    "summary(de <- decideTestsDGE(et, p=.05))\n",
    "detags <- rownames(y)[as.logical(de)]\n",
    "\n",
    "\n",
    "# Output DE genes\n",
    "# Matrix of significantly DE genes\n",
    "mat <- cbind(\n",
    " genes,gene_names,\n",
    " sprintf('%0.3f',log10(et$table$PValue)),\n",
    " sprintf('%0.3f',et$table$logFC)\n",
    ")[as.logical(de),]\n",
    "colnames(mat) <- c(\"Gene\", \"Gene_Name\", \"Log10_Pvalue\", \"Log_fold_change\")\n",
    "\n",
    "# Order by log fold change\n",
    "o <- order(et$table$logFC[as.logical(de)],decreasing=TRUE)\n",
    "mat <- mat[o,]\n",
    "\n",
    "# Save table\n",
    "write.table(mat, file=\"DE_genes.txt\", quo> keep <- which((quant >= 25) == 1)\n",
    "> rawdata <- rawdata[keep,]\n",
    "> dim(rawdata)\n",
    "[1] 345   6\n",
    "> \n",
    "> #################\n",
    "> # Running edgeR #\n",
    "> #################\n",
    "> \n",
    "> # load edgeR\n",
    "> library('edgeR')\n",
    "Loading required package: limma\n",
    "te=FALSE, row.names=FALSE, sep=\"\\t\")\n",
    "\n",
    "#To exit R type the following\n",
    "quit(save=\"no\")\n",
    "> \n",
    "> # make class labels\n",
    "> class <- factor( c( rep(\"UHR\",3), rep(\"HBR\",3) ))\n",
    "> \n",
    "> # Get common gene names\n",
    "> genes=rownames(rawdata)\n",
    "> gene_names=mapping[genes,1]\n",
    "> \n",
    "> \n",
    "> # Make DGEList object\n",
    "> y <- DGEList(counts=rawdata, genes=genes, group=class)\n",
    "> nrow(y)\n",
    "[1] 345\n",
    "> \n",
    "> # TMM Normalization\n",
    "> y <- calcNormFactors(y)\n",
    "> \n",
    "> # Estimate dispersion\n",
    "> y <- estimateCommonDisp(y, verbose=TRUE)\n",
    "Disp = 0.00202 , BCV = 0.0449 \n",
    "> y <- estimateTagwiseDisp(y)\n",
    "> \n",
    "> # Differential expression test\n",
    "> et <- exactTest(y)\n",
    "> \n",
    "> # Print top genes\n",
    "> topTags(et)\n",
    "Comparison of groups:  UHR-HBR \n",
    "                          genes     logFC   logCPM        PValue           FDR\n",
    "ENSG00000211677 ENSG00000211677 12.625129 11.74636  0.000000e+00  0.000000e+00\n",
    "ENSG00000008735 ENSG00000008735 -5.715293 10.99925  0.000000e+00  0.000000e+00\n",
    "ENSG00000100095 ENSG00000100095 -5.355988 10.94973  0.000000e+00  0.000000e+00\n",
    "ENSG00000100321 ENSG00000100321 -4.760565 11.96301  0.000000e+00  0.000000e+00\n",
    "ENSG00000100167 ENSG00000100167 -4.616857 11.93340  0.000000e+00  0.000000e+00\n",
    "ENSG00000225783 ENSG00000225783 -4.042781 11.58429  0.000000e+00  0.000000e+00\n",
    "ERCC-00004           ERCC-00004  2.436832 13.94830  0.000000e+00  0.000000e+00\n",
    "ERCC-00130           ERCC-00130  2.110204 16.85187  0.000000e+00  0.000000e+00\n",
    "ERCC-00136           ERCC-00136  2.250670 12.93333 9.651521e-306 3.699750e-304\n",
    "ENSG00000128245 ENSG00000128245 -2.599032 12.61920 1.727421e-304 5.959602e-303\n",
    "> \n",
    "> # Print number of up/down significant genes at FDR = 0.05  significance level\n",
    "> summary(de <- decideTestsDGE(et, p=.05))\n",
    "       HBR+UHR\n",
    "Down       136\n",
    "NotSig      65\n",
    "Up         144\n",
    "> detags <- rownames(y)[as.logical(de)]\n",
    "> \n",
    "> \n",
    "> # Output DE genes\n",
    "> # Matrix of significantly DE genes\n",
    "> mat <- cbind(\n",
    "+  genes,gene_names,\n",
    "+  sprintf('%0.3f',log10(et$table$PValue)),\n",
    "+  sprintf('%0.3f',et$table$logFC)\n",
    "+ )[as.logical(de),]\n",
    "> colnames(mat) <- c(\"Gene\", \"Gene_Name\", \"Log10_Pvalue\", \"Log_fold_change\")\n",
    "> \n",
    "> # Order by log fold change\n",
    "> o <- order(et$table$logFC[as.logical(de)],decreasing=TRUE)\n",
    "> mat <- mat[o,]\n",
    "> \n",
    "> # Save table\n",
    "> write.table(mat, file=\"DE_genes.txt\", quote=FALSE, row.names=FALSE, sep=\"\\t\")\n",
    "> \n",
    "> #To exit R type the following\n",
    "> quit(save=\"no\")\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 48K\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 35K Nov 13 20:32 ENSG_ID2Name.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 11K Nov 13 20:34 DE_genes.txt\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  head $RNA_HOME/de/ballgown/ref_only/DE_genes.txt $RNA_HOME/de/htseq_counts/DE_genes.txt\n",
    "==> /home/ubuntu/workspace/rnaseq/de/ballgown/ref_only/DE_genes.txt <==\n",
    "MAPK8IP2\n",
    "NCAPH2\n",
    "RTN4R\n",
    "UFD1L\n",
    "CLTCL1\n",
    "DGCR2\n",
    "PANX2\n",
    "GTSE1\n",
    "TTC38\n",
    "CELSR1\n",
    "\n",
    "==> /home/ubuntu/workspace/rnaseq/de/htseq_counts/DE_genes.txt <==\n",
    "Gene\tGene_Name\tLog10_Pvalue\tLog_fold_change\n",
    "ENSG00000211677\tIGLC2\t-Inf\t12.625\n",
    "ENSG00000185686\tPRAME\t-295.609\t12.020\n",
    "ENSG00000211666\tIGLV2-14\t-77.190\t10.158\n",
    "ENSG00000093009\tCDC45\t-96.498\t10.029\n",
    "ENSG00000211679\tIGLC3\t-60.325\t9.322\n",
    "ENSG00000133454\tMYO18B\t-37.420\t8.821\n",
    "ENSG00000280623\tPCAT14\t-89.490\t7.890\n",
    "ENSG00000075275\tCELSR1\t-94.964\t7.354\n",
    "ENSG00000099937\tSERPIND1\t-76.893\t6.263\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME/de/\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/de\n",
    "\n",
    "[nathalia@aws] $  cut -f 1 $RNA_HOME/de/ballgown/ref_only/DE_genes.txt | sort  > ballgown_DE_gene_symbols.txt\n",
    "[nathalia@aws] $  cut -f 2 $RNA_HOME/de/htseq_counts/DE_genes.txt | sort > htseq_counts_edgeR_DE_gene_symbols.txt\n",
    "[nathalia@aws] $  lhtr\n",
    "total 16K\n",
    "drwxrwxr-x 3 ubuntu ubuntu 4.0K Nov 13 20:17 ballgown\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 13 20:34 htseq_counts\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.5K Nov 13 20:36 ballgown_DE_gene_symbols.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.0K Nov 13 20:36 htseq_counts_edgeR_DE_gene_symbols.txt\n",
    "[nathalia@aws] $  head ballgown_DE_gene_symbols.txt htseq_counts_edgeR_DE_gene_symbols.txt \n",
    "==> ballgown_DE_gene_symbols.txt <==\n",
    "A4GALT\n",
    "AC000095.11\n",
    "AC000095.12\n",
    "ACO2\n",
    "ADM2\n",
    "ADSL\n",
    "AIFM3\n",
    "AP000347.4\n",
    "AP000350.10\n",
    "AP000350.5\n",
    "\n",
    "==> htseq_counts_edgeR_DE_gene_symbols.txt <==\n",
    "ACO2\n",
    "ADM2\n",
    "ADSL\n",
    "AIFM3\n",
    "AP1B1\n",
    "APOBEC3C\n",
    "APOL1\n",
    "APOL2\n",
    "APOL6\n",
    "ARFGAP3\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/de/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://bioinfogp.cnb.csic.es/tools/venny/\n",
    "\n",
    "Copied the entire list of gene in ballgown and htseq into venny."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![venn](venn_diagram.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DE Visualization\n",
    "\n",
    "https://github.com/griffithlab/rnaseq_tutorial/blob/master/scripts/Tutorial_Part2_ballgown.R\n",
    "\n",
    "https://github.com/griffithlab/rnaseq_tutorial/blob/master/scripts/Tutorial_Supplementary_R.R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME/de/ballgown/ref_only\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/de/ballgown/ref_only\n",
    "\n",
    "[nathalia@aws] $  R\n",
    "\n",
    "R version 3.4.4 (2018-03-15) -- \"Someone to Lean On\"\n",
    "Copyright (C) 2018 The R Foundation for Statistical Computing\n",
    "Platform: x86_64-pc-linux-gnu (64-bit)\n",
    "\n",
    "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
    "You are welcome to redistribute it under certain conditions.\n",
    "Type 'license()' or 'licence()' for distribution details.\n",
    "\n",
    "R is a collaborative project with many contributors.\n",
    "Type 'contributors()' for more information and\n",
    "'citation()' on how to cite R or R packages in publications.\n",
    "\n",
    "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
    "'help.start()' for an HTML browser interface to help.\n",
    "Type 'q()' to quit R.\n",
    "\n",
    "[Previously saved workspace restored]\n",
    "\n",
    "> # Load libraries needed for this analysis\n",
    "> library(ballgown)\n",
    "pkm)\n",
    "\n",
    "# Create boxplots to display summary statistics for the FPKM values for each sample\n",
    "boxplot(fpkm,col=as.numeric(pheno_data$type)+1,las=2,ylab='log2(FPKM+1)')\n",
    "\n",
    "# Display the transcript ID for a single row of data\n",
    "ballgown::transcriptNames(bg)[2763]\n",
    "\n",
    "# Display the gene name for a single row of data \n",
    "ballgown::geneNames(bg)[2763]\n",
    "\n",
    "# Create a BoxPlot comparing the expression of a single gene for all replicates of both conditions\n",
    "plot(fpkm[2763,] ~ pheno_data$type, border=c(2,3), main=paste(ballgown::geneNames(bg)[2763],' : ', ballgown::transcriptNames(bg)[2763]),pch=19, xlab=\"Type\", ylab='log2(FPKM+1)')\n",
    "\n",
    "# Add the FPKM values for each sample onto the plot \n",
    "points(fpkm[2763,] ~ jitter(as.numeric(pheno_data$type)), col=as.numeric(pheno_data$type)+1, pch=16)\n",
    "\n",
    "# Create a plot of transcript structures observed in each replicate and color transcripts by expression level\n",
    "plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main=c('Gene in sample HBR_Rep1'), sample=c('HBR_Rep1'))\n",
    "plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main=c('Gene in sample HBR_Rep2'), sample=c('HBR_Rep2'))\n",
    "plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main=c('Gene in sample HBR_Rep3'), sample=c('HBR_Rep3'))\n",
    "plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main=c('Gene in sample UHR_Rep1'), sample=c('UHR_Rep1'))\n",
    "plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main=c('Gene in sample UHR_Rep2'), sample=c('UHR_Rep2'))\n",
    "plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main=c('Gene in sample UHR_Rep3'), sample=c('UHR_Rep3'))\n",
    "\n",
    "#plotMeans('TST',bg,groupvar=\"type\",legend=FALSE)\n",
    "\n",
    "# Close the PDF device where we have been saving our plots\n",
    "dev.off()\n",
    "\n",
    "# Exit the R session\n",
    "quit(save=\"no\")\n",
    "\n",
    "Attaching package: ballgown\n",
    "\n",
    "The following object is masked from package:base:\n",
    "\n",
    "    structure\n",
    "\n",
    "> library(genefilter)\n",
    "> library(dplyr)\n",
    "\n",
    "Attaching package: dplyr\n",
    "\n",
    "The following objects are masked from package:ballgown:\n",
    "\n",
    "    contains, expr, last\n",
    "\n",
    "The following objects are masked from package:stats:\n",
    "\n",
    "    filter, lag\n",
    "\n",
    "The following objects are masked from package:base:\n",
    "\n",
    "    intersect, setdiff, setequal, union\n",
    "\n",
    "> library(devtools)\n",
    "> \n",
    "> # Define a path for the output PDF to be written\n",
    "> outfile=\"~/workspace/rnaseq/de/ballgown/ref_only/Tutorial_Part2_ballgown_output.pdf\"\n",
    "> \n",
    "> # Load phenotype data\n",
    "> pheno_data = read.csv(\"UHR_vs_HBR.csv\")\n",
    "> \n",
    "> # Display the phenotype data\n",
    "> pheno_data\n",
    "       ids type\n",
    "1 UHR_Rep1  UHR\n",
    "2 UHR_Rep2  UHR\n",
    "3 UHR_Rep3  UHR\n",
    "4 HBR_Rep1  HBR\n",
    "5 HBR_Rep2  HBR\n",
    "6 HBR_Rep3  HBR\n",
    "                                                                  path\n",
    "1 /home/ubuntu/workspace/rnaseq/expression/stringtie/ref_only/UHR_Rep1\n",
    "2 /home/ubuntu/workspace/rnaseq/expression/stringtie/ref_only/UHR_Rep2\n",
    "3 /home/ubuntu/workspace/rnaseq/expression/stringtie/ref_only/UHR_Rep3\n",
    "4 /home/ubuntu/workspace/rnaseq/expression/stringtie/ref_only/HBR_Rep1\n",
    "5 /home/ubuntu/workspace/rnaseq/expression/stringtie/ref_only/HBR_Rep2\n",
    "6 /home/ubuntu/workspace/rnaseq/expression/stringtie/ref_only/HBR_Rep3\n",
    "> \n",
    "> # Load the ballgown object from file\n",
    "> load('bg.rda')\n",
    "> \n",
    "> # The load command, loads an R object from a file into memory in our R session. \n",
    "> # You can use ls() to view the names of variables that have been loaded\n",
    "> ls()\n",
    " [1] \"bg\"                  \"bg_filt\"             \"bg_filt_gene_names\" \n",
    " [4] \"bg_filt_table\"       \"bg_gene_names\"       \"bg_table\"           \n",
    " [7] \"outfile\"             \"pheno_data\"          \"results_genes\"      \n",
    "[10] \"results_transcripts\" \"sig_genes\"           \"sig_transcripts\"    \n",
    "> \n",
    "> # Print a summary of the ballgown object\n",
    "> bg\n",
    "ballgown instance with 4564 transcripts and 6 samples\n",
    "> \n",
    "> # Open a PDF file where we will save some plots. We will save all figures and then view the PDF at the end\n",
    "> pdf(file=outfile)\n",
    "> \n",
    "> # Extract FPKM values from the 'bg' object\n",
    "> fpkm = texpr(bg,meas=\"FPKM\")\n",
    "> \n",
    "> # View the last several rows of the FPKM table\n",
    "> tail(fpkm)\n",
    "     FPKM.UHR_Rep1 FPKM.UHR_Rep2 FPKM.UHR_Rep3 FPKM.HBR_Rep1 FPKM.HBR_Rep2\n",
    "4559      57.64166     58.148632     105.70252      94.66223    107.827049\n",
    "4560       0.00000      0.000000       0.00000       0.00000      8.190107\n",
    "4561     305.15756    253.149765     270.60403     389.02515    613.900330\n",
    "4562       0.00000      7.708646       0.00000      20.09885      8.174078\n",
    "4563      91.73609    108.026817      93.40608      10.05922     16.364185\n",
    "4564   19674.26562  20861.417969   20929.00391   18689.37500  20731.685547\n",
    "     FPKM.HBR_Rep3\n",
    "4559    225.034378\n",
    "4560      0.000000\n",
    "4561    430.962921\n",
    "4562      9.186255\n",
    "4563     27.585764\n",
    "4564  19666.533203\n",
    "> \n",
    "> # Transform the FPKM values by adding 1 and convert to a log2 scale\n",
    "> fpkm = log2(fpkm+1)\n",
    "> \n",
    "> # View the last several rows of the transformed FPKM table\n",
    "> tail(fpkm)\n",
    "     FPKM.UHR_Rep1 FPKM.UHR_Rep2 FPKM.UHR_Rep3 FPKM.HBR_Rep1 FPKM.HBR_Rep2\n",
    "4559      5.873854      5.886273      6.737450      6.579878      6.765893\n",
    "4560      0.000000      0.000000      0.000000      0.000000      3.200082\n",
    "4561      8.258131      7.989535      8.085361      8.607423      9.264209\n",
    "4562      0.000000      3.122448      0.000000      4.399092      3.197563\n",
    "4563      6.535059      6.768539      6.560808      3.467178      4.118043\n",
    "4564     14.264095     14.348619     14.353285     14.190008     14.339619\n",
    "     FPKM.HBR_Rep3\n",
    "4559      7.820398\n",
    "4560      0.000000\n",
    "4561      8.754764\n",
    "4562      3.348552\n",
    "4563      4.837225\n",
    "4564     14.263528\n",
    "> \n",
    "> # Create boxplots to display summary statistics for the FPKM values for each sample\n",
    "> boxplot(fpkm,col=as.numeric(pheno_data$type)+1,las=2,ylab='log2(FPKM+1)')\n",
    "> \n",
    "> # Display the transcript ID for a single row of data\n",
    "> ballgown::transcriptNames(bg)[2763]\n",
    "             2763 \n",
    "\"ENST00000249042\" \n",
    "> \n",
    "> # Display the gene name for a single row of data \n",
    "> ballgown::geneNames(bg)[2763]\n",
    " 2763 \n",
    "\"TST\" \n",
    "> \n",
    "> # Create a BoxPlot comparing the expression of a single gene for all replicates of both conditions\n",
    "> plot(fpkm[2763,] ~ pheno_data$type, border=c(2,3), main=paste(ballgown::geneNames(bg)[2763],' : ', ballgown::transcriptNames(bg)[2763]),pch=19, xlab=\"Type\", ylab='log2(FPKM+1)')\n",
    "> \n",
    "> # Add the FPKM values for each sample onto the plot \n",
    "> points(fpkm[2763,] ~ jitter(as.numeric(pheno_data$type)), col=as.numeric(pheno_data$type)+1, pch=16)\n",
    "> \n",
    "> # Create a plot of transcript structures observed in each replicate and color transcripts by expression level\n",
    "> plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main=c('Gene in sample HBR_Rep1'), sample=c('HBR_Rep1'))\n",
    "> plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main=c('Gene in sample HBR_Rep2'), sample=c('HBR_Rep2'))\n",
    "> plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main=c('Gene in sample HBR_Rep3'), sample=c('HBR_Rep3'))\n",
    "> plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main=c('Gene in sample UHR_Rep1'), sample=c('UHR_Rep1'))\n",
    "> plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main=c('Gene in sample UHR_Rep2'), sample=c('UHR_Rep2'))\n",
    "> plotTranscripts(ballgown::geneIDs(bg)[2763], bg, main=c('Gene in sample UHR_Rep3'), sample=c('UHR_Rep3'))\n",
    "> \n",
    "> #plotMeans('TST',bg,groupvar=\"type\",legend=FALSE)\n",
    "> \n",
    "> # Close the PDF device where we have been saving our plots\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "> # Exit the R session\n",
    "> quit(save=\"no\")\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  R\n",
    "\n",
    "R version 3.4.4 (2018-03-15) -- \"Someone to Lean On\"\n",
    "Copyright (C) 2018 The R Foundation for Statistical Computing\n",
    "Platform: x86_64-pc-linux-gnu (64-bit)\n",
    "\n",
    "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
    "You are welcome to redistribute it under certain conditions.\n",
    "Type 'license()' or 'licence()' for distribution details.\n",
    "\n",
    "R is a collaborative project with many contributors.\n",
    "Type 'contributors()' for more information and\n",
    "'citation()' on how to cite R or R packages in publications.\n",
    "\n",
    "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
    "'help.start()' for an HTML browser interface to help.\n",
    "Type 'q()' to quit R.\n",
    "\n",
    "[Previously saved workspace restored]\n",
    "\n",
    "> #Tutorial_Part3_Supplementary_R.R\n",
    "> \n",
    "> #Malachi Griffith, mgriffit[AT]wustl.edu\n",
    "> #Obi Griffith, obigriffith[AT]wustl.edu\n",
    "> #Jason Walker, jason.walker[AT]wustl.edu\n",
    "> #McDonnell Genome Institute, Washington University School of Medicine\n",
    "> \n",
    "> #R tutorial for Informatics for RNA-sequence Analysis workshops\n",
    "> \n",
    "> #Starting from the output of the RNA-seq Tutorial Part 1.\n",
    "> \n",
    "> #Load libraries\n",
    "> library(ggplot2)\n",
    "he tutorial\n",
    "bg_table = texpr(bg, 'all')\n",
    "bg_gene_names = unique(bg_table[, 9:10])\n",
    "\n",
    "# Pull the gene_expression data frame from the ballgown object\n",
    "gene_expression = as.data.frame(gexpr(bg))\n",
    "\n",
    "#### Working with 'dataframes'\n",
    "#View the first five rows of data (all columns) in one of the dataframes created\n",
    "head(gene_expression)\n",
    "\n",
    "#View the column names\n",
    "colnames(gene_expression)\n",
    "\n",
    "#View the row names\n",
    "row.names(gene_expression)\n",
    "\n",
    "#Determine the dimensions of the dataframe.  'dim()' will return the number of rows and columns\n",
    "dim(gene_expression)\n",
    "\n",
    "#Get the first 3 rows of data and a selection of columns\n",
    "gene_expression[1:3,c(1:3,6)]\n",
    "\n",
    "#Do the same thing, but using the column names instead of numbers\n",
    "gene_expression[1:3, c(\"FPKM.UHR_Rep1\",\"FPKM.UHR_Rep2\",\"FPKM.UHR_Rep3\",\"FPKM.HBR_Rep3\")]\n",
    "\n",
    "#Assign colors to each.  You can specify color by RGB, Hex code, or name\n",
    "#To get a list of color names:\n",
    "colours()\n",
    "data_colors=c(\"tomato1\",\"tomato2\",\"tomato3\",\"royalblue1\",\"royalblue2\",\"royalblue3\")\n",
    "\n",
    "#View expression values for the transcripts of a particular gene symbol of chromosome 22.  e.g. 'TST'\n",
    "#First determine the rows in the data.frame that match 'TST', aka. ENSG00000128311, then display only those rows of the data.frame\n",
    "i = row.names(gene_expression) == \"ENSG00000128311\"\n",
    "gene_expression[i,]\n",
    "\n",
    "#What if we want to view values for a list of genes of interest all at once? \n",
    "#genes_of_interest = c(\"TST\", \"MMP11\", \"LGALS2\", \"ISX\")\n",
    "genes_of_interest = c(\"ENSG00000128311\",\"ENSG00000099953\",\"ENSG00000100079\",\"ENSG00000175329\")\n",
    "i = which(row.names(gene_expression) %in% genes_of_interest)\n",
    "gene_expression[i,]\n",
    "\n",
    "# Load the transcript to gene index from the ballgown object\n",
    "transcript_gene_table = indexes(bg)$t2g\n",
    "head(transcript_gene_table)\n",
    "\n",
    "#Each row of data represents a transcript. Many of these transcripts represent the same gene. Determine the numbers of transcripts and unique genes  \n",
    "length(row.names(transcript_gene_table)) #Transcript count\n",
    "length(unique(transcript_gene_table[,\"g_id\"])) #Unique Gene count\n",
    "\n",
    "#### Plot #1 - the number of transcripts per gene.  \n",
    "#Many genes will have only 1 transcript, some genes will have several transcripts\n",
    "#Use the 'table()' command to count the number of times each gene symbol occurs (i.e. the # of transcripts that have each gene symbol)\n",
    "#Then use the 'hist' command to create a histogram of these counts\n",
    "#How many genes have 1 transcript?  More than one transcript?  What is the maximum number of transcripts for a single gene?\n",
    "counts=table(transcript_gene_table[,\"g_id\"])\n",
    "c_one = length(which(counts == 1))\n",
    "c_more_than_one = length(which(counts > 1))\n",
    "c_max = max(counts)\n",
    "hist(counts, breaks=50, col=\"bisque4\", xlab=\"Transcripts per gene\", main=\"Distribution of transcript count per gene\")\n",
    "legend_text = c(paste(\"Genes with one transcript =\", c_one), paste(\"Genes with more than one transcript =\", c_more_than_one), paste(\"Max transcripts for single gene = \", c_max))\n",
    "legend(\"topright\", legend_text, lty=NULL)\n",
    "\n",
    "\n",
    "#### Plot #2 - the distribution of transcript sizes as a histogram\n",
    "#In this analysis we supplied StringTie with tra> library(gplots)\n",
    "nscript models so the lengths will be those of known transcripts\n",
    "#However, if we had used a de novo transcript discovery mode, this step would give us some idea of how well transcripts were being assembled\n",
    "#If we had a low coverage library, or other problems, we might get short 'transcripts' that are actually only pieces of real transcripts\n",
    "\n",
    "full_table <- texpr(bg , 'all')\n",
    "hist(full_table$length, breaks=5\n",
    "Attaching package: gplots\n",
    "\n",
    "The following object is masked from package:stats:\n",
    "\n",
    "    lowess\n",
    "\n",
    "> library(GenomicRanges)\n",
    "Loading required package: stats4\n",
    "Loading required package: BiocGenerics\n",
    "Loading required package: parallel\n",
    "\n",
    "Attaching package: BiocGenerics\n",
    "\n",
    "The following objects are masked from package:parallel:\n",
    "\n",
    "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
    "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
    "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
    "\n",
    "The following objects are masked from package:stats:\n",
    "\n",
    "    IQR, mad, sd, var, xtabs\n",
    "\n",
    "The following objects are masked from package:base:\n",
    "\n",
    "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
    "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
    "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
    "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
    "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
    "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
    "\n",
    "Loading required package: S4Vectors\n",
    "\n",
    "Attaching package: S4Vectors\n",
    "\n",
    "The following object is masked from package:gplots:\n",
    "\n",
    "    space\n",
    "\n",
    "The following object is masked from package:base:\n",
    "\n",
    "    expand.grid\n",
    "\n",
    "Loading required package: IRanges\n",
    "Loading required package: GenomeInfoDb\n",
    "> library(ballgown)\n",
    "\n",
    "Attaching package: ballgown\n",
    "\n",
    "The following object is masked from package:ggplot2:\n",
    "\n",
    "    expr\n",
    "\n",
    "The following object is masked from package:base:\n",
    "\n",
    "    structure\n",
    "\n",
    "> \n",
    "i> #If X11 not available, open a pdf device for output of all plots\n",
    "> pdf(file=\"Tutorial_Part3_Supplementary_R_output.pdf\")\n",
    "> \n",
    "m> #### Import the gene expression data from the HISAT2/StringTie/Ballgown tutorial\n",
    "> \n",
    "H> #Set working directory where results files exist\n",
    "> working_dir = \"~/workspace/rnaseq/de/ballgown/ref_only\" \n",
    "> setwd(working_dir)\n",
    "> \n",
    "> # List the current contents of this directory\n",
    "> dir()\n",
    " [1] \"bg.rda\"                                    \n",
    " [2] \"DE_genes.txt\"                              \n",
    " [3] \"Tutorial_ERCC_DE.pdf\"                      \n",
    " [4] \"Tutorial_ERCC_DE.R\"                        \n",
    " [5] \"Tutorial_Part2_ballgown_output.pdf\"        \n",
    " [6] \"Tutorial_Part3_Supplementary_R_output.pdf\" \n",
    " [7] \"UHR_vs_HBR_gene_results_filtered.tsv\"      \n",
    " [8] \"UHR_vs_HBR_gene_results_sig.tsv\"           \n",
    " [9] \"UHR_vs_HBR_gene_results.tsv\"               \n",
    "[10] \"UHR_vs_HBR_transcript_results_filtered.tsv\"\n",
    "[11] \"UHR_vs_HBR_transcript_results_sig.tsv\"     \n",
    "[12] \"UHR_vs_HBR_transcript_results.tsv\"         \n",
    "[13] \"UHR_vs_HBR.csv\"                            \n",
    "> \n",
    "l> #Import expression and differential expression results from the HISAT2/StringTie/Ballgown pipeline\n",
    "> load('bg.rda')\n",
    "> \n",
    "n> # View a summary of the ballgown object\n",
    "> bg\n",
    "sballgown instance with 4564 transcripts and 6 samples\n",
    "> \n",
    "t> # Load gene names for lookup later in the tutorial\n",
    "> bg_table = texpr(bg, 'all')\n",
    "> bg_gene_names = unique(bg_table[, 9:10])\n",
    "> \n",
    "c> # Pull the gene_expression data frame from the ballgown object\n",
    "> gene_expression = as.data.frame(gexpr(bg))\n",
    "> \n",
    "u> #### Working with 'dataframes'\n",
    "> #View the first five rows of data (all columns) in one of the dataframes created\n",
    "> head(gene_expression)\n",
    "                FPKM.UHR_Rep1 FPKM.UHR_Rep2 FPKM.UHR_Rep3 FPKM.HBR_Rep1\n",
    "ENSG00000008735       9.38558      14.53502      4.809072     703.06514\n",
    "ENSG00000015475     106.93594     115.94829    124.577798      85.26480\n",
    "ENSG00000025708      73.77805      30.93387     43.370466      39.30164\n",
    "ENSG00000025770     328.52167     308.14258    331.613213     251.23852\n",
    "ENSG00000040608      28.28600      23.21874     21.021328     221.51516\n",
    "ENSG00000054611      73.01579      65.71288     82.782018      69.67217\n",
    "                FPKM.HBR_Rep2 FPKM.HBR_Rep3\n",
    "ENSG00000008735     768.34990     759.68554\n",
    "ENSG00000015475      80.12762     102.07289\n",
    "ENSG00000025708      21.91103      55.85629\n",
    "ENSG00000025770     221.04940     226.36098\n",
    "ENSG00000040608     230.12622     230.43580\n",
    "ENSG00000054611      74.37402      77.11958\n",
    "> \n",
    "m> #View the column names\n",
    "> colnames(gene_expression)\n",
    "[1] \"FPKM.UHR_Rep1\" \"FPKM.UHR_Rep2\" \"FPKM.UHR_Rep3\" \"FPKM.HBR_Rep1\"\n",
    "[5] \"FPKM.HBR_Rep2\" \"FPKM.HBR_Rep3\"\n",
    "> \n",
    "g> #View the row names\n",
    "> row.names(gene_expression)\n",
    "   [1] \"ENSG00000008735\" \"ENSG00000015475\" \"ENSG00000025708\" \"ENSG00000025770\"\n",
    "   [5] \"ENSG00000040608\" \"ENSG00000054611\" \"ENSG00000056487\" \"ENSG00000063515\"\n",
    "   [9] \"ENSG00000069998\" \"ENSG00000070010\" \"ENSG00000070371\" \"ENSG00000070413\"\n",
    "  [13] \"ENSG00000073146\" \"ENSG00000073150\" \"ENSG00000073169\" \"ENSG00000075218\"\n",
    "  [17] \"ENSG00000075234\" \"ENSG00000075240\" \"ENSG00000075275\" \"ENSG00000077935\"\n",
    "  [21] \"ENSG00000077942\" \"ENSG00000079974\" \"ENSG00000093000\" \"ENSG00000093009\"\n",
    "  [25] \"ENSG00000093010\" \"ENSG00000093072\" \"ENSG00000093100\" \"ENSG00000099889\"\n",
    "  [29] \"ENSG00000099899\" \"ENSG00000099901\" \"ENSG00000099904\" \"ENSG00000099910\"\n",
    "  [33] \"ENSG00000099917\" \"ENSG00000099937\" \"ENSG00000099940\" \"ENSG00000099942\"\n",
    "  [37] \"ENSG00000099949\" \"ENSG00000099953\" \"ENSG00000099954\" \"ENSG00000099956\"\n",
    "  [41] \"ENSG00000099957\" \"ENSG00000099958\" \"ENSG00000099960\" \"ENSG00000099968\"\n",
    "  [45] \"ENSG00000099974\" \"ENSG00000099977\" \"ENSG00000099984\" \"ENSG00000099985\"\n",
    "  [49] \"ENSG00000099991\" \"ENSG00000099992\" \"ENSG00000099994\" \"ENSG00000099995\"\n",
    "  [53] \"ENSG00000099998\" \"ENSG00000099999\" \"ENSG00000100003\" \"ENSG00000100012\"\n",
    "  [57] \"ENSG00000100014\" \"ENSG00000100023\" \"ENSG00000100024\" \"ENSG00000100027\"\n",
    "  [61] \"ENSG00000100028\" \"ENSG00000100029\" \"ENSG00000100030\" \"ENSG00000100031\"\n",
    "  [65] \"ENSG00000100033\" \"ENSG00000100034\" \"ENSG00000100036\" \"ENSG00000100038\"\n",
    "  [69] \"ENSG00000100053\" \"ENSG00000100055\" \"ENSG00000100056\" \"ENSG00000100058\"\n",
    "  [73] \"ENSG00000100060\" \"ENSG00000100065\" \"ENSG00000100068\" \"ENSG00000100075\"\n",
    "  [77] \"ENSG00000100077\" \"ENSG00000100078\" \"ENSG00000100079\" \"ENSG00000100083\"\n",
    "  [81] \"ENSG00000100084\" \"ENSG00000100092\" \"ENSG00000100095\" \"ENSG00000100097\"\n",
    "  [85] \"ENSG00000100099\" \"ENSG00000100100\" \"ENSG00000100101\" \"ENSG00000100104\"\n",
    "  [89] \"ENSG00000100105\" \"ENSG00000100106\" \"ENSG00000100109\" \"ENSG00000100116\"\n",
    "  [93] \"ENSG00000100121\" \"ENSG00000100122\" \"ENSG00000100124\" \"ENSG00000100129\"\n",
    "  [97] \"ENSG00000100138\" \"ENSG00000100139\" \"ENSG00000100142\" \"ENSG00000100146\"\n",
    " [101] \"ENSG00000100147\" \"ENSG00000100150\" \"ENSG00000100151\" \"ENSG00000100154\"\n",
    " [105] \"ENSG00000100156\" \"ENSG00000100162\" \"ENSG00000100167\" \"ENSG00000100170\"\n",
    " [109] \"ENSG00000100181\" \"ENSG00000100191\" \"ENSG00000100196\" \"ENSG00000100197\"\n",
    " [113] \"ENSG00000100201\" \"ENSG00000100206\" \"ENSG00000100207\" \"ENSG00000100209\"\n",
    " [117] \"ENSG00000100211\" \"ENSG00000100216\" \"ENSG00000100218\" \"ENSG00000100219\"\n",
    " [121] \"ENSG00000100220\" \"ENSG00000100221\" \"ENSG00000100225\" \"ENSG00000100226\"\n",
    " [125] \"ENSG00000100227\" \"ENSG00000100228\" \"ENSG00000100234\" \"ENSG00000100239\"\n",
    " [129] \"ENSG00000100241\" \"ENSG00000100242\" \"ENSG00000100243\" \"ENSG00000100246\"\n",
    " [133] \"ENSG00000100249\" \"ENSG00000100253\" \"ENSG00000100258\" \"ENSG00000100263\"\n",
    " [137] \"ENSG00000100266\" \"ENSG00000100271\" \"ENSG00000100276\" \"ENSG00000100280\"\n",
    " [141] \"ENSG00000100281\" \"ENSG00000100284\" \"ENSG00000100285\" \"ENSG00000100288\"\n",
    " [145] \"ENSG00000100290\" \"ENSG00000100292\" \"ENSG00000100294\" \"ENSG00000100296\"\n",
    " [149] \"ENSG00000100297\" \"ENSG00000100298\" \"ENSG00000100299\" \"ENSG00000100300\"\n",
    " [153] \"ENSG00000100302\" \"ENSG00000100304\" \"ENSG00000100307\" \"ENSG00000100311\"\n",
    " [157] \"ENSG00000100312\" \"ENSG00000100314\" \"ENSG00000100316\" \"ENSG00000100319\"\n",
    " [161] \"ENSG00000100320\" \"ENSG00000100321\" \"ENSG00000100324\" \"ENSG00000100325\"\n",
    " [165] \"ENSG00000100330\" \"ENSG00000100335\" \"ENSG00000100336\" \"ENSG00000100341\"\n",
    " [169] \"ENSG00000100342\" \"ENSG00000100344\" \"ENSG00000100345\" \"ENSG00000100346\"\n",
    " [173] \"ENSG00000100347\" \"ENSG00000100348\" \"ENSG00000100350\" \"ENSG00000100351\"\n",
    " [177] \"ENSG00000100353\" \"ENSG00000100354\" \"ENSG00000100359\" \"ENSG00000100360\"\n",
    " [181] \"ENSG00000100362\" \"ENSG00000100364\" \"ENSG00000100365\" \"ENSG00000100368\"\n",
    " [185] \"ENSG00000100372\" \"ENSG00000100373\" \"ENSG00000100376\" \"ENSG00000100379\"\n",
    " [189] \"ENSG00000100380\" \"ENSG00000100385\" \"ENSG00000100387\" \"ENSG00000100393\"\n",
    " [193] \"ENSG00000100395\" \"ENSG00000100399\" \"ENSG00000100401\" \"ENSG00000100403\"\n",
    " [197] \"ENSG00000100410\" \"ENSG00000100412\" \"ENSG00000100413\" \"ENSG00000100416\"\n",
    " [201] \"ENSG00000100417\" \"ENSG00000100418\" \"ENSG00000100422\" \"ENSG00000100425\"\n",
    " [205] \"ENSG00000100426\" \"ENSG00000100427\" \"ENSG00000100429\" \"ENSG00000128159\"\n",
    " [209] \"ENSG00000128165\" \"ENSG00000128185\" \"ENSG00000128191\" \"ENSG00000128203\"\n",
    " [213] \"ENSG00000128218\" \"ENSG00000128228\" \"ENSG00000128242\" \"ENSG00000128245\"\n",
    " [217] \"ENSG00000128250\" \"ENSG00000128253\" \"ENSG00000128254\" \"ENSG00000128262\"\n",
    " [221] \"ENSG00000128266\" \"ENSG00000128268\" \"ENSG00000128271\" \"ENSG00000128272\"\n",
    " [225] \"ENSG00000128274\" \"ENSG00000128276\" \"ENSG00000128283\" \"ENSG00000128284\"\n",
    " [229] \"ENSG00000128285\" \"ENSG00000128294\" \"ENSG00000128298\" \"ENSG00000128309\"\n",
    " [233] \"ENSG00000128310\" \"ENSG00000128311\" \"ENSG00000128313\" \"ENSG00000128322\"\n",
    " [237] \"ENSG00000128335\" \"ENSG00000128340\" \"ENSG00000128342\" \"ENSG00000128346\"\n",
    " [241] \"ENSG00000128383\" \"ENSG00000128394\" \"ENSG00000128408\" \"ENSG00000130487\"\n",
    " [245] \"ENSG00000130489\" \"ENSG00000130538\" \"ENSG00000130540\" \"ENSG00000130638\"\n",
    " [249] \"ENSG00000130943\" \"ENSG00000131100\" \"ENSG00000133422\" \"ENSG00000133424\"\n",
    " [253] \"ENSG00000133433\" \"ENSG00000133454\" \"ENSG00000133460\" \"ENSG00000133466\"\n",
    " [257] \"ENSG00000133475\" \"ENSG00000133477\" \"ENSG00000133488\" \"ENSG00000133519\"\n",
    " [261] \"ENSG00000138867\" \"ENSG00000138892\" \"ENSG00000138942\" \"ENSG00000138944\"\n",
    " [265] \"ENSG00000138964\" \"ENSG00000159307\" \"ENSG00000159496\" \"ENSG00000159873\"\n",
    " [269] \"ENSG00000159958\" \"ENSG00000161103\" \"ENSG00000161132\" \"ENSG00000161133\"\n",
    " [273] \"ENSG00000161149\" \"ENSG00000161179\" \"ENSG00000161180\" \"ENSG00000166862\"\n",
    " [277] \"ENSG00000166897\" \"ENSG00000167037\" \"ENSG00000167065\" \"ENSG00000167074\"\n",
    " [281] \"ENSG00000167077\" \"ENSG00000168135\" \"ENSG00000169184\" \"ENSG00000169314\"\n",
    " [285] \"ENSG00000169548\" \"ENSG00000169575\" \"ENSG00000169635\" \"ENSG00000169662\"\n",
    " [289] \"ENSG00000169668\" \"ENSG00000170638\" \"ENSG00000172250\" \"ENSG00000172346\"\n",
    " [293] \"ENSG00000172404\" \"ENSG00000172912\" \"ENSG00000172967\" \"ENSG00000175329\"\n",
    " [297] \"ENSG00000176177\" \"ENSG00000176635\" \"ENSG00000177096\" \"ENSG00000177663\"\n",
    " [301] \"ENSG00000177989\" \"ENSG00000177993\" \"ENSG00000178026\" \"ENSG00000178248\"\n",
    " [305] \"ENSG00000178803\" \"ENSG00000179750\" \"ENSG00000180150\" \"ENSG00000180957\"\n",
    " [309] \"ENSG00000181123\" \"ENSG00000182057\" \"ENSG00000182257\" \"ENSG00000182351\"\n",
    " [313] \"ENSG00000182541\" \"ENSG00000182824\" \"ENSG00000182841\" \"ENSG00000182858\"\n",
    " [317] \"ENSG00000182902\" \"ENSG00000182944\" \"ENSG00000183066\" \"ENSG00000183169\"\n",
    " [321] \"ENSG00000183172\" \"ENSG00000183246\" \"ENSG00000183307\" \"ENSG00000183506\"\n",
    " [325] \"ENSG00000183530\" \"ENSG00000183569\" \"ENSG00000183579\" \"ENSG00000183597\"\n",
    " [329] \"ENSG00000183628\" \"ENSG00000183741\" \"ENSG00000183762\" \"ENSG00000183765\"\n",
    " [333] \"ENSG00000183773\" \"ENSG00000183785\" \"ENSG00000183822\" \"ENSG00000183864\"\n",
    " [337] \"ENSG00000183963\" \"ENSG00000184058\" \"ENSG00000184068\" \"ENSG00000184076\"\n",
    " [341] \"ENSG00000184113\" \"ENSG00000184117\" \"ENSG00000184164\" \"ENSG00000184208\"\n",
    " [345] \"ENSG00000184319\" \"ENSG00000184381\" \"ENSG00000184436\" \"ENSG00000184459\"\n",
    " [349] \"ENSG00000184470\" \"ENSG00000184571\" \"ENSG00000184624\" \"ENSG00000184702\"\n",
    " [353] \"ENSG00000184708\" \"ENSG00000184792\" \"ENSG00000184949\" \"ENSG00000184979\"\n",
    " [357] \"ENSG00000184983\" \"ENSG00000185022\" \"ENSG00000185065\" \"ENSG00000185133\"\n",
    " [361] \"ENSG00000185252\" \"ENSG00000185264\" \"ENSG00000185339\" \"ENSG00000185340\"\n",
    " [365] \"ENSG00000185386\" \"ENSG00000185608\" \"ENSG00000185651\" \"ENSG00000185666\"\n",
    " [369] \"ENSG00000185686\" \"ENSG00000185721\" \"ENSG00000185837\" \"ENSG00000185838\"\n",
    " [373] \"ENSG00000186575\" \"ENSG00000186654\" \"ENSG00000186716\" \"ENSG00000186732\"\n",
    " [377] \"ENSG00000186951\" \"ENSG00000186976\" \"ENSG00000186998\" \"ENSG00000187012\"\n",
    " [381] \"ENSG00000187045\" \"ENSG00000187051\" \"ENSG00000187792\" \"ENSG00000187860\"\n",
    " [385] \"ENSG00000187905\" \"ENSG00000187979\" \"ENSG00000188064\" \"ENSG00000188078\"\n",
    " [389] \"ENSG00000188130\" \"ENSG00000188263\" \"ENSG00000188280\" \"ENSG00000188511\"\n",
    " [393] \"ENSG00000188636\" \"ENSG00000188677\" \"ENSG00000189060\" \"ENSG00000189269\"\n",
    " [397] \"ENSG00000189295\" \"ENSG00000189306\" \"ENSG00000196236\" \"ENSG00000196419\"\n",
    " [401] \"ENSG00000196431\" \"ENSG00000196576\" \"ENSG00000196588\" \"ENSG00000197077\"\n",
    " [405] \"ENSG00000197182\" \"ENSG00000197210\" \"ENSG00000197421\" \"ENSG00000197549\"\n",
    " [409] \"ENSG00000198062\" \"ENSG00000198089\" \"ENSG00000198125\" \"ENSG00000198355\"\n",
    " [413] \"ENSG00000198445\" \"ENSG00000198792\" \"ENSG00000198832\" \"ENSG00000198911\"\n",
    " [417] \"ENSG00000198951\" \"ENSG00000199248\" \"ENSG00000199515\" \"ENSG00000199695\"\n",
    " [421] \"ENSG00000199783\" \"ENSG00000199797\" \"ENSG00000199865\" \"ENSG00000200057\"\n",
    " [425] \"ENSG00000200332\" \"ENSG00000200443\" \"ENSG00000200448\" \"ENSG00000200683\"\n",
    " [429] \"ENSG00000200871\" \"ENSG00000200976\" \"ENSG00000200985\" \"ENSG00000201078\"\n",
    " [433] \"ENSG00000201209\" \"ENSG00000201916\" \"ENSG00000202019\" \"ENSG00000202058\"\n",
    " [437] \"ENSG00000203280\" \"ENSG00000203416\" \"ENSG00000203527\" \"ENSG00000203618\"\n",
    " [441] \"ENSG00000205559\" \"ENSG00000205560\" \"ENSG00000205593\" \"ENSG00000205632\"\n",
    " [445] \"ENSG00000205634\" \"ENSG00000205643\" \"ENSG00000205702\" \"ENSG00000205704\"\n",
    " [449] \"ENSG00000205853\" \"ENSG00000205856\" \"ENSG00000206028\" \"ENSG00000206066\"\n",
    " [453] \"ENSG00000206069\" \"ENSG00000206090\" \"ENSG00000206140\" \"ENSG00000206142\"\n",
    " [457] \"ENSG00000206145\" \"ENSG00000206195\" \"ENSG00000206203\" \"ENSG00000206615\"\n",
    " [461] \"ENSG00000206739\" \"ENSG00000206813\" \"ENSG00000206841\" \"ENSG00000207227\"\n",
    " [465] \"ENSG00000207343\" \"ENSG00000207455\" \"ENSG00000207457\" \"ENSG00000207575\"\n",
    " [469] \"ENSG00000207696\" \"ENSG00000207751\" \"ENSG00000207780\" \"ENSG00000207932\"\n",
    " [473] \"ENSG00000208023\" \"ENSG00000209480\" \"ENSG00000209482\" \"ENSG00000211637\"\n",
    " [477] \"ENSG00000211638\" \"ENSG00000211639\" \"ENSG00000211640\" \"ENSG00000211641\"\n",
    " [481] \"ENSG00000211642\" \"ENSG00000211643\" \"ENSG00000211644\" \"ENSG00000211645\"\n",
    " [485] \"ENSG00000211647\" \"ENSG00000211648\" \"ENSG00000211649\" \"ENSG00000211650\"\n",
    " [489] \"ENSG00000211651\" \"ENSG00000211652\" \"ENSG00000211653\" \"ENSG00000211654\"\n",
    " [493] \"ENSG00000211655\" \"ENSG00000211656\" \"ENSG00000211657\" \"ENSG00000211658\"\n",
    " [497] \"ENSG00000211659\" \"ENSG00000211660\" \"ENSG00000211661\" \"ENSG00000211662\"\n",
    " [501] \"ENSG00000211663\" \"ENSG00000211664\" \"ENSG00000211665\" \"ENSG00000211666\"\n",
    " [505] \"ENSG00000211667\" \"ENSG00000211668\" \"ENSG00000211669\" \"ENSG00000211670\"\n",
    " [509] \"ENSG00000211672\" \"ENSG00000211673\" \"ENSG00000211674\" \"ENSG00000211675\"\n",
    " [513] \"ENSG00000211676\" \"ENSG00000211677\" \"ENSG00000211678\" \"ENSG00000211679\"\n",
    " [517] \"ENSG00000211680\" \"ENSG00000211681\" \"ENSG00000211682\" \"ENSG00000211683\"\n",
    " [521] \"ENSG00000211684\" \"ENSG00000211685\" \"ENSG00000212102\" \"ENSG00000212216\"\n",
    " [525] \"ENSG00000212542\" \"ENSG00000212569\" \"ENSG00000212939\" \"ENSG00000213279\"\n",
    " [529] \"ENSG00000213683\" \"ENSG00000213707\" \"ENSG00000213727\" \"ENSG00000213752\"\n",
    " [533] \"ENSG00000213790\" \"ENSG00000213857\" \"ENSG00000213888\" \"ENSG00000213923\"\n",
    " [537] \"ENSG00000214076\" \"ENSG00000214093\" \"ENSG00000214125\" \"ENSG00000214161\"\n",
    " [541] \"ENSG00000214491\" \"ENSG00000214526\" \"ENSG00000214727\" \"ENSG00000215012\"\n",
    " [545] \"ENSG00000215193\" \"ENSG00000215268\" \"ENSG00000215270\" \"ENSG00000215333\"\n",
    " [549] \"ENSG00000215347\" \"ENSG00000215403\" \"ENSG00000215456\" \"ENSG00000215464\"\n",
    " [553] \"ENSG00000215478\" \"ENSG00000215481\" \"ENSG00000215493\" \"ENSG00000215498\"\n",
    " [557] \"ENSG00000215544\" \"ENSG00000215568\" \"ENSG00000215771\" \"ENSG00000217261\"\n",
    " [561] \"ENSG00000217442\" \"ENSG00000218357\" \"ENSG00000218537\" \"ENSG00000219438\"\n",
    " [565] \"ENSG00000220248\" \"ENSG00000220702\" \"ENSG00000220891\" \"ENSG00000221039\"\n",
    " [569] \"ENSG00000221598\" \"ENSG00000221760\" \"ENSG00000221890\" \"ENSG00000221963\"\n",
    " [573] \"ENSG00000222037\" \"ENSG00000222044\" \"ENSG00000222352\" \"ENSG00000222585\"\n",
    " [577] \"ENSG00000222630\" \"ENSG00000222698\" \"ENSG00000222915\" \"ENSG00000223056\"\n",
    " [581] \"ENSG00000223142\" \"ENSG00000223350\" \"ENSG00000223461\" \"ENSG00000223553\"\n",
    " [585] \"ENSG00000223679\" \"ENSG00000223695\" \"ENSG00000223704\" \"ENSG00000223726\"\n",
    " [589] \"ENSG00000223730\" \"ENSG00000223831\" \"ENSG00000223843\" \"ENSG00000223875\"\n",
    " [593] \"ENSG00000223999\" \"ENSG00000224003\" \"ENSG00000224027\" \"ENSG00000224050\"\n",
    " [597] \"ENSG00000224086\" \"ENSG00000224124\" \"ENSG00000224144\" \"ENSG00000224192\"\n",
    " [601] \"ENSG00000224205\" \"ENSG00000224256\" \"ENSG00000224271\" \"ENSG00000224277\"\n",
    " [605] \"ENSG00000224279\" \"ENSG00000224334\" \"ENSG00000224404\" \"ENSG00000224435\"\n",
    " [609] \"ENSG00000224465\" \"ENSG00000224688\" \"ENSG00000224715\" \"ENSG00000224794\"\n",
    " [613] \"ENSG00000224806\" \"ENSG00000224856\" \"ENSG00000224883\" \"ENSG00000224973\"\n",
    " [617] \"ENSG00000225007\" \"ENSG00000225098\" \"ENSG00000225225\" \"ENSG00000225255\"\n",
    " [621] \"ENSG00000225282\" \"ENSG00000225293\" \"ENSG00000225335\" \"ENSG00000225413\"\n",
    " [625] \"ENSG00000225450\" \"ENSG00000225465\" \"ENSG00000225480\" \"ENSG00000225528\"\n",
    " [629] \"ENSG00000225544\" \"ENSG00000225557\" \"ENSG00000225676\" \"ENSG00000225720\"\n",
    " [633] \"ENSG00000225724\" \"ENSG00000225741\" \"ENSG00000225774\" \"ENSG00000225783\"\n",
    " [637] \"ENSG00000225867\" \"ENSG00000225929\" \"ENSG00000225971\" \"ENSG00000226024\"\n",
    " [641] \"ENSG00000226085\" \"ENSG00000226142\" \"ENSG00000226160\" \"ENSG00000226169\"\n",
    " [645] \"ENSG00000226233\" \"ENSG00000226287\" \"ENSG00000226328\" \"ENSG00000226336\"\n",
    " [649] \"ENSG00000226420\" \"ENSG00000226444\" \"ENSG00000226450\" \"ENSG00000226471\"\n",
    " [653] \"ENSG00000226474\" \"ENSG00000226477\" \"ENSG00000226534\" \"ENSG00000226573\"\n",
    " [657] \"ENSG00000226595\" \"ENSG00000226738\" \"ENSG00000226741\" \"ENSG00000226772\"\n",
    " [661] \"ENSG00000226872\" \"ENSG00000226885\" \"ENSG00000226912\" \"ENSG00000226954\"\n",
    " [665] \"ENSG00000226989\" \"ENSG00000227009\" \"ENSG00000227117\" \"ENSG00000227188\"\n",
    " [669] \"ENSG00000227201\" \"ENSG00000227367\" \"ENSG00000227370\" \"ENSG00000227413\"\n",
    " [673] \"ENSG00000227484\" \"ENSG00000227519\" \"ENSG00000227710\" \"ENSG00000227755\"\n",
    " [677] \"ENSG00000227813\" \"ENSG00000227838\" \"ENSG00000227880\" \"ENSG00000227895\"\n",
    " [681] \"ENSG00000228039\" \"ENSG00000228045\" \"ENSG00000228050\" \"ENSG00000228122\"\n",
    " [685] \"ENSG00000228161\" \"ENSG00000228194\" \"ENSG00000228274\" \"ENSG00000228315\"\n",
    " [689] \"ENSG00000228587\" \"ENSG00000228599\" \"ENSG00000228620\" \"ENSG00000228622\"\n",
    " [693] \"ENSG00000228719\" \"ENSG00000228807\" \"ENSG00000228839\" \"ENSG00000228923\"\n",
    " [697] \"ENSG00000229027\" \"ENSG00000229088\" \"ENSG00000229107\" \"ENSG00000229169\"\n",
    " [701] \"ENSG00000229266\" \"ENSG00000229275\" \"ENSG00000229286\" \"ENSG00000229409\"\n",
    " [705] \"ENSG00000229492\" \"ENSG00000229598\" \"ENSG00000229608\" \"ENSG00000229658\"\n",
    " [709] \"ENSG00000229673\" \"ENSG00000229770\" \"ENSG00000229891\" \"ENSG00000229955\"\n",
    " [713] \"ENSG00000229971\" \"ENSG00000229999\" \"ENSG00000230051\" \"ENSG00000230071\"\n",
    " [717] \"ENSG00000230107\" \"ENSG00000230120\" \"ENSG00000230137\" \"ENSG00000230149\"\n",
    " [721] \"ENSG00000230194\" \"ENSG00000230203\" \"ENSG00000230319\" \"ENSG00000230345\"\n",
    " [725] \"ENSG00000230471\" \"ENSG00000230481\" \"ENSG00000230485\" \"ENSG00000230513\"\n",
    " [729] \"ENSG00000230634\" \"ENSG00000230637\" \"ENSG00000230643\" \"ENSG00000230701\"\n",
    " [733] \"ENSG00000230712\" \"ENSG00000230736\" \"ENSG00000230741\" \"ENSG00000230769\"\n",
    " [737] \"ENSG00000230821\" \"ENSG00000230866\" \"ENSG00000230912\" \"ENSG00000230922\"\n",
    " [741] \"ENSG00000230947\" \"ENSG00000231004\" \"ENSG00000231010\" \"ENSG00000231086\"\n",
    " [745] \"ENSG00000231253\" \"ENSG00000231261\" \"ENSG00000231271\" \"ENSG00000231369\"\n",
    " [749] \"ENSG00000231392\" \"ENSG00000231405\" \"ENSG00000231466\" \"ENSG00000231467\"\n",
    " [753] \"ENSG00000231495\" \"ENSG00000231565\" \"ENSG00000231576\" \"ENSG00000231711\"\n",
    " [757] \"ENSG00000231801\" \"ENSG00000231907\" \"ENSG00000231933\" \"ENSG00000231993\"\n",
    " [761] \"ENSG00000232050\" \"ENSG00000232073\" \"ENSG00000232081\" \"ENSG00000232218\"\n",
    " [765] \"ENSG00000232254\" \"ENSG00000232305\" \"ENSG00000232346\" \"ENSG00000232350\"\n",
    " [769] \"ENSG00000232363\" \"ENSG00000232396\" \"ENSG00000232464\" \"ENSG00000232515\"\n",
    " [773] \"ENSG00000232530\" \"ENSG00000232545\" \"ENSG00000232564\" \"ENSG00000232603\"\n",
    " [777] \"ENSG00000232655\" \"ENSG00000232707\" \"ENSG00000232710\" \"ENSG00000232754\"\n",
    " [781] \"ENSG00000232771\" \"ENSG00000232775\" \"ENSG00000232897\" \"ENSG00000232926\"\n",
    " [785] \"ENSG00000233010\" \"ENSG00000233080\" \"ENSG00000233157\" \"ENSG00000233179\"\n",
    " [789] \"ENSG00000233360\" \"ENSG00000233388\" \"ENSG00000233408\" \"ENSG00000233471\"\n",
    " [793] \"ENSG00000233521\" \"ENSG00000233574\" \"ENSG00000233577\" \"ENSG00000233603\"\n",
    " [797] \"ENSG00000233632\" \"ENSG00000233720\" \"ENSG00000233733\" \"ENSG00000233739\"\n",
    " [801] \"ENSG00000233764\" \"ENSG00000233866\" \"ENSG00000233903\" \"ENSG00000233995\"\n",
    " [805] \"ENSG00000234009\" \"ENSG00000234121\" \"ENSG00000234208\" \"ENSG00000234300\"\n",
    " [809] \"ENSG00000234353\" \"ENSG00000234381\" \"ENSG00000234409\" \"ENSG00000234479\"\n",
    " [813] \"ENSG00000234503\" \"ENSG00000234626\" \"ENSG00000234630\" \"ENSG00000234688\"\n",
    " [817] \"ENSG00000234726\" \"ENSG00000234735\" \"ENSG00000234764\" \"ENSG00000234869\"\n",
    " [821] \"ENSG00000234884\" \"ENSG00000234891\" \"ENSG00000234892\" \"ENSG00000234913\"\n",
    " [825] \"ENSG00000234928\" \"ENSG00000234965\" \"ENSG00000234979\" \"ENSG00000235062\"\n",
    " [829] \"ENSG00000235091\" \"ENSG00000235111\" \"ENSG00000235154\" \"ENSG00000235159\"\n",
    " [833] \"ENSG00000235209\" \"ENSG00000235237\" \"ENSG00000235246\" \"ENSG00000235271\"\n",
    " [837] \"ENSG00000235295\" \"ENSG00000235328\" \"ENSG00000235343\" \"ENSG00000235347\"\n",
    " [841] \"ENSG00000235445\" \"ENSG00000235478\" \"ENSG00000235485\" \"ENSG00000235513\"\n",
    " [845] \"ENSG00000235568\" \"ENSG00000235573\" \"ENSG00000235578\" \"ENSG00000235617\"\n",
    " [849] \"ENSG00000235647\" \"ENSG00000235689\" \"ENSG00000235759\" \"ENSG00000235776\"\n",
    " [853] \"ENSG00000235786\" \"ENSG00000235954\" \"ENSG00000235969\" \"ENSG00000235989\"\n",
    " [857] \"ENSG00000235992\" \"ENSG00000236003\" \"ENSG00000236052\" \"ENSG00000236054\"\n",
    " [861] \"ENSG00000236079\" \"ENSG00000236097\" \"ENSG00000236117\" \"ENSG00000236118\"\n",
    " [865] \"ENSG00000236132\" \"ENSG00000236235\" \"ENSG00000236272\" \"ENSG00000236323\"\n",
    " [869] \"ENSG00000236325\" \"ENSG00000236464\" \"ENSG00000236499\" \"ENSG00000236540\"\n",
    " [873] \"ENSG00000236611\" \"ENSG00000236641\" \"ENSG00000236666\" \"ENSG00000236670\"\n",
    " [877] \"ENSG00000236754\" \"ENSG00000236794\" \"ENSG00000236831\" \"ENSG00000236850\"\n",
    " [881] \"ENSG00000236858\" \"ENSG00000236867\" \"ENSG00000236890\" \"ENSG00000237015\"\n",
    " [885] \"ENSG00000237037\" \"ENSG00000237127\" \"ENSG00000237129\" \"ENSG00000237214\"\n",
    " [889] \"ENSG00000237299\" \"ENSG00000237316\" \"ENSG00000237347\" \"ENSG00000237358\"\n",
    " [893] \"ENSG00000237387\" \"ENSG00000237406\" \"ENSG00000237407\" \"ENSG00000237428\"\n",
    " [897] \"ENSG00000237438\" \"ENSG00000237439\" \"ENSG00000237446\" \"ENSG00000237472\"\n",
    " [901] \"ENSG00000237476\" \"ENSG00000237517\" \"ENSG00000237601\" \"ENSG00000237668\"\n",
    " [905] \"ENSG00000237689\" \"ENSG00000237835\" \"ENSG00000237862\" \"ENSG00000237948\"\n",
    " [909] \"ENSG00000237977\" \"ENSG00000238120\" \"ENSG00000238125\" \"ENSG00000238153\"\n",
    " [913] \"ENSG00000238192\" \"ENSG00000238195\" \"ENSG00000238498\" \"ENSG00000238584\"\n",
    " [917] \"ENSG00000238875\" \"ENSG00000239127\" \"ENSG00000239249\" \"ENSG00000239282\"\n",
    " [921] \"ENSG00000239435\" \"ENSG00000239446\" \"ENSG00000239511\" \"ENSG00000239547\"\n",
    " [925] \"ENSG00000239674\" \"ENSG00000239713\" \"ENSG00000239900\" \"ENSG00000239989\"\n",
    " [929] \"ENSG00000240122\" \"ENSG00000240160\" \"ENSG00000240186\" \"ENSG00000240591\"\n",
    " [933] \"ENSG00000240647\" \"ENSG00000240972\" \"ENSG00000241333\" \"ENSG00000241360\"\n",
    " [937] \"ENSG00000241484\" \"ENSG00000241527\" \"ENSG00000241693\" \"ENSG00000241717\"\n",
    " [941] \"ENSG00000241832\" \"ENSG00000241838\" \"ENSG00000241878\" \"ENSG00000241954\"\n",
    " [945] \"ENSG00000241973\" \"ENSG00000241990\" \"ENSG00000242082\" \"ENSG00000242114\"\n",
    " [949] \"ENSG00000242156\" \"ENSG00000242247\" \"ENSG00000242251\" \"ENSG00000242259\"\n",
    " [953] \"ENSG00000242876\" \"ENSG00000243156\" \"ENSG00000243453\" \"ENSG00000243519\"\n",
    " [957] \"ENSG00000243762\" \"ENSG00000243811\" \"ENSG00000243902\" \"ENSG00000244296\"\n",
    " [961] \"ENSG00000244425\" \"ENSG00000244486\" \"ENSG00000244491\" \"ENSG00000244509\"\n",
    " [965] \"ENSG00000244625\" \"ENSG00000244627\" \"ENSG00000244671\" \"ENSG00000244723\"\n",
    " [969] \"ENSG00000244752\" \"ENSG00000248405\" \"ENSG00000248409\" \"ENSG00000248751\"\n",
    " [973] \"ENSG00000249222\" \"ENSG00000249263\" \"ENSG00000249288\" \"ENSG00000249310\"\n",
    " [977] \"ENSG00000249333\" \"ENSG00000249590\" \"ENSG00000249680\" \"ENSG00000249923\"\n",
    " [981] \"ENSG00000250261\" \"ENSG00000250318\" \"ENSG00000250470\" \"ENSG00000250479\"\n",
    " [985] \"ENSG00000251322\" \"ENSG00000251357\" \"ENSG00000251890\" \"ENSG00000251913\"\n",
    " [989] \"ENSG00000251940\" \"ENSG00000251952\" \"ENSG00000251985\" \"ENSG00000252020\"\n",
    " [993] \"ENSG00000252143\" \"ENSG00000252225\" \"ENSG00000252267\" \"ENSG00000252412\"\n",
    " [997] \"ENSG00000252603\" \"ENSG00000252859\" \"ENSG00000252909\" \"ENSG00000253007\"\n",
    "[1001] \"ENSG00000253120\" \"ENSG00000253126\" \"ENSG00000253152\" \"ENSG00000253234\"\n",
    "[1005] \"ENSG00000253239\" \"ENSG00000253242\" \"ENSG00000253329\" \"ENSG00000253338\"\n",
    "[1009] \"ENSG00000253352\" \"ENSG00000253448\" \"ENSG00000253451\" \"ENSG00000253460\"\n",
    "[1013] \"ENSG00000253481\" \"ENSG00000253546\" \"ENSG00000253590\" \"ENSG00000253631\"\n",
    "[1017] \"ENSG00000253637\" \"ENSG00000253691\" \"ENSG00000253752\" \"ENSG00000253779\"\n",
    "[1021] \"ENSG00000253786\" \"ENSG00000253794\" \"ENSG00000253818\" \"ENSG00000253822\"\n",
    "[1025] \"ENSG00000253823\" \"ENSG00000253874\" \"ENSG00000253889\" \"ENSG00000253913\"\n",
    "[1029] \"ENSG00000253920\" \"ENSG00000253935\" \"ENSG00000253963\" \"ENSG00000254029\"\n",
    "[1033] \"ENSG00000254030\" \"ENSG00000254073\" \"ENSG00000254075\" \"ENSG00000254077\"\n",
    "[1037] \"ENSG00000254127\" \"ENSG00000254161\" \"ENSG00000254175\" \"ENSG00000254240\"\n",
    "[1041] \"ENSG00000254264\" \"ENSG00000254308\" \"ENSG00000254355\" \"ENSG00000254413\"\n",
    "[1045] \"ENSG00000254499\" \"ENSG00000254709\" \"ENSG00000254835\" \"ENSG00000255156\"\n",
    "[1049] \"ENSG00000258555\" \"ENSG00000259271\" \"ENSG00000259421\" \"ENSG00000260613\"\n",
    "[1053] \"ENSG00000260708\" \"ENSG00000260924\" \"ENSG00000261188\" \"ENSG00000261202\"\n",
    "[1057] \"ENSG00000261251\" \"ENSG00000261675\" \"ENSG00000263366\" \"ENSG00000263764\"\n",
    "[1061] \"ENSG00000264073\" \"ENSG00000264139\" \"ENSG00000264141\" \"ENSG00000264160\"\n",
    "[1065] \"ENSG00000264346\" \"ENSG00000264661\" \"ENSG00000264757\" \"ENSG00000264824\"\n",
    "[1069] \"ENSG00000266012\" \"ENSG00000266320\" \"ENSG00000266503\" \"ENSG00000266508\"\n",
    "[1073] \"ENSG00000266533\" \"ENSG00000266594\" \"ENSG00000266887\" \"ENSG00000267338\"\n",
    "[1077] \"ENSG00000268292\" \"ENSG00000268812\" \"ENSG00000268818\" \"ENSG00000269220\"\n",
    "[1081] \"ENSG00000269972\" \"ENSG00000269987\" \"ENSG00000270022\" \"ENSG00000270083\"\n",
    "[1085] \"ENSG00000270226\" \"ENSG00000270393\" \"ENSG00000271093\" \"ENSG00000271127\"\n",
    "[1089] \"ENSG00000271138\" \"ENSG00000271181\" \"ENSG00000271275\" \"ENSG00000271672\"\n",
    "[1093] \"ENSG00000272578\" \"ENSG00000272582\" \"ENSG00000272600\" \"ENSG00000272666\"\n",
    "[1097] \"ENSG00000272669\" \"ENSG00000272682\" \"ENSG00000272689\" \"ENSG00000272694\"\n",
    "[1101] \"ENSG00000272720\" \"ENSG00000272733\" \"ENSG00000272779\" \"ENSG00000272787\"\n",
    "[1105] \"ENSG00000272798\" \"ENSG00000272821\" \"ENSG00000272829\" \"ENSG00000272834\"\n",
    "[1109] \"ENSG00000272836\" \"ENSG00000272858\" \"ENSG00000272872\" \"ENSG00000272940\"\n",
    "[1113] \"ENSG00000272942\" \"ENSG00000272954\" \"ENSG00000272973\" \"ENSG00000272977\"\n",
    "[1117] \"ENSG00000273000\" \"ENSG00000273032\" \"ENSG00000273044\" \"ENSG00000273068\"\n",
    "[1121] \"ENSG00000273076\" \"ENSG00000273082\" \"ENSG00000273096\" \"ENSG00000273137\"\n",
    "[1125] \"ENSG00000273139\" \"ENSG00000273145\" \"ENSG00000273164\" \"ENSG00000273176\"\n",
    "[1129] \"ENSG00000273188\" \"ENSG00000273192\" \"ENSG00000273203\" \"ENSG00000273212\"\n",
    "[1133] \"ENSG00000273214\" \"ENSG00000273216\" \"ENSG00000273243\" \"ENSG00000273244\"\n",
    "[1137] \"ENSG00000273253\" \"ENSG00000273272\" \"ENSG00000273287\" \"ENSG00000273289\"\n",
    "[1141] \"ENSG00000273295\" \"ENSG00000273300\" \"ENSG00000273311\" \"ENSG00000273325\"\n",
    "[1145] \"ENSG00000273342\" \"ENSG00000273343\" \"ENSG00000273350\" \"ENSG00000273353\"\n",
    "[1149] \"ENSG00000273362\" \"ENSG00000273366\" \"ENSG00000273387\" \"ENSG00000273424\"\n",
    "[1153] \"ENSG00000273428\" \"ENSG00000273442\" \"ENSG00000273643\" \"ENSG00000273846\"\n",
    "[1157] \"ENSG00000273866\" \"ENSG00000273899\" \"ENSG00000273907\" \"ENSG00000273979\"\n",
    "[1161] \"ENSG00000274044\" \"ENSG00000274170\" \"ENSG00000274252\" \"ENSG00000274280\"\n",
    "[1165] \"ENSG00000274342\" \"ENSG00000274381\" \"ENSG00000274422\" \"ENSG00000274457\"\n",
    "[1169] \"ENSG00000274552\" \"ENSG00000274600\" \"ENSG00000274602\" \"ENSG00000274617\"\n",
    "[1173] \"ENSG00000274625\" \"ENSG00000274717\" \"ENSG00000275004\" \"ENSG00000275060\"\n",
    "[1177] \"ENSG00000275293\" \"ENSG00000275307\" \"ENSG00000275319\" \"ENSG00000275361\"\n",
    "[1181] \"ENSG00000275362\" \"ENSG00000275609\" \"ENSG00000275635\" \"ENSG00000275656\"\n",
    "[1185] \"ENSG00000275772\" \"ENSG00000275793\" \"ENSG00000275818\" \"ENSG00000275840\"\n",
    "[1189] \"ENSG00000275942\" \"ENSG00000276027\" \"ENSG00000276057\" \"ENSG00000276095\"\n",
    "[1193] \"ENSG00000276138\" \"ENSG00000276150\" \"ENSG00000276160\" \"ENSG00000276162\"\n",
    "[1197] \"ENSG00000276400\" \"ENSG00000276427\" \"ENSG00000276664\" \"ENSG00000276675\"\n",
    "[1201] \"ENSG00000276753\" \"ENSG00000276800\" \"ENSG00000276871\" \"ENSG00000276874\"\n",
    "[1205] \"ENSG00000276950\" \"ENSG00000276965\" \"ENSG00000276991\" \"ENSG00000277017\"\n",
    "[1209] \"ENSG00000277232\" \"ENSG00000277248\" \"ENSG00000277385\" \"ENSG00000277460\"\n",
    "[1213] \"ENSG00000277690\" \"ENSG00000277771\" \"ENSG00000277870\" \"ENSG00000277872\"\n",
    "[1217] \"ENSG00000277941\" \"ENSG00000277950\" \"ENSG00000277971\" \"ENSG00000278008\"\n",
    "[1221] \"ENSG00000278044\" \"ENSG00000278110\" \"ENSG00000278188\" \"ENSG00000278195\"\n",
    "[1225] \"ENSG00000278196\" \"ENSG00000278278\" \"ENSG00000278292\" \"ENSG00000278404\"\n",
    "[1229] \"ENSG00000278420\" \"ENSG00000278558\" \"ENSG00000278655\" \"ENSG00000278657\"\n",
    "[1233] \"ENSG00000278667\" \"ENSG00000278869\" \"ENSG00000278881\" \"ENSG00000278890\"\n",
    "[1237] \"ENSG00000278920\" \"ENSG00000278948\" \"ENSG00000278960\" \"ENSG00000279006\"\n",
    "[1241] \"ENSG00000279010\" \"ENSG00000279080\" \"ENSG00000279085\" \"ENSG00000279110\"\n",
    "[1245] \"ENSG00000279159\" \"ENSG00000279175\" \"ENSG00000279182\" \"ENSG00000279184\"\n",
    "[1249] \"ENSG00000279217\" \"ENSG00000279219\" \"ENSG00000279278\" \"ENSG00000279298\"\n",
    "[1253] \"ENSG00000279345\" \"ENSG00000279440\" \"ENSG00000279442\" \"ENSG00000279467\"\n",
    "[1257] \"ENSG00000279494\" \"ENSG00000279548\" \"ENSG00000279560\" \"ENSG00000279597\"\n",
    "[1261] \"ENSG00000279642\" \"ENSG00000279652\" \"ENSG00000279699\" \"ENSG00000279712\"\n",
    "[1265] \"ENSG00000279714\" \"ENSG00000279738\" \"ENSG00000279805\" \"ENSG00000279833\"\n",
    "[1269] \"ENSG00000279848\" \"ENSG00000279914\" \"ENSG00000279920\" \"ENSG00000279927\"\n",
    "[1273] \"ENSG00000279933\" \"ENSG00000279954\" \"ENSG00000279973\" \"ENSG00000279987\"\n",
    "[1277] \"ENSG00000280007\" \"ENSG00000280011\" \"ENSG00000280025\" \"ENSG00000280080\"\n",
    "[1281] \"ENSG00000280156\" \"ENSG00000280178\" \"ENSG00000280199\" \"ENSG00000280216\"\n",
    "[1285] \"ENSG00000280224\" \"ENSG00000280263\" \"ENSG00000280312\" \"ENSG00000280341\"\n",
    "[1289] \"ENSG00000280356\" \"ENSG00000280363\" \"ENSG00000280383\" \"ENSG00000280384\"\n",
    "[1293] \"ENSG00000280395\" \"ENSG00000280418\" \"ENSG00000280424\" \"ENSG00000280434\"\n",
    "[1297] \"ENSG00000280445\" \"ENSG00000280623\" \"ENSG00000280968\" \"ENSG00000281295\"\n",
    "[1301] \"ENSG00000281530\" \"ENSG00000281538\" \"ENSG00000281548\" \"ENSG00000281732\"\n",
    "[1305] \"ENSG00000282012\" \"ENSG00000282041\" \"ENSG00000282602\" \"ENSG00000282816\"\n",
    "[1309] \"ENSG00000283023\" \"ENSG00000283047\" \"ENSG00000283145\" \"ENSG00000283225\"\n",
    "[1313] \"ENSG00000283366\" \"ENSG00000283406\" \"ENSG00000283438\" \"ENSG00000283535\"\n",
    "[1317] \"ENSG00000283583\" \"ENSG00000283633\" \"ERCC-00002\"      \"ERCC-00003\"     \n",
    "[1321] \"ERCC-00004\"      \"ERCC-00009\"      \"ERCC-00012\"      \"ERCC-00013\"     \n",
    "[1325] \"ERCC-00014\"      \"ERCC-00016\"      \"ERCC-00017\"      \"ERCC-00019\"     \n",
    "[1329] \"ERCC-00022\"      \"ERCC-00024\"      \"ERCC-00025\"      \"ERCC-00028\"     \n",
    "[1333] \"ERCC-00031\"      \"ERCC-00033\"      \"ERCC-00034\"      \"ERCC-00035\"     \n",
    "[1337] \"ERCC-00039\"      \"ERCC-00040\"      \"ERCC-00041\"      \"ERCC-00042\"     \n",
    "[1341] \"ERCC-00043\"      \"ERCC-00044\"      \"ERCC-00046\"      \"ERCC-00048\"     \n",
    "[1345] \"ERCC-00051\"      \"ERCC-00053\"      \"ERCC-00054\"      \"ERCC-00057\"     \n",
    "[1349] \"ERCC-00058\"      \"ERCC-00059\"      \"ERCC-00060\"      \"ERCC-00061\"     \n",
    "[1353] \"ERCC-00062\"      \"ERCC-00067\"      \"ERCC-00069\"      \"ERCC-00071\"     \n",
    "[1357] \"ERCC-00073\"      \"ERCC-00074\"      \"ERCC-00075\"      \"ERCC-00076\"     \n",
    "[1361] \"ERCC-00077\"      \"ERCC-00078\"      \"ERCC-00079\"      \"ERCC-00081\"     \n",
    "[1365] \"ERCC-00083\"      \"ERCC-00084\"      \"ERCC-00085\"      \"ERCC-00086\"     \n",
    "[1369] \"ERCC-00092\"      \"ERCC-00095\"      \"ERCC-00096\"      \"ERCC-00097\"     \n",
    "[1373] \"ERCC-00098\"      \"ERCC-00099\"      \"ERCC-00104\"      \"ERCC-00108\"     \n",
    "[1377] \"ERCC-00109\"      \"ERCC-00111\"      \"ERCC-00112\"      \"ERCC-00113\"     \n",
    "[1381] \"ERCC-00116\"      \"ERCC-00117\"      \"ERCC-00120\"      \"ERCC-00123\"     \n",
    "[1385] \"ERCC-00126\"      \"ERCC-00130\"      \"ERCC-00131\"      \"ERCC-00134\"     \n",
    "[1389] \"ERCC-00136\"      \"ERCC-00137\"      \"ERCC-00138\"      \"ERCC-00142\"     \n",
    "[1393] \"ERCC-00143\"      \"ERCC-00144\"      \"ERCC-00145\"      \"ERCC-00147\"     \n",
    "[1397] \"ERCC-00148\"      \"ERCC-00150\"      \"ERCC-00154\"      \"ERCC-00156\"     \n",
    "[1401] \"ERCC-00157\"      \"ERCC-00158\"      \"ERCC-00160\"      \"ERCC-00162\"     \n",
    "[1405] \"ERCC-00163\"      \"ERCC-00164\"      \"ERCC-00165\"      \"ERCC-00168\"     \n",
    "[1409] \"ERCC-00170\"      \"ERCC-00171\"     \n",
    "> \n",
    "c> #Determine the dimensions of the dataframe.  'dim()' will return the number of rows and columns\n",
    "> dim(gene_expression)\n",
    "[1] 1410    6\n",
    "> \n",
    "t> #Get the first 3 rows of data and a selection of columns\n",
    "> gene_expression[1:3,c(1:3,6)]\n",
    "                FPKM.UHR_Rep1 FPKM.UHR_Rep2 FPKM.UHR_Rep3 FPKM.HBR_Rep3\n",
    "ENSG00000008735       9.38558      14.53502      4.809072     759.68554\n",
    "ENSG00000015475     106.93594     115.94829    124.577798     102.07289\n",
    "ENSG00000025708      73.77805      30.93387     43.370466      55.85629\n",
    "> \n",
    "> #Do the same thing, but using the column names instead of numbers\n",
    "> gene_expression[1:3, c(\"FPKM.UHR_Rep1\",\"FPKM.UHR_Rep2\",\"FPKM.UHR_Rep3\",\"FPKM.HBR_Rep3\")]\n",
    "                FPKM.UHR_Rep1 FPKM.UHR_Rep2 FPKM.UHR_Rep3 FPKM.HBR_Rep3\n",
    "ENSG00000008735       9.38558      14.53502      4.809072     759.68554\n",
    "ENSG00000015475     106.93594     115.94829    124.577798     102.07289\n",
    "ENSG00000025708      73.77805      30.93387     43.370466      55.85629\n",
    "> \n",
    "d> #Assign colors to each.  You can specify color by RGB, Hex code, or name\n",
    "> #To get a list of color names:\n",
    "> colours()\n",
    "  [1] \"white\"                \"aliceblue\"            \"antiquewhite\"        \n",
    "  [4] \"antiquewhite1\"        \"antiquewhite2\"        \"antiquewhite3\"       \n",
    "  [7] \"antiquewhite4\"        \"aquamarine\"           \"aquamarine1\"         \n",
    " [10] \"aquamarine2\"          \"aquamarine3\"          \"aquamarine4\"         \n",
    " [13] \"azure\"                \"azure1\"               \"azure2\"              \n",
    " [16] \"azure3\"               \"azure4\"               \"beige\"               \n",
    " [19] \"bisque\"               \"bisque1\"              \"bisque2\"             \n",
    " [22] \"bisque3\"              \"bisque4\"              \"black\"               \n",
    " [25] \"blanchedalmond\"       \"blue\"                 \"blue1\"               \n",
    " [28] \"blue2\"                \"blue3\"                \"blue4\"               \n",
    " [31] \"blueviolet\"           \"brown\"                \"brown1\"              \n",
    " [34] \"brown2\"               \"brown3\"               \"brown4\"              \n",
    " [37] \"burlywood\"            \"burlywood1\"           \"burlywood2\"          \n",
    " [40] \"burlywood3\"           \"burlywood4\"           \"cadetblue\"           \n",
    " [43] \"cadetblue1\"           \"cadetblue2\"           \"cadetblue3\"          \n",
    " [46] \"cadetblue4\"           \"chartreuse\"           \"chartreuse1\"         \n",
    " [49] \"chartreuse2\"          \"chartreuse3\"          \"chartreuse4\"         \n",
    " [52] \"chocolate\"            \"chocolate1\"           \"chocolate2\"          \n",
    " [55] \"chocolate3\"           \"chocolate4\"           \"coral\"               \n",
    " [58] \"coral1\"               \"coral2\"               \"coral3\"              \n",
    " [61] \"coral4\"               \"cornflowerblue\"       \"cornsilk\"            \n",
    " [64] \"cornsilk1\"            \"cornsilk2\"            \"cornsilk3\"           \n",
    " [67] \"cornsilk4\"            \"cyan\"                 \"cyan1\"               \n",
    " [70] \"cyan2\"                \"cyan3\"                \"cyan4\"               \n",
    " [73] \"darkblue\"             \"darkcyan\"             \"darkgoldenrod\"       \n",
    " [76] \"darkgoldenrod1\"       \"darkgoldenrod2\"       \"darkgoldenrod3\"      \n",
    " [79] \"darkgoldenrod4\"       \"darkgray\"             \"darkgreen\"           \n",
    " [82] \"darkgrey\"             \"darkkhaki\"            \"darkmagenta\"         \n",
    " [85] \"darkolivegreen\"       \"darkolivegreen1\"      \"darkolivegreen2\"     \n",
    " [88] \"darkolivegreen3\"      \"darkolivegreen4\"      \"darkorange\"          \n",
    " [91] \"darkorange1\"          \"darkorange2\"          \"darkorange3\"         \n",
    " [94] \"darkorange4\"          \"darkorchid\"           \"darkorchid1\"         \n",
    " [97] \"darkorchid2\"          \"darkorchid3\"          \"darkorchid4\"         \n",
    "[100] \"darkred\"              \"darksalmon\"           \"darkseagreen\"        \n",
    "[103] \"darkseagreen1\"        \"darkseagreen2\"        \"darkseagreen3\"       \n",
    "[106] \"darkseagreen4\"        \"darkslateblue\"        \"darkslategray\"       \n",
    "[109] \"darkslategray1\"       \"darkslategray2\"       \"darkslategray3\"      \n",
    "[112] \"darkslategray4\"       \"darkslategrey\"        \"darkturquoise\"       \n",
    "[115] \"darkviolet\"           \"deeppink\"             \"deeppink1\"           \n",
    "[118] \"deeppink2\"            \"deeppink3\"            \"deeppink4\"           \n",
    "[121] \"deepskyblue\"          \"deepskyblue1\"         \"deepskyblue2\"        \n",
    "[124] \"deepskyblue3\"         \"deepskyblue4\"         \"dimgray\"             \n",
    "[127] \"dimgrey\"              \"dodgerblue\"           \"dodgerblue1\"         \n",
    "[130] \"dodgerblue2\"          \"dodgerblue3\"          \"dodgerblue4\"         \n",
    "[133] \"firebrick\"            \"firebrick1\"           \"firebrick2\"          \n",
    "[136] \"firebrick3\"           \"firebrick4\"           \"floralwhite\"         \n",
    "[139] \"forestgreen\"          \"gainsboro\"            \"ghostwhite\"          \n",
    "[142] \"gold\"                 \"gold1\"                \"gold2\"               \n",
    "[145] \"gold3\"                \"gold4\"                \"goldenrod\"           \n",
    "[148] \"goldenrod1\"           \"goldenrod2\"           \"goldenrod3\"          \n",
    "[151] \"goldenrod4\"           \"gray\"                 \"gray0\"               \n",
    "[154] \"gray1\"                \"gray2\"                \"gray3\"               \n",
    "[157] \"gray4\"                \"gray5\"                \"gray6\"               \n",
    "[160] \"gray7\"                \"gray8\"                \"gray9\"               \n",
    "[163] \"gray10\"               \"gray11\"               \"gray12\"              \n",
    "[166] \"gray13\"               \"gray14\"               \"gray15\"              \n",
    "[169] \"gray16\"               \"gray17\"               \"gray18\"              \n",
    "[172] \"gray19\"               \"gray20\"               \"gray21\"              \n",
    "[175] \"gray22\"               \"gray23\"               \"gray24\"              \n",
    "[178] \"gray25\"               \"gray26\"               \"gray27\"              \n",
    "[181] \"gray28\"               \"gray29\"               \"gray30\"              \n",
    "[184] \"gray31\"               \"gray32\"               \"gray33\"              \n",
    "[187] \"gray34\"               \"gray35\"               \"gray36\"              \n",
    "[190] \"gray37\"               \"gray38\"               \"gray39\"              \n",
    "[193] \"gray40\"               \"gray41\"               \"gray42\"              \n",
    "[196] \"gray43\"               \"gray44\"               \"gray45\"              \n",
    "[199] \"gray46\"               \"gray47\"               \"gray48\"              \n",
    "[202] \"gray49\"               \"gray50\"               \"gray51\"              \n",
    "[205] \"gray52\"               \"gray53\"               \"gray54\"              \n",
    "[208] \"gray55\"               \"gray56\"               \"gray57\"              \n",
    "[211] \"gray58\"               \"gray59\"               \"gray60\"              \n",
    "[214] \"gray61\"               \"gray62\"               \"gray63\"              \n",
    "[217] \"gray64\"               \"gray65\"               \"gray66\"              \n",
    "[220] \"gray67\"               \"gray68\"               \"gray69\"              \n",
    "[223] \"gray70\"               \"gray71\"               \"gray72\"              \n",
    "[226] \"gray73\"               \"gray74\"               \"gray75\"              \n",
    "[229] \"gray76\"               \"gray77\"               \"gray78\"              \n",
    "[232] \"gray79\"               \"gray80\"               \"gray81\"              \n",
    "[235] \"gray82\"               \"gray83\"               \"gray84\"              \n",
    "[238] \"gray85\"               \"gray86\"               \"gray87\"              \n",
    "[241] \"gray88\"               \"gray89\"               \"gray90\"              \n",
    "[244] \"gray91\"               \"gray92\"               \"gray93\"              \n",
    "[247] \"gray94\"               \"gray95\"               \"gray96\"              \n",
    "[250] \"gray97\"               \"gray98\"               \"gray99\"              \n",
    "[253] \"gray100\"              \"green\"                \"green1\"              \n",
    "[256] \"green2\"               \"green3\"               \"green4\"              \n",
    "[259] \"greenyellow\"          \"grey\"                 \"grey0\"               \n",
    "[262] \"grey1\"                \"grey2\"                \"grey3\"               \n",
    "[265] \"grey4\"                \"grey5\"                \"grey6\"               \n",
    "[268] \"grey7\"                \"grey8\"                \"grey9\"               \n",
    "[271] \"grey10\"               \"grey11\"               \"grey12\"              \n",
    "[274] \"grey13\"               \"grey14\"               \"grey15\"              \n",
    "[277] \"grey16\"               \"grey17\"               \"grey18\"              \n",
    "[280] \"grey19\"               \"grey20\"               \"grey21\"              \n",
    "[283] \"grey22\"               \"grey23\"               \"grey24\"              \n",
    "[286] \"grey25\"               \"grey26\"               \"grey27\"              \n",
    "[289] \"grey28\"               \"grey29\"               \"grey30\"              \n",
    "[292] \"grey31\"               \"grey32\"               \"grey33\"              \n",
    "[295] \"grey34\"               \"grey35\"               \"grey36\"              \n",
    "[298] \"grey37\"               \"grey38\"               \"grey39\"              \n",
    "[301] \"grey40\"               \"grey41\"               \"grey42\"              \n",
    "[304] \"grey43\"               \"grey44\"               \"grey45\"              \n",
    "[307] \"grey46\"               \"grey47\"               \"grey48\"              \n",
    "[310] \"grey49\"               \"grey50\"               \"grey51\"              \n",
    "[313] \"grey52\"               \"grey53\"               \"grey54\"              \n",
    "[316] \"grey55\"               \"grey56\"               \"grey57\"              \n",
    "[319] \"grey58\"               \"grey59\"               \"grey60\"              \n",
    "[322] \"grey61\"               \"grey62\"               \"grey63\"              \n",
    "[325] \"grey64\"               \"grey65\"               \"grey66\"              \n",
    "[328] \"grey67\"               \"grey68\"               \"grey69\"              \n",
    "[331] \"grey70\"               \"grey71\"               \"grey72\"              \n",
    "[334] \"grey73\"               \"grey74\"               \"grey75\"              \n",
    "[337] \"grey76\"               \"grey77\"               \"grey78\"              \n",
    "[340] \"grey79\"               \"grey80\"               \"grey81\"              \n",
    "[343] \"grey82\"               \"grey83\"               \"grey84\"              \n",
    "[346] \"grey85\"               \"grey86\"               \"grey87\"              \n",
    "[349] \"grey88\"               \"grey89\"               \"grey90\"              \n",
    "[352] \"grey91\"               \"grey92\"               \"grey93\"              \n",
    "[355] \"grey94\"               \"grey95\"               \"grey96\"              \n",
    "[358] \"grey97\"               \"grey98\"               \"grey99\"              \n",
    "[361] \"grey100\"              \"honeydew\"             \"honeydew1\"           \n",
    "[364] \"honeydew2\"            \"honeydew3\"            \"honeydew4\"           \n",
    "[367] \"hotpink\"              \"hotpink1\"             \"hotpink2\"            \n",
    "[370] \"hotpink3\"             \"hotpink4\"             \"indianred\"           \n",
    "[373] \"indianred1\"           \"indianred2\"           \"indianred3\"          \n",
    "[376] \"indianred4\"           \"ivory\"                \"ivory1\"              \n",
    "[379] \"ivory2\"               \"ivory3\"               \"ivory4\"              \n",
    "[382] \"khaki\"                \"khaki1\"               \"khaki2\"              \n",
    "[385] \"khaki3\"               \"khaki4\"               \"lavender\"            \n",
    "[388] \"lavenderblush\"        \"lavenderblush1\"       \"lavenderblush2\"      \n",
    "[391] \"lavenderblush3\"       \"lavenderblush4\"       \"lawngreen\"           \n",
    "[394] \"lemonchiffon\"         \"lemonchiffon1\"        \"lemonchiffon2\"       \n",
    "[397] \"lemonchiffon3\"        \"lemonchiffon4\"        \"lightblue\"           \n",
    "[400] \"lightblue1\"           \"lightblue2\"           \"lightblue3\"          \n",
    "[403] \"lightblue4\"           \"lightcoral\"           \"lightcyan\"           \n",
    "[406] \"lightcyan1\"           \"lightcyan2\"           \"lightcyan3\"          \n",
    "[409] \"lightcyan4\"           \"lightgoldenrod\"       \"lightgoldenrod1\"     \n",
    "[412] \"lightgoldenrod2\"      \"lightgoldenrod3\"      \"lightgoldenrod4\"     \n",
    "[415] \"lightgoldenrodyellow\" \"lightgray\"            \"lightgreen\"          \n",
    "[418] \"lightgrey\"            \"lightpink\"            \"lightpink1\"          \n",
    "[421] \"lightpink2\"           \"lightpink3\"           \"lightpink4\"          \n",
    "[424] \"lightsalmon\"          \"lightsalmon1\"         \"lightsalmon2\"        \n",
    "[427] \"lightsalmon3\"         \"lightsalmon4\"         \"lightseagreen\"       \n",
    "[430] \"lightskyblue\"         \"lightskyblue1\"        \"lightskyblue2\"       \n",
    "[433] \"lightskyblue3\"        \"lightskyblue4\"        \"lightslateblue\"      \n",
    "[436] \"lightslategray\"       \"lightslategrey\"       \"lightsteelblue\"      \n",
    "[439] \"lightsteelblue1\"      \"lightsteelblue2\"      \"lightsteelblue3\"     \n",
    "[442] \"lightsteelblue4\"      \"lightyellow\"          \"lightyellow1\"        \n",
    "[445] \"lightyellow2\"         \"lightyellow3\"         \"lightyellow4\"        \n",
    "[448] \"limegreen\"            \"linen\"                \"magenta\"             \n",
    "[451] \"magenta1\"             \"magenta2\"             \"magenta3\"            \n",
    "[454] \"magenta4\"             \"maroon\"               \"maroon1\"             \n",
    "[457] \"maroon2\"              \"maroon3\"              \"maroon4\"             \n",
    "[460] \"mediumaquamarine\"     \"mediumblue\"           \"mediumorchid\"        \n",
    "[463] \"mediumorchid1\"        \"mediumorchid2\"        \"mediumorchid3\"       \n",
    "[466] \"mediumorchid4\"        \"mediumpurple\"         \"mediumpurple1\"       \n",
    "[469] \"mediumpurple2\"        \"mediumpurple3\"        \"mediumpurple4\"       \n",
    "[472] \"mediumseagreen\"       \"mediumslateblue\"      \"mediumspringgreen\"   \n",
    "[475] \"mediumturquoise\"      \"mediumvioletred\"      \"midnightblue\"        \n",
    "[478] \"mintcream\"            \"mistyrose\"            \"mistyrose1\"          \n",
    "[481] \"mistyrose2\"           \"mistyrose3\"           \"mistyrose4\"          \n",
    "[484] \"moccasin\"             \"navajowhite\"          \"navajowhite1\"        \n",
    "[487] \"navajowhite2\"         \"navajowhite3\"         \"navajowhite4\"        \n",
    "[490] \"navy\"                 \"navyblue\"             \"oldlace\"             \n",
    "[493] \"olivedrab\"            \"olivedrab1\"           \"olivedrab2\"          \n",
    "[496] \"olivedrab3\"           \"olivedrab4\"           \"orange\"              \n",
    "[499] \"orange1\"              \"orange2\"              \"orange3\"             \n",
    "[502] \"orange4\"              \"orangered\"            \"orangered1\"          \n",
    "[505] \"orangered2\"           \"orangered3\"           \"orangered4\"          \n",
    "[508] \"orchid\"               \"orchid1\"              \"orchid2\"             \n",
    "[511] \"orchid3\"              \"orchid4\"              \"palegoldenrod\"       \n",
    "[514] \"palegreen\"            \"palegreen1\"           \"palegreen2\"          \n",
    "[517] \"palegreen3\"           \"palegreen4\"           \"paleturquoise\"       \n",
    "[520] \"paleturquoise1\"       \"paleturquoise2\"       \"paleturquoise3\"      \n",
    "[523] \"paleturquoise4\"       \"palevioletred\"        \"palevioletred1\"      \n",
    "[526] \"palevioletred2\"       \"palevioletred3\"       \"palevioletred4\"      \n",
    "[529] \"papayawhip\"           \"peachpuff\"            \"peachpuff1\"          \n",
    "[532] \"peachpuff2\"           \"peachpuff3\"           \"peachpuff4\"          \n",
    "[535] \"peru\"                 \"pink\"                 \"pink1\"               \n",
    "[538] \"pink2\"                \"pink3\"                \"pink4\"               \n",
    "[541] \"plum\"                 \"plum1\"                \"plum2\"               \n",
    "[544] \"plum3\"                \"plum4\"                \"powderblue\"          \n",
    "[547] \"purple\"               \"purple1\"              \"purple2\"             \n",
    "[550] \"purple3\"              \"purple4\"              \"red\"                 \n",
    "[553] \"red1\"                 \"red2\"                 \"red3\"                \n",
    "[556] \"red4\"                 \"rosybrown\"            \"rosybrown1\"          \n",
    "[559] \"rosybrown2\"           \"rosybrown3\"           \"rosybrown4\"          \n",
    "[562] \"royalblue\"            \"royalblue1\"           \"royalblue2\"          \n",
    "[565] \"royalblue3\"           \"royalblue4\"           \"saddlebrown\"         \n",
    "[568] \"salmon\"               \"salmon1\"              \"salmon2\"             \n",
    "[571] \"salmon3\"              \"salmon4\"              \"sandybrown\"          \n",
    "[574] \"seagreen\"             \"seagreen1\"            \"seagreen2\"           \n",
    "[577] \"seagreen3\"            \"seagreen4\"            \"seashell\"            \n",
    "[580] \"seashell1\"            \"seashell2\"            \"seashell3\"           \n",
    "[583] \"seashell4\"            \"sienna\"               \"sienna1\"             \n",
    "[586] \"sienna2\"              \"sienna3\"              \"sienna4\"             \n",
    "[589] \"skyblue\"              \"skyblue1\"             \"skyblue2\"            \n",
    "[592] \"skyblue3\"             \"skyblue4\"             \"slateblue\"           \n",
    "[595] \"slateblue1\"           \"slateblue2\"           \"slateblue3\"          \n",
    "[598] \"slateblue4\"           \"slategray\"            \"slategray1\"          \n",
    "[601] \"slategray2\"           \"slategray3\"           \"slategray4\"          \n",
    "[604] \"slategrey\"            \"snow\"                 \"snow1\"               \n",
    "[607] \"snow2\"                \"snow3\"                \"snow4\"               \n",
    "[610] \"springgreen\"          \"springgreen1\"         \"springgreen2\"        \n",
    "[613] \"springgreen3\"         \"springgreen4\"         \"steelblue\"           \n",
    "[616] \"steelblue1\"           \"steelblue2\"           \"steelblue3\"          \n",
    "[619] \"steelblue4\"           \"tan\"                  \"tan1\"                \n",
    "[622] \"tan2\"                 \"tan3\"                 \"tan4\"                \n",
    "[625] \"thistle\"              \"thistle1\"             \"thistle2\"            \n",
    "[628] \"thistle3\"             \"thistle4\"             \"tomato\"              \n",
    "[631] \"tomato1\"              \"tomato2\"              \"tomato3\"             \n",
    "[634] \"tomato4\"              \"turquoise\"            \"turquoise1\"          \n",
    "[637] \"turquoise2\"           \"turquoise3\"           \"turquoise4\"          \n",
    "[640] \"violet\"               \"violetred\"            \"violetred1\"          \n",
    "[643] \"violetred2\"           \"violetred3\"           \"violetred4\"          \n",
    "[646] \"wheat\"                \"wheat1\"               \"wheat2\"              \n",
    "[649] \"wheat3\"               \"wheat4\"               \"whitesmoke\"          \n",
    "[652] \"yellow\"               \"yellow1\"              \"yellow2\"             \n",
    "[655] \"yellow3\"              \"yellow4\"              \"yellowgreen\"         \n",
    "> data_colors=c(\"tomato1\",\"tomato2\",\"tomato3\",\"royalblue1\",\"royalblue2\",\"royalblue3\")\n",
    "> \n",
    "y> #View expression values for the transcripts of a particular gene symbol of chromosome 22.  e.g. 'TST'\n",
    "> #First determine the rows in the data.frame that match 'TST', aka. ENSG00000128311, then display only those rows of the data.frame\n",
    "> i = row.names(gene_expression) == \"ENSG00000128311\"\n",
    "> gene_expression[i,]\n",
    "                FPKM.UHR_Rep1 FPKM.UHR_Rep2 FPKM.UHR_Rep3 FPKM.HBR_Rep1\n",
    "ENSG00000128311       236.999       241.619      164.5634      92.49329\n",
    "                FPKM.HBR_Rep2 FPKM.HBR_Rep3\n",
    "ENSG00000128311      90.61679      84.91327\n",
    "> \n",
    "\"> #What if we want to view values for a list of genes of interest all at once? \n",
    "> #genes_of_interest = c(\"TST\", \"MMP11\", \"LGALS2\", \"ISX\")\n",
    "> genes_of_interest = c(\"ENSG00000128311\",\"ENSG00000099953\",\"ENSG00000100079\",\"ENSG00000175329\")\n",
    "> i = which(row.names(gene_expression) %in% genes_of_interest)\n",
    "> gene_expression[i,]\n",
    "                FPKM.UHR_Rep1 FPKM.UHR_Rep2 FPKM.UHR_Rep3 FPKM.HBR_Rep1\n",
    "ENSG00000099953     30.647211     28.038704      28.87433      3.603737\n",
    "ENSG00000100079     12.621452      9.533409       0.00000      0.000000\n",
    "ENSG00000128311    236.998977    241.618972     164.56340     92.493287\n",
    "ENSG00000175329      1.544123      4.665280       0.00000      0.000000\n",
    "                FPKM.HBR_Rep2 FPKM.HBR_Rep3\n",
    "ENSG00000099953       0.00000      3.634759\n",
    "ENSG00000100079       0.00000      0.000000\n",
    "ENSG00000128311      90.61679     84.913273\n",
    "ENSG00000175329       0.00000      0.000000\n",
    "> \n",
    " > # Load the transcript to gene index from the ballgown object\n",
    "> transcript_gene_table = indexes(bg)$t2g\n",
    "> head(transcript_gene_table)\n",
    "  t_id            g_id\n",
    "1    1 ENSG00000277248\n",
    "2    2 ENSG00000283047\n",
    "3    3 ENSG00000280363\n",
    "4    4 ENSG00000279973\n",
    "5    5 ENSG00000226444\n",
    "6    6 ENSG00000276871\n",
    "> \n",
    "a> #Each row of data represents a transcript. Many of these transcripts represent the same gene. Determie the numbers of transcripts and unique genes  \n",
    "> length(row.names(transcript_gene_table)) #Transcript count\n",
    "[1] 4564\n",
    "> length(unique(transcript_gene_table[,\"g_id\"])) #Unique Gene count\n",
    "[1] 1410\n",
    "> \n",
    " > #### Plot #1 - the number of transcripts per gene.  \n",
    "> #Many genes will have only 1 transcript, some genes will have several transcripts\n",
    "> #Use the 'table()' command to count the number of times each gene symbol occurs (i.e. the # of transcripts that have each gene symbol)\n",
    "> #Then use the 'hist' command to create a histogram of these counts\n",
    "> #How many genes have 1 transcript?  More than one transcript?  What is the maximum number of transcripts for a single gene?\n",
    "> counts=table(transcript_gene_table[,\"g_id\"])\n",
    "> c_one = length(which(counts == 1))\n",
    "> c_more_than_one = length(which(counts > 1))\n",
    "> c_max = max(counts)\n",
    "> hist(counts, breaks=50, col=\"bisque4\", xlab=\"Transcripts per gene\", main=\"Distribution of transcript count per gene\")\n",
    "> legend_text = c(paste(\"Genes with one transcript =\", c_one), paste(\"Genes with more than one transcript =\", c_more_than_one), paste(\"Max transcripts for single gene = \", c_max))\n",
    "> legend(\"topright\", legend_text, lty=NULL)\n",
    "> \n",
    "t> \n",
    "> #### Plot #2 - the distribution of transcript sizes as a histogram\n",
    "> #In this analysis we supplied StringTie with transcript models so the lengths will be those of known transcripts\n",
    "> #However, if we had used a de novo transcript discovery mode, this step would give us some idea of how well transcripts were being assembled\n",
    "> #If we had a low coverage library, or other problems, we might get short 'transcripts' that are actually only pieces of real transcripts\n",
    "> \n",
    "e> full_table <- texpr(bg , 'all')\n",
    "> hist(full_table$length, breaks=50, xlab=\"Transcript length (bp)\", main=\"Distribution of transcript lengths\", col=\"steelblue\")\n",
    "> \n",
    "=> #### Summarize FPKM values for all 6 replicates\n",
    "> #What are the minimum and maximum FPKM values for a particular library?\n",
    "> min(gene_expression[,\"FPKM.UHR_Rep1\"])\n",
    "[1] 0\n",
    "> max(gene_expression[,\"FPKM.UHR_Rep1\"])\n",
    "[1] 171926.5\n",
    "> \n",
    "H> #Set the minimum non-zero FPKM values for use later.\n",
    "> #Do this by grabbing a copy of all data values, coverting 0's to NA, and calculating the minimum or all non NA values\n",
    "> #zz = fpkm_matrix[,data_columns]\n",
    "> #zz[zz==0] = NA\n",
    "> #min_nonzero = min(zz, na.rm=TRUE)\n",
    "> #min_nonzero\n",
    "> \n",
    "> #Alternatively just set min value to 1\n",
    "> min_nonzero=1\n",
    "l> \n",
    "> # Set the columns for finding FPKM and create shorter names for figures\n",
    "> data_columns=c(1:6)\n",
    "> short_names=c(\"UHR_1\",\"UHR_2\",\"UHR_3\",\"HBR_1\",\"HBR_2\",\"HBR_3\")\n",
    "> \n",
    "> #### Plot #3 - View the range of values and general distribution of FPKM values for all 4 libraries\n",
    "> #Create boxplots for this purpose\n",
    "> #Display on a log2 scale and add the minimum non-zero value to avoid log2(0)\n",
    "> boxplot(log2(gene_expression[,data_columns]+min_nonzero), col=data_colors, names=short_names, las=2, ylab=\"log2(FPKM)\", main=\"Distribution of FPKMs for all 6 libraries\")\n",
    "> #Note that the bold horizontal line on each boxplot is the median\n",
    "> \n",
    "n> #### Plot #4 - plot a pair of replicates to assess reproducibility of technical replicates\n",
    "> #Tranform the data by converting to log2 scale after adding an arbitrary small value to avoid log2(0)\n",
    "> x = gene_expression[,\"FPKM.UHR_Rep1\"]\n",
    "> y = gene_expression[,\"FPKM.UHR_Rep2\"]\n",
    "> plot(x=log2(x+min_nonzero), y=log2(y+min_nonzero), pch=16, col=\"blue\", cex=0.25, xlab=\"FPKM (UHR, Replicate 1)\", ylab=\"FPKM (UHR, Replicate 2)\", main=\"Comparison of expression values for a pair of replicates\")\n",
    "> \n",
    "n> #Add a straight line of slope 1, and intercept 0\n",
    "> abline(a=0,b=1)\n",
    "> \n",
    "s> #Calculate the correlation coefficient and display in a legend\n",
    "> rs=cor(x,y)^2\n",
    "n> legend(\"topleft\", paste(\"R squared = \", round(rs, digits=3), sep=\"\"), lwd=1, col=\"black\")\n",
    "> \n",
    ",> #### Plot #5 - Scatter plots with a large number of data points can be misleading ... regenerate thisfigure as a density scatter plot\n",
    "> colors = colorRampPalette(c(\"white\", \"blue\", \"#007FFF\", \"cyan\",\"#7FFF7F\", \"yellow\", \"#FF7F00\", \"red\", \"#7F0000\"))\n",
    "> smoothScatter(x=log2(x+min_nonzero), y=log2(y+min_nonzero), xlab=\"FPKM (UHR, Replicate 1)\", ylab=\"FPKM (UHR, Replicate 2)\", main=\"Comparison of expression values for a pair of replicates\", colramp=colors, nbin=200)\n",
    "> \n",
    "s> \n",
    "> #### Plot all sets of replicates on a single plot\n",
    "> #Create an function that generates an R plot.  This function will take as input the two libraries to be compared and a plot name and color\n",
    "> plotCor = function(lib1, lib2, name, color){\n",
    "+ x=gene_expression[,lib1]\n",
    "+ y=gene_expression[,lib2]\n",
    "+ zero_count = length(which(x==0)) + length(which(y==0))\n",
    "+ plot(x=log2(x+min_nonzero), y=log2(y+min_nonzero), pch=16, col=color, cex=0.25, xlab=lib1, ylab=lib2, main=name)\n",
    "+ abline(a=0,b=1)\n",
    "+ rs=cor(x,y, method=\"pearson\")^2\n",
    "+ legend_text = c(paste(\"R squared = \", round(rs, digits=3), sep=\"\"), paste(\"Zero count = \", zero_count, sep=\"\"))\n",
    "+ legend(\"topleft\", legend_text, lwd=c(1,NA), col=\"black\", bg=\"white\", cex=0.8)\n",
    "+ }\n",
    "> #Open a plotting page with room for two plots on one page\n",
    "> par(mfrow=c(1,2))\n",
    "> \n",
    " > #Plot #6 - Now make a call to our custom function created above, once for each library comparison\n",
    "> plotCor(\"FPKM.UHR_Rep1\", \"FPKM.HBR_Rep1\", \"UHR_1 vs HBR_1\", \"tomato2\")\n",
    "> plotCor(\"FPKM.UHR_Rep2\", \"FPKM.HBR_Rep2\", \"UHR_2 vs HBR_2\", \"royalblue2\")\n",
    "> \n",
    "e> \n",
    "s> ##### One problem with these plots is that there are so many data points on top of each other, that iformation is being lost\n",
    "> #Regenerate these plots using a density scatter plot\n",
    "> plotCor2 = function(lib1, lib2, name, color){\n",
    "+ x=gene_expression[,lib1]\n",
    "+ y=gene_expression[,lib2]\n",
    "+ zero_count = length(which(x==0)) + length(which(y==0))\n",
    "+ colors = colorRampPalette(c(\"white\", \"blue\", \"#007FFF\", \"cyan\",\"#7FFF7F\", \"yellow\", \"#FF7F00\", \"red\", \"#7F0000\"))\n",
    "+ smoothScatter(x=log2(x+min_nonzero), y=log2(y+min_nonzero), xlab=lib1, ylab=lib2, main=name, colramp=colors, nbin=275)\n",
    "+ abline(a=0,b=1)\n",
    "+ rs=cor(x,y, method=\"pearson\")^2\n",
    "+ legend_text = c(paste(\"R squared = \", round(rs, digits=3), sep=\"\"), paste(\"Zero count = \", zero_count, sep=\"\"))\n",
    "+ legend(\"topleft\", legend_text, lwd=c(1,NA), col=\"black\", bg=\"white\", cex=0.8)\n",
    "+ }\n",
    "> \n",
    " > #### Plot #7 - Now make a call to our custom function created above, once for each library comparison\n",
    "> par(mfrow=c(1,2))\n",
    "i> plotCor2(\"FPKM.UHR_Rep1\", \"FPKM.HBR_Rep1\", \"UHR_1 vs HBR_1\", \"tomato2\")\n",
    "> plotCor2(\"FPKM.UHR_Rep2\", \"FPKM.HBR_Rep2\", \"UHR_2 vs HBR_2\", \"royalblue2\")\n",
    "> \n",
    "m> \n",
    "e> #### Compare the correlation 'distance' between all replicates\n",
    "> #Do we see the expected pattern for all eight libraries (i.e. replicates most similar, then tumor vs. normal)?\n",
    "> \n",
    "s> #Calculate the FPKM sum for all 6 libraries\n",
    "> gene_expression[,\"sum\"]=apply(gene_expression[,data_columns], 1, sum)\n",
    "> \n",
    "m> #Identify the genes with a grand sum FPKM of at least 5 - we will filter out the genes with very low xpression across the board\n",
    "> i = which(gene_expression[,\"sum\"] > 5)\n",
    "> \n",
    "\"> #Calculate the correlation between all pairs of data\n",
    "> r=cor(gene_expression[i,data_columns], use=\"pairwise.complete.obs\", method=\"pearson\")\n",
    "> \n",
    "v> #Print out these correlation values\n",
    "> r\n",
    "              FPKM.UHR_Rep1 FPKM.UHR_Rep2 FPKM.UHR_Rep3 FPKM.HBR_Rep1\n",
    "FPKM.UHR_Rep1     1.0000000     0.9975915     0.9996930     0.8208764\n",
    "FPKM.UHR_Rep2     0.9975915     1.0000000     0.9981973     0.8377259\n",
    "FPKM.UHR_Rep3     0.9996930     0.9981973     1.0000000     0.8298098\n",
    "FPKM.HBR_Rep1     0.8208764     0.8377259     0.8298098     1.0000000\n",
    "FPKM.HBR_Rep2     0.8248637     0.8400775     0.8337725     0.9994727\n",
    "FPKM.HBR_Rep3     0.8187110     0.8351467     0.8276353     0.9998584\n",
    "              FPKM.HBR_Rep2 FPKM.HBR_Rep3\n",
    "FPKM.UHR_Rep1     0.8248637     0.8187110\n",
    "FPKM.UHR_Rep2     0.8400775     0.8351467\n",
    "FPKM.UHR_Rep3     0.8337725     0.8276353\n",
    "FPKM.HBR_Rep1     0.9994727     0.9998584\n",
    "FPKM.HBR_Rep2     1.0000000     0.9996791\n",
    "FPKM.HBR_Rep3     0.9996791     1.0000000\n",
    "> \n",
    " > #### Plot #8 - Convert correlation to 'distance', and use 'multi-dimensional scaling' to display the elative differences between libraries\n",
    "> #This step calculates 2-dimensional coordinates to plot points for each library\n",
    "> #Libraries with similar expression patterns (highly correlated to each other) should group together\n",
    "> #What pattern do we expect to see, given the types of libraries we have (technical replicates, biologal replicates, tumor/normal)?\n",
    "> d=1-r\n",
    "> mds=cmdscale(d, k=2, eig=TRUE)\n",
    "> par(mfrow=c(1,1))\n",
    "> plot(mds$points, type=\"n\", xlab=\"\", ylab=\"\", main=\"MDS distance plot (all non-zero genes)\", xlim=c(-0.12,0.12), ylim=c(-0.12,0.12))\n",
    "> points(mds$points[,1], mds$points[,2], col=\"grey\", cex=2, pch=16)\n",
    "> text(mds$points[,1], mds$points[,2], short_names, col=data_colors)\n",
    "> \n",
    "> # Calculate the differential expression results including significance\n",
    "> results_genes = stattest(bg, feature=\"gene\", covariate=\"type\", getFC=TRUE, meas=\"FPKM\")\n",
    "> results_genes = merge(results_genes,bg_gene_names,by.x=c(\"id\"),by.y=c(\"gene_id\"))\n",
    "> \n",
    "> #### Plot #9 - View the distribution of differential expression values as a histogram\n",
    "> #Display only those that are significant according to Ballgown\n",
    "> \n",
    "> sig=which(results_genes$pval<0.05)\n",
    "> results_genes[,\"de\"] = log2(results_genes[,\"fc\"])\n",
    "> hist(results_genes[sig,\"de\"], breaks=50, col=\"seagreen\", xlab=\"log2(Fold change) UHR vs HBR\", main=\"Distribution of differential expression values\")\n",
    "> abline(v=-2, col=\"black\", lwd=2, lty=2)\n",
    "> abline(v=2, col=\"black\", lwd=2, lty=2)\n",
    "> legend(\"topleft\", \"Fold-change > 4\", lwd=2, lty=2)\n",
    "> \n",
    "> #### Plot #10 - Display the grand expression values from UHR and HBR and mark those that are significantly differentially expressed\n",
    "> gene_expression[,\"UHR\"]=apply(gene_expression[,c(1:3)], 1, mean)\n",
    "> gene_expression[,\"HBR\"]=apply(gene_expression[,c(4:6)], 1, mean)\n",
    "> \n",
    "> x=log2(gene_expression[,\"UHR\"]+min_nonzero)\n",
    "> y=log2(gene_expression[,\"HBR\"]+min_nonzero)\n",
    "> plot(x=x, y=y, pch=16, cex=0.25, xlab=\"UHR FPKM (log2)\", ylab=\"HBR FPKM (log2)\", main=\"UHR vs HBR FPKMs\")\n",
    "> abline(a=0, b=1)\n",
    "> xsig=x[sig]\n",
    "> ysig=y[sig]\n",
    "> points(x=xsig, y=ysig, col=\"magenta\", pch=16, cex=0.5)\n",
    "> legend(\"topleft\", \"Significant\", col=\"magenta\", pch=16)\n",
    "> \n",
    "> #Get the gene symbols for the top N (according to corrected p-value) and display them on the plot\n",
    "> topn = order(abs(results_genes[sig,\"fc\"]), decreasing=TRUE)[1:25]\n",
    "> topn = order(results_genes[sig,\"qval\"])[1:25]\n",
    "> text(x[topn], y[topn], results_genes[topn,\"gene_name\"], col=\"black\", cex=0.75, srt=45)\n",
    "> \n",
    "> \n",
    "> #### Write a simple table of differentially expressed transcripts to an output file\n",
    "> #Each should be significant with a log2 fold-change >= 2\n",
    "> sigpi = which(results_genes[,\"pval\"]<0.05)\n",
    "> sigp = results_genes[sigpi,]\n",
    "> sigde = which(abs(sigp[,\"de\"]) >= 2)\n",
    "> sig_tn_de = sigp[sigde,]\n",
    "> \n",
    "> #Order the output by or p-value and then break ties using fold-change\n",
    "> o = order(sig_tn_de[,\"qval\"], -abs(sig_tn_de[,\"de\"]), decreasing=FALSE)\n",
    "> \n",
    "> output = sig_tn_de[o,c(\"gene_name\",\"id\",\"fc\",\"pval\",\"qval\",\"de\")]\n",
    "> write.table(output, file=\"SigDE_supplementary_R.txt\", sep=\"\\t\", row.names=FALSE, quote=FALSE)\n",
    "> \n",
    "> #View selected columns of the first 25 lines of output\n",
    "> output[1:25,c(1,4,5)]\n",
    "         gene_name         pval         qval\n",
    "516          IGLC3 2.269820e-09 1.983823e-06\n",
    "369          PRAME 1.179221e-07 5.153196e-05\n",
    "24           CDC45 1.261807e-06 2.757048e-04\n",
    "306       APOBEC3B 9.794817e-07 2.757048e-04\n",
    "1058 RP5-1119A7.17 2.374988e-06 4.151479e-04\n",
    "162         SYNGR1 3.061438e-06 4.459495e-04\n",
    "504       IGLV2-14 5.721194e-06 5.534999e-04\n",
    "111         KDELR3 5.812050e-06 5.534999e-04\n",
    "254         MYO18B 6.332951e-06 5.534999e-04\n",
    "206           MLC1 4.433320e-06 5.534999e-04\n",
    "985         SHANK3 8.245864e-06 6.551714e-04\n",
    "216          YWHAH 1.256224e-05 7.842428e-04\n",
    "1321    ERCC-00004 1.240150e-05 7.842428e-04\n",
    "30          RANBP1 1.077009e-05 7.842428e-04\n",
    "221           GNAZ 1.839617e-05 1.004891e-03\n",
    "143           NEFH 1.986105e-05 1.021092e-03\n",
    "278          SGSM1 2.288711e-05 1.023408e-03\n",
    "107          SEPT3 2.269393e-05 1.023408e-03\n",
    "1386    ERCC-00130 3.054813e-05 1.270870e-03\n",
    "247        SULT4A1 5.819090e-05 1.756677e-03\n",
    "172        CACNA1I 5.815224e-05 1.756677e-03\n",
    "43          SLC7A4 5.757371e-05 1.756677e-03\n",
    "498       IGLV2-23 6.029783e-05 1.756677e-03\n",
    "280            TEF 5.973668e-05 1.756677e-03\n",
    "571          NPTXR 5.352581e-05 1.756677e-03\n",
    "> \n",
    "> #You can open the file \"SigDE.txt\" in Excel, Calc, etc.\n",
    "> #It should have been written to the current working directory that you set at the beginning of the R tutorial\n",
    "> dir()\n",
    " [1] \"bg.rda\"                                    \n",
    " [2] \"DE_genes.txt\"                              \n",
    " [3] \"SigDE_supplementary_R.txt\"                 \n",
    " [4] \"Tutorial_ERCC_DE.pdf\"                      \n",
    " [5] \"Tutorial_ERCC_DE.R\"                        \n",
    " [6] \"Tutorial_Part2_ballgown_output.pdf\"        \n",
    " [7] \"Tutorial_Part3_Supplementary_R_output.pdf\" \n",
    " [8] \"UHR_vs_HBR_gene_results_filtered.tsv\"      \n",
    " [9] \"UHR_vs_HBR_gene_results_sig.tsv\"           \n",
    "[10] \"UHR_vs_HBR_gene_results.tsv\"               \n",
    "[11] \"UHR_vs_HBR_transcript_results_filtered.tsv\"\n",
    "[12] \"UHR_vs_HBR_transcript_results_sig.tsv\"     \n",
    "[13] \"UHR_vs_HBR_transcript_results.tsv\"         \n",
    "[14] \"UHR_vs_HBR.csv\"                            \n",
    "> \n",
    "> \n",
    "> #### Plot #11 - Create a heatmap to vizualize expression differences between the eight samples\n",
    "> #Define custom dist and hclust functions for use with heatmaps\n",
    "> mydist=function(c) {dist(c,method=\"euclidian\")}\n",
    "> myclust=function(c) {hclust(c,method=\"average\")}\n",
    "> \n",
    "> main_title=\"sig DE Transcripts\"\n",
    "> par(cex.main=0.8)\n",
    "> sig_genes=results_genes[sig,\"id\"]\n",
    "> sig_gene_names=results_genes[sig,\"gene_name\"]\n",
    "> data=log2(as.matrix(gene_expression[sig_genes,data_columns])+1)\n",
    "> heatmap.2(data, hclustfun=myclust, distfun=mydist, na.rm = TRUE, scale=\"none\", dendrogram=\"both\", margins=c(6,7), Rowv=TRUE, Colv=TRUE, symbreaks=FALSE, key=TRUE, symkey=FALSE, density.info=\"none\", trace=\"none\", main=main_title, cexRow=0.3, cexCol=1, labRow=sig_gene_names,col=rev(heat.colors(75)))\n",
    "> \n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "> #The output file can be viewed in your browser at the following url:\n",
    "> #Note, you must replace __YOUR_IP_ADDRESS__ with your own amazon instance IP\n",
    "> #http://__YOUR_IP_ADDRESS__/workspace/rnaseq/de/ballgown/ref_only/Tutorial_Part3_Supplementary_R_output.pdf\n",
    "> #To exit R type:\n",
    "> quit(save=\"no\")\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "More: https://genviz.org/course/\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
