{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 11/13/2018\n",
    "\n",
    "RNA-seq continued\n",
    "\n",
    "Griffith lab https://github.com/griffithlab/rnaseq_tutorial/wiki/Alignment\n",
    "\n",
    "nathalia."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alignment with hisat2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  echo $RNA_ALIGN_DIR\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "[nathalia@aws] $  mkdir -p $RNA_ALIGN_DIR\n",
    "[nathalia@aws] $  cd $RNA_ALIGN_DIR\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  emacs alignment.sh \n",
    "#!/bin/bash\n",
    "\n",
    "hisat2 -p 8 --rg-id=UHR_Rep1 --rg SM:UHR --rg LB:UHR_Rep1_ERCC-Mix1 --rg PL:ILLUMINA --rg PU:CXX1234-ACTGAC.1 -x $RNA_REF_INDEX --dta --rna-strandness RF -1 $RNA_DATA_DIR/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz -2 $RNA_DATA_DIR/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz -S ./UHR_Rep1.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=UHR_Rep2 --rg SM:UHR --rg LB:UHR_Rep2_ERCC-Mix1 --rg PL:ILLUMINA --rg PU:CXX1234-TGACAC.1 -x $RNA_REF_INDEX --dta --rna-strandness RF -1 $RNA_DATA_DIR/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz -2 $RNA_DATA_DIR/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz -S ./UHR_Rep2.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=UHR_Rep3 --rg SM:UHR --rg LB:UHR_Rep3_ERCC-Mix1 --rg PL:ILLUMINA --rg PU:CXX1234-CTGACA.1 -x $RNA_REF_INDEX --dta --rna-strandness RF -1 $RNA_DATA_DIR/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz -2 $RNA_DATA_DIR/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz -S ./UHR_Rep3.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=HBR_Rep1 --rg SM:HBR --rg LB:HBR_Rep1_ERCC-Mix2 --rg PL:ILLUMINA --rg PU:CXX1234-TGACAC.1 -x $RNA_REF_INDEX --dta --rna-strandness RF -1 $RNA_DATA_DIR/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz -2 $RNA_DATA_DIR/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz -S ./HBR_Rep1.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=HBR_Rep2 --rg SM:HBR --rg LB:HBR_Rep2_ERCC-Mix2 --rg PL:ILLUMINA --rg PU:CXX1234-GACACT.1 -x $RNA_REF_INDEX --dta --rna-strandness RF -1 $RNA_DATA_DIR/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz -2 $RNA_DATA_DIR/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz -S ./HBR_Rep2.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=HBR_Rep3 --rg SM:HBR --rg LB:HBR_Rep3_ERCC-Mix2 --rg PL:ILLUMINA --rg PU:CXX1234-ACACTG.1 -x $RNA_REF_INDEX --dta --rna-strandness RF -1 $RNA_DATA_DIR/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz -2 $RNA_DATA_DIR/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz -S ./HBR_Rep3.sam\n",
    "\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bash alignment.sh \n",
    "227392 reads; of these:\n",
    "  227392 (100.00%) were paired; of these:\n",
    "    1167 (0.51%) aligned concordantly 0 times\n",
    "    222479 (97.84%) aligned concordantly exactly 1 time\n",
    "    3746 (1.65%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1167 pairs aligned concordantly 0 times; of these:\n",
    "      534 (45.76%) aligned discordantly 1 time\n",
    "    ----\n",
    "    633 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      1266 mates make up the pairs; of these:\n",
    "        515 (40.68%) aligned 0 times\n",
    "        648 (51.18%) aligned exactly 1 time\n",
    "        103 (8.14%) aligned >1 times\n",
    "99.89% overall alignment rate\n",
    "162373 reads; of these:\n",
    "  162373 (100.00%) were paired; of these:\n",
    "    971 (0.60%) aligned concordantly 0 times\n",
    "    158727 (97.75%) aligned concordantly exactly 1 time\n",
    "    2675 (1.65%) aligned concordantly >1 times\n",
    "    ----\n",
    "    971 pairs aligned concordantly 0 times; of these:\n",
    "      586 (60.35%) aligned discordantly 1 time\n",
    "    ----\n",
    "    385 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      770 mates make up the pairs; of these:\n",
    "        324 (42.08%) aligned 0 times\n",
    "        406 (52.73%) aligned exactly 1 time\n",
    "        40 (5.19%) aligned >1 times\n",
    "99.90% overall alignment rate\n",
    "185442 reads; of these:\n",
    "  185442 (100.00%) were paired; of these:\n",
    "    1162 (0.63%) aligned concordantly 0 times\n",
    "    181295 (97.76%) aligned concordantly exactly 1 time\n",
    "    2985 (1.61%) aligned concordantly >1 times\n",
    "    ----\n",
    "    1162 pairs aligned concordantly 0 times; of these:\n",
    "      718 (61.79%) aligned discordantly 1 time\n",
    "    ----\n",
    "    444 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      888 mates make up the pairs; of these:\n",
    "        381 (42.91%) aligned 0 times\n",
    "        452 (50.90%) aligned exactly 1 time\n",
    "        55 (6.19%) aligned >1 times\n",
    "99.90% overall alignment rate\n",
    "118571 reads; of these:\n",
    "  118571 (100.00%) were paired; of these:\n",
    "    486 (0.41%) aligned concordantly 0 times\n",
    "    117363 (98.98%) aligned concordantly exactly 1 time\n",
    "    722 (0.61%) aligned concordantly >1 times\n",
    "    ----\n",
    "    486 pairs aligned concordantly 0 times; of these:\n",
    "      331 (68.11%) aligned discordantly 1 time\n",
    "    ----\n",
    "    155 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      310 mates make up the pairs; of these:\n",
    "        188 (60.65%) aligned 0 times\n",
    "        112 (36.13%) aligned exactly 1 time\n",
    "        10 (3.23%) aligned >1 times\n",
    "99.92% overall alignment rate\n",
    "144826 reads; of these:\n",
    "  144826 (100.00%) were paired; of these:\n",
    "    622 (0.43%) aligned concordantly 0 times\n",
    "    143265 (98.92%) aligned concordantly exactly 1 time\n",
    "    939 (0.65%) aligned concordantly >1 times\n",
    "    ----\n",
    "    622 pairs aligned concordantly 0 times; of these:\n",
    "      454 (72.99%) aligned discordantly 1 time\n",
    "    ----\n",
    "    168 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      336 mates make up the pairs; of these:\n",
    "        215 (63.99%) aligned 0 times\n",
    "        111 (33.04%) aligned exactly 1 time\n",
    "        10 (2.98%) aligned >1 times\n",
    "99.93% overall alignment rate\n",
    "129786 reads; of these:\n",
    "  129786 (100.00%) were paired; of these:\n",
    "    518 (0.40%) aligned concordantly 0 times\n",
    "    128459 (98.98%) aligned concordantly exactly 1 time\n",
    "    809 (0.62%) aligned concordantly >1 times\n",
    "    ----\n",
    "    518 pairs aligned concordantly 0 times; of these:\n",
    "      353 (68.15%) aligned discordantly 1 time\n",
    "    ----\n",
    "    165 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      330 mates make up the pairs; of these:\n",
    "        206 (62.42%) aligned 0 times\n",
    "        116 (35.15%) aligned exactly 1 time\n",
    "        8 (2.42%) aligned >1 times\n",
    "99.92% overall alignment rate\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# sam to bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  emacs sam_bam.sh\n",
    "\n",
    "#!/bin/bash\n",
    "\n",
    "samtools sort -@ 8 -o UHR_Rep1.bam UHR_Rep1.sam\n",
    "samtools sort -@ 8 -o UHR_Rep2.bam UHR_Rep2.sam \n",
    "samtools sort -@ 8 -o UHR_Rep3.bam UHR_Rep3.sam \n",
    "samtools sort -@ 8 -o HBR_Rep1.bam HBR_Rep1.sam\n",
    "samtools sort -@ 8 -o HBR_Rep2.bam HBR_Rep2.sam\n",
    "samtools sort -@ 8 -o HBR_Rep3.bam HBR_Rep3.sam\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bash sam_bam.sh \n",
    "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
    "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
    "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
    "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
    "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
    "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 843M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.0K Nov 13 14:24 alignment.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 166M Nov 13 14:24 UHR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 119M Nov 13 14:24 UHR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 136M Nov 13 14:24 UHR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  86M Nov 13 14:25 HBR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105M Nov 13 14:25 HBR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  94M Nov 13 14:25 HBR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  303 Nov 13 14:27 sam_bam.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  34M Nov 13 14:28 UHR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25M Nov 13 14:28 UHR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  28M Nov 13 14:28 UHR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  17M Nov 13 14:28 HBR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20M Nov 13 14:28 HBR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  18M Nov 13 14:28 HBR_Rep3.bam\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge hisat2 bam files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME/alignments/hisat2\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  java -Xmx2g -jar $RNA_HOME/tools/picard.jar MergeSamFiles OUTPUT=UHR.bam INPUT=UHR_Rep1.bam INPUT=UHR_Rep2.bam INPUT=UHR_Rep3.bam\n",
    "INFO\t2018-11-13 14:29:56\tMergeSamFiles\t\n",
    "\n",
    "********** NOTE: Picard's command line syntax is changing.\n",
    "**********\n",
    "********** For more information, please see:\n",
    "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
    "**********\n",
    "********** The command line looks like this in the new syntax:\n",
    "**********\n",
    "**********    MergeSamFiles -OUTPUT UHR.bam -INPUT UHR_Rep1.bam -INPUT UHR_Rep2.bam -INPUT UHR_Rep3.bam\n",
    "**********\n",
    "\n",
    "\n",
    "14:30:02.450 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/workspace/rnaseq/tools/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
    "[Tue Nov 13 14:30:02 UTC 2018] MergeSamFiles INPUT=[UHR_Rep1.bam, UHR_Rep2.bam, UHR_Rep3.bam] OUTPUT=UHR.bam    SORT_ORDER=coordinate ASSUME_SORTED=false MERGE_SEQUENCE_DICTIONARIES=false USE_THREADING=false VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
    "[Tue Nov 13 14:30:02 UTC 2018] Executing as ubuntu@ip-172-31-39-185 on Linux 4.15.0-1025-aws amd64; OpenJDK 64-Bit Server VM 10.0.2+13-Ubuntu-1ubuntu0.18.04.3; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.15-SNAPSHOT\n",
    "INFO\t2018-11-13 14:30:02\tMergeSamFiles\tInput files are in same order as output so sorting to temp directory is not needed.\n",
    "INFO\t2018-11-13 14:30:13\tMergeSamFiles\tProcessed     1,000,000 records.  Elapsed time: 00:00:10s.  Time for last 1,000,000:   10s.  Last read position: ERCC-00130:135\n",
    "INFO\t2018-11-13 14:30:14\tMergeSamFiles\tFinished reading inputs.\n",
    "[Tue Nov 13 14:30:14 UTC 2018] picard.sam.MergeSamFiles done. Elapsed time: 0.21 minutes.\n",
    "Runtime.totalMemory()=515899392\n",
    "\n",
    "[nathalia@aws] $  java -Xmx2g -jar $RNA_HOME/tools/picard.jar MergeSamFiles OUTPUT=HBR.bam INPUT=HBR_Rep1.bam INPUT=HBR_Rep2.bam INPUT=HBR_Rep3.bam\n",
    "INFO\t2018-11-13 14:30:37\tMergeSamFiles\t\n",
    "\n",
    "********** NOTE: Picard's command line syntax is changing.\n",
    "**********\n",
    "********** For more information, please see:\n",
    "********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
    "**********\n",
    "********** The command line looks like this in the new syntax:\n",
    "**********\n",
    "**********    MergeSamFiles -OUTPUT HBR.bam -INPUT HBR_Rep1.bam -INPUT HBR_Rep2.bam -INPUT HBR_Rep3.bam\n",
    "**********\n",
    "\n",
    "\n",
    "14:30:37.524 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/workspace/rnaseq/tools/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
    "[Tue Nov 13 14:30:37 UTC 2018] MergeSamFiles INPUT=[HBR_Rep1.bam, HBR_Rep2.bam, HBR_Rep3.bam] OUTPUT=HBR.bam    SORT_ORDER=coordinate ASSUME_SORTED=false MERGE_SEQUENCE_DICTIONARIES=false USE_THREADING=false VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
    "[Tue Nov 13 14:30:37 UTC 2018] Executing as ubuntu@ip-172-31-39-185 on Linux 4.15.0-1025-aws amd64; OpenJDK 64-Bit Server VM 10.0.2+13-Ubuntu-1ubuntu0.18.04.3; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.15-SNAPSHOT\n",
    "INFO\t2018-11-13 14:30:37\tMergeSamFiles\tInput files are in same order as output so sorting to temp directory is not needed.\n",
    "INFO\t2018-11-13 14:30:45\tMergeSamFiles\tFinished reading inputs.\n",
    "[Tue Nov 13 14:30:45 UTC 2018] picard.sam.MergeSamFiles done. Elapsed time: 0.14 minutes.\n",
    "Runtime.totalMemory()=515899392\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 981M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.0K Nov 13 14:24 alignment.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 166M Nov 13 14:24 UHR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 119M Nov 13 14:24 UHR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 136M Nov 13 14:24 UHR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  86M Nov 13 14:25 HBR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105M Nov 13 14:25 HBR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  94M Nov 13 14:25 HBR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  303 Nov 13 14:27 sam_bam.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  34M Nov 13 14:28 UHR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25M Nov 13 14:28 UHR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  28M Nov 13 14:28 UHR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  17M Nov 13 14:28 HBR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20M Nov 13 14:28 HBR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  18M Nov 13 14:28 HBR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  85M Nov 13 14:30 UHR.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53M Nov 13 14:30 HBR.bam\n",
    "\n",
    "[nathalia@aws] $  ls -l *.bam | wc -l\n",
    "8\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Practice 6"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/practice3\n",
    "[nathalia@aws] $  cd data/\n",
    "[nathalia@aws] $  ln -s ../trimmed/*.fastq.gz .\n",
    "[nathalia@aws] $  ls\n",
    "hcc1395_normal_rep1_1.fastq.gz  hcc1395_normal_rep3_1.fastq.gz  hcc1395_tumor_rep2_1.fastq.gz\n",
    "hcc1395_normal_rep1_2.fastq.gz  hcc1395_normal_rep3_2.fastq.gz  hcc1395_tumor_rep2_2.fastq.gz\n",
    "hcc1395_normal_rep2_1.fastq.gz  hcc1395_tumor_rep1_1.fastq.gz   hcc1395_tumor_rep3_1.fastq.gz\n",
    "hcc1395_normal_rep2_2.fastq.gz  hcc1395_tumor_rep1_2.fastq.gz   hcc1395_tumor_rep3_2.fastq.gz\n",
    "\n",
    "[nathalia@aws] $  cp ../../alignments/hisat2/alignment.sh .\n",
    "\n",
    "#!/bin/bash\n",
    "\n",
    "hisat2 -p 8 --rg-id=hcc1395_normal_rep1 --rg SM:hcc1395_normal --rg LB:hcc1395_normal_rep1 --rg PL:ILLUMINA  -x $RNA_REF_INDEX --dta --rna-strandness RF -1 hcc1395_normal_rep1_1.fastq.gz -2 hcc1395_normal_rep1_2.fastq.gz -S ./hcc1395_normal_rep1.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=hcc1395_normal_rep2 --rg SM:hcc1395_normal --rg LB:hcc1395_normal_rep2 --rg PL:ILLUMINA  -x $RNA_REF_INDEX --dta --rna-strandness RF -1 hcc1395_normal_rep2_1.fastq.gz -2 hcc1395_normal_rep2_2.fastq.gz -S ./hcc1395_normal_rep2.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=hcc1395_normal_rep3 --rg SM:hcc1395_normal --rg LB:hcc1395_normal_rep3 --rg PL:ILLUMINA  -x $RNA_REF_INDEX --dta --rna-strandness RF -1 hcc1395_normal_rep3_1.fastq.gz -2 hcc1395_normal_rep3_2.fastq.gz -S ./hcc1395_normal_rep3.sam\n",
    "\n",
    "##\n",
    "\n",
    "hisat2 -p 8 --rg-id=hcc1395_tumor_rep1 --rg SM:hcc1395_tumor --rg LB:hcc1395_tumor_rep1 --rg PL:ILLUMINA  -x $RNA_REF_INDEX --dta --rna-strandness RF -1 hcc1395_tumor_rep1_1.fastq.gz -2 hcc1395_tumor_rep1_2.fastq.gz -S ./hcc1395_tumor_rep1.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=hcc1395_tumor_rep2 --rg SM:hcc1395_tumor --rg LB:hcc1395_tumor_rep2 --rg PL:ILLUMINA  -x $RNA_REF_INDEX --dta --rna-strandness RF -1 hcc1395_tumor_rep2_1.fastq.gz -2 hcc1395_tumor_rep2_2.fastq.gz -S ./hcc1395_tumor_rep2.sam\n",
    "\n",
    "hisat2 -p 8 --rg-id=hcc1395_tumor_rep3 --rg SM:hcc1395_tumor --rg LB:hcc1395_tumor_rep3 --rg PL:ILLUMINA  -x $RNA_REF_INDEX --dta --rna-strandness RF -1 hcc1395_tumor_rep3_1.fastq.gz -2 hcc1395_tumor_rep3_2.fastq.gz -S ./hcc1395_tumor_rep3.sam\n",
    "\n",
    "\n",
    "samtools sort -@ 8 -o hcc1395_normal_rep1.bam hcc1395_normal_rep1.sam\n",
    "samtools sort -@ 8 -o hcc1395_normal_rep2.bam hcc1395_normal_rep2.sam \n",
    "samtools sort -@ 8 -o hcc1395_normal_rep3.bam hcc1395_normal_rep3.sam \n",
    "samtools sort -@ 8 -o hcc1395_tumor_rep1.bam hcc1395_tumor_rep1.sam\n",
    "samtools sort -@ 8 -o hcc1395_tumor_rep2.bam hcc1395_tumor_rep2.sam\n",
    "samtools sort -@ 8 -o hcc1395_tumor_rep3.bam hcc1395_tumor_rep3.sam\n",
    "\n",
    "\n",
    "java -Xmx2g -jar $RNA_HOME/tools/picard.jar MergeSamFiles OUTPUT=hcc_normal.bam INPUT=hcc1395_normal_rep1.bam INPUT=hcc1395_normal_rep2.bam INPUT=hcc1395_normal_rep3.bam\n",
    "\n",
    "java -Xmx2g -jar $RNA_HOME/tools/picard.jar MergeSamFiles OUTPUT=hcc_tumor.bam INPUT=hcc1395_tumor_rep1.bam INPUT=hcc1395_tumor_rep2.bam INPUT=hcc1395_tumor_rep3.bam\n",
    "\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 2.6G\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep3_2.fastq.gz -> ../trimmed/hcc1395_tumor_rep3_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep3_1.fastq.gz -> ../trimmed/hcc1395_tumor_rep3_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep2_2.fastq.gz -> ../trimmed/hcc1395_tumor_rep2_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep2_1.fastq.gz -> ../trimmed/hcc1395_tumor_rep2_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep1_2.fastq.gz -> ../trimmed/hcc1395_tumor_rep1_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   40 Nov 13 14:39 hcc1395_tumor_rep1_1.fastq.gz -> ../trimmed/hcc1395_tumor_rep1_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep3_2.fastq.gz -> ../trimmed/hcc1395_normal_rep3_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep3_1.fastq.gz -> ../trimmed/hcc1395_normal_rep3_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep2_2.fastq.gz -> ../trimmed/hcc1395_normal_rep2_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep2_1.fastq.gz -> ../trimmed/hcc1395_normal_rep2_1.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep1_2.fastq.gz -> ../trimmed/hcc1395_normal_rep1_2.fastq.gz\n",
    "lrwxrwxrwx 1 ubuntu ubuntu   41 Nov 13 14:39 hcc1395_normal_rep1_1.fastq.gz -> ../trimmed/hcc1395_normal_rep1_1.fastq.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.3K Nov 13 15:15 alignment.sh~\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.4K Nov 13 15:22 alignment.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 314M Nov 13 15:22 hcc1395_normal_rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 315M Nov 13 15:23 hcc1395_normal_rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 315M Nov 13 15:23 hcc1395_normal_rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 369M Nov 13 15:26 hcc1395_tumor_rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 368M Nov 13 15:26 hcc1395_tumor_rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 368M Nov 13 15:27 hcc1395_tumor_rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  49M Nov 13 15:27 hcc1395_normal_rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  47M Nov 13 15:27 hcc1395_normal_rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  47M Nov 13 15:27 hcc1395_normal_rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  55M Nov 13 15:27 hcc1395_tumor_rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53M Nov 13 15:27 hcc1395_tumor_rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53M Nov 13 15:27 hcc1395_tumor_rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 141M Nov 13 15:28 hcc_normal.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 159M Nov 13 15:28 hcc_tumor.bam\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ?\n",
    "```\n",
    "[nathalia@aws] $  zcat hcc1395_normal_rep1_1.fastq.gz | head -n 5\n",
    "@K00193:38:H3MYFBBXX:4:1101:10003:44458/1\n",
    "TTCCTTATGAAACAGGAAGAGTCCCTGGGCCCAGGCCTGGCCCACGGTTGTCAAGGCACATCATTGCCAGCAAGCTGAAGCATACCAGCAGCCACAACCTAGATCTCATTCCCAACCCAAAGTTCTGACTTCTGTACAAACTCGTTTCCAG\n",
    "+\n",
    "AAFFFKKKKKKKKKKKKKKKKKKKKKKKKFKKFKKKKF<AAKKKKKKKKKKKKKKKKFKKKFKKKKKKKKKKKFKAFKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKFKKKKKKKKKKKKFKFFKKKKKKKKKKKKFKKKK\n",
    "@K00193:38:H3MYFBBXX:4:1101:10003:46427/1\n",
    "[nathalia@aws] $  zcat /home/ubuntu/workspace/rnaseq/data/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz | head -n 5\n",
    "@HWI-ST718_146963544:6:1213:8996:10047/1\n",
    "CTTTTTTATTTTTGTCTGACTGGGTTGATTCAAAGGTCTGGTCTTTGAGCTCTTAAATTAGTTCTTCTATTTGGCCTAGTCTGTTGCTAAGGCTGCCAAC\n",
    "+\n",
    "CCCFFFFFHHHHGJHIIJHIHIIIFHIJJJJIJJGIBBFGEGGHIIHGGIJJIIHGGHIIIFGCGHHIIHIHHEEE?DFEFFFEEDCEEDDDDDDDBCDD\n",
    "@HWI-ST718_146963544:5:2303:11793:37095/1\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alignment visualization"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  echo $RNA_ALIGN_DIR\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "[nathalia@aws] $  cd $RNA_ALIGN_DIR\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "\n",
    "[nathalia@aws] $  find *.bam -exec echo samtools index {} \\; | sh\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 981M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.0K Nov 13 14:24 alignment.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 166M Nov 13 14:24 UHR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 119M Nov 13 14:24 UHR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 136M Nov 13 14:24 UHR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  86M Nov 13 14:25 HBR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105M Nov 13 14:25 HBR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  94M Nov 13 14:25 HBR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  303 Nov 13 14:27 sam_bam.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  34M Nov 13 14:28 UHR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25M Nov 13 14:28 UHR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  28M Nov 13 14:28 UHR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  17M Nov 13 14:28 HBR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20M Nov 13 14:28 HBR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  18M Nov 13 14:28 HBR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  85M Nov 13 14:30 UHR.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53M Nov 13 14:30 HBR.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  47K Nov 13 15:34 HBR.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  39K Nov 13 15:34 HBR_Rep1.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  40K Nov 13 15:34 HBR_Rep2.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  40K Nov 13 15:34 HBR_Rep3.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  51K Nov 13 15:34 UHR.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  42K Nov 13 15:34 UHR_Rep1.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  44K Nov 13 15:34 UHR_Rep2.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  43K Nov 13 15:34 UHR_Rep3.bam.bai\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In IGV, change the reference to hg38, and go to chr22, then\n",
    "\n",
    "`File` > `Load from URL` \n",
    "\n",
    "http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/alignments/hisat2/HBR.bam\n",
    "\n",
    "http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/alignments/hisat2/UHR.bam\n",
    "\n",
    "In the genomic region box, find `EIF3L`, hit Go\n",
    "\n",
    "Right-click, click on `Show splice-juction track`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![igv1](alignment_igv.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![variant](variant.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BAM read count"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME\n",
    "[nathalia@aws] $  mkdir bam_readcount\n",
    "[nathalia@aws] $  cd bam_readcount\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/bam_readcount\n",
    "\n",
    "[nathalia@aws] $  echo $RNA_REF_FASTA\n",
    "/home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92.fa\n",
    "\n",
    "[nathalia@aws] $  samtools faidx $RNA_REF_FASTA\n",
    "\n",
    "[nathalia@aws] $  ls /home/ubuntu/workspace/rnaseq/refs/\n",
    "chr22_with_ERCC92.1.ht2  chr22_with_ERCC92.6.ht2   chr22_with_ERCC92.gtf  splicesites.tsv\n",
    "chr22_with_ERCC92.2.ht2  chr22_with_ERCC92.7.ht2   exons.tsv              \n",
    "chr22_with_ERCC92.3.ht2  chr22_with_ERCC92.8.ht2   illumina_multiplex.fa\n",
    "chr22_with_ERCC92.4.ht2  chr22_with_ERCC92.fa      \n",
    "chr22_with_ERCC92.5.ht2  chr22_with_ERCC92.fa.fai  \n",
    "\n",
    "# produced the .fai file\n",
    "\n",
    "[nathalia@aws] $  samtools mpileup -f $RNA_REF_FASTA -r 22:18918457-18918467 $RNA_ALIGN_DIR/UHR.bam $RNA_ALIGN_DIR/HBR.bam \n",
    "[mpileup] 2 samples in 2 input files\n",
    "22\t18918457\tA\t7\t..,.,.,\t3EJJDD?\t32\t>.$.,.,,.,,..,,....,....,,,,.,.,.\tICD@mAJCIJDDJJDDDDJEEHJJIJEGEIDI\n",
    "22\t18918458\tG\t7\t..,.,.,\t8EIIEDC\t31\t>.$,.,,.,,..,,....,....,,,,.,.,.\tIB@k:JBJJDDJJDDDDJDEGHJGJFGEIDJ\n",
    "22\t18918459\tC\t7\t..,.,.,\t8DJJDH?\t30\t>,$.,,.,,..,,....,....,,,,.,.,.\tI@mAIDIIDDIJBDDDJDDFHJIJFIHJDJ\n",
    "22\t18918460\tT\t7\t..,.,.,\t4DIIAEB\t29\t>.,,.,,..,,....,....,,,,.,.,.\tImFIDGJDDIICDDCJDEFGIHJFIHJDJ\n",
    "22\t18918461\tC\t7\t..,.,.,\t4DJIDH@\t29\t>.,,.,,..,,....,....,,,,.,.,.\tImHJDJJDDIJDDDDJDDFHJFJFIEJDJ\n",
    "22\t18918462\tC\t8\t..,.,.,^].\t:DJJFE?@\t29\t>.,,.,,..,,....,....,,,,.,.,.\tImHJDIJDDIJDDDCIDDFHIGJHIDIDJ\n",
    "22\t18918463\tA\t8\t..,.,.,.\t>DIJBH5@\t29\t>.,,.,,..,,....,....,,,,.,.,.\tIkDIaHJDDEICDDCGDDFFHFHHGBGFJ\n",
    "22\t18918464\tC\t8\t..,.,.,.\t?DJJFG@@\t29\t>.,,.,t..,,....,....,,,,.,.,.\tIkDJgJJDDJJCDDCJDDEFHHJHGHHFJ\n",
    "22\t18918465\tG\t9\t..,.,.,.^].\t?DJJHG?FF\t30\t>.,a.,,..,,....,....,,,,.,A,.^].\tIkBIgJJDDJJBDDDHDDDFJBJJIFIFJC\n",
    "22\t18918466\tT\t9\t..,.,.,..\t2BJEHF5DF\t30\t>.,,.,,..,,....,....,,,,.,.,..\tIkBFeJJBDIIABBDJDDDFI?JJ@HJHHB\n",
    "22\t18918467\tC\t9\t..,.,.,..\t8DJHHG=DF\t30\t>.$,,.,,..,,....,....,,,,.,.,..\tIdDGgJJDDJJBDDDJDDDDHBJJEHJHIC\n",
    "```\n",
    "\n",
    "Each line consists of chromosome, 1-based coordinate, reference base, the number of reads covering the site, read bases and base qualities. At the read base column, a dot stands for a match to the reference base on the forward strand, a comma for a match on the reverse strand, ACGTN for a mismatch on the forward strand and acgtn for a mismatch on the reverse strand. A pattern \\+[0-9]+[ACGTNacgtn]+ indicates there is an insertion between this reference position and the next reference position.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  echo \"22 38483683 38483683\" > snvs.bed\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bam-readcount -l snvs.bed -f $RNA_REF_FASTA $RNA_ALIGN_DIR/UHR.bam 2>/dev/null\n",
    "22\t38483683\tG\t326\t=:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\tA:163:60.00:37.15:0.00:94:69:0.53:0.01:38.37:96:0.56:99.98:0.54\tC:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\tG:163:60.00:37.06:0.74:84:79:0.45:0.00:1.89:89:0.53:99.85:0.54\tT:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\tN:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\n",
    "\n",
    "#326 = depth\n",
    "#A/G 50/50\n",
    "\n",
    "#IGV chr22:38483683\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bam-readcount -l snvs.bed -f $RNA_REF_FASTA $RNA_ALIGN_DIR/HBR.bam 2>/dev/null\n",
    "22\t38483683\tG\t206\t=:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\tA:75:60.00:38.41:0.00:44:31:0.52:0.01:38.99:44:0.55:99.99:0.50\tC:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\tG:131:60.00:37.31:0.00:76:55:0.50:0.00:1.22:77:0.56:99.95:0.54\tT:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\tN:0:0.00:0.00:0.00:0:0:0.00:0.00:0.00:0:0.00:0.00:0.00\n",
    "\n",
    "#Here, is not exactly 50/50\n",
    "\n",
    "[nathalia@aws] $  bam-readcount -l snvs.bed -f $RNA_REF_FASTA $RNA_ALIGN_DIR/UHR.bam 2>/dev/null 1>UHR_bam-readcounts.txt\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bam-readcount -l snvs.bed -f $RNA_REF_FASTA $RNA_ALIGN_DIR/HBR.bam 2>/dev/null 1>HBR_bam-readcounts.txt\n",
    "\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 12K\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  21 Nov 13 15:53 snvs.bed\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 365 Nov 13 15:54 UHR_bam-readcounts.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 364 Nov 13 15:54 HBR_bam-readcounts.txt\n",
    "\n",
    "\n",
    "[nathalia@aws] $  cat UHR_bam-readcounts.txt | perl -ne '@data=split(\"\\t\", $_); @Adata=split(\":\", $data[5]); @Cdata=split(\":\", $data[6]); @Gdata=split(\":\", $data[7]); @Tdata=split(\":\", $data[8]); print \"UHR Counts\\t$data[0]\\t$data[1]\\tA: $Adata[1]\\tC: $Cdata[1]\\tT: $Tdata[1]\\tG: $Gdata[1]\\n\";'\n",
    "UHR Counts\t22\t38483683\tA: 163\tC: 0\tT: 0\tG: 163\n",
    "\n",
    "# uses perl to split the last columns, so we can easily see the counts of nt bases\n",
    "\n",
    "[nathalia@aws] $  cat HBR_bam-readcounts.txt | perl -ne '@data=split(\"\\t\", $_); @Adata=split(\":\", $data[5]); @Cdata=split(\":\", $data[6]); @Gdata=split(\":\", $data[7]); @Tdata=split(\":\", $data[8]); print \"HBR Counts\\t$data[0]\\t$data[1]\\tA: $Adata[1]\\tC: $Cdata[1]\\tT: $Tdata[1]\\tG: $Gdata[1]\\n\";'\n",
    "HBR Counts\t22\t38483683\tA: 75\tC: 0\tT: 0\tG: 131\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Practice 6"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME/practice3/data\n",
    "\n",
    "[nathalia@aws] $  find *.bam -exec samtools index {} \\; | sh\n",
    "```\n",
    "\n",
    "Loading into IGV\n",
    "\n",
    "http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/practice3/data/hcc_normal.bam\n",
    "\n",
    "\n",
    "http://ec2-54-184-243-221.us-west-2.compute.amazonaws.com/workspace/rnaseq/practice3/data/hcc_tumor.bam\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![practice](alignment_practice_igv.png) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![splices](Sashimi_.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Alignment QC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_ALIGN_DIR\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "\n",
    "[nathalia@aws] $  samtools view UHR.bam | head\n",
    "HWI-ST718_146963544:5:1311:6035:51578\t419\t22\t10649269\t1\t100M\t=\t10649312\t143\tTGGGTCTATTTCTTTAAAACAAAGGGAGGGGAGTCTCTCATTTACATTAGTTTTTTTCATAGCCTTTTGGACTTTGCAATTTCTATGTTTTGGAACCTAT\t@@CFDDFFHHHHHJJJJJIJJJJJJJFHIJJBGFHIIJJJJJJJJJJJJJIIJHJIHFFFFFFFEEEEEDDDDDDDDEDDDDDEEEECDEDDDDDDDDDD\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:1311:6035:51578\t339\t22\t10649312\t1\t100M\t=\t10649269\t-143\tACATTAGTTTTTTTCATAGCCTTTTGGACTTTGCAATTTCTATGTTTTGGAACCTATTTCTTACAGTTTTTCTATGCTAAACTCTGTCCTGGTCAGTTCC\tCDDDDDDDDEEEEEEEFFFFHHHHEJHHIJJIIIGJJJIJJJJGHEJIJHGBGGHFIIIIIIGJJJJJJJJJJGJJJJJJIJJJJJJHHHHHFFFFFCCC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:6:1310:5357:21076\t355\t22\t10834840\t1\t100M\t=\t10834935\t195\tTGAATAGTGCATTCAAGTAATCTGTAGAACAAAGTTTGTATTTTATATTTTGGTGGGAGGGAGAAACCAGTTAATTTTCCCCTCTTAACTTCAGAAAGCA\tCCCFFFFEHHHHHJJJJGIJJJJJJJJJJJJJJJHIJJHIJJJJIJIJJJJIIGHIJIJJJIJJJJJJHHHHHFFDFFFFEEEDEDDEDDEDDDDDDDDC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:-\tNH:i:2\n",
    "HWI-ST718_146963544:6:1310:5357:21076\t403\t22\t10834935\t1\t100M\t=\t10834840\t-195\tAAGCATACTTGTTCAAATGTTTATAGATCATTTGTATTTTTCTATACTTTAGAAAAAAATAGTTCTATATTCTCTATTTTAGGGTATTAACTCTCAAGAG\tEEDEEDCDDDEEEEEEEFFFFFFFHHHHHHJJJJJJJJJJJJJIJHJJIIJJJJJJJJJJJJJJJIJIJJIJJJJJIJIJJJJJJJJHHHHHFFFFFCCC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:-\tNH:i:2\n",
    "HWI-ST718_146963544:5:2216:3503:44168\t419\t22\t10835740\t1\t100M\t=\t10835762\t122\tGCTACTTCTTCTTTCATGGGTCTTTTTGTTGGGTTTGCATCCTGGTTCTTCCATTGTTGATCTTGCATAGGAATATTTTTGTAATTCACATTTTTTATTA\tBBCDFFFFHHHHHJJEHIJJAFHJJJJJJJJJJHJJJJJJJJJJJJJJJJHIHIIJHIJHIJJJJJJJJJJIJHHHHHHHFBDFDFEEEEEEFEDDDDDF\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2216:3503:44168\t339\t22\t10835762\t1\t100M\t=\t10835740\t-122\tTTTTTGTTGGGTTTGCATCCTGGTTCTTCCATTGTTGATCTTGCATAGGAATATTTTTGTAATTCACATTTTTTATTAATATGCTGCCTGCTTTTCTTTC\tDDDDDDDDDDDDDDDDDDDDEDFFDFHHHHGIJJJJJJIJJJJIJIIJJJJJJJJJJJJJJJJIHGJJJJJJJJIJIJJJJJIJIIJHHHHHFFFFFCCC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2208:13984:35958\t163\t22\t10836084\t1\t100M\t=\t10836228\t244\tCACAAAGAAGCAAACTGTGTCCTGAACAGTTTTGCAAACTCTGTAGTTGGTAGTATTCTTTTCACACGTCTTTCTTGTCGTTCTTTTTAACAACAGTCAT\tCCCFFFDFHHHHHJJIIHGIJJJJIJIIJHGHIIIIIJJIIIGGIIIIIICFGIHIJEIHIIDGIIJJHGIJJIGJIHHHEFBBDEEECCACDDDDA:AD\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:-6\tYT:Z:CP\tRG:Z:UHR_Rep1\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2208:13984:35958\t83\t22\t10836228\t1\t100M\t=\t10836084\t-244\tGTGACATAGTATAGTATCCTTAGTATCCTAACACACAGTAGAGTACCTAAAGTCAGAACTTTTAAGAGAGATACCTGTAAATTTGGACATTACATCAGAT\tEDFCDDDEEHFHEHHIGIJJIHEHHIJIHHGIIJGIIIJJIHIHGFJHGIICGIGIJIIHIFJJJJJJIHHFBJJIJIJJJJJIJIJGHHHHFFFFF@CC\tAS:i:-6\tZS:i:-6\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:86G13\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep1\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:6:1103:13485:78315\t419\t22\t10837443\t1\t100M\t=\t10837541\t198\tCTAACTGATAAGAGAAAAAGTATAACACAACCTCCAAAATTAAAAATCACTTCAGAGGATTTCCAATACTTGTGTGGAGGGGTGAGCTCCTAACAAACTG\t@C@FFFFDHDHHBHIGIJJGCFHIIIIIIJGGIGGIJJJJJGIIJIJIJJIIIGIGHCGGIJJIDHIIIGGIIF?CCFFFB@>?BAC:ACCCDDCCBBD@\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:6:1103:13485:78315\t339\t22\t10837541\t1\t100M\t=\t10837443\t-198\tTGATCTTCTCACAAATAACCATTTGTAAACTCTGCACATAATATAGATAACATCTATCTGAGGGTTGTGGAGATTGAATAAAAGCAGGCAAGCTTTGGAG\tDCACC@CCCFFFFDDE?=FHHHAGGIHIHGIGGGIHGGHHIJJGJIGIHDGCIIHBGGEJJJJIJJIJJIIJJJIGIIIJJIJJJIIHHHHGFFFFF@C@\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://broadinstitute.github.io/picard/explain-flags.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  samtools view -f 3 -F 268 UHR.bam | head\n",
    "HWI-ST718_146963544:5:2208:13984:35958\t163\t22\t10836084\t1\t100M\t=\t10836228\t244\tCACAAAGAAGCAAACTGTGTCCTGAACAGTTTTGCAAACTCTGTAGTTGGTAGTATTCTTTTCACACGTCTTTCTTGTCGTTCTTTTTAACAACAGTCAT\tCCCFFFDFHHHHHJJIIHGIJJJJIJIIJHGHIIIIIJJIIIGGIIIIIICFGIHIJEIHIIDGIIJJHGIJJIGJIHHHEFBBDEEECCACDDDDA:AD\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:-6\tYT:Z:CP\tRG:Z:UHR_Rep1\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2208:13984:35958\t83\t22\t10836228\t1\t100M\t=\t10836084\t-244\tGTGACATAGTATAGTATCCTTAGTATCCTAACACACAGTAGAGTACCTAAAGTCAGAACTTTTAAGAGAGATACCTGTAAATTTGGACATTACATCAGAT\tEDFCDDDEEHFHEHHIGIJJIHEHHIJIHHGIIJGIIIJJIHIHGFJHGIICGIGIJIIHIFJJJJJJIHHFBJJIJIJJJJJIJIJGHHHHFFFFF@CC\tAS:i:-6\tZS:i:-6\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:86G13\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep1\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:1116:3649:92911\t163\t22\t10837841\t1\t100M\t=\t10837925\t184\tATATACCAACTCTAAGATAACCCACATGTTGGAAATATCACATGAGGTCTATTGTAACTAGCGAGGTAGAGAAAAGTTTACTTATAATGAATTAAAAGAT\tCCCFFFFFHHHHHIIHHIJJHHIIDHIJJIJJGIJJIHIJJGHIHGIFHIGHIIJJIJJJIJJFHIHIHHFHHHFFEFFEEEEEEEEEDDDCDDDDDDDD\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:1116:3649:92911\t83\t22\t10837925\t1\t100M\t=\t10837841\t-184\tTAATGAATTAAAAGATAGGAAGTACCAGCCGTGAAATAAAACAAAATCAGATGGCAATTCTAGAGCTGAAAAATATATCAGAATTAAAAATAGGCTCTAT\tEFDCEEEEDEEDEFFEFFFHHHHHHJJJJJJGHJJJIJJJIJJJIIIJJIIJJJJIIIJJJJJJJJIHJJJJJIJJJJJJJJJIHJJHHHHHFFFFFCCC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:6:2107:2342:27323\t163\t22\t10838055\t1\t100M\t=\t10838144\t189\tAAGTAAATGAACTTGAATATAGATTGATAGAAAATCTGAAGAGAGTTATGAAAGATTGGGGTGAAAAAATAGAACCATAGAGATGTATGGGGGCAGTTTT\t?B@DDFDFHHHHHJJJJJJJJJIJJIJJIIIJJJJJJJIIJGIEHHIIIJGIIJJJJJJJJFHIJJJJHGHHFFFFFDEEEECDDDDFEDDDDDBBCDDD\tAS:i:-6\tZS:i:-6\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:16T83\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep3\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:6:2107:2342:27323\t83\t22\t10838144\t1\t100M\t=\t10838055\t-189\tGGGGCAGTTTTGAAAGGTCTAATAGGCACAGTATAAGAGAAAGAGCCAGAAAAAATATTTGACTAAATGATGACAGAAAACTTTTCAGATTTGGTAAAAG\tDDEEEEEEFEDFFFHHHHGHJJIHEJJJJJJJJJJJJJJIJJJIGIGJJJJJJJJJJJIHHGJIJJJJJJJIIJJJJIHIJJJJJJJHHHHHFFFFFCCC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:-6\tYT:Z:CP\tRG:Z:UHR_Rep3\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2312:11656:62706\t163\t22\t10838443\t1\t100M\t=\t10838707\t364\tCCATAGAGGTGGAACAAAATCTTTAAAGTTCTGAAAGGAAGAAAAAAATCTGTCAACCTAGAATTTTTTATCCAGTGAAAATATTCTTTAAGCCTTTTGA\t@@7ADDDD?DDF;CBC@:;<ECHIAAFCCDFCBBCEFE;;DFGIF?DAFE>CFEE>@FFEE@EFF<CEDCBAAA>AB@@ABBB;>>BDBBBBBB@ABB><\tAS:i:-5\tZS:i:-5\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:92T7\tYS:i:-4\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2308:15338:36631\t99\t22\t10838665\t1\t100M\t=\t10838833\t268\tCTATAAATGCACCTATGTAGTTGAAAAGTTATGTATTTTGCAATGTGTCTTGATAGAATGTTGACTACAATTCAATGGCTAAAACAAAATTATCAAAGTC\tCCCFFFFFGHHHHJJJJIJJIJIJJJJJGHGJIIIIJJJJJJJJJIIIIIIJHIIIGJJJJJJJJJJJJGIJJJJHIIJJJJJJJHHHHHFFFFFFEEEC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:-\tNH:i:2\n",
    "HWI-ST718_146963544:5:2312:11656:62706\t83\t22\t10838707\t1\t96M4S\t=\t10838443\t-364\tATGTGTCTTGATAGAATGTTGACTACAATTCAATGGCTAAAACAAAATTATCAAAGTCAAAATTGCATGTCTTTTCTTGAATGATAGGTATGTATCCGGT\t>?CA?@;CCCADECDCBC?>FEHHFIHGGECFHIIGHHHF@BB*EFDCIIHDBF?GHGGGGGHD>?C???CIHF9<FA,HHGGGAFGBDGHHFDDFFCCC\tAS:i:-4\tZS:i:-4\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:96\tYS:i:-5\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:+\tNH:i:2\n",
    "HWI-ST718_146963544:5:2308:15338:36631\t147\t22\t10838833\t1\t100M\t=\t10838665\t-268\tTAGACCCCTCTTTCAAATTAAAGATAACAAAAGCCCAATTGAAAGGAAAACAAAATACATCAATGCTAACTAAATACAAATAATTAGCCTTCACTAATTC\tDDDDDDFFDHHCGHHHJIJJJJJJJJJJIIJJJJIJJIIIJJJJJJJJJJJJJIIIFJIIIJJJJJIIIJJJJJIHJJJJJIJJJJJHHGHHEFFFFCCC\tAS:i:0\tZS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:100\tYS:i:0\tYT:Z:CP\tRG:Z:UHR_Rep2\tXS:A:-\tNH:i:2\n",
    "\n",
    "\n",
    "[nathalia@aws] $  samtools view -f 1024 UHR.bam | head\n",
    "\n",
    "# no stdout\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_ALIGN_DIR\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/alignments/hisat2\n",
    "\n",
    "\n",
    "[nathalia@aws] $  samtools flagstat UHR.bam\n",
    "1174925 + 0 in total (QC-passed reads + QC-failed reads)\n",
    "24511 + 0 secondary\n",
    "0 + 0 supplementary\n",
    "0 + 0 duplicates\n",
    "1173705 + 0 mapped (99.90% : N/A)\n",
    "1150414 + 0 paired in sequencing\n",
    "575207 + 0 read1\n",
    "575207 + 0 read2\n",
    "1143814 + 0 properly paired (99.43% : N/A)\n",
    "1148584 + 0 with itself and mate mapped\n",
    "610 + 0 singletons (0.05% : N/A)\n",
    "6 + 0 with mate mapped to a different chr\n",
    "6 + 0 with mate mapped to a different chr (mapQ>=5)\n",
    "\n",
    "\n",
    "[nathalia@aws] $  samtools flagstat HBR.bam\n",
    "793931 + 0 in total (QC-passed reads + QC-failed reads)\n",
    "7565 + 0 secondary\n",
    "0 + 0 supplementary\n",
    "0 + 0 duplicates\n",
    "793322 + 0 mapped (99.92% : N/A)\n",
    "786366 + 0 paired in sequencing\n",
    "393183 + 0 read1\n",
    "393183 + 0 read2\n",
    "783114 + 0 properly paired (99.59% : N/A)\n",
    "785492 + 0 with itself and mate mapped\n",
    "265 + 0 singletons (0.03% : N/A)\n",
    "0 + 0 with mate mapped to a different chr\n",
    "0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  ls *.bam\n",
    "HBR.bam  HBR_Rep1.bam  HBR_Rep2.bam  HBR_Rep3.bam  UHR.bam  UHR_Rep1.bam  UHR_Rep2.bam  UHR_Rep3.bam\n",
    "[nathalia@aws] $  fastqc *.bam\n",
    "Started analysis of HBR.bam\n",
    "Approx 5% complete for HBR.bam\n",
    "Approx 10% complete for HBR.bam\n",
    "Approx 15% complete for HBR.bam\n",
    "Approx 20% complete for HBR.bam\n",
    "Approx 25% complete for HBR.bam\n",
    "Approx 30% complete for HBR.bam\n",
    "Approx 35% complete for HBR.bam\n",
    "Approx 40% complete for HBR.bam\n",
    "Approx 45% complete for HBR.bam\n",
    "Approx 50% complete for HBR.bam\n",
    "Approx 55% complete for HBR.bam\n",
    "Approx 60% complete for HBR.bam\n",
    "Approx 65% complete for HBR.bam\n",
    "Approx 70% complete for HBR.bam\n",
    "Approx 75% complete for HBR.bam\n",
    "Approx 80% complete for HBR.bam\n",
    "Approx 85% complete for HBR.bam\n",
    "Approx 90% complete for HBR.bam\n",
    "Approx 95% complete for HBR.bam\n",
    "Approx 100% complete for HBR.bam\n",
    "Analysis complete for HBR.bam\n",
    "Started analysis of HBR_Rep1.bam\n",
    "Approx 5% complete for HBR_Rep1.bam\n",
    "Approx 10% complete for HBR_Rep1.bam\n",
    "Approx 15% complete for HBR_Rep1.bam\n",
    "Approx 20% complete for HBR_Rep1.bam\n",
    "Approx 25% complete for HBR_Rep1.bam\n",
    "Approx 30% complete for HBR_Rep1.bam\n",
    "Approx 35% complete for HBR_Rep1.bam\n",
    "Approx 40% complete for HBR_Rep1.bam\n",
    "Approx 45% complete for HBR_Rep1.bam\n",
    "Approx 50% complete for HBR_Rep1.bam\n",
    "Approx 55% complete for HBR_Rep1.bam\n",
    "Approx 60% complete for HBR_Rep1.bam\n",
    "Approx 65% complete for HBR_Rep1.bam\n",
    "Approx 70% complete for HBR_Rep1.bam\n",
    "Approx 75% complete for HBR_Rep1.bam\n",
    "Approx 80% complete for HBR_Rep1.bam\n",
    "Approx 85% complete for HBR_Rep1.bam\n",
    "Approx 90% complete for HBR_Rep1.bam\n",
    "Approx 95% complete for HBR_Rep1.bam\n",
    "Approx 100% complete for HBR_Rep1.bam\n",
    "Analysis complete for HBR_Rep1.bam\n",
    "Started analysis of HBR_Rep2.bam\n",
    "Approx 5% complete for HBR_Rep2.bam\n",
    "Approx 10% complete for HBR_Rep2.bam\n",
    "Approx 15% complete for HBR_Rep2.bam\n",
    "Approx 20% complete for HBR_Rep2.bam\n",
    "Approx 25% complete for HBR_Rep2.bam\n",
    "Approx 30% complete for HBR_Rep2.bam\n",
    "Approx 35% complete for HBR_Rep2.bam\n",
    "Approx 40% complete for HBR_Rep2.bam\n",
    "Approx 45% complete for HBR_Rep2.bam\n",
    "Approx 50% complete for HBR_Rep2.bam\n",
    "Approx 55% complete for HBR_Rep2.bam\n",
    "Approx 60% complete for HBR_Rep2.bam\n",
    "Approx 65% complete for HBR_Rep2.bam\n",
    "Approx 70% complete for HBR_Rep2.bam\n",
    "Approx 75% complete for HBR_Rep2.bam\n",
    "Approx 80% complete for HBR_Rep2.bam\n",
    "Approx 85% complete for HBR_Rep2.bam\n",
    "Approx 90% complete for HBR_Rep2.bam\n",
    "Approx 95% complete for HBR_Rep2.bam\n",
    "Approx 100% complete for HBR_Rep2.bam\n",
    "Analysis complete for HBR_Rep2.bam\n",
    "Started analysis of HBR_Rep3.bam\n",
    "Approx 5% complete for HBR_Rep3.bam\n",
    "Approx 10% complete for HBR_Rep3.bam\n",
    "Approx 15% complete for HBR_Rep3.bam\n",
    "Approx 20% complete for HBR_Rep3.bam\n",
    "Approx 25% complete for HBR_Rep3.bam\n",
    "Approx 30% complete for HBR_Rep3.bam\n",
    "Approx 35% complete for HBR_Rep3.bam\n",
    "Approx 40% complete for HBR_Rep3.bam\n",
    "Approx 45% complete for HBR_Rep3.bam\n",
    "Approx 50% complete for HBR_Rep3.bam\n",
    "Approx 55% complete for HBR_Rep3.bam\n",
    "Approx 60% complete for HBR_Rep3.bam\n",
    "Approx 65% complete for HBR_Rep3.bam\n",
    "Approx 70% complete for HBR_Rep3.bam\n",
    "Approx 75% complete for HBR_Rep3.bam\n",
    "Approx 80% complete for HBR_Rep3.bam\n",
    "Approx 85% complete for HBR_Rep3.bam\n",
    "Approx 90% complete for HBR_Rep3.bam\n",
    "Approx 95% complete for HBR_Rep3.bam\n",
    "Approx 100% complete for HBR_Rep3.bam\n",
    "Analysis complete for HBR_Rep3.bam\n",
    "Started analysis of UHR.bam\n",
    "Approx 5% complete for UHR.bam\n",
    "Approx 10% complete for UHR.bam\n",
    "Approx 15% complete for UHR.bam\n",
    "Approx 20% complete for UHR.bam\n",
    "Approx 25% complete for UHR.bam\n",
    "Approx 30% complete for UHR.bam\n",
    "Approx 35% complete for UHR.bam\n",
    "Approx 40% complete for UHR.bam\n",
    "Approx 45% complete for UHR.bam\n",
    "Approx 50% complete for UHR.bam\n",
    "Approx 55% complete for UHR.bam\n",
    "Approx 60% complete for UHR.bam\n",
    "Approx 65% complete for UHR.bam\n",
    "Approx 70% complete for UHR.bam\n",
    "Approx 75% complete for UHR.bam\n",
    "Approx 80% complete for UHR.bam\n",
    "Approx 85% complete for UHR.bam\n",
    "Approx 90% complete for UHR.bam\n",
    "Approx 95% complete for UHR.bam\n",
    "Analysis complete for UHR.bam\n",
    "Started analysis of UHR_Rep1.bam\n",
    "Approx 5% complete for UHR_Rep1.bam\n",
    "Approx 10% complete for UHR_Rep1.bam\n",
    "Approx 15% complete for UHR_Rep1.bam\n",
    "Approx 20% complete for UHR_Rep1.bam\n",
    "Approx 25% complete for UHR_Rep1.bam\n",
    "Approx 30% complete for UHR_Rep1.bam\n",
    "Approx 35% complete for UHR_Rep1.bam\n",
    "Approx 40% complete for UHR_Rep1.bam\n",
    "Approx 45% complete for UHR_Rep1.bam\n",
    "Approx 50% complete for UHR_Rep1.bam\n",
    "Approx 55% complete for UHR_Rep1.bam\n",
    "Approx 60% complete for UHR_Rep1.bam\n",
    "Approx 65% complete for UHR_Rep1.bam\n",
    "Approx 70% complete for UHR_Rep1.bam\n",
    "Approx 75% complete for UHR_Rep1.bam\n",
    "Approx 80% complete for UHR_Rep1.bam\n",
    "Approx 85% complete for UHR_Rep1.bam\n",
    "Approx 90% complete for UHR_Rep1.bam\n",
    "Approx 95% complete for UHR_Rep1.bam\n",
    "Approx 100% complete for UHR_Rep1.bam\n",
    "Analysis complete for UHR_Rep1.bam\n",
    "Started analysis of UHR_Rep2.bam\n",
    "Approx 5% complete for UHR_Rep2.bam\n",
    "Approx 10% complete for UHR_Rep2.bam\n",
    "Approx 15% complete for UHR_Rep2.bam\n",
    "Approx 20% complete for UHR_Rep2.bam\n",
    "Approx 25% complete for UHR_Rep2.bam\n",
    "Approx 30% complete for UHR_Rep2.bam\n",
    "Approx 35% complete for UHR_Rep2.bam\n",
    "Approx 40% complete for UHR_Rep2.bam\n",
    "Approx 45% complete for UHR_Rep2.bam\n",
    "Approx 50% complete for UHR_Rep2.bam\n",
    "Approx 55% complete for UHR_Rep2.bam\n",
    "Approx 60% complete for UHR_Rep2.bam\n",
    "Approx 65% complete for UHR_Rep2.bam\n",
    "Approx 70% complete for UHR_Rep2.bam\n",
    "Approx 75% complete for UHR_Rep2.bam\n",
    "Approx 80% complete for UHR_Rep2.bam\n",
    "Approx 85% complete for UHR_Rep2.bam\n",
    "Approx 90% complete for UHR_Rep2.bam\n",
    "Approx 95% complete for UHR_Rep2.bam\n",
    "Approx 100% complete for UHR_Rep2.bam\n",
    "Analysis complete for UHR_Rep2.bam\n",
    "Started analysis of UHR_Rep3.bam\n",
    "Approx 5% complete for UHR_Rep3.bam\n",
    "Approx 10% complete for UHR_Rep3.bam\n",
    "Approx 15% complete for UHR_Rep3.bam\n",
    "Approx 20% complete for UHR_Rep3.bam\n",
    "Approx 25% complete for UHR_Rep3.bam\n",
    "Approx 30% complete for UHR_Rep3.bam\n",
    "Approx 35% complete for UHR_Rep3.bam\n",
    "Approx 40% complete for UHR_Rep3.bam\n",
    "Approx 45% complete for UHR_Rep3.bam\n",
    "Approx 50% complete for UHR_Rep3.bam\n",
    "Approx 55% complete for UHR_Rep3.bam\n",
    "Approx 60% complete for UHR_Rep3.bam\n",
    "Approx 65% complete for UHR_Rep3.bam\n",
    "Approx 70% complete for UHR_Rep3.bam\n",
    "Approx 75% complete for UHR_Rep3.bam\n",
    "Approx 80% complete for UHR_Rep3.bam\n",
    "Approx 85% complete for UHR_Rep3.bam\n",
    "Approx 90% complete for UHR_Rep3.bam\n",
    "Approx 95% complete for UHR_Rep3.bam\n",
    "Approx 100% complete for UHR_Rep3.bam\n",
    "Analysis complete for UHR_Rep3.bam\n",
    "[nathalia@aws] $  lhtr\n",
    "total 990M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.0K Nov 13 14:24 alignment.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 166M Nov 13 14:24 UHR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 119M Nov 13 14:24 UHR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 136M Nov 13 14:24 UHR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  86M Nov 13 14:25 HBR_Rep1.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105M Nov 13 14:25 HBR_Rep2.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  94M Nov 13 14:25 HBR_Rep3.sam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  303 Nov 13 14:27 sam_bam.sh\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  34M Nov 13 14:28 UHR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25M Nov 13 14:28 UHR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  28M Nov 13 14:28 UHR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  17M Nov 13 14:28 HBR_Rep1.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20M Nov 13 14:28 HBR_Rep2.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  18M Nov 13 14:28 HBR_Rep3.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  85M Nov 13 14:30 UHR.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  53M Nov 13 14:30 HBR.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  47K Nov 13 15:34 HBR.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  39K Nov 13 15:34 HBR_Rep1.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  40K Nov 13 15:34 HBR_Rep2.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  40K Nov 13 15:34 HBR_Rep3.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  51K Nov 13 15:34 UHR.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  42K Nov 13 15:34 UHR_Rep1.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  44K Nov 13 15:34 UHR_Rep2.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  43K Nov 13 15:34 UHR_Rep3.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 447K Nov 13 17:29 HBR_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 638K Nov 13 17:29 HBR_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 454K Nov 13 17:29 HBR_Rep1_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 645K Nov 13 17:29 HBR_Rep1_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 452K Nov 13 17:29 HBR_Rep2_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 643K Nov 13 17:29 HBR_Rep2_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 453K Nov 13 17:29 HBR_Rep3_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 643K Nov 13 17:29 HBR_Rep3_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 462K Nov 13 17:29 UHR_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 646K Nov 13 17:29 UHR_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 469K Nov 13 17:29 UHR_Rep1_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 649K Nov 13 17:29 UHR_Rep1_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 472K Nov 13 17:29 UHR_Rep2_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 651K Nov 13 17:29 UHR_Rep2_fastqc.html\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 470K Nov 13 17:29 UHR_Rep3_fastqc.zip\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 653K Nov 13 17:29 UHR_Rep3_fastqc.html\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME\n",
    "\n",
    "[nathalia@aws] $  wget http://genomedata.org/rnaseq-tutorial/RSeQC.zip\n",
    "--2018-11-13 17:21:07--  http://genomedata.org/rnaseq-tutorial/RSeQC.zip\n",
    "Resolving genomedata.org (genomedata.org)... 3.8.108.6\n",
    "Connecting to genomedata.org (genomedata.org)|3.8.108.6|:80... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 442199597 (422M) [application/zip]\n",
    "Saving to: ‘RSeQC.zip’\n",
    "\n",
    "RSeQC.zip                  100%[========================================>] 421.71M  2.05MB/s    in 6m 39s  \n",
    "\n",
    "2018-11-13 17:27:52 (1.06 MB/s) - ‘RSeQC.zip’ saved [442199597/442199597]\n",
    "\n",
    "[nathalia@aws] $  unzip RSeQC.zip\n",
    "Archive:  RSeQC.zip\n",
    "   creating: RSeQC/\n",
    "  inflating: RSeQC/Pairend_nonStrandSpecific_36mer_Human_hg19.bam  \n",
    "  inflating: RSeQC/hg19_UCSC_knownGene.bed.gz  \n",
    "  inflating: RSeQC/Pairend_nonStrandSpecific_36mer_Human_hg19.bam.bai  \n",
    "\n",
    "\n",
    "[nathalia@aws] $  cd RSeQC/\n",
    "\n",
    "[nathalia@aws] $  gunzip hg19_UCSC_knownGene.bed.gz\n",
    "\n",
    "[nathalia@aws] $  emacs rseqc.sh\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
