{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 11/14/2018\n",
    "\n",
    "Kallisto https://github.com/griffithlab/rnaseq_tutorial/wiki/Kallisto\n",
    "\n",
    "extra \n",
    "\n",
    "nathalia"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Get transcript sequences in fasta format"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME/refs\n",
    "[nathalia@aws] $  gtf_to_fasta $RNA_REF_GTF $RNA_REF_FASTA chr22_ERCC92_transcripts.fa\n",
    "Reading the annotation file: /home/ubuntu/workspace/rnaseq/refs/chr22_with_ERCC92.gtf\n",
    "\n",
    "[nathalia@aws] $  less chr22_ERCC92_transcripts.fa\n",
    "\n",
    "# messy headers\n",
    "\n",
    "[nathalia@aws] $  cat chr22_ERCC92_transcripts.fa | perl -ne 'if ($_ =~/^\\>\\d+\\s+\\w+\\s+(ERCC\\S+)[\\+\\-]/){print \">$1\\n\"}elsif($_ =~ /\\d+\\s+(ENST\\d+)/){print \">$1\\n\"}else{print $_}' > chr22_ERCC92_transcripts.clean.fa\n",
    "\n",
    "[nathalia@aws] $  wc -l chr22_ERCC92_transcripts*.fa\n",
    "  126025 chr22_ERCC92_transcripts.clean.fa\n",
    "  126025 chr22_ERCC92_transcripts.fa\n",
    "  252050 total\n",
    "\n",
    "[nathalia@aws] $  cat chr22_ERCC92_transcripts.clean.fa | grep \">\" | perl -ne '$_ =~ s/\\>//; print $_' | sort | uniq > transcript_id_list.txt\n",
    "\n",
    "[nathalia@aws] $  less transcript_id_list.txt \n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 175M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  50M Oct 23 16:48 chr22_with_ERCC92.fa\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  161 Oct 23 16:49 illumina_multiplex.fa\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  30M Oct 23 16:49 chr22_with_ERCC92.gtf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 172K Nov 13 00:26 splicesites.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 148K Nov 13 00:26 exons.tsv\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 9.4M Nov 13 00:26 chr22_with_ERCC92.4.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.3K Nov 13 00:26 chr22_with_ERCC92.3.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  55K Nov 13 00:26 chr22_with_ERCC92.8.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 278K Nov 13 00:26 chr22_with_ERCC92.7.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 9.4M Nov 13 00:27 chr22_with_ERCC92.2.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  28M Nov 13 00:27 chr22_with_ERCC92.1.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 9.6M Nov 13 00:27 chr22_with_ERCC92.6.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  25M Nov 13 00:27 chr22_with_ERCC92.5.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  12K Nov 13 13:48 test_chr22.fa\n",
    "-rwxrwxr-x 1 ubuntu ubuntu  799 Nov 13 13:48 practice2.py~\n",
    "-rwxrwxr-x 1 ubuntu ubuntu  844 Nov 13 13:54 practice2.py\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.8K Nov 13 15:49 chr22_with_ERCC92.fa.fai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 570K Nov 14 13:31 chr22_ERCC92_transcripts.fa.tlst\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 7.6M Nov 14 13:31 chr22_ERCC92_transcripts.fa\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 7.1M Nov 14 13:32 chr22_ERCC92_transcripts.clean.fa\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  71K Nov 14 13:33 transcript_id_list.txt\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Transcriptome index"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  mkdir kallisto\n",
    "[nathalia@aws] $  cd kallisto/\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/refs/kallisto\n",
    "[nathalia@aws] $  kallisto index --index=chr22_ERCC92_transcripts_kallisto_index ../chr22_ERCC92_transcripts.clean.fa\n",
    "\n",
    "[build] loading fasta file ../chr22_ERCC92_transcripts.clean.fa\n",
    "[build] k-mer length: 31\n",
    "[build] warning: clipped off poly-A tail (longer than 10)\n",
    "        from 102 target sequences\n",
    "[build] warning: replaced 1 non-ACGUT characters in the input sequence\n",
    "        with pseudorandom nucleotides\n",
    "[build] counting k-mers ... done.\n",
    "[build] building target de Bruijn graph ...  done \n",
    "[build] creating equivalence classes ...  done\n",
    "[build] target de Bruijn graph has 27640 contigs and contains 3293113 k-mers \n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 69M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 69M Nov 14 13:35 chr22_ERCC92_transcripts_kallisto_index\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Abundance estimates for all samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME/expression\n",
    "[nathalia@aws] $  ls\n",
    "htseq_counts  stringtie\n",
    "[nathalia@aws] $  mkdir kallisto\n",
    "[nathalia@aws] $  cd kallisto/\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/expression/kallisto\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  emacs abundance.sh \n",
    "#!/bin/bash\n",
    "\n",
    "kallisto quant --index=$RNA_HOME/refs/kallisto/chr22_ERCC92_transcripts_kallisto_index --output-dir=UHR_Rep1_ERCC-Mix1 --threads=4 --plaintext $RNA_DATA_DIR/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz $RNA_DATA_DIR/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "\n",
    "kallisto quant --index=$RNA_HOME/refs/kallisto/chr22_ERCC92_transcripts_kallisto_index --output-dir=UHR_Rep2_ERCC-Mix1 --threads=4 --plaintext $RNA_DATA_DIR/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz $RNA_DATA_DIR/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "\n",
    "kallisto quant --index=$RNA_HOME/refs/kallisto/chr22_ERCC92_transcripts_kallisto_index --output-dir=UHR_Rep3_ERCC-Mix1 --threads=4 --plaintext $RNA_DATA_DIR/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz $RNA_DATA_DIR/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "\n",
    "kallisto quant --index=$RNA_HOME/refs/kallisto/chr22_ERCC92_transcripts_kallisto_index --output-dir=HBR_Rep1_ERCC-Mix2 --threads=4 --plaintext $RNA_DATA_DIR/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz $RNA_DATA_DIR/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "\n",
    "kallisto quant --index=$RNA_HOME/refs/kallisto/chr22_ERCC92_transcripts_kallisto_index --output-dir=HBR_Rep2_ERCC-Mix2 --threads=4 --plaintext $RNA_DATA_DIR/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz $RNA_DATA_DIR/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "\n",
    "kallisto quant --index=$RNA_HOME/refs/kallisto/chr22_ERCC92_transcripts_kallisto_index --output-dir=HBR_Rep3_ERCC-Mix2 --threads=4 --plaintext $RNA_DATA_DIR/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz $RNA_DATA_DIR/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "\n",
    "\n",
    "\n",
    "[nathalia@aws] $  bash abundance.sh \n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 4,564\n",
    "[index] number of k-mers: 3,293,113\n",
    "[index] number of equivalence classes: 17,221\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /home/ubuntu/workspace/rnaseq/data/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "                             /home/ubuntu/workspace/rnaseq/data/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 227,392 reads, 193,188 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 181.255\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 834 rounds\n",
    "\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 4,564\n",
    "[index] number of k-mers: 3,293,113\n",
    "[index] number of equivalence classes: 17,221\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /home/ubuntu/workspace/rnaseq/data/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "                             /home/ubuntu/workspace/rnaseq/data/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 162,373 reads, 129,440 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 169.788\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 760 rounds\n",
    "\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 4,564\n",
    "[index] number of k-mers: 3,293,113\n",
    "[index] number of equivalence classes: 17,221\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /home/ubuntu/workspace/rnaseq/data/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "                             /home/ubuntu/workspace/rnaseq/data/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 185,442 reads, 158,183 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 167.718\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 854 rounds\n",
    "\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 4,564\n",
    "[index] number of k-mers: 3,293,113\n",
    "[index] number of equivalence classes: 17,221\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /home/ubuntu/workspace/rnaseq/data/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "                             /home/ubuntu/workspace/rnaseq/data/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 118,571 reads, 99,270 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 169.016\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 736 rounds\n",
    "\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 4,564\n",
    "[index] number of k-mers: 3,293,113\n",
    "[index] number of equivalence classes: 17,221\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /home/ubuntu/workspace/rnaseq/data/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "                             /home/ubuntu/workspace/rnaseq/data/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 144,826 reads, 121,976 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 169.638\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 833 rounds\n",
    "\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 4,564\n",
    "[index] number of k-mers: 3,293,113\n",
    "[index] number of equivalence classes: 17,221\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /home/ubuntu/workspace/rnaseq/data/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "                             /home/ubuntu/workspace/rnaseq/data/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 129,786 reads, 108,536 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 171.252\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 756 rounds\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "#creating single tsv file that has the tpm abundance estimates for all six samples\n",
    "\n",
    "[nathalia@aws] $  paste */abundance.tsv | cut -f 1,2,5,10,15,20,25,30 > transcript_tpms_all_samples.tsv\n",
    "\n",
    "[nathalia@aws] $  ls -1 */abundance.tsv | perl -ne 'chomp $_; if ($_ =~ /(\\S+)\\/abundance\\.tsv/){print \"\\t$1\"}' | perl -ne 'print \"target_id\\tlength$_\\n\"' > header.tsv\n",
    "\n",
    "[nathalia@aws] $  cat header.tsv transcript_tpms_all_samples.tsv | grep -v \"tpm\" > transcript_tpms_all_samples.tsv2\n",
    "\n",
    "[nathalia@aws] $  mv transcript_tpms_all_samples.tsv2 transcript_tpms_all_samples.tsv\n",
    "\n",
    "[nathalia@aws] $  rm -f header.tsv\n",
    "\n",
    "[nathalia@aws] $  head transcript_tpms_all_samples.tsv \n",
    "target_id\tlength\tHBR_Rep1_ERCC-Mix2\tHBR_Rep2_ERCC-Mix2\tHBR_Rep3_ERCC-Mix2\tUHR_Rep1_ERCC-Mix1\tUHR_Rep2_ERCC-Mix1\tUHR_Rep3_ERCC-Mix1\n",
    "ENST00000615943\t113\t0\t0\t0\t0\t0\t0\n",
    "ENST00000635667\t749\t0\t0\t0\t0\t0\t0\n",
    "ENST00000623473\t54\t0\t0\t0\t0\t0\t0\n",
    "ENST00000624155\t120\t0\t0\t0\t0\t0\t0\n",
    "ENST00000422332\t1241\t0\t0\t0\t0\t5.78142\t0\n",
    "ENST00000612732\t151\t0\t0\t0\t0\t0\t0\n",
    "ENST00000634617\t774\t0\t0\t0\t0\t0\t0\n",
    "ENST00000613107\t107\t0\t0\t0\t0\t0\t0\n",
    "ENST00000623391\t6114\t1.62081\t12.0816\t5.22447\t7.92662\t15.2975\t8.18668\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Compare transcript and gene abundance estimates from Kallisto to isoform abundance estimates from StringTie and counts from HtSe-Count"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  wget https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/kallisto_gene_matrix.pl\n",
    "--2018-11-14 13:40:49--  https://raw.githubusercontent.com/griffithlab/rnaseq_tutorial/master/scripts/kallisto_gene_matrix.pl\n",
    "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.52.133\n",
    "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.52.133|:443... connected.\n",
    "HTTP request sent, awaiting response... 200 OK\n",
    "Length: 3175 (3.1K) [text/plain]\n",
    "Saving to: ‘kallisto_gene_matrix.pl’\n",
    "\n",
    "kallisto_gene_matrix.pl 100%[=============================>]   3.10K  --.-KB/s    in 0s      \n",
    "\n",
    "2018-11-14 13:40:54 (130 MB/s) - ‘kallisto_gene_matrix.pl’ saved [3175/3175]\n",
    "\n",
    "[nathalia@aws] $  chmod +x kallisto_gene_matrix.pl\n",
    "[nathalia@aws] $  ./kallisto_gene_matrix.pl --gtf_file=$RNA_HOME/refs/chr22_with_ERCC92.gtf  --kallisto_transcript_matrix_in=transcript_tpms_all_samples.tsv --kallisto_transcript_matrix_out=gene_tpms_all_samples.tsv\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 344K\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.8K Nov 14 13:29 abundance.sh\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 13:36 UHR_Rep1_ERCC-Mix1\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 13:36 UHR_Rep2_ERCC-Mix1\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 13:36 UHR_Rep3_ERCC-Mix1\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 13:36 HBR_Rep1_ERCC-Mix2\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 13:36 HBR_Rep2_ERCC-Mix2\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 13:36 HBR_Rep3_ERCC-Mix2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 221K Nov 14 13:38 transcript_tpms_all_samples.tsv\n",
    "-rwxrwxr-x 1 ubuntu ubuntu 3.2K Nov 14 13:40 kallisto_gene_matrix.pl\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  77K Nov 14 13:41 gene_tpms_all_samples.tsv\n",
    "\n",
    "[nathalia@aws] $  cd ..\n",
    "[nathalia@aws] $  ls\n",
    "htseq_counts  kallisto  stringtie\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/expression\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://github.com/griffithlab/rnaseq_tutorial/blob/master/scripts/Tutorial_comparisons.R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  R\n",
    "\n",
    "R version 3.4.4 (2018-03-15) -- \"Someone to Lean On\"\n",
    "Copyright (C) 2018 The R Foundation for Statistical Computing\n",
    "Platform: x86_64-pc-linux-gnu (64-bit)\n",
    "\n",
    "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
    "You are welcome to redistribute it under certain conditions.\n",
    "Type 'license()' or 'licence()' for distribution details.\n",
    "\n",
    "R is a collaborative project with many contributors.\n",
    "Type 'contributors()' for more information and\n",
    "'citation()' on how to cite R or R packages in publications.\n",
    "\n",
    "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
    "'help.start()' for an HTML browser interface to help.\n",
    "Type 'q()' to quit R.\n",
    "\n",
    "> #Load libraries\n",
    "> library(ggplot2)\n",
    "ipt_tpms_all_samples.tsv', sep=\"\\t\", header=TRUE, as.is=1, row.names=1)\n",
    "\n",
    "#Summarize the data.frames created\n",
    "dim(htseq_gene_counts)\n",
    "dim(stringtie_gene)\n",
    "dim(stringtie_gene_fpkm)\n",
    "dim(kallisto_gene)\n",
    "\n",
    "dim(stringtie_tran)\n",
    "dim(stringtie_tran_fpkm)\n",
    "dim(kallisto_tran)\n",
    "\n",
    "#Reorganize the data.frames for total consistency\n",
    "kallisto_names = c(\"length\", \"HBR_Rep1\", \"HBR_Rep2\", \"HBR_Rep3\", \"UHR_Rep1\", \"UHR_Rep2\", \"UHR_Rep3\") \n",
    "names(kallisto_gene) = kallisto_names\n",
    "names(kallisto_tran) = kallisto_names\n",
    "\n",
    "sample_names = c(\"HBR_Rep1\", \"HBR_Rep2\", \"HBR_Rep3\", \"UHR_Rep1\", \"UHR_Rep2\", \"UHR_Rep3\")\n",
    "gene_names = row.names(kallisto_gene)\n",
    "tran_names = row.names(kallisto_tran)\n",
    "\n",
    "htseq_gene_counts = htseq_gene_counts[gene_names, sample_names]\n",
    "stringtie_gene = stringtie_gene[gene_names, sample_names]\n",
    "stringtie_gene_fpkm = stringtie_gene_fpkm[gene_names, sample_names]\n",
    "kallisto_gene = kallisto_gene[gene_names, sample_names]\n",
    "stringtie_tran = stringtie_tran[tran_names, sample_names]\n",
    "stringtie_tran_fpkm = stringtie_tran_fpkm[tran_names, sample_names]\n",
    "kallisto_tran = kallisto_tran[tran_names, sample_names]\n",
    "\n",
    "#Take a look at the top of each data.frame\n",
    "head(htseq_gene_counts)\n",
    "head(stringtie_gene)\n",
    "head(stringtie_gene_fpkm)\n",
    "head(kallisto_gene)\n",
    "head(stringtie_tran)\n",
    "head(stringtie_tran_fpkm)\n",
    "head(kallisto_tran)\n",
    "\n",
    "#1. Plot kallisto gene TPMs vs stringtie gene TPMs - Pick HBR_Rep1 data arbitrarily\n",
    "stabvar = 0.1\n",
    "HBR1_gene_data = data.frame(kallisto_gene[,\"HBR_Rep1\"], stringtie_gene[,\"HBR_Rep1\"], htseq_gene_counts[,\"HBR_Rep1\"])\n",
    "names(HBR1_gene_data) = c(\"kallisto\", \"stringtie\", \"htseq\")\n",
    "p1 = ggplot(HBR1_gene_data, aes(log2(kallisto+stabvar), log2(stringtie+stabvar)))\n",
    "p1 = p1 + geom_point()\n",
    "p1 = p1 + geom_point(aes(colour = log2(htseq+stabvar))) + scale_colour_gradient(low = \"yellow\", high = \"red\")\n",
    "p1 = p1 + xlab(\"Kallisto TPM\") + ylab(\"StringTie TPM\") + labs(colour = \"HtSeq Counts\")\n",
    "p1 = p1 + labs(title = \"HBR1 gene expression values [log2(value + 0.1) scaled]\")\n",
    "\n",
    "#2. Plot kallisto transcript TPMs vs stringtie transcript TPMs - Pick HBR_Rep1 data arbitrarily\n",
    "# Indicate with the points whether the data are real transcripts vs. spike-in controls\n",
    "HBR1_tran_data = data.frame(kallisto_tran[,\"HBR_Rep1\"], stringtie_tran[,\"HBR_Rep1\"])\n",
    "names(HBR1_tran_data) = c(\"kallisto\", \"stringtie\")\n",
    "spikein_status=grepl(\"ERCC\",tran_names)\n",
    "p2 = ggplot(HBR1_tran_data, aes(log2(kallisto+stabvar), log2(stringtie+stabvar)))\n",
    "p2 = p2 + geom_point()\n",
    "p2 = p2 + geom_point(aes(colour = spikein_status)) \n",
    "p2 = p2 + xlab(\"Kallisto TPM\") + ylab(\"StringTie TPM\") + labs(colour = \"SpikeIn Status\")\n",
    "p2 = p2 + labs(title = \"HBR1 transcript expression values [log2(value + 0.1) scaled]\")\n",
    "\n",
    "#3. Plot stringtie transcript TPMs vs. stringtie transcript FPKMs - Pick HBR_Rep1 data arbitrarily\n",
    "# Indicate with the points whether the data are real transcripts vs. spike-in controls\n",
    "HBR1_tran_data2 = data.frame(stringtie_tran[,\"HBR_Rep1\"], stringtie_tran_fpkm[,\"HBR_Rep1\"])\n",
    "names(HBR1_tran_data2) = c(\"stringtie_TPM\", \"stringtie_FPKM\")\n",
    "p3 = ggplot(HBR1_tran_data2, aes(log2(stringtie_TPM+stabvar), log2(stringtie_FPKM+stabvar)))\n",
    "p3 = p3 + geom_point()\n",
    "p3 = p3 + g> \n",
    "e> #Set the base working dir from which to access the input files\n",
    "> working_dir = '/home/ubuntu/workspace/rnaseq/expression'\n",
    "> setwd(working_dir)\n",
    "> \n",
    "K> #Load in expression matrix files from each expression method\n",
    "> htseq_gene_counts = read.table('htseq_counts/gene_read_counts_table_all_final.tsv', sep=\"\\t\", header=TRUE, as.is=1, row.names=1)\n",
    "> stringtie_gene = read.table('stringtie/ref_only/gene_tpm_all_samples.tsv', sep=\"\\t\", header=TRUE, as.is=1, row.names=1)\n",
    "> stringtie_tran = read.table('stringtie/ref_only/transcript_tpm_all_samples.tsv', sep=\"\\t\", header=TRUE, as.is=1, row.names=1)\n",
    "> stringtie_gene_fpkm = read.table('stringtie/ref_only/gene_fpkm_all_samples.tsv', sep=\"\\t\", header=TRUE, as.is=1, row.names=1)\n",
    "> stringtie_tran_fpkm = read.table('stringtie/ref_only/transcript_fpkm_all_samples.tsv', sep=\"\\t\", header=TRUE, as.is=1, row.names=1)\n",
    "> kallisto_gene = read.table('kallisto/gene_tpms_all_samples.tsv', sep=\"\\t\", header=TRUE, as.is=1, row.names=1)\n",
    "> kallisto_tran = read.table('kallisto/transcript_tpms_all_samples.tsv', sep=\"\\t\", header=TRUE, as.is=1, row.names=1)\n",
    "> \n",
    "> #Summarize the data.frames created\n",
    "> dim(htseq_gene_counts)\n",
    "[1] 1410    6\n",
    "> dim(stringtie_gene)\n",
    "[1] 1410    6\n",
    "> dim(stringtie_gene_fpkm)\n",
    "[1] 1410    6\n",
    "> dim(kallisto_gene)\n",
    "[1] 1410    7\n",
    "> \n",
    "> dim(stringtie_tran)\n",
    "[1] 4564    6\n",
    "> dim(stringtie_tran_fpkm)\n",
    "[1] 4564    6\n",
    "> dim(kallisto_tran)\n",
    "[1] 4564    7\n",
    "> \n",
    "> #Reorganize the data.frames for total consistency\n",
    "> kallisto_names = c(\"length\", \"HBR_Rep1\", \"HBR_Rep2\", \"HBR_Rep3\", \"UHR_Rep1\", \"UHR_Rep2\", \"UHR_Rep3\") \n",
    "> names(kallisto_gene) = kallisto_names\n",
    "> names(kallisto_tran) = kallisto_names\n",
    "> \n",
    "> sample_names = c(\"HBR_Rep1\", \"HBR_Rep2\", \"HBR_Rep3\", \"UHR_Rep1\", \"UHR_Rep2\", \"UHR_Rep3\")\n",
    "> gene_names = row.names(kallisto_gene)\n",
    "> tran_names = row.names(kallisto_tran)\n",
    "> \n",
    "> htseq_gene_counts = htseq_gene_counts[gene_names, sample_names]\n",
    "> stringtie_gene = stringtie_gene[gene_names, sample_names]\n",
    "> stringtie_gene_fpkm = stringtie_gene_fpkm[gene_names, sample_names]\n",
    "> kallisto_gene = kallisto_gene[gene_names, sample_names]\n",
    "> stringtie_tran = stringtie_tran[tran_names, sample_names]\n",
    "> stringtie_tran_fpkm = stringtie_tran_fpkm[tran_names, sample_names]\n",
    "> kallisto_tran = kallisto_tran[tran_names, sample_names]\n",
    "> \n",
    "> #Take a look at the top of each data.frame\n",
    "> head(htseq_gene_counts)\n",
    "                HBR_Rep1 HBR_Rep2 HBR_Rep3 UHR_Rep1 UHR_Rep2 UHR_Rep3\n",
    "ENSG00000008735      399      533      467       13       17        8\n",
    "ENSG00000015475       40       43       48      103       65      100\n",
    "ENSG00000025708       13       11       19       46       16       26\n",
    "ENSG00000025770       73       78       71      189      118      154\n",
    "ENSG00000040608       69       92       81       18        8       11\n",
    "ENSG00000054611       36       43       40       70       47       66\n",
    "> head(stringtie_gene)\n",
    "                  HBR_Rep1   HBR_Rep2  HBR_Rep3  UHR_Rep1  UHR_Rep2   UHR_Rep3\n",
    "ENSG00000008735 1378.35828 1502.89465 1497.1288  18.05805  27.21023   9.181973\n",
    "ENSG00000015475  183.29393  171.64763  254.2153 420.35126 262.01462 307.026245\n",
    "ENSG00000025708   91.45995   53.57246  133.4411 164.33978  67.97478 100.541763\n",
    "ENSG00000025770  543.73023  481.48813  503.9516 788.37329 596.40881 792.156189\n",
    "ENSG00000040608  424.65552  436.60098  429.4858  48.79994  49.20034  35.597595\n",
    "ENSG00000054611  119.96106  137.55148  143.3566 114.56329 105.67173 128.256882\n",
    "> head(stringtie_gene_fpkm)\n",
    "                 HBR_Rep1   HBR_Rep2  HBR_Rep3  UHR_Rep1  UHR_Rep2   UHR_Rep3\n",
    "ENSG00000008735 1154.9430 1251.98816 1251.0580  16.38104  25.36857   8.393474\n",
    "ENSG00000015475  153.5842  142.99124  212.4321 381.31439 244.28082 280.660492\n",
    "ENSG00000025708   76.6354   44.62861  111.5084 149.07800  63.37408  91.907776\n",
    "ENSG00000025770  455.5981  401.10425  421.1212 715.15924 556.04236 724.130066\n",
    "ENSG00000040608  355.8240  363.71094  358.8947  44.26803  45.87033  32.540665\n",
    "ENSG00000054611  100.5168  114.58743  119.7943 103.92411  98.51960 117.242867\n",
    "> head(kallisto_gene)\n",
    "                 HBR_Rep1  HBR_Rep2  HBR_Rep3  UHR_Rep1  UHR_Rep2 UHR_Rep3\n",
    "ENSG00000008735 1045.4520 1115.9110 1132.7170  17.28662  29.60714  12.6840\n",
    "ENSG00000015475  300.7760  287.1842  328.9308 518.08519 470.71818 511.2855\n",
    "ENSG00000025708  129.6910  130.9991  172.0632 164.98254 101.01418 133.5296\n",
    "ENSG00000025770  459.3089  420.6720  462.6208 636.11700 517.41460 574.2223\n",
    "ENSG00000040608  422.5272  479.4329  419.7968  44.82300  48.94940  35.5732\n",
    "ENSG00000054611  145.7266  211.7900  160.5624 144.22880 124.27750 180.3427\n",
    "> head(stringtie_tran)\n",
    "                HBR_Rep1 HBR_Rep2 HBR_Rep3 UHR_Rep1 UHR_Rep2 UHR_Rep3\n",
    "ENST00000615943        0        0        0        0        0        0\n",
    "ENST00000635667        0        0        0        0        0        0\n",
    "ENST00000623473        0        0        0        0        0        0\n",
    "ENST00000624155        0        0        0        0        0        0\n",
    "ENST00000422332        0        0        0        0        0        0\n",
    "ENST00000612732        0        0        0        0        0        0\n",
    "> head(stringtie_tran_fpkm)\n",
    "                HBR_Rep1 HBR_Rep2 HBR_Rep3 UHR_Rep1 UHR_Rep2 UHR_Rep3\n",
    "ENST00000615943        0        0        0        0        0        0\n",
    "ENST00000635667        0        0        0        0        0        0\n",
    "ENST00000623473        0        0        0        0        0        0\n",
    "ENST00000624155        0        0        0        0        0        0\n",
    "ENST00000422332        0        0        0        0        0        0\n",
    "ENST00000612732        0        0        0        0        0        0\n",
    "> head(kallisto_tran)\n",
    "                HBR_Rep1 HBR_Rep2 HBR_Rep3 UHR_Rep1 UHR_Rep2 UHR_Rep3\n",
    "ENST00000615943        0        0        0        0  0.00000        0\n",
    "ENST00000635667        0        0        0        0  0.00000        0\n",
    "ENST00000623473        0        0        0        0  0.00000        0\n",
    "ENST00000624155        0        0        0        0  0.00000        0\n",
    "ENST00000422332        0        0        0        0  5.78142        0\n",
    "ENST00000612732        0        0        0        0  0.00000        0\n",
    "> \n",
    "> #1. Plot kallisto gene TPMs vs stringtie gene TPMs - Pick HBR_Rep1 data arbitrarily\n",
    "> stabvar = 0.1\n",
    "> HBR1_gene_data = data.frame(kallisto_gene[,\"HBR_Rep1\"], stringtie_gene[,\"HBR_Rep1\"], htseq_gene_counts[,\"HBR_Rep1\"])\n",
    "> names(HBR1_gene_data) = c(\"kallisto\", \"stringtie\", \"htseq\")\n",
    "> p1 = ggplot(HBR1_gene_data, aes(log2(kallisto+stabvar), log2(stringtie+stabvar)))\n",
    "> p1 = p1 + geom_point()\n",
    "> p1 = p1 + geom_point(aes(colour = log2(htseq+stabvar))) + scale_colour_gradient(low = \"yellow\", high = \"red\")\n",
    "> p1 = p1 + xlab(\"Kallisto TPM\") + ylab(\"StringTie TPM\") + labs(colour = \"HtSeq Counts\")\n",
    "> p1 = p1 + labs(title = \"HBR1 gene expression values [log2(value + 0.1) scaled]\")\n",
    "> \n",
    "> #2. Plot kallisto transcript TPMs vs stringtie transcript TPMs - Pick HBR_Rep1 data arbitrarily\n",
    "> # Indicate with the points whether the data are real transcripts vs. spike-in controls\n",
    "> HBR1_tran_data = data.frame(kallisto_tran[,\"HBR_Rep1\"], stringtie_tran[,\"HBR_Rep1\"])\n",
    "> names(HBR1_tran_data) = c(\"kallisto\", \"stringtie\")\n",
    "> spikein_status=grepl(\"ERCC\",tran_names)\n",
    "> p2 = ggplot(HBR1_tran_data, aes(log2(kallisto+stabvar), log2(stringtie+stabvar)))\n",
    "> p2 = p2 + geom_point()\n",
    "> p2 = p2 + geom_point(aes(colour = spikein_status)) \n",
    "> p2 = p2 + xlab(\"Kallisto TPM\") + ylab(\"StringTie TPM\") + labs(colour = \"SpikeIn Status\")\n",
    "> p2 = p2 + labs(title = \"HBR1 transcript expression values [log2(value + 0.1) scaled]\")\n",
    "> \n",
    "> #3. Plot stringtie transcript TPMs vs. stringtie transcript FPKMs - Pick HBR_Rep1 data arbitrarily\n",
    "> # Indicate with the points whether the data are real transcripts vs. spike-in controls\n",
    "> HBR1_tran_data2 = data.frame(stringtie_tran[,\"HBR_Rep1\"], stringtie_tran_fpkm[,\"HBR_Rep1\"])\n",
    "> names(HBR1_tran_data2) = c(\"stringtie_TPM\", \"stringtie_FPKM\")\n",
    "> p3 = ggplot(HBR1_tran_data2, aes(log2(stringtie_TPM+stabvar), log2(stringtie_FPKM+stabvar)))\n",
    "> p3 = p3 + geom_point()\n",
    "> p3 = p3 + geom_point(aes(colour = spikein_status))\n",
    "> p3 = p3 + geom_abline(intercept = 0, slope = 1)\n",
    "> p3 = p3 + xlab(\"StringTie TPM\") + ylab(\"StringTie FPKM\") + labs(colour = \"SpikeIn Status\")\n",
    "> p3 = p3 + labs(title = \"HBR1 transcript expression values [log2(value + 0.1) scaled]\")\n",
    "> \n",
    "> #Print out the plots created above and store in a single PDF file\n",
    "> pdf(file=\"Tutorial_comparisons.pdf\")\n",
    "> print(p1)\n",
    "> print(p2)\n",
    "> print(p3)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "> #Quit R\n",
    "> quit(save=\"no\")\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://ec2-34-221-102-104.us-west-2.compute.amazonaws.com/workspace/rnaseq/expression/Tutorial_comparisons.pdf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Custom transcriptome database to examine a specific set of genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME/refs\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/refs\n",
    "[nathalia@aws] $  grep rRNA $RNA_REF_GTF > genes_chr22_ERCC92_rRNA.gtf \n",
    "[nathalia@aws] $  gtf_to_fasta genes_chr22_ERCC92_rRNA.gtf $RNA_REF_FASTA chr22_rRNA_transcripts.fa\n",
    "Reading the annotation file: genes_chr22_ERCC92_rRNA.gtf\n",
    "[nathalia@aws] $  cat chr22_rRNA_transcripts.fa | perl -ne 'if ($_ =~/^\\>\\d+\\s+\\w+\\s+(ERCC\\S+)[\\+\\-]/){print \">$1\\n\"}elsif($_ =~ /\\d+\\s+(ENST\\d+)/){print \">$1\\n\"}else{print $_}' > chr22_rRNA_transcripts.clean.fa\n",
    "[nathalia@aws] $  cat chr22_rRNA_transcripts.clean.fa\n",
    ">ENST00000612732\n",
    "CGACGCTCAGACAGGCATAGCCCTGGGAGGAACCCAGGCCACAAGTGCATTCAAGTGTCG\n",
    "ATGATCAATGTGTCCTGCAGTTCACATTAATTCTCCCAGCTAGCTGCATTCTTCATTGAT\n",
    "GCAGGAGCTGAGTGATCCACCACTAAAAGTC\n",
    ">ENST00000364115\n",
    "tcctctggccataccaccctgaacacgcccgatcttgtctgatctcggaagctaagcagg\n",
    "gttggcctggttagtgcttggatgggaggaa\n",
    ">ENST00000410653\n",
    "AACTTAGCACGACATGTCACTTGTCAAGAGAGGTAAATgtctcccattcaagtactaacc\n",
    "aggcccaaccctgcttagcttccaagatcagatgacatcGGGTGTGTTAAGGGTTGTATG\n",
    "GCCATAGAC\n",
    ">ENST00000516458\n",
    "acttgtaggcttttgtactcctaggcagtctcccatccaagacactaagtaggcccgatc\n",
    "ctgcttagcttccaagatcagataagatcGGGCACATTTACAGTGGCAAG\n",
    ">ENST00000391240\n",
    "gtctacagccataccaccctgaacatgtgcaatcttgtctgatctcagaagctaagcagg\n",
    "gttgagtctggttagtactcagatgggagaaacctagtacctgttggca\n",
    ">ENST00000516081\n",
    "GGTCTACAtctcccatccaaatactaaccaggcccgaccctgcttagtttccaagatcag\n",
    "acaagagcggtgtgttaagggtggtatgtctgtaggc\n",
    "[nathalia@aws] $  cd $RNA_HOME/refs/kallisto\n",
    "[nathalia@aws] $  kallisto index --index=chr22_rRNA_transcripts_kallisto_index ../chr22_rRNA_transcripts.clean.fa\n",
    "\n",
    "[build] loading fasta file ../chr22_rRNA_transcripts.clean.fa\n",
    "[build] k-mer length: 31\n",
    "[build] counting k-mers ... done.\n",
    "[build] building target de Bruijn graph ...  done \n",
    "[build] creating equivalence classes ...  done\n",
    "[build] target de Bruijn graph has 6 contigs and contains 507 k-mers \n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "And then repeat steps we did before?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DE analysis of Kallisto expression using Sleuth"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd $RNA_HOME/de\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/de\n",
    "[nathalia@aws] $  mkdir -p sleuth/input\n",
    "[nathalia@aws] $  mkdir -p sleuth/results\n",
    "[nathalia@aws] $  cd sleuth/input\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/rnaseq/de/sleuth/input\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  emacs de_kallisto.sh\n",
    "#!/bin/bash\n",
    "\n",
    "kallisto quant -b 100 --index=$RNA_HOME/refs/kallisto/chr22_ERCC92_transcripts_kallisto_index --output-dir=UHR_Rep1_ERCC-Mix1 --threads=4 $RNA_DATA_DIR/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz $RNA_DATA_DIR/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "kallisto quant -b 100 --index=$RNA_HOME/refs/kallisto/chr22_ERCC92_transcripts_kallisto_index --output-dir=UHR_Rep2_ERCC-Mix1 --threads=4 $RNA_DATA_DIR/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz $RNA_DATA_DIR/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "kallisto quant -b 100 --index=$RNA_HOME/refs/kallisto/chr22_ERCC92_transcripts_kallisto_index --output-dir=UHR_Rep3_ERCC-Mix1 --threads=4 $RNA_DATA_DIR/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz $RNA_DATA_DIR/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "\n",
    "kallisto quant -b 100 --index=$RNA_HOME/refs/kallisto/chr22_ERCC92_transcripts_kallisto_index --output-dir=HBR_Rep1_ERCC-Mix2 --threads=4 $RNA_DATA_DIR/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz $RNA_DATA_DIR/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "kallisto quant -b 100 --index=$RNA_HOME/refs/kallisto/chr22_ERCC92_transcripts_kallisto_index --output-dir=HBR_Rep2_ERCC-Mix2 --threads=4 $RNA_DATA_DIR/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz $RNA_DATA_DIR/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "kallisto quant -b 100 --index=$RNA_HOME/refs/kallisto/chr22_ERCC92_transcripts_kallisto_index --output-dir=HBR_Rep3_ERCC-Mix2 --threads=4 $RNA_DATA_DIR/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz $RNA_DATA_DIR/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "\n",
    "[nathalia@aws] $  bash de_kallisto.sh \n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 4,564\n",
    "[index] number of k-mers: 3,293,113\n",
    "[index] number of equivalence classes: 17,221\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /home/ubuntu/workspace/rnaseq/data/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "                             /home/ubuntu/workspace/rnaseq/data/UHR_Rep1_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 227,392 reads, 193,188 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 181.255\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 834 rounds\n",
    "[bstrp] number of EM bootstraps complete: 100\n",
    "\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 4,564\n",
    "[index] number of k-mers: 3,293,113\n",
    "[index] number of equivalence classes: 17,221\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /home/ubuntu/workspace/rnaseq/data/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "                             /home/ubuntu/workspace/rnaseq/data/UHR_Rep2_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 162,373 reads, 129,440 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 169.788\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 760 rounds\n",
    "[bstrp] number of EM bootstraps complete: 100\n",
    "\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 4,564\n",
    "[index] number of k-mers: 3,293,113\n",
    "[index] number of equivalence classes: 17,221\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /home/ubuntu/workspace/rnaseq/data/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "                             /home/ubuntu/workspace/rnaseq/data/UHR_Rep3_ERCC-Mix1_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 185,442 reads, 158,183 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 167.718\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 854 rounds\n",
    "[bstrp] number of EM bootstraps complete: 100\n",
    "\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 4,564\n",
    "[index] number of k-mers: 3,293,113\n",
    "[index] number of equivalence classes: 17,221\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /home/ubuntu/workspace/rnaseq/data/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "                             /home/ubuntu/workspace/rnaseq/data/HBR_Rep1_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 118,571 reads, 99,270 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 169.016\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 736 rounds\n",
    "[bstrp] number of EM bootstraps complete: 100\n",
    "\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 4,564\n",
    "[index] number of k-mers: 3,293,113\n",
    "[index] number of equivalence classes: 17,221\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /home/ubuntu/workspace/rnaseq/data/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "                             /home/ubuntu/workspace/rnaseq/data/HBR_Rep2_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 144,826 reads, 121,976 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 169.638\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 833 rounds\n",
    "[bstrp] number of EM bootstraps complete: 100\n",
    "\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 4,564\n",
    "[index] number of k-mers: 3,293,113\n",
    "[index] number of equivalence classes: 17,221\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /home/ubuntu/workspace/rnaseq/data/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read1.fastq.gz\n",
    "                             /home/ubuntu/workspace/rnaseq/data/HBR_Rep3_ERCC-Mix2_Build37-ErccTranscripts-chr22.read2.fastq.gz\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 129,786 reads, 108,536 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 171.252\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 756 rounds\n",
    "[bstrp] number of EM bootstraps complete: 100\n",
    "\n",
    "[nathalia@aws] $  ls -R\n",
    ".:\n",
    "HBR_Rep1_ERCC-Mix2  HBR_Rep3_ERCC-Mix2  UHR_Rep2_ERCC-Mix1  de_kallisto.sh\n",
    "HBR_Rep2_ERCC-Mix2  UHR_Rep1_ERCC-Mix1  UHR_Rep3_ERCC-Mix1\n",
    "\n",
    "./HBR_Rep1_ERCC-Mix2:\n",
    "abundance.h5  abundance.tsv  run_info.json\n",
    "\n",
    "./HBR_Rep2_ERCC-Mix2:\n",
    "abundance.h5  abundance.tsv  run_info.json\n",
    "\n",
    "./HBR_Rep3_ERCC-Mix2:\n",
    "abundance.h5  abundance.tsv  run_info.json\n",
    "\n",
    "./UHR_Rep1_ERCC-Mix1:\n",
    "abundance.h5  abundance.tsv  run_info.json\n",
    "\n",
    "./UHR_Rep2_ERCC-Mix1:\n",
    "abundance.h5  abundance.tsv  run_info.json\n",
    "\n",
    "./UHR_Rep3_ERCC-Mix1:\n",
    "abundance.h5  abundance.tsv  run_info.json\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://github.com/griffithlab/rnaseq_tutorial/blob/master/scripts/Tutorial_KallistoSleuth.R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Didn't work..."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
