{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 11/14/2018\n",
    "\n",
    "RNAseq \n",
    "\n",
    "Brian Haas\n",
    "\n",
    "Genome guided and genome free assembly"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://rrwick.github.io/Bandage/\n",
    "\n",
    "https://github.com/trinityrnaseq/CSHLadvSeqTech/wiki/Home/229b766e058e6798fb87024cfb1c0d9e4fb77114"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd ~/workspace\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace\n",
    "\n",
    "[nathalia@aws] $  mkdir transreco\n",
    "[nathalia@aws] $  cd transreco/\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/transreco\n",
    "\n",
    "[nathalia@aws] $  cp -r ~/workspace/data/trinity/rnaseq/* .\n",
    "[nathalia@aws] $  lhtr\n",
    "total 2.0M\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 15:24 data\n",
    "-rw-r--r-- 1 ubuntu ubuntu 777K Nov 14 15:24 minigenome.fa\n",
    "-rw-r--r-- 1 ubuntu ubuntu  354 Nov 14 15:24 samples.txt\n",
    "-rw-r--r-- 1 ubuntu ubuntu 1.2M Nov 14 15:24 minigenome.gtf\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IGV"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  samtools faidx minigenome.fa\n",
    "[nathalia@aws] $  lhtr\n",
    "total 2.0M\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 15:24 data\n",
    "-rw-r--r-- 1 ubuntu ubuntu 777K Nov 14 15:24 minigenome.fa\n",
    "-rw-r--r-- 1 ubuntu ubuntu  354 Nov 14 15:24 samples.txt\n",
    "-rw-r--r-- 1 ubuntu ubuntu 1.2M Nov 14 15:24 minigenome.gtf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   27 Nov 14 15:25 minigenome.fa.fai\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://ec2-34-221-102-104.us-west-2.compute.amazonaws.com/workspace/transreco/\n",
    "\n",
    "Get `.fa` and `.gtf` links\n",
    "\n",
    "Genome tab > Load from URL > Load `.fa`   \n",
    "\n",
    "File tab > Load from URL > Load `.gtf`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![igv1](igv1_minigenome.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome-guided Transcript Reconstruction"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Align with HISAT2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  hisat2_extract_splice_sites.py minigenome.gtf  > minigenome.gtf.ss\n",
    "[nathalia@aws] $  hisat2_extract_exons.py minigenome.gtf > minigenome.gtf.exons\n",
    "[nathalia@aws] $  hisat2-build --exon minigenome.gtf.exons \\\n",
    ">        --ss minigenome.gtf.ss \\\n",
    ">        -p 2 minigenome.fa minigenome.fa\n",
    "Settings:\n",
    "  Output files: \"minigenome.fa.*.ht2\"\n",
    "  Line rate: 7 (line is 128 bytes)\n",
    "  Lines per side: 1 (side is 128 bytes)\n",
    "  Offset rate: 4 (one in 16)\n",
    "  FTable chars: 10\n",
    "  Strings: unpacked\n",
    "  Local offset rate: 3 (one in 8)\n",
    "  Local fTable chars: 6\n",
    "  Local sequence length: 57344\n",
    "  Local sequence overlap between two consecutive indexes: 1024\n",
    "  Endianness: little\n",
    "  Actual local endianness: little\n",
    "  Sanity checking: disabled\n",
    "  Assertions: disabled\n",
    "  Random seed: 0\n",
    "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
    "Input files DNA, FASTA:\n",
    "  minigenome.fa\n",
    "Reading reference sizes\n",
    "  Time reading reference sizes: 00:00:00\n",
    "Calculating joined length\n",
    "Writing header\n",
    "Reserving space for joined string\n",
    "Joining reference sequences\n",
    "  Time to join reference sequences: 00:00:00\n",
    "  Time to read SNPs and splice sites: 00:00:00\n",
    "Generation 0 (713858 -> 713858 nodes, 0 ranks)\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "Generation 1 (714349 -> 714349 nodes, 0 ranks)\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "Generation 2 (715331 -> 715331 nodes, 0 ranks)\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "Generation 3 (717296 -> 717296 nodes, 0 ranks)\n",
    "BUILT FROM_INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "RESIZE NODES: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "92 361077\n",
    "274 360163\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "SORT NODES: 0\n",
    "MERGE, UPDATE RANK: 0\n",
    "Generation 4 (721240 -> 716850 nodes, 708915 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "175 360205\n",
    "174 356645\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 0\n",
    "TOTAL TIME: 0\n",
    "Generation 5 (716893 -> 716881 nodes, 711511 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 1\n",
    "175 360205\n",
    "174 356676\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 1\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 0\n",
    "TOTAL TIME: 1\n",
    "Generation 6 (716954 -> 716942 nodes, 712221 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "175 360205\n",
    "174 356737\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 0\n",
    "TOTAL TIME: 0\n",
    "Generation 7 (717076 -> 717006 nodes, 713114 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "175 360205\n",
    "174 356801\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 0\n",
    "TOTAL TIME: 0\n",
    "Generation 8 (717195 -> 717134 nodes, 714405 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "175 360205\n",
    "174 356929\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 0\n",
    "TOTAL TIME: 0\n",
    "Generation 9 (717451 -> 717390 nodes, 715920 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "175 360205\n",
    "174 357185\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 0\n",
    "TOTAL TIME: 0\n",
    "Generation 10 (717961 -> 717900 nodes, 717651 ranks)\n",
    "ALLOCATE FROM_TABLE: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "175 360205\n",
    "174 357695\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "BUILD TABLE: 0\n",
    "BUILD INDEX: 0\n",
    "COUNTED NEW NODES: 0\n",
    "COUNTED TEMP NODES: 0\n",
    "RESIZED NODES: 0\n",
    "RESIZED NODES: 0\n",
    "MADE NEW NODES: 0\n",
    "MERGEUPDATERANK: 0\n",
    "TOTAL TIME: 0\n",
    "Generation 11 (717952 -> 717899 nodes, 717899 ranks)\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "175 360205\n",
    "174 357694\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "Generating edges... \n",
    "NODE.TO -> GENOME POS: 0\n",
    "BUILD FROM_INDEX 0\n",
    "COUNTED NEW EDGES: 0\n",
    "MADE NEW EDGES: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "165 103481\n",
    "186 102910\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "163 77015\n",
    "188 76207\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "166 76679\n",
    "185 76245\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "204 103369\n",
    "147 102482\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "SORTED NEW EDGES: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "176 360448\n",
    "175 357451\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "RE-SORTED NODES: 0\n",
    "PROCESS EDGES: 0\n",
    "REMOVE Y: 0\n",
    "COUNT NUMBER IN EACH BIN: 0\n",
    "FINISHED FIRST ROUND: 0\n",
    "176 360797\n",
    "175 357592\n",
    "FINISHED RECURSIVE SORTS: 0\n",
    "SORT, Make index: 0\n",
    "TOTAL: 0\n",
    "Allocating ftab, absorbFtab\n",
    "Entering GFM loop\n",
    "Exited GFM loop\n",
    "fchr[A]: 0\n",
    "fchr[C]: 206391\n",
    "fchr[G]: 359613\n",
    "fchr[T]: 512537\n",
    "fchr[$]: 718388\n",
    "Exiting GFM::buildToDisk()\n",
    "Returning from initFromVector\n",
    "Wrote 4639748 bytes to primary GFM file: minigenome.fa.1.ht2\n",
    "Wrote 179480 bytes to secondary GFM file: minigenome.fa.2.ht2\n",
    "Re-opening _in1 and _in2 as input streams\n",
    "Returning from GFM constructor\n",
    "Returning from initFromVector\n",
    "Wrote 520257 bytes to primary GFM file: minigenome.fa.5.ht2\n",
    "Wrote 182410 bytes to secondary GFM file: minigenome.fa.6.ht2\n",
    "Re-opening _in5 and _in5 as input streams\n",
    "Returning from HierEbwt constructor\n",
    "Headers:\n",
    "    len: 713338\n",
    "    gbwtLen: 718389\n",
    "    nodes: 717898\n",
    "    sz: 356669\n",
    "    gbwtSz: 359195\n",
    "    lineRate: 7\n",
    "    offRate: 4\n",
    "    offMask: 0xfffffff0\n",
    "    ftabChars: 10\n",
    "    eftabLen: 0\n",
    "    eftabSz: 0\n",
    "    ftabLen: 1048577\n",
    "    ftabSz: 4194308\n",
    "    offsLen: 44869\n",
    "    offsSz: 179476\n",
    "    lineSz: 128\n",
    "    sideSz: 128\n",
    "    sideGbwtSz: 104\n",
    "    sideGbwtLen: 208\n",
    "    numSides: 3454\n",
    "    numLines: 3454\n",
    "    gbwtTotLen: 442112\n",
    "    gbwtTotSz: 442112\n",
    "    reverse: 0\n",
    "    linearFM: No\n",
    "Total time for call to driver() for forward index: 00:00:02\n",
    "[nathalia@aws] $  lhtr\n",
    "total 7.5M\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 15:24 data\n",
    "-rw-r--r-- 1 ubuntu ubuntu 777K Nov 14 15:24 minigenome.fa\n",
    "-rw-r--r-- 1 ubuntu ubuntu  354 Nov 14 15:24 samples.txt\n",
    "-rw-r--r-- 1 ubuntu ubuntu 1.2M Nov 14 15:24 minigenome.gtf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   27 Nov 14 15:25 minigenome.fa.fai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  14K Nov 14 15:33 minigenome.gtf.ss\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13K Nov 14 15:33 minigenome.gtf.exons\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 175K Nov 14 15:33 minigenome.fa.4.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  944 Nov 14 15:33 minigenome.fa.3.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 3.8K Nov 14 15:33 minigenome.fa.8.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  19K Nov 14 15:33 minigenome.fa.7.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 176K Nov 14 15:33 minigenome.fa.2.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.5M Nov 14 15:33 minigenome.fa.1.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 179K Nov 14 15:33 minigenome.fa.6.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 509K Nov 14 15:33 minigenome.fa.5.ht2\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  hisat2 --dta -x minigenome.fa -p 2 \\\n",
    ">     -1 data/sA_rep1_1.fastq.gz -2 data/sA_rep1_2.fastq.gz \\\n",
    ">     | samtools view -Sb -F 4 \\\n",
    ">     | samtools sort  -o sA_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "25000 reads; of these:\n",
    "  25000 (100.00%) were paired; of these:\n",
    "    827 (3.31%) aligned concordantly 0 times\n",
    "    24093 (96.37%) aligned concordantly exactly 1 time\n",
    "    80 (0.32%) aligned concordantly >1 times\n",
    "    ----\n",
    "    827 pairs aligned concordantly 0 times; of these:\n",
    "      2 (0.24%) aligned discordantly 1 time\n",
    "    ----\n",
    "    825 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      1650 mates make up the pairs; of these:\n",
    "        822 (49.82%) aligned 0 times\n",
    "        814 (49.33%) aligned exactly 1 time\n",
    "        14 (0.85%) aligned >1 times\n",
    "98.36% overall alignment rate\n",
    "[nathalia@aws] $  hisat2 --dta -x minigenome.fa -p 2 \\\n",
    ">       -1 data/sA_rep2_1.fastq.gz -2 data/sA_rep2_2.fastq.gz \\\n",
    ">       | samtools view -Sb -F 4 \\\n",
    ">       | samtools sort  -o sA_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "25000 reads; of these:\n",
    "  25000 (100.00%) were paired; of these:\n",
    "    794 (3.18%) aligned concordantly 0 times\n",
    "    24143 (96.57%) aligned concordantly exactly 1 time\n",
    "    63 (0.25%) aligned concordantly >1 times\n",
    "    ----\n",
    "    794 pairs aligned concordantly 0 times; of these:\n",
    "      6 (0.76%) aligned discordantly 1 time\n",
    "    ----\n",
    "    788 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      1576 mates make up the pairs; of these:\n",
    "        787 (49.94%) aligned 0 times\n",
    "        782 (49.62%) aligned exactly 1 time\n",
    "        7 (0.44%) aligned >1 times\n",
    "98.43% overall alignment rate\n",
    "[nathalia@aws] $  hisat2 --dta -x minigenome.fa -p 2 \\\n",
    ">        -1 data/sA_rep3_1.fastq.gz -2 data/sA_rep3_2.fastq.gz \\\n",
    ">        | samtools view -Sb -F 4 \\\n",
    ">        | samtools sort  -o sA_rep3.cSorted.hisat2.minigenome.fa.bam \n",
    "25000 reads; of these:\n",
    "  25000 (100.00%) were paired; of these:\n",
    "    791 (3.16%) aligned concordantly 0 times\n",
    "    24099 (96.40%) aligned concordantly exactly 1 time\n",
    "    110 (0.44%) aligned concordantly >1 times\n",
    "    ----\n",
    "    791 pairs aligned concordantly 0 times; of these:\n",
    "      2 (0.25%) aligned discordantly 1 time\n",
    "    ----\n",
    "    789 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      1578 mates make up the pairs; of these:\n",
    "        785 (49.75%) aligned 0 times\n",
    "        781 (49.49%) aligned exactly 1 time\n",
    "        12 (0.76%) aligned >1 times\n",
    "98.43% overall alignment rate\n",
    "[nathalia@aws] $  hisat2 --dta -x minigenome.fa -p 2 \\\n",
    ">        -1 data/sB_rep1_1.fastq.gz -2 data/sB_rep1_2.fastq.gz \\\n",
    ">        | samtools view -Sb -F 4 \\\n",
    ">        | samtools sort  -o sB_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "25000 reads; of these:\n",
    "  25000 (100.00%) were paired; of these:\n",
    "    835 (3.34%) aligned concordantly 0 times\n",
    "    24071 (96.28%) aligned concordantly exactly 1 time\n",
    "    94 (0.38%) aligned concordantly >1 times\n",
    "    ----\n",
    "    835 pairs aligned concordantly 0 times; of these:\n",
    "      5 (0.60%) aligned discordantly 1 time\n",
    "    ----\n",
    "    830 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      1660 mates make up the pairs; of these:\n",
    "        828 (49.88%) aligned 0 times\n",
    "        828 (49.88%) aligned exactly 1 time\n",
    "        4 (0.24%) aligned >1 times\n",
    "98.34% overall alignment rate\n",
    "[nathalia@aws] $  hisat2 --dta -x minigenome.fa -p 2 \\\n",
    ">        -1 data/sB_rep2_1.fastq.gz -2 data/sB_rep2_2.fastq.gz \\\n",
    ">        | samtools view -Sb -F 4 \\\n",
    ">        | samtools sort  -o sB_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "25000 reads; of these:\n",
    "  25000 (100.00%) were paired; of these:\n",
    "    736 (2.94%) aligned concordantly 0 times\n",
    "    24188 (96.75%) aligned concordantly exactly 1 time\n",
    "    76 (0.30%) aligned concordantly >1 times\n",
    "    ----\n",
    "    736 pairs aligned concordantly 0 times; of these:\n",
    "      3 (0.41%) aligned discordantly 1 time\n",
    "    ----\n",
    "    733 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      1466 mates make up the pairs; of these:\n",
    "        733 (50.00%) aligned 0 times\n",
    "        722 (49.25%) aligned exactly 1 time\n",
    "        11 (0.75%) aligned >1 times\n",
    "98.53% overall alignment rate\n",
    "[nathalia@aws] $  hisat2 --dta -x minigenome.fa -p 2 \\\n",
    ">        -1 data/sB_rep3_1.fastq.gz -2 data/sB_rep3_2.fastq.gz \\\n",
    ">        | samtools view -Sb -F 4 \\\n",
    ">        | samtools sort  -o sB_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "25000 reads; of these:\n",
    "  25000 (100.00%) were paired; of these:\n",
    "    720 (2.88%) aligned concordantly 0 times\n",
    "    24203 (96.81%) aligned concordantly exactly 1 time\n",
    "    77 (0.31%) aligned concordantly >1 times\n",
    "    ----\n",
    "    720 pairs aligned concordantly 0 times; of these:\n",
    "      3 (0.42%) aligned discordantly 1 time\n",
    "    ----\n",
    "    717 pairs aligned 0 times concordantly or discordantly; of these:\n",
    "      1434 mates make up the pairs; of these:\n",
    "        717 (50.00%) aligned 0 times\n",
    "        710 (49.51%) aligned exactly 1 time\n",
    "        7 (0.49%) aligned >1 times\n",
    "98.57% overall alignment rate\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Index each of the bam files for IGV"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  for file in *.bam  \n",
    ">    do \n",
    ">    samtools index $file\n",
    ">    echo indexed $file\n",
    ">    done\n",
    "indexed sA_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "indexed sA_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "indexed sA_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "indexed sB_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "indexed sB_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "indexed sB_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://ec2-34-221-102-104.us-west-2.compute.amazonaws.com/workspace/transreco/sA_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "\n",
    "http://ec2-34-221-102-104.us-west-2.compute.amazonaws.com/workspace/transreco/sB_rep1.cSorted.hisat2.minigenome.fa.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![igv2](bam_igv.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Reconstruct transcripts using StringTie"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  for file in *.bam\n",
    ">    do \n",
    ">    stringtie $file -o $file.stringtie.GTF\n",
    ">    echo done running stringtie on $file\n",
    ">    done\n",
    "done running stringtie on sA_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "done running stringtie on sA_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "done running stringtie on sA_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "done running stringtie on sB_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "done running stringtie on sB_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "done running stringtie on sB_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "\n",
    "[nathalia@aws] $  stringtie --merge -o stringtie.final.GTF  *.stringtie.GTF\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "http://ec2-34-221-102-104.us-west-2.compute.amazonaws.com/workspace/transreco/stringtie.final.GTF\n",
    "\n",
    "Load into IGV"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![stringtie](stringtie_final.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome-free de novo transcript reconstruction with TRINITY"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  export TRINITY_HOME=~/trinityrnaseq-Trinity-v2.5.1\n",
    "\n",
    "[nathalia@aws] $  $TRINITY_HOME/Trinity --samples_file samples.txt \\\n",
    ">       --seqType fq --CPU 2 --max_memory 1G\n",
    "\n",
    "\n",
    "     ______  ____   ____  ____   ____  ______  __ __\n",
    "    |      ||    \\ |    ||    \\ |    ||      ||  |  |\n",
    "    |      ||  D  ) |  | |  _  | |  | |      ||  |  |\n",
    "    |_|  |_||    /  |  | |  |  | |  | |_|  |_||  ~  |\n",
    "      |  |  |    \\  |  | |  |  | |  |   |  |  |___, |\n",
    "      |  |  |  .  \\ |  | |  |  | |  |   |  |  |     |\n",
    "      |__|  |__|\\_||____||__|__||____|  |__|  |____/\n",
    "\n",
    "\n",
    "Left read files: $VAR1 = [\n",
    "          'data/sA_rep1_1.fastq.gz',\n",
    "          'data/sA_rep2_1.fastq.gz',\n",
    "          'data/sA_rep3_1.fastq.gz',\n",
    "          'data/sB_rep1_1.fastq.gz',\n",
    "          'data/sB_rep2_1.fastq.gz',\n",
    "          'data/sB_rep3_1.fastq.gz'\n",
    "        ];\n",
    "Right read files: $VAR1 = [\n",
    "          'data/sA_rep1_2.fastq.gz',\n",
    "          'data/sA_rep2_2.fastq.gz',\n",
    "          'data/sA_rep3_2.fastq.gz',\n",
    "          'data/sB_rep1_2.fastq.gz',\n",
    "          'data/sB_rep2_2.fastq.gz',\n",
    "          'data/sB_rep3_2.fastq.gz'\n",
    "        ];\n",
    "\n",
    "\n",
    "\n",
    "********************************************************************\n",
    "** Warning, Trinity cannot determine which version of Java is being used.  Version 1.8 is required. \n",
    "\n",
    "Attempting to continue in 30 seconds\n",
    "********************************************************************\n",
    "\n",
    "\n",
    "Trinity version: Trinity-v2.5.1\n",
    "-ERROR: couldn't run the network check to confirm latest Trinity software version.\n",
    "\n",
    "Wednesday, November 14, 2018: 15:51:22\tCMD: java -Xmx64m -XX:ParallelGCThreads=2  -jar /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/ExitTester.jar 0\n",
    "Wednesday, November 14, 2018: 15:51:22\tCMD: java -Xmx64m -XX:ParallelGCThreads=2  -jar /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/ExitTester.jar 1\n",
    "Wednesday, November 14, 2018: 15:51:22\tCMD: mkdir -p /workspace/transreco/trinity_out_dir\n",
    "Wednesday, November 14, 2018: 15:51:22\tCMD: mkdir -p /workspace/transreco/trinity_out_dir/chrysalis\n",
    "\n",
    "\n",
    "----------------------------------------------------------------------------------\n",
    "-------------- Trinity Phase 1: Clustering of RNA-Seq Reads  ---------------------\n",
    "----------------------------------------------------------------------------------\n",
    "\n",
    "---------------------------------------------------------------\n",
    "------------ In silico Read Normalization ---------------------\n",
    "-- (Removing Excess Reads Beyond 50 Coverage --\n",
    "---------------------------------------------------------------\n",
    "\n",
    "# running normalization on reads: $VAR1 = [\n",
    "          [\n",
    "            '/workspace/transreco/data/sA_rep1_1.fastq.gz',\n",
    "            '/workspace/transreco/data/sA_rep2_1.fastq.gz',\n",
    "            '/workspace/transreco/data/sA_rep3_1.fastq.gz',\n",
    "            '/workspace/transreco/data/sB_rep1_1.fastq.gz',\n",
    "            '/workspace/transreco/data/sB_rep2_1.fastq.gz',\n",
    "            '/workspace/transreco/data/sB_rep3_1.fastq.gz'\n",
    "          ],\n",
    "          [\n",
    "            '/workspace/transreco/data/sA_rep1_2.fastq.gz',\n",
    "            '/workspace/transreco/data/sA_rep2_2.fastq.gz',\n",
    "            '/workspace/transreco/data/sA_rep3_2.fastq.gz',\n",
    "            '/workspace/transreco/data/sB_rep1_2.fastq.gz',\n",
    "            '/workspace/transreco/data/sB_rep2_2.fastq.gz',\n",
    "            '/workspace/transreco/data/sB_rep3_2.fastq.gz'\n",
    "          ]\n",
    "        ];\n",
    "\n",
    "\n",
    "Wednesday, November 14, 2018: 15:51:22\tCMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/insilico_read_normalization.pl --seqType fq --JM 1G  --max_cov 50 --CPU 2 --output /workspace/transreco/trinity_out_dir/insilico_read_normalization   --max_pct_stdev 10000  --left /workspace/transreco/data/sA_rep1_1.fastq.gz,/workspace/transreco/data/sA_rep2_1.fastq.gz,/workspace/transreco/data/sA_rep3_1.fastq.gz,/workspace/transreco/data/sB_rep1_1.fastq.gz,/workspace/transreco/data/sB_rep2_1.fastq.gz,/workspace/transreco/data/sB_rep3_1.fastq.gz --right /workspace/transreco/data/sA_rep1_2.fastq.gz,/workspace/transreco/data/sA_rep2_2.fastq.gz,/workspace/transreco/data/sA_rep3_2.fastq.gz,/workspace/transreco/data/sB_rep1_2.fastq.gz,/workspace/transreco/data/sB_rep2_2.fastq.gz,/workspace/transreco/data/sB_rep3_2.fastq.gz --pairs_together --PARALLEL_STATS  \n",
    "Converting input files. (both directions in parallel)CMD: seqtk-trinity seq -A <(gunzip -c /workspace/transreco/data/sA_rep1_1.fastq.gz) >> left.fa\n",
    "CMD: seqtk-trinity seq -A <(gunzip -c /workspace/transreco/data/sA_rep1_2.fastq.gz) >> right.fa\n",
    "CMD finished (0 seconds)\n",
    "CMD: seqtk-trinity seq -A <(gunzip -c /workspace/transreco/data/sA_rep2_1.fastq.gz) >> left.fa\n",
    "CMD finished (0 seconds)\n",
    "CMD: seqtk-trinity seq -A <(gunzip -c /workspace/transreco/data/sA_rep2_2.fastq.gz) >> right.fa\n",
    "CMD finished (0 seconds)\n",
    "CMD: seqtk-trinity seq -A <(gunzip -c /workspace/transreco/data/sA_rep3_2.fastq.gz) >> right.fa\n",
    "CMD finished (0 seconds)\n",
    "CMD: seqtk-trinity seq -A <(gunzip -c /workspace/transreco/data/sA_rep3_1.fastq.gz) >> left.fa\n",
    "CMD finished (1 seconds)\n",
    "CMD: seqtk-trinity seq -A <(gunzip -c /workspace/transreco/data/sB_rep1_2.fastq.gz) >> right.fa\n",
    "CMD finished (1 seconds)\n",
    "CMD: seqtk-trinity seq -A <(gunzip -c /workspace/transreco/data/sB_rep1_1.fastq.gz) >> left.fa\n",
    "CMD finished (0 seconds)\n",
    "CMD: seqtk-trinity seq -A <(gunzip -c /workspace/transreco/data/sB_rep2_2.fastq.gz) >> right.fa\n",
    "CMD finished (0 seconds)\n",
    "CMD: seqtk-trinity seq -A <(gunzip -c /workspace/transreco/data/sB_rep2_1.fastq.gz) >> left.fa\n",
    "CMD finished (0 seconds)\n",
    "CMD: seqtk-trinity seq -A <(gunzip -c /workspace/transreco/data/sB_rep3_2.fastq.gz) >> right.fa\n",
    "CMD finished (0 seconds)\n",
    "CMD: seqtk-trinity seq -A <(gunzip -c /workspace/transreco/data/sB_rep3_1.fastq.gz) >> left.fa\n",
    "CMD finished (0 seconds)\n",
    "CMD finished (0 seconds)\n",
    "CMD: touch left.fa.ok\n",
    "CMD finished (0 seconds)\n",
    "CMD: touch right.fa.ok\n",
    "CMD finished (0 seconds)\n",
    "Done converting input files.CMD: cat left.fa right.fa > both.fa\n",
    "CMD finished (0 seconds)\n",
    "CMD: touch both.fa.ok\n",
    "CMD finished (0 seconds)\n",
    "-------------------------------------------\n",
    "----------- Jellyfish  --------------------\n",
    "-- (building a k-mer catalog from reads) --\n",
    "-------------------------------------------\n",
    "\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/..//trinity-plugins/jellyfish/bin/jellyfish count -t 2 -m 25 -s 100000000  --canonical  both.fa\n",
    "CMD finished (3 seconds)\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/..//trinity-plugins/jellyfish/bin/jellyfish histo -t 2 -o jellyfish.K25.min2.kmers.fa.histo mer_counts.jf\n",
    "CMD finished (0 seconds)\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/..//trinity-plugins/jellyfish/bin/jellyfish dump -L 2 mer_counts.jf > jellyfish.K25.min2.kmers.fa\n",
    "CMD finished (0 seconds)\n",
    "CMD: touch jellyfish.K25.min2.kmers.fa.success\n",
    "CMD finished (0 seconds)\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/..//Inchworm/bin/fastaToKmerCoverageStats --reads left.fa --kmers jellyfish.K25.min2.kmers.fa --kmer_size 25  --num_threads 2  --DS  > left.fa.K25.stats\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/..//Inchworm/bin/fastaToKmerCoverageStats --reads right.fa --kmers jellyfish.K25.min2.kmers.fa --kmer_size 25  --num_threads 2  --DS  > right.fa.K25.stats\n",
    "-reading Kmer occurrences...\n",
    "-reading Kmer occurrences...\n",
    "\n",
    " done parsing 1006423 Kmers, 1006423 added, taking 1 seconds.\n",
    "\n",
    " done parsing 1006423 Kmers, 1006423 added, taking 1 seconds.\n",
    "STATS_GENERATION_TIME: 2 seconds.\n",
    "STATS_GENERATION_TIME: 2 seconds.\n",
    "CMD finished (4 seconds)\n",
    "CMD finished (4 seconds)\n",
    "CMD: touch left.fa.K25.stats.ok\n",
    "CMD finished (0 seconds)\n",
    "CMD: touch right.fa.K25.stats.ok\n",
    "CMD finished (0 seconds)\n",
    "-sorting each stats file by read name.\n",
    "CMD: /usr/bin/sort --parallel=2 -k5,5 -T . -S 1G left.fa.K25.stats > left.fa.K25.stats.sort\n",
    "CMD: /usr/bin/sort --parallel=2 -k5,5 -T . -S 1G right.fa.K25.stats > right.fa.K25.stats.sort\n",
    "CMD finished (0 seconds)\n",
    "CMD finished (0 seconds)\n",
    "CMD: touch left.fa.K25.stats.sort.ok\n",
    "CMD finished (0 seconds)\n",
    "CMD: touch right.fa.K25.stats.sort.ok\n",
    "CMD finished (0 seconds)\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/..//util/support_scripts//nbkc_merge_left_right_stats.pl --left left.fa.K25.stats.sort --right right.fa.K25.stats.sort --sorted > pairs.K25.stats\n",
    "-opening left.fa.K25.stats.sort\n",
    "-opening right.fa.K25.stats.sort\n",
    "-done opening files.\n",
    "CMD finished (0 seconds)\n",
    "CMD: touch pairs.K25.stats.ok\n",
    "CMD finished (0 seconds)\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/..//util/support_scripts//nbkc_normalize.pl pairs.K25.stats 50 10000 > pairs.K25.stats.C50.pctSD10000.accs\n",
    "76630 / 150000 = 51.09% reads selected during normalization.\n",
    "0 / 150000 = 0.00% reads discarded as likely aberrant based on coverage profiles.\n",
    "0 / 150000 = 0.00% reads missing kmer coverage (N chars included?).\n",
    "CMD finished (0 seconds)\n",
    "CMD: touch pairs.K25.stats.C50.pctSD10000.accs.ok\n",
    "CMD finished (0 seconds)\n",
    "CMD: touch /workspace/transreco/trinity_out_dir/insilico_read_normalization/sA_rep1_1.fastq.gz_ext_all_reads.normalized_K25_C50_pctSD10000.fq.ok\n",
    "CMD finished (0 seconds)\n",
    "CMD: touch /workspace/transreco/trinity_out_dir/insilico_read_normalization/sA_rep1_2.fastq.gz_ext_all_reads.normalized_K25_C50_pctSD10000.fq.ok\n",
    "CMD finished (0 seconds)\n",
    "CMD: ln -sf /workspace/transreco/trinity_out_dir/insilico_read_normalization/sA_rep1_1.fastq.gz_ext_all_reads.normalized_K25_C50_pctSD10000.fq left.norm.fq\n",
    "CMD finished (0 seconds)\n",
    "CMD: ln -sf /workspace/transreco/trinity_out_dir/insilico_read_normalization/sA_rep1_2.fastq.gz_ext_all_reads.normalized_K25_C50_pctSD10000.fq right.norm.fq\n",
    "CMD finished (0 seconds)\n",
    "-removing tmp dir /workspace/transreco/trinity_out_dir/insilico_read_normalization/tmp_normalized_reads\n",
    "\n",
    "\n",
    "Normalization complete. See outputs: \n",
    "\t/workspace/transreco/trinity_out_dir/insilico_read_normalization/sA_rep1_1.fastq.gz_ext_all_reads.normalized_K25_C50_pctSD10000.fq\n",
    "\t/workspace/transreco/trinity_out_dir/insilico_read_normalization/sA_rep1_2.fastq.gz_ext_all_reads.normalized_K25_C50_pctSD10000.fq\n",
    "Wednesday, November 14, 2018: 15:51:32\tCMD: touch /workspace/transreco/trinity_out_dir/insilico_read_normalization/normalization.ok\n",
    "Converting input files. (in parallel)Wednesday, November 14, 2018: 15:51:32\tCMD: cat /workspace/transreco/trinity_out_dir/insilico_read_normalization/left.norm.fq | seqtk-trinity seq -A - >> left.fa\n",
    "Wednesday, November 14, 2018: 15:51:32\tCMD: cat /workspace/transreco/trinity_out_dir/insilico_read_normalization/right.norm.fq | seqtk-trinity seq -A - >> right.fa\n",
    "Wednesday, November 14, 2018: 15:51:32\tCMD: touch left.fa.ok\n",
    "Wednesday, November 14, 2018: 15:51:32\tCMD: touch right.fa.ok\n",
    "Wednesday, November 14, 2018: 15:51:32\tCMD: touch left.fa.ok right.fa.ok\n",
    "Wednesday, November 14, 2018: 15:51:32\tCMD: cat left.fa right.fa > /workspace/transreco/trinity_out_dir/both.fa\n",
    "Wednesday, November 14, 2018: 15:51:32\tCMD: touch /workspace/transreco/trinity_out_dir/both.fa.ok\n",
    "-------------------------------------------\n",
    "----------- Jellyfish  --------------------\n",
    "-- (building a k-mer catalog from reads) --\n",
    "-------------------------------------------\n",
    "\n",
    "* Running CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/trinity-plugins/jellyfish/bin/jellyfish count -t 2 -m 25 -s 100000000  --canonical  /workspace/transreco/trinity_out_dir/both.fa\n",
    "* Running CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/trinity-plugins/jellyfish/bin/jellyfish dump -L 1 mer_counts.jf > jellyfish.kmers.fa\n",
    "* Running CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/trinity-plugins/jellyfish/bin/jellyfish histo -t 2 -o jellyfish.kmers.fa.histo mer_counts.jf\n",
    "----------------------------------------------\n",
    "--------------- Inchworm ---------------------\n",
    "-- (Linear contig construction from k-mers) --\n",
    "----------------------------------------------\n",
    "\n",
    "* Running CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Inchworm/bin//inchworm --kmers jellyfish.kmers.fa --run_inchworm -K 25 -L 25 --monitor 1   --DS  --num_threads 2  --PARALLEL_IWORM  > /workspace/transreco/trinity_out_dir/inchworm.K25.L25.DS.fa.tmp\n",
    "* Running CMD: mv /workspace/transreco/trinity_out_dir/inchworm.K25.L25.DS.fa.tmp /workspace/transreco/trinity_out_dir/inchworm.K25.L25.DS.fa\n",
    "Wednesday, November 14, 2018: 15:51:43\tCMD: touch /workspace/transreco/trinity_out_dir/inchworm.K25.L25.DS.fa.finished\n",
    "--------------------------------------------------------\n",
    "-------------------- Chrysalis -------------------------\n",
    "-- (Contig Clustering & de Bruijn Graph Construction) --\n",
    "--------------------------------------------------------\n",
    "\n",
    "inchworm_target: /workspace/transreco/trinity_out_dir/both.fa\n",
    "bowite_reads_fa: /workspace/transreco/trinity_out_dir/both.fa\n",
    "chrysalis_reads_fa: /workspace/transreco/trinity_out_dir/both.fa\n",
    "* Running CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/filter_iworm_by_min_length_or_cov.pl /workspace/transreco/trinity_out_dir/inchworm.K25.L25.DS.fa 100 10 > /workspace/transreco/trinity_out_dir/chrysalis/inchworm.K25.L25.DS.fa.min100\n",
    "* Running CMD: bowtie2-build --threads 2 -o 3 /workspace/transreco/trinity_out_dir/chrysalis/inchworm.K25.L25.DS.fa.min100 /workspace/transreco/trinity_out_dir/chrysalis/inchworm.K25.L25.DS.fa.min100 1>/dev/null\n",
    "* Running CMD: bash -c \" set -o pipefail;bowtie2 --local -k 2 --threads 2 -f --score-min G,46,0 -x /workspace/transreco/trinity_out_dir/chrysalis/inchworm.K25.L25.DS.fa.min100 /workspace/transreco/trinity_out_dir/both.fa  | samtools view -@ 2 -F4 -Sb - | samtools sort -m 268435456 -@ 2 -no - - > /workspace/transreco/trinity_out_dir/chrysalis/iworm.bowtie.nameSorted.bam\" \n",
    "* Running CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/scaffold_iworm_contigs.pl /workspace/transreco/trinity_out_dir/chrysalis/iworm.bowtie.nameSorted.bam /workspace/transreco/trinity_out_dir/inchworm.K25.L25.DS.fa > /workspace/transreco/trinity_out_dir/chrysalis/iworm_scaffolds.txt\n",
    "* Running CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Chrysalis/GraphFromFasta -i /workspace/transreco/trinity_out_dir/inchworm.K25.L25.DS.fa -r /workspace/transreco/trinity_out_dir/both.fa -min_contig_length 200 -min_glue 2 -glue_factor 0.05 -min_iso_ratio 0.05 -t 2 -k 24 -kk 48  -scaffolding /workspace/transreco/trinity_out_dir/chrysalis/iworm_scaffolds.txt  > /workspace/transreco/trinity_out_dir/chrysalis/iworm_cluster_welds_graph.txt\n",
    "* Running CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Chrysalis/BubbleUpClustering -i /workspace/transreco/trinity_out_dir/inchworm.K25.L25.DS.fa  -weld_graph /workspace/transreco/trinity_out_dir/chrysalis/iworm_cluster_welds_graph.txt -min_contig_length 200  > /workspace/transreco/trinity_out_dir/chrysalis/GraphFromIwormFasta.out\n",
    "* Running CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Chrysalis/CreateIwormFastaBundle -i /workspace/transreco/trinity_out_dir/chrysalis/GraphFromIwormFasta.out -o /workspace/transreco/trinity_out_dir/chrysalis/bundled_iworm_contigs.fasta -min 200\n",
    "* Running CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Chrysalis/ReadsToTranscripts -i /workspace/transreco/trinity_out_dir/both.fa -f /workspace/transreco/trinity_out_dir/chrysalis/bundled_iworm_contigs.fasta -o /workspace/transreco/trinity_out_dir/chrysalis/readsToComponents.out -t 2 -max_mem_reads 50000000 \n",
    "* Running CMD: /usr/bin/sort --parallel=2 -T . -S 1G -k 1,1n /workspace/transreco/trinity_out_dir/chrysalis/readsToComponents.out > /workspace/transreco/trinity_out_dir/chrysalis/readsToComponents.out.sort\n",
    "Wednesday, November 14, 2018: 15:52:07\tCMD: mkdir -p /workspace/transreco/trinity_out_dir/read_partitions/Fb_0/CBin_0\n",
    "Wednesday, November 14, 2018: 15:52:07\tCMD: mkdir -p /workspace/transreco/trinity_out_dir/read_partitions/Fb_0/CBin_1\n",
    "Wednesday, November 14, 2018: 15:52:07\tCMD: touch /workspace/transreco/trinity_out_dir/partitioned_reads.files.list.ok\n",
    "Wednesday, November 14, 2018: 15:52:07\tCMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/write_partitioned_trinity_cmds.pl --reads_list_file /workspace/transreco/trinity_out_dir/partitioned_reads.files.list --CPU 1 --max_memory 1G  --run_as_paired  --seqType fa --trinity_complete --full_cleanup  > recursive_trinity.cmds\n",
    "Wednesday, November 14, 2018: 15:52:07\tCMD: touch recursive_trinity.cmds.ok\n",
    "Wednesday, November 14, 2018: 15:52:07\tCMD: touch recursive_trinity.cmds.ok\n",
    "\n",
    "\n",
    "--------------------------------------------------------------------------------\n",
    "------------ Trinity Phase 2: Assembling Clusters of Reads ---------------------\n",
    "--------------------------------------------------------------------------------\n",
    "\n",
    "Wednesday, November 14, 2018: 15:52:07\tCMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/trinity-plugins/BIN/ParaFly -c recursive_trinity.cmds -CPU 2 -v -shuffle \n",
    "Number of Commands: 168\n",
    "succeeded(168)   100% completed.        \n",
    "\n",
    "All commands completed successfully. :-)\n",
    "\n",
    "\n",
    "\n",
    "** Harvesting all assembled transcripts into a single multi-fasta file...\n",
    "\n",
    "Wednesday, November 14, 2018: 15:54:59\tCMD: find /workspace/transreco/trinity_out_dir/read_partitions/ -name '*inity.fasta'  | /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/partitioned_trinity_aggregator.pl TRINITY_DN > /workspace/transreco/trinity_out_dir/Trinity.fasta.tmp\n",
    "-relocating /workspace/transreco/trinity_out_dir/Trinity.fasta.tmp to /workspace/transreco/trinity_out_dir/Trinity.fasta\n",
    "Wednesday, November 14, 2018: 15:54:59\tCMD: mv /workspace/transreco/trinity_out_dir/Trinity.fasta.tmp /workspace/transreco/trinity_out_dir/Trinity.fasta\n",
    "\n",
    "\n",
    "###################################################################\n",
    "Trinity assemblies are written to /workspace/transreco/trinity_out_dir/Trinity.fasta\n",
    "###################################################################\n",
    "\n",
    "\n",
    "Wednesday, November 14, 2018: 15:54:59\tCMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/get_Trinity_gene_to_trans_map.pl /workspace/transreco/trinity_out_dir/Trinity.fasta > /workspace/transreco/trinity_out_dir/Trinity.fasta.gene_trans_map\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  lhtr\n",
    "total 36M\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 15:24 data\n",
    "-rw-r--r-- 1 ubuntu ubuntu 777K Nov 14 15:24 minigenome.fa\n",
    "-rw-r--r-- 1 ubuntu ubuntu  354 Nov 14 15:24 samples.txt\n",
    "-rw-r--r-- 1 ubuntu ubuntu 1.2M Nov 14 15:24 minigenome.gtf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   27 Nov 14 15:25 minigenome.fa.fai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  14K Nov 14 15:33 minigenome.gtf.ss\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13K Nov 14 15:33 minigenome.gtf.exons\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 175K Nov 14 15:33 minigenome.fa.4.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  944 Nov 14 15:33 minigenome.fa.3.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 3.8K Nov 14 15:33 minigenome.fa.8.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  19K Nov 14 15:33 minigenome.fa.7.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 176K Nov 14 15:33 minigenome.fa.2.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.5M Nov 14 15:33 minigenome.fa.1.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 179K Nov 14 15:33 minigenome.fa.6.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 509K Nov 14 15:33 minigenome.fa.5.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:40 sB_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.7K Nov 14 15:41 sA_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sB_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.4K Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.8K Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sA_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 106K Nov 14 15:45 sA_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105K Nov 14 15:45 sA_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sB_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  99K Nov 14 15:45 sB_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 103K Nov 14 15:45 sB_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 141K Nov 14 15:45 stringtie.final.GTF\n",
    "drwxrwxr-x 5 ubuntu ubuntu 4.0K Nov 14 15:54 trinity_out_dir\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Trinity stats**\n",
    "```\n",
    "[nathalia@aws] $  $TRINITY_HOME/util/TrinityStats.pl trinity_out_dir/Trinity.fasta\n",
    "\n",
    "\n",
    "################################\n",
    "## Counts of transcripts, etc.\n",
    "################################\n",
    "Total trinity 'genes':\t189\n",
    "Total trinity transcripts:\t309\n",
    "Percent GC: 48.07\n",
    "\n",
    "########################################\n",
    "Stats based on ALL transcript contigs:\n",
    "########################################\n",
    "\n",
    "\tContig N10: 6202\n",
    "\tContig N20: 5321\n",
    "\tContig N30: 4245\n",
    "\tContig N40: 3770\n",
    "\tContig N50: 3242\n",
    "\n",
    "\tMedian contig length: 1259\n",
    "\tAverage contig: 1964.40\n",
    "\tTotal assembled bases: 607000\n",
    "\n",
    "\n",
    "#####################################################\n",
    "## Stats based on ONLY LONGEST ISOFORM per 'GENE':\n",
    "#####################################################\n",
    "\n",
    "\tContig N10: 6408\n",
    "\tContig N20: 5223\n",
    "\tContig N30: 4407\n",
    "\tContig N40: 3880\n",
    "\tContig N50: 2774\n",
    "\n",
    "\tMedian contig length: 923\n",
    "\tAverage contig: 1597.84\n",
    "\tTotal assembled bases: 301992\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Transcript-to-genome alignment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  gmap_build -D . -T . -d minigenome.fa.gmap -k 13 minigenome.fa\n",
    "Note: the -T flag is no longer necessary, since gmap_build now builds directly in the destination directory\n",
    "Sorting chromosomes in chrom order.  To turn off or sort other ways, use the -s flag.\n",
    "Creating files in directory ./minigenome.fa.gmap\n",
    "Running \"/usr/lib/gmap/fa_coords\"     -o \"./minigenome.fa.gmap.coords\" -f \"./minigenome.fa.gmap.sources\"\n",
    "Opening file minigenome.fa\n",
    "  Processed short contigs (<1000000 nt): .\n",
    "============================================================\n",
    "Contig mapping information has been written to file ./minigenome.fa.gmap.coords.\n",
    "You should look at this file, and edit it if necessary\n",
    "If everything is okay, you should proceed by running\n",
    "    make gmapdb\n",
    "============================================================\n",
    "Running \"/usr/lib/gmap/gmap_process\"  -c \"./minigenome.fa.gmap.coords\" -f \"./minigenome.fa.gmap.sources\" | \"/usr/lib/gmap/gmapindex\"  -d minigenome.fa.gmap -D \"./minigenome.fa.gmap\" -A \n",
    "Reading coordinates from file ./minigenome.fa.gmap.coords\n",
    "Logging contig minigenome at minigenome:1..781778 in genome minigenome.fa.gmap\n",
    " => primary (linear) chromosome\n",
    "Total genomic length = 781778 bp\n",
    "Have a total of 1 chromosomes\n",
    "Writing chromosome file ./minigenome.fa.gmap/minigenome.fa.gmap.chromosome\n",
    "Chromosome minigenome has universal coordinates 1..781778\n",
    "Writing chromosome IIT file ./minigenome.fa.gmap/minigenome.fa.gmap.chromosome.iit\n",
    "Writing IIT file header information...coordinates require 4 bytes each...done\n",
    "Processing null division/chromosome...sorting...writing...done (1 intervals)\n",
    "Writing IIT file footer information...done\n",
    "Writing IIT file header information...coordinates require 4 bytes each...done\n",
    "Processing null division/chromosome...sorting...writing...done (1 intervals)\n",
    "Writing IIT file footer information...done\n",
    "No alternate scaffolds observed\n",
    "Running \"/usr/lib/gmap/gmap_process\"  -c \"./minigenome.fa.gmap.coords\" -f \"./minigenome.fa.gmap.sources\" | \"/usr/lib/gmap/gmapindex\"  -d minigenome.fa.gmap -F \"./minigenome.fa.gmap\" -D \"./minigenome.fa.gmap\" -G\n",
    "Genome length is 781778 nt\n",
    "Trying to allocate 73293*4 bytes of memory...succeeded.  Building genome in memory.\n",
    "Reading coordinates from file ./minigenome.fa.gmap.coords\n",
    "Writing contig minigenome to universal coordinates 1..781778\n",
    "A total of 0 non-ACGTNX characters were seen in the genome.\n",
    "Running cat \"./minigenome.fa.gmap/minigenome.fa.gmap.genomecomp\" | \"/usr/lib/gmap/gmapindex\" -d minigenome.fa.gmap -U > \"./minigenome.fa.gmap/minigenome.fa.gmap.genomebits128\"\n",
    "Running cat \"./minigenome.fa.gmap/minigenome.fa.gmap.genomecomp\" | \"/usr/lib/gmap/gmapindex\" -k 13 -q 3  -d minigenome.fa.gmap -F \"./minigenome.fa.gmap\" -D \"./minigenome.fa.gmap\" -N\n",
    "Counting positions in genome minigenome.fa.gmap (13 bp every 3 bp), position 0\n",
    "Number of offsets: 237366 => pages file not required\n",
    "Running \"/usr/lib/gmap/gmapindex\" -k 13 -q 3  -d minigenome.fa.gmap -F \"./minigenome.fa.gmap\" -D \"./minigenome.fa.gmap\" -O  \"./minigenome.fa.gmap/minigenome.fa.gmap.genomecomp\"\n",
    "Offset compression types: bitpack64\n",
    "Allocating 1048576*1 bytes for packsizes\n",
    "Allocating 1048576*8 bytes for bitpacks\n",
    "Indexing offsets of oligomers in genome minigenome.fa.gmap (13 bp every 3 bp), position 0\n",
    "Writing 67108865 offsets compressed via bitpack64...done\n",
    "Running \"/usr/lib/gmap/gmapindex\" -k 13 -q 3  -d minigenome.fa.gmap -F \"./minigenome.fa.gmap\" -D \"./minigenome.fa.gmap\" -P \"./minigenome.fa.gmap/minigenome.fa.gmap.genomecomp\"\n",
    "Looking for index files in directory ./minigenome.fa.gmap\n",
    "  Pointers file is minigenome.fa.gmap.ref133offsets64meta\n",
    "  Offsets file is minigenome.fa.gmap.ref133offsets64strm\n",
    "  Positions file is minigenome.fa.gmap.ref133positions\n",
    "Expanding offsetsstrm into counters...done\n",
    "Allocating 3103296 bytes for counterstrm\n",
    "Trying to allocate 237366*4 bytes of memory for positions...succeeded.  Building positions in memory.\n",
    "Indexing positions of oligomers in genome minigenome.fa.gmap (13 bp every 3 bp), position 0\n",
    "Writing 237366 genomic positions to file ./minigenome.fa.gmap/minigenome.fa.gmap.ref133positions ...\n",
    "done\n",
    "Running \"/usr/lib/gmap/gmapindex\" -d minigenome.fa.gmap -F \"./minigenome.fa.gmap\" -D \"./minigenome.fa.gmap\" -S\n",
    "Genome length is 781778\n",
    "Building suffix array\n",
    "SACA_K called with n = 781779, K = 5, level 0\n",
    "SACA_K called with n = 200750, K = 0, level 1\n",
    "SACA_K called with n = 63708, K = 0, level 2\n",
    "SACA_K called with n = 20596, K = 0, level 3\n",
    "SACA_K called with n = 6738, K = 0, level 4\n",
    "SACA_K called with n = 2155, K = 0, level 5\n",
    "For indexsize 12, occupied 664233/16777216\n",
    "Optimal indexsize = 12\n",
    "Running \"/usr/lib/gmap/gmapindex\" -d minigenome.fa.gmap -F \"./minigenome.fa.gmap\" -D \"./minigenome.fa.gmap\" -L\n",
    "Building LCP array\n",
    "Writing temporary file for rank...done\n",
    "Writing temporary file for permuted sarray...done\n",
    "Found 53776 exceptions\n",
    "Byte-coding: 728003 values < 255, 53776 exceptions >= 255 (6.9%)\n",
    "Building DC array\n",
    "Building child array\n",
    "Byte-coding: 779508 values < 255, 2271 exceptions >= 255 (0.3%)\n",
    "Writing file ./minigenome.fa.gmap/minigenome.fa.gmap.salcpchilddcdone\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  lhtr\n",
    "total 36M\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 15:24 data\n",
    "-rw-r--r-- 1 ubuntu ubuntu 777K Nov 14 15:24 minigenome.fa\n",
    "-rw-r--r-- 1 ubuntu ubuntu  354 Nov 14 15:24 samples.txt\n",
    "-rw-r--r-- 1 ubuntu ubuntu 1.2M Nov 14 15:24 minigenome.gtf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   27 Nov 14 15:25 minigenome.fa.fai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  14K Nov 14 15:33 minigenome.gtf.ss\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13K Nov 14 15:33 minigenome.gtf.exons\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 175K Nov 14 15:33 minigenome.fa.4.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  944 Nov 14 15:33 minigenome.fa.3.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 3.8K Nov 14 15:33 minigenome.fa.8.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  19K Nov 14 15:33 minigenome.fa.7.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 176K Nov 14 15:33 minigenome.fa.2.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.5M Nov 14 15:33 minigenome.fa.1.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 179K Nov 14 15:33 minigenome.fa.6.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 509K Nov 14 15:33 minigenome.fa.5.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:40 sB_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.7K Nov 14 15:41 sA_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sB_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.4K Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.8K Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sA_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 106K Nov 14 15:45 sA_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105K Nov 14 15:45 sA_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sB_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  99K Nov 14 15:45 sB_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 103K Nov 14 15:45 sB_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 141K Nov 14 15:45 stringtie.final.GTF\n",
    "drwxrwxr-x 5 ubuntu ubuntu 4.0K Nov 14 15:54 trinity_out_dir\n",
    "drwxrwxr-x 3 ubuntu ubuntu 4.0K Nov 14 15:57 minigenome.fa.gmap\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  gmap -D . -d minigenome.fa.gmap trinity_out_dir/Trinity.fasta \\\n",
    ">       -f samse -n 0 -x 50 -t 2 -B 5 \\\n",
    ">       | samtools view -Sb | samtools sort -o Trinity.gmap.bam\n",
    "Note: gmap.avx512 does not exist.  For faster speed, may want to compile package on an AVX512 machine\n",
    "Note: gmap.avx2 does not exist.  For faster speed, may want to compile package on an AVX2 machine\n",
    "GMAP version 2017-11-15 called with args: gmap.sse42 -D . -d minigenome.fa.gmap trinity_out_dir/Trinity.fasta -f samse -n 0 -x 50 -t 2 -B 5\n",
    "Note: Batch mode 5 is now the same as batch mode 4.\n",
    "Expansion of offsets is now controlled separately by --expand-offsets (default=1).\n",
    "Checking compiler assumptions for SSE2: 6B8B4567 327B23C6 xor=59F066A1\n",
    "Checking compiler assumptions for SSE4.1: -103 -58 max=198 => compiler zero extends\n",
    "Checking compiler options for SSE4.2: 6B8B4567 __builtin_clz=1 __builtin_ctz=0 _mm_popcnt_u32=17 __builtin_popcount=17 \n",
    "Finished checking compiler assumptions\n",
    "Allocating memory for compressed genome (oligos)...Attached new memory for ./minigenome.fa.gmap/minigenome.fa.gmap.genomecomp...done (293,172 bytes, 0.00 sec)\n",
    "Allocating memory for compressed genome (bits)...Attached new memory for ./minigenome.fa.gmap/minigenome.fa.gmap.genomebits128...done (293,184 bytes, 0.00 sec)\n",
    "Looking for index files in directory ./minigenome.fa.gmap\n",
    "  Pointers file is minigenome.fa.gmap.ref133offsets64meta\n",
    "  Offsets file is minigenome.fa.gmap.ref133offsets64strm\n",
    "  Positions file is minigenome.fa.gmap.ref133positions\n",
    "Offsets compression type: bitpack64\n",
    "Allocating memory for ref offset pointers, kmer 13, interval 3...Attached new memory for ./minigenome.fa.gmap/minigenome.fa.gmap.ref133offsets64meta...done (8,388,624 bytes, 0.00 sec)\n",
    "Allocating memory for ref offsets, kmer 13, interval 3...Attached new memory for ./minigenome.fa.gmap/minigenome.fa.gmap.ref133offsets64strm...done (3,103,296 bytes, 0.00 sec)\n",
    "Allocating memory for ref positions, kmer 13, interval 3...Attached new memory for ./minigenome.fa.gmap/minigenome.fa.gmap.ref133positions...done (949,464 bytes, 0.00 sec)\n",
    "Starting alignment\n",
    "No paths found for TRINITY_DN33_c0_g2_i1\n",
    "Processed 309 queries in 2.62 seconds (117.94 queries/sec)\n",
    "Removed existing memory for shmid 131076\n",
    "Removed existing memory for shmid 98307\n",
    "Removed existing memory for shmid 65538\n",
    "Removed existing memory for shmid 0\n",
    "Removed existing memory for shmid 32769\n",
    "\n",
    "[nathalia@aws] $  samtools index Trinity.gmap.bam\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 36M\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 15:24 data\n",
    "-rw-r--r-- 1 ubuntu ubuntu 777K Nov 14 15:24 minigenome.fa\n",
    "-rw-r--r-- 1 ubuntu ubuntu  354 Nov 14 15:24 samples.txt\n",
    "-rw-r--r-- 1 ubuntu ubuntu 1.2M Nov 14 15:24 minigenome.gtf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   27 Nov 14 15:25 minigenome.fa.fai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  14K Nov 14 15:33 minigenome.gtf.ss\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13K Nov 14 15:33 minigenome.gtf.exons\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 175K Nov 14 15:33 minigenome.fa.4.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  944 Nov 14 15:33 minigenome.fa.3.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 3.8K Nov 14 15:33 minigenome.fa.8.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  19K Nov 14 15:33 minigenome.fa.7.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 176K Nov 14 15:33 minigenome.fa.2.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.5M Nov 14 15:33 minigenome.fa.1.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 179K Nov 14 15:33 minigenome.fa.6.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 509K Nov 14 15:33 minigenome.fa.5.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:40 sB_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.7K Nov 14 15:41 sA_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sB_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.4K Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.8K Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sA_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 106K Nov 14 15:45 sA_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105K Nov 14 15:45 sA_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sB_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  99K Nov 14 15:45 sB_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 103K Nov 14 15:45 sB_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 141K Nov 14 15:45 stringtie.final.GTF\n",
    "drwxrwxr-x 5 ubuntu ubuntu 4.0K Nov 14 15:54 trinity_out_dir\n",
    "drwxrwxr-x 3 ubuntu ubuntu 4.0K Nov 14 15:57 minigenome.fa.gmap\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 134K Nov 14 15:58 Trinity.gmap.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  472 Nov 14 15:58 Trinity.gmap.bam.bai\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load into IGV\n",
    "\n",
    "http://ec2-34-221-102-104.us-west-2.compute.amazonaws.com/workspace/transreco/Trinity.gmap.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![gmap](gmap.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Estimating abundance\n",
    "\n",
    "https://github.com/trinityrnaseq/trinityrnaseq/wiki/Trinity-Transcript-Quantification#estimating-transcript-abundance"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  mkdir abundance-kallisto\n",
    "[nathalia@aws] $  $TRINITY_HOME/util/align_and_estimate_abundance.pl  --transcripts trinity_out_dir/Trinity.fasta --seqType fq --samples_file samples.txt --est_method kallisto --output_dir abundance-kallisto --trinity_mode \n",
    "Error, no kallisto index file: /workspace/transreco/trinity_out_dir/Trinity.fasta.kallisto_idx, and --prep_reference not set.  Re-run with --prep_reference at /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/align_and_estimate_abundance.pl line 838.\n",
    "\tmain::run_kallisto(HASH(0x5636714f45f8), HASH(0x5636712c49b8), HASH(0x5636713eaa20), HASH(0x5636713b8c80), HASH(0x563671579d00), HASH(0x563671579f10)) called at /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/align_and_estimate_abundance.pl line 455\n",
    "\tmain::run_alignment_FREE_estimation(HASH(0x5636714f45f8), HASH(0x5636712c49b8), HASH(0x5636713eaa20), HASH(0x5636713b8c80), HASH(0x563671579d00), HASH(0x563671579f10)) called at /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/align_and_estimate_abundance.pl line 441\n",
    "\n",
    "# need to add --prep_reference\n",
    "\n",
    "[nathalia@aws] $  $TRINITY_HOME/util/align_and_estimate_abundance.pl  --transcripts trinity_out_dir/Trinity.fasta --seqType fq --samples_file samples.txt --est_method kallisto --output_dir abundance-kallisto --trinity_mode --prep_reference\n",
    "CMD: kallisto index -i /workspace/transreco/trinity_out_dir/Trinity.fasta.kallisto_idx /workspace/transreco/trinity_out_dir/Trinity.fasta\n",
    "\n",
    "[build] loading fasta file /workspace/transreco/trinity_out_dir/Trinity.fasta\n",
    "[build] k-mer length: 31\n",
    "[build] counting k-mers ... done.\n",
    "[build] building target de Bruijn graph ...  done \n",
    "[build] creating equivalence classes ...  done\n",
    "[build] target de Bruijn graph has 483 contigs and contains 316749 k-mers \n",
    "\n",
    "CMD: kallisto quant -i /workspace/transreco/trinity_out_dir/Trinity.fasta.kallisto_idx  -o sA_rep1 <(gunzip -c /workspace/transreco/data/sA_rep1_1.fastq.gz) <(gunzip -c /workspace/transreco/data/sA_rep1_2.fastq.gz)\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 309\n",
    "[index] number of k-mers: 316,749\n",
    "[index] number of equivalence classes: 472\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /dev/fd/63\n",
    "                             /dev/fd/62\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 25,000 reads, 24,914 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 218.862\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 459 rounds\n",
    "\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/kallisto_trans_to_gene_results.pl sA_rep1/abundance.tsv /workspace/transreco/trinity_out_dir/Trinity.fasta.gene_trans_map > sA_rep1/abundance.tsv.genes\n",
    "CMD: kallisto quant -i /workspace/transreco/trinity_out_dir/Trinity.fasta.kallisto_idx  -o sA_rep2 <(gunzip -c /workspace/transreco/data/sA_rep2_1.fastq.gz) <(gunzip -c /workspace/transreco/data/sA_rep2_2.fastq.gz)\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 309\n",
    "[index] number of k-mers: 316,749\n",
    "[index] number of equivalence classes: 472\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /dev/fd/63\n",
    "                             /dev/fd/62\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 25,000 reads, 24,944 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 218.873\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 443 rounds\n",
    "\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/kallisto_trans_to_gene_results.pl sA_rep2/abundance.tsv /workspace/transreco/trinity_out_dir/Trinity.fasta.gene_trans_map > sA_rep2/abundance.tsv.genes\n",
    "CMD: kallisto quant -i /workspace/transreco/trinity_out_dir/Trinity.fasta.kallisto_idx  -o sA_rep3 <(gunzip -c /workspace/transreco/data/sA_rep3_1.fastq.gz) <(gunzip -c /workspace/transreco/data/sA_rep3_2.fastq.gz)\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 309\n",
    "[index] number of k-mers: 316,749\n",
    "[index] number of equivalence classes: 472\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /dev/fd/63\n",
    "                             /dev/fd/62\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 25,000 reads, 24,933 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 217.832\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 611 rounds\n",
    "\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/kallisto_trans_to_gene_results.pl sA_rep3/abundance.tsv /workspace/transreco/trinity_out_dir/Trinity.fasta.gene_trans_map > sA_rep3/abundance.tsv.genes\n",
    "CMD: kallisto quant -i /workspace/transreco/trinity_out_dir/Trinity.fasta.kallisto_idx  -o sB_rep1 <(gunzip -c /workspace/transreco/data/sB_rep1_1.fastq.gz) <(gunzip -c /workspace/transreco/data/sB_rep1_2.fastq.gz)\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 309\n",
    "[index] number of k-mers: 316,749\n",
    "[index] number of equivalence classes: 472\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /dev/fd/63\n",
    "                             /dev/fd/62\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 25,000 reads, 24,957 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 217.786\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 692 rounds\n",
    "\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/kallisto_trans_to_gene_results.pl sB_rep1/abundance.tsv /workspace/transreco/trinity_out_dir/Trinity.fasta.gene_trans_map > sB_rep1/abundance.tsv.genes\n",
    "CMD: kallisto quant -i /workspace/transreco/trinity_out_dir/Trinity.fasta.kallisto_idx  -o sB_rep2 <(gunzip -c /workspace/transreco/data/sB_rep2_1.fastq.gz) <(gunzip -c /workspace/transreco/data/sB_rep2_2.fastq.gz)\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 309\n",
    "[index] number of k-mers: 316,749\n",
    "[index] number of equivalence classes: 472\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /dev/fd/63\n",
    "                             /dev/fd/62\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 25,000 reads, 24,965 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 218.906\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 449 rounds\n",
    "\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/kallisto_trans_to_gene_results.pl sB_rep2/abundance.tsv /workspace/transreco/trinity_out_dir/Trinity.fasta.gene_trans_map > sB_rep2/abundance.tsv.genes\n",
    "CMD: kallisto quant -i /workspace/transreco/trinity_out_dir/Trinity.fasta.kallisto_idx  -o sB_rep3 <(gunzip -c /workspace/transreco/data/sB_rep3_1.fastq.gz) <(gunzip -c /workspace/transreco/data/sB_rep3_2.fastq.gz)\n",
    "\n",
    "[quant] fragment length distribution will be estimated from the data\n",
    "[index] k-mer length: 31\n",
    "[index] number of targets: 309\n",
    "[index] number of k-mers: 316,749\n",
    "[index] number of equivalence classes: 472\n",
    "[quant] running in paired-end mode\n",
    "[quant] will process pair 1: /dev/fd/63\n",
    "                             /dev/fd/62\n",
    "[quant] finding pseudoalignments for the reads ... done\n",
    "[quant] processed 25,000 reads, 24,935 reads pseudoaligned\n",
    "[quant] estimated average fragment length: 216.619\n",
    "[   em] quantifying the abundances ... done\n",
    "[   em] the Expectation-Maximization algorithm ran for 834 rounds\n",
    "\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/kallisto_trans_to_gene_results.pl sB_rep3/abundance.tsv /workspace/transreco/trinity_out_dir/Trinity.fasta.gene_trans_map > sB_rep3/abundance.tsv.genes\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  lhtr\n",
    "total 36M\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 15:24 data\n",
    "-rw-r--r-- 1 ubuntu ubuntu 777K Nov 14 15:24 minigenome.fa\n",
    "-rw-r--r-- 1 ubuntu ubuntu  354 Nov 14 15:24 samples.txt\n",
    "-rw-r--r-- 1 ubuntu ubuntu 1.2M Nov 14 15:24 minigenome.gtf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   27 Nov 14 15:25 minigenome.fa.fai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  14K Nov 14 15:33 minigenome.gtf.ss\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13K Nov 14 15:33 minigenome.gtf.exons\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 175K Nov 14 15:33 minigenome.fa.4.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  944 Nov 14 15:33 minigenome.fa.3.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 3.8K Nov 14 15:33 minigenome.fa.8.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  19K Nov 14 15:33 minigenome.fa.7.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 176K Nov 14 15:33 minigenome.fa.2.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.5M Nov 14 15:33 minigenome.fa.1.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 179K Nov 14 15:33 minigenome.fa.6.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 509K Nov 14 15:33 minigenome.fa.5.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:40 sB_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.7K Nov 14 15:41 sA_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sB_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.4K Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.8K Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sA_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 106K Nov 14 15:45 sA_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105K Nov 14 15:45 sA_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sB_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  99K Nov 14 15:45 sB_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 103K Nov 14 15:45 sB_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 141K Nov 14 15:45 stringtie.final.GTF\n",
    "drwxrwxr-x 3 ubuntu ubuntu 4.0K Nov 14 15:57 minigenome.fa.gmap\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 134K Nov 14 15:58 Trinity.gmap.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  472 Nov 14 15:58 Trinity.gmap.bam.bai\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:09 abundance-kallisto\n",
    "drwxrwxr-x 5 ubuntu ubuntu 4.0K Nov 14 16:11 trinity_out_dir\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sA_rep1\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sA_rep2\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sA_rep3\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sB_rep1\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sB_rep2\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sB_rep3\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Build transcript and gene expression matrices"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  $TRINITY_HOME/util/abundance_estimates_to_matrix.pl --est_method kallisto --gene_trans_map trinity_out_dir/Trinity.fasta.gene_trans_map --out_prefix kallisto --name_sample_by_basedir sA_rep1/abundance.tsv sA_rep2/abundance.tsv sA_rep3/abundance.tsv sB_rep1/abundance.tsv sB_rep2/abundance.tsv sB_rep3/abundance.tsv\n",
    "-reading file: sA_rep1/abundance.tsv\n",
    "-reading file: sA_rep2/abundance.tsv\n",
    "-reading file: sA_rep3/abundance.tsv\n",
    "-reading file: sB_rep1/abundance.tsv\n",
    "-reading file: sB_rep2/abundance.tsv\n",
    "-reading file: sB_rep3/abundance.tsv\n",
    "\n",
    "\n",
    "* Outputting combined matrix.\n",
    "\n",
    "/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/run_TMM_scale_matrix.pl --matrix kallisto.isoform.TPM.not_cross_norm > kallisto.isoform.TMM.EXPR.matrixCMD: R --vanilla -q < __tmp_runTMM.R 1>&2 \n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> rnaseqMatrix = read.table(\"kallisto.isoform.TPM.not_cross_norm\", header=T, row.names=1, com='', check.names=F)\n",
    "> rnaseqMatrix = as.matrix(rnaseqMatrix)\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=factor(colnames(rnaseqMatrix)))\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> exp_study$samples$eff.lib.size = exp_study$samples$lib.size * exp_study$samples$norm.factors\n",
    "> write.table(exp_study$samples, file=\"kallisto.isoform.TPM.not_cross_norm.TMM_info.txt\", quote=F, sep=\"\\t\", row.names=F)\n",
    "> \n",
    "/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/util/support_scripts/run_TMM_scale_matrix.pl --matrix kallisto.gene.TPM.not_cross_norm > kallisto.gene.TMM.EXPR.matrixCMD: R --vanilla -q < __tmp_runTMM.R 1>&2 \n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> rnaseqMatrix = read.table(\"kallisto.gene.TPM.not_cross_norm\", header=T, row.names=1, com='', check.names=F)\n",
    "> rnaseqMatrix = as.matrix(rnaseqMatrix)\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=factor(colnames(rnaseqMatrix)))\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> exp_study$samples$eff.lib.size = exp_study$samples$lib.size * exp_study$samples$norm.factors\n",
    "> write.table(exp_study$samples, file=\"kallisto.gene.TPM.not_cross_norm.TMM_info.txt\", quote=F, sep=\"\\t\", row.names=F)\n",
    "> \n",
    "\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 37M\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 15:24 data\n",
    "-rw-r--r-- 1 ubuntu ubuntu 777K Nov 14 15:24 minigenome.fa\n",
    "-rw-r--r-- 1 ubuntu ubuntu  354 Nov 14 15:24 samples.txt\n",
    "-rw-r--r-- 1 ubuntu ubuntu 1.2M Nov 14 15:24 minigenome.gtf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   27 Nov 14 15:25 minigenome.fa.fai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  14K Nov 14 15:33 minigenome.gtf.ss\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13K Nov 14 15:33 minigenome.gtf.exons\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 175K Nov 14 15:33 minigenome.fa.4.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  944 Nov 14 15:33 minigenome.fa.3.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 3.8K Nov 14 15:33 minigenome.fa.8.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  19K Nov 14 15:33 minigenome.fa.7.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 176K Nov 14 15:33 minigenome.fa.2.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.5M Nov 14 15:33 minigenome.fa.1.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 179K Nov 14 15:33 minigenome.fa.6.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 509K Nov 14 15:33 minigenome.fa.5.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:40 sB_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.7K Nov 14 15:41 sA_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sB_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.4K Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.8K Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sA_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 106K Nov 14 15:45 sA_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105K Nov 14 15:45 sA_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sB_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  99K Nov 14 15:45 sB_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 103K Nov 14 15:45 sB_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 141K Nov 14 15:45 stringtie.final.GTF\n",
    "drwxrwxr-x 3 ubuntu ubuntu 4.0K Nov 14 15:57 minigenome.fa.gmap\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 134K Nov 14 15:58 Trinity.gmap.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  472 Nov 14 15:58 Trinity.gmap.bam.bai\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:09 abundance-kallisto\n",
    "drwxrwxr-x 5 ubuntu ubuntu 4.0K Nov 14 16:11 trinity_out_dir\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sA_rep1\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sA_rep2\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sA_rep3\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sB_rep1\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sB_rep2\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sB_rep3\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  17K Nov 14 16:22 kallisto.isoform.counts.matrix\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20K Nov 14 16:22 kallisto.isoform.TPM.not_cross_norm\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  11K Nov 14 16:22 kallisto.gene.counts.matrix\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13K Nov 14 16:22 kallisto.gene.TPM.not_cross_norm\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  335 Nov 14 16:22 kallisto.isoform.TPM.not_cross_norm.TMM_info.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  22K Nov 14 16:22 kallisto.isoform.TMM.EXPR.matrix\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  526 Nov 14 16:22 __tmp_runTMM.R\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  337 Nov 14 16:22 kallisto.gene.TPM.not_cross_norm.TMM_info.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  14K Nov 14 16:22 kallisto.gene.TMM.EXPR.matrix\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Counting number of expressed transcripts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  $TRINITY_HOME/util/misc/count_matrix_features_given_MIN_TPM_threshold.pl kallisto.gene.TPM.not_cross_norm | tee genes_matrix.TPM.not_cross_norm.counts_by_min_TPM\n",
    "neg_min_tpm\tnum_features\n",
    "-102417\t1\n",
    "-98625\t2\n",
    "-79903\t3\n",
    "-70405\t4\n",
    "-64163\t5\n",
    "-46860\t6\n",
    "-44959\t7\n",
    "-41042\t8\n",
    "-39478\t9\n",
    "-39192\t10\n",
    "-35702\t11\n",
    "-31449\t12\n",
    "-30667\t13\n",
    "-29997\t14\n",
    "-29834\t15\n",
    "-28664\t16\n",
    "-20570\t17\n",
    "-19929\t18\n",
    "-19243\t19\n",
    "-19039\t20\n",
    "-19026\t21\n",
    "-16731\t22\n",
    "-16150\t23\n",
    "-14600\t24\n",
    "-14555\t25\n",
    "-14219\t26\n",
    "-14082\t27\n",
    "-13899\t28\n",
    "-13578\t29\n",
    "-13458\t30\n",
    "-13401\t31\n",
    "-13362\t32\n",
    "-13322\t33\n",
    "-12434\t34\n",
    "-11835\t35\n",
    "-11488\t36\n",
    "-11332\t37\n",
    "-11271\t38\n",
    "-11081\t39\n",
    "-10841\t40\n",
    "-10780\t41\n",
    "-10185\t42\n",
    "-9573\t43\n",
    "-8900\t44\n",
    "-8494\t45\n",
    "-8358\t46\n",
    "-8246\t47\n",
    "-8128\t48\n",
    "-7864\t49\n",
    "-7856\t50\n",
    "-7723\t51\n",
    "-7647\t52\n",
    "-7323\t53\n",
    "-6707\t54\n",
    "-6595\t55\n",
    "-6506\t56\n",
    "-6465\t57\n",
    "-6266\t58\n",
    "-6013\t59\n",
    "-5981\t60\n",
    "-5553\t61\n",
    "-5491\t62\n",
    "-5344\t63\n",
    "-5330\t64\n",
    "-5314\t65\n",
    "-4741\t66\n",
    "-4593\t67\n",
    "-4282\t68\n",
    "-4278\t69\n",
    "-4117\t70\n",
    "-4096\t71\n",
    "-3998\t72\n",
    "-3881\t73\n",
    "-3875\t74\n",
    "-3804\t75\n",
    "-3740\t76\n",
    "-3666\t77\n",
    "-3443\t78\n",
    "-3309\t79\n",
    "-3108\t80\n",
    "-3076\t81\n",
    "-3036\t82\n",
    "-3027\t83\n",
    "-3015\t84\n",
    "-2868\t85\n",
    "-2862\t86\n",
    "-2824\t87\n",
    "-2818\t88\n",
    "-2749\t89\n",
    "-2742\t90\n",
    "-2699\t91\n",
    "-2694\t92\n",
    "-2654\t93\n",
    "-2617\t94\n",
    "-2552\t95\n",
    "-2387\t96\n",
    "-2370\t97\n",
    "-2367\t98\n",
    "-2355\t99\n",
    "-2345\t100\n",
    "-2325\t101\n",
    "-2285\t102\n",
    "-2268\t103\n",
    "-2215\t104\n",
    "-2170\t105\n",
    "-2088\t106\n",
    "-2065\t107\n",
    "-2056\t108\n",
    "-2015\t109\n",
    "-1988\t110\n",
    "-1986\t111\n",
    "-1963\t112\n",
    "-1901\t113\n",
    "-1879\t114\n",
    "-1878\t115\n",
    "-1837\t116\n",
    "-1803\t117\n",
    "-1801\t119\n",
    "-1796\t120\n",
    "-1750\t121\n",
    "-1732\t122\n",
    "-1686\t123\n",
    "-1680\t124\n",
    "-1665\t125\n",
    "-1638\t126\n",
    "-1631\t127\n",
    "-1617\t128\n",
    "-1608\t129\n",
    "-1593\t130\n",
    "-1556\t131\n",
    "-1498\t132\n",
    "-1485\t133\n",
    "-1459\t134\n",
    "-1413\t135\n",
    "-1398\t136\n",
    "-1395\t137\n",
    "-1394\t138\n",
    "-1392\t140\n",
    "-1390\t141\n",
    "-1369\t142\n",
    "-1342\t143\n",
    "-1318\t144\n",
    "-1287\t145\n",
    "-1220\t146\n",
    "-1150\t147\n",
    "-1144\t148\n",
    "-1143\t149\n",
    "-1132\t150\n",
    "-1109\t151\n",
    "-1106\t152\n",
    "-1090\t153\n",
    "-1026\t154\n",
    "-1017\t155\n",
    "-989\t156\n",
    "-960\t157\n",
    "-954\t158\n",
    "-945\t159\n",
    "-940\t160\n",
    "-919\t161\n",
    "-917\t162\n",
    "-908\t163\n",
    "-904\t164\n",
    "-862\t165\n",
    "-845\t166\n",
    "-830\t167\n",
    "-793\t168\n",
    "-783\t169\n",
    "-759\t170\n",
    "-750\t171\n",
    "-745\t172\n",
    "-736\t173\n",
    "-734\t174\n",
    "-719\t175\n",
    "-713\t176\n",
    "-682\t177\n",
    "-681\t178\n",
    "-657\t179\n",
    "-636\t180\n",
    "-623\t181\n",
    "-601\t182\n",
    "-591\t183\n",
    "-586\t184\n",
    "-560\t185\n",
    "-534\t186\n",
    "-503\t187\n",
    "-492\t188\n",
    "0\t189\n",
    "[nathalia@aws] $  lhtr\n",
    "total 37M\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 15:24 data\n",
    "-rw-r--r-- 1 ubuntu ubuntu 777K Nov 14 15:24 minigenome.fa\n",
    "-rw-r--r-- 1 ubuntu ubuntu  354 Nov 14 15:24 samples.txt\n",
    "-rw-r--r-- 1 ubuntu ubuntu 1.2M Nov 14 15:24 minigenome.gtf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   27 Nov 14 15:25 minigenome.fa.fai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  14K Nov 14 15:33 minigenome.gtf.ss\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13K Nov 14 15:33 minigenome.gtf.exons\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 175K Nov 14 15:33 minigenome.fa.4.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  944 Nov 14 15:33 minigenome.fa.3.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 3.8K Nov 14 15:33 minigenome.fa.8.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  19K Nov 14 15:33 minigenome.fa.7.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 176K Nov 14 15:33 minigenome.fa.2.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.5M Nov 14 15:33 minigenome.fa.1.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 179K Nov 14 15:33 minigenome.fa.6.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 509K Nov 14 15:33 minigenome.fa.5.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:40 sB_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.7K Nov 14 15:41 sA_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sB_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.4K Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.8K Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sA_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 106K Nov 14 15:45 sA_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105K Nov 14 15:45 sA_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sB_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  99K Nov 14 15:45 sB_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 103K Nov 14 15:45 sB_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 141K Nov 14 15:45 stringtie.final.GTF\n",
    "drwxrwxr-x 3 ubuntu ubuntu 4.0K Nov 14 15:57 minigenome.fa.gmap\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 134K Nov 14 15:58 Trinity.gmap.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  472 Nov 14 15:58 Trinity.gmap.bam.bai\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:09 abundance-kallisto\n",
    "drwxrwxr-x 5 ubuntu ubuntu 4.0K Nov 14 16:11 trinity_out_dir\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sA_rep1\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sA_rep2\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sA_rep3\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sB_rep1\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sB_rep2\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sB_rep3\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  17K Nov 14 16:22 kallisto.isoform.counts.matrix\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20K Nov 14 16:22 kallisto.isoform.TPM.not_cross_norm\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  11K Nov 14 16:22 kallisto.gene.counts.matrix\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13K Nov 14 16:22 kallisto.gene.TPM.not_cross_norm\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  335 Nov 14 16:22 kallisto.isoform.TPM.not_cross_norm.TMM_info.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  22K Nov 14 16:22 kallisto.isoform.TMM.EXPR.matrix\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  526 Nov 14 16:22 __tmp_runTMM.R\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  337 Nov 14 16:22 kallisto.gene.TPM.not_cross_norm.TMM_info.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  14K Nov 14 16:22 kallisto.gene.TMM.EXPR.matrix\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.8K Nov 14 16:25 genes_matrix.TPM.not_cross_norm.counts_by_min_TPM\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  $TRINITY_HOME/util/misc/count_matrix_features_given_MIN_TPM_threshold.pl kallisto.isoform.TPM.not_cross_norm | tee isoform_matrix.TPM.not_cross_norm.counts_by_min_TPM\n",
    "neg_min_tpm\tnum_features\n",
    "-102417\t1\n",
    "-98625\t2\n",
    "-61862\t3\n",
    "-56379\t4\n",
    "-49690\t5\n",
    "-47736\t6\n",
    "-41042\t7\n",
    "-36709\t8\n",
    "-36363\t9\n",
    "-35452\t10\n",
    "-35010\t11\n",
    "-31833\t12\n",
    "-31449\t13\n",
    "-31196\t14\n",
    "-28051\t15\n",
    "-25339\t16\n",
    "-23815\t17\n",
    "-22599\t18\n",
    "-22503\t19\n",
    "-20570\t20\n",
    "-19929\t21\n",
    "-19802\t22\n",
    "-19026\t23\n",
    "-16892\t24\n",
    "-16667\t25\n",
    "-16150\t26\n",
    "-15822\t27\n",
    "-15131\t28\n",
    "-14986\t29\n",
    "-14682\t30\n",
    "-14062\t31\n",
    "-13578\t32\n",
    "-13322\t33\n",
    "-13034\t34\n",
    "-12861\t35\n",
    "-12667\t36\n",
    "-12555\t37\n",
    "-12322\t38\n",
    "-12258\t39\n",
    "-11919\t41\n",
    "-11488\t42\n",
    "-11192\t43\n",
    "-11081\t44\n",
    "-10979\t45\n",
    "-10841\t46\n",
    "-10822\t47\n",
    "-10185\t48\n",
    "-9888\t49\n",
    "-9663\t50\n",
    "-9573\t51\n",
    "-9504\t52\n",
    "-9437\t53\n",
    "-9414\t54\n",
    "-9260\t55\n",
    "-9226\t56\n",
    "-9141\t57\n",
    "-9076\t58\n",
    "-8940\t59\n",
    "-8900\t60\n",
    "-8436\t61\n",
    "-8358\t62\n",
    "-8266\t63\n",
    "-8190\t64\n",
    "-8152\t65\n",
    "-7915\t66\n",
    "-7864\t67\n",
    "-7856\t68\n",
    "-7608\t69\n",
    "-7323\t70\n",
    "-7236\t71\n",
    "-7110\t72\n",
    "-6905\t73\n",
    "-6558\t74\n",
    "-6506\t75\n",
    "-6465\t76\n",
    "-6340\t77\n",
    "-6266\t78\n",
    "-6166\t79\n",
    "-6117\t80\n",
    "-6101\t81\n",
    "-5999\t82\n",
    "-5981\t83\n",
    "-5928\t84\n",
    "-5749\t85\n",
    "-5553\t86\n",
    "-5491\t87\n",
    "-5431\t88\n",
    "-5344\t89\n",
    "-5314\t90\n",
    "-5197\t91\n",
    "-4947\t92\n",
    "-4573\t93\n",
    "-4439\t94\n",
    "-4385\t95\n",
    "-4381\t96\n",
    "-4361\t97\n",
    "-4285\t98\n",
    "-4278\t99\n",
    "-4243\t100\n",
    "-4195\t101\n",
    "-4091\t102\n",
    "-3993\t103\n",
    "-3885\t104\n",
    "-3806\t105\n",
    "-3751\t106\n",
    "-3740\t107\n",
    "-3646\t108\n",
    "-3496\t109\n",
    "-3485\t110\n",
    "-3449\t111\n",
    "-3367\t112\n",
    "-3355\t113\n",
    "-3316\t114\n",
    "-3309\t115\n",
    "-3278\t116\n",
    "-3277\t117\n",
    "-3253\t118\n",
    "-3235\t119\n",
    "-3198\t120\n",
    "-3112\t121\n",
    "-3110\t122\n",
    "-3109\t123\n",
    "-3106\t124\n",
    "-3076\t125\n",
    "-3036\t126\n",
    "-3027\t127\n",
    "-3015\t128\n",
    "-3010\t129\n",
    "-2950\t130\n",
    "-2868\t131\n",
    "-2862\t132\n",
    "-2837\t133\n",
    "-2824\t134\n",
    "-2818\t135\n",
    "-2780\t136\n",
    "-2749\t137\n",
    "-2742\t138\n",
    "-2699\t139\n",
    "-2696\t140\n",
    "-2694\t141\n",
    "-2654\t142\n",
    "-2648\t143\n",
    "-2639\t144\n",
    "-2617\t145\n",
    "-2514\t146\n",
    "-2484\t147\n",
    "-2453\t148\n",
    "-2444\t149\n",
    "-2387\t150\n",
    "-2370\t151\n",
    "-2355\t152\n",
    "-2347\t153\n",
    "-2345\t154\n",
    "-2340\t155\n",
    "-2325\t156\n",
    "-2310\t157\n",
    "-2286\t158\n",
    "-2268\t159\n",
    "-2226\t160\n",
    "-2215\t161\n",
    "-2170\t162\n",
    "-2142\t163\n",
    "-2123\t164\n",
    "-2088\t165\n",
    "-2065\t167\n",
    "-2064\t168\n",
    "-2056\t169\n",
    "-2015\t170\n",
    "-1986\t171\n",
    "-1969\t172\n",
    "-1965\t173\n",
    "-1963\t175\n",
    "-1957\t176\n",
    "-1901\t177\n",
    "-1879\t178\n",
    "-1878\t179\n",
    "-1869\t180\n",
    "-1837\t181\n",
    "-1830\t182\n",
    "-1816\t183\n",
    "-1803\t184\n",
    "-1802\t185\n",
    "-1801\t187\n",
    "-1796\t188\n",
    "-1760\t189\n",
    "-1750\t190\n",
    "-1732\t192\n",
    "-1707\t193\n",
    "-1694\t194\n",
    "-1686\t195\n",
    "-1684\t196\n",
    "-1680\t197\n",
    "-1673\t198\n",
    "-1665\t199\n",
    "-1638\t200\n",
    "-1631\t201\n",
    "-1617\t202\n",
    "-1616\t203\n",
    "-1608\t204\n",
    "-1593\t205\n",
    "-1588\t206\n",
    "-1556\t207\n",
    "-1550\t208\n",
    "-1498\t209\n",
    "-1485\t210\n",
    "-1441\t211\n",
    "-1420\t212\n",
    "-1413\t213\n",
    "-1404\t214\n",
    "-1401\t215\n",
    "-1398\t216\n",
    "-1395\t217\n",
    "-1394\t218\n",
    "-1392\t220\n",
    "-1390\t221\n",
    "-1389\t222\n",
    "-1369\t223\n",
    "-1352\t224\n",
    "-1350\t225\n",
    "-1342\t226\n",
    "-1318\t227\n",
    "-1273\t228\n",
    "-1266\t229\n",
    "-1243\t230\n",
    "-1220\t231\n",
    "-1192\t232\n",
    "-1186\t233\n",
    "-1173\t234\n",
    "-1158\t235\n",
    "-1150\t236\n",
    "-1144\t237\n",
    "-1143\t238\n",
    "-1132\t239\n",
    "-1117\t240\n",
    "-1109\t241\n",
    "-1107\t242\n",
    "-1106\t243\n",
    "-1104\t244\n",
    "-1096\t245\n",
    "-1090\t246\n",
    "-1036\t247\n",
    "-1017\t248\n",
    "-1011\t249\n",
    "-989\t250\n",
    "-960\t251\n",
    "-954\t252\n",
    "-945\t253\n",
    "-940\t254\n",
    "-938\t255\n",
    "-921\t256\n",
    "-919\t257\n",
    "-917\t258\n",
    "-908\t259\n",
    "-904\t260\n",
    "-862\t261\n",
    "-845\t262\n",
    "-830\t263\n",
    "-801\t264\n",
    "-793\t265\n",
    "-783\t266\n",
    "-759\t267\n",
    "-750\t268\n",
    "-749\t269\n",
    "-745\t270\n",
    "-736\t271\n",
    "-734\t272\n",
    "-713\t273\n",
    "-682\t274\n",
    "-681\t275\n",
    "-673\t276\n",
    "-657\t277\n",
    "-648\t278\n",
    "-644\t279\n",
    "-636\t280\n",
    "-630\t281\n",
    "-623\t282\n",
    "-622\t283\n",
    "-608\t284\n",
    "-604\t285\n",
    "-601\t286\n",
    "-591\t287\n",
    "-586\t288\n",
    "-584\t289\n",
    "-578\t290\n",
    "-560\t291\n",
    "-555\t292\n",
    "-534\t293\n",
    "-515\t294\n",
    "-512\t295\n",
    "-504\t296\n",
    "-503\t297\n",
    "-492\t299\n",
    "-447\t300\n",
    "-441\t301\n",
    "-405\t302\n",
    "-349\t303\n",
    "-301\t304\n",
    "-288\t305\n",
    "-128\t306\n",
    "0\t309\n",
    "[nathalia@aws] $  lhtr\n",
    "total 37M\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 15:24 data\n",
    "-rw-r--r-- 1 ubuntu ubuntu 777K Nov 14 15:24 minigenome.fa\n",
    "-rw-r--r-- 1 ubuntu ubuntu  354 Nov 14 15:24 samples.txt\n",
    "-rw-r--r-- 1 ubuntu ubuntu 1.2M Nov 14 15:24 minigenome.gtf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   27 Nov 14 15:25 minigenome.fa.fai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  14K Nov 14 15:33 minigenome.gtf.ss\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13K Nov 14 15:33 minigenome.gtf.exons\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 175K Nov 14 15:33 minigenome.fa.4.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  944 Nov 14 15:33 minigenome.fa.3.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 3.8K Nov 14 15:33 minigenome.fa.8.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  19K Nov 14 15:33 minigenome.fa.7.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 176K Nov 14 15:33 minigenome.fa.2.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.5M Nov 14 15:33 minigenome.fa.1.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 179K Nov 14 15:33 minigenome.fa.6.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 509K Nov 14 15:33 minigenome.fa.5.ht2\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:39 sA_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:40 sB_rep1.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 4.6M Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sA_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.7K Nov 14 15:41 sA_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.6K Nov 14 15:41 sB_rep1.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.4K Nov 14 15:41 sB_rep2.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.8K Nov 14 15:41 sB_rep3.cSorted.hisat2.minigenome.fa.bam.bai\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sA_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 106K Nov 14 15:45 sA_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 105K Nov 14 15:45 sA_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 108K Nov 14 15:45 sB_rep1.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  99K Nov 14 15:45 sB_rep2.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 103K Nov 14 15:45 sB_rep3.cSorted.hisat2.minigenome.fa.bam.stringtie.GTF\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 141K Nov 14 15:45 stringtie.final.GTF\n",
    "drwxrwxr-x 3 ubuntu ubuntu 4.0K Nov 14 15:57 minigenome.fa.gmap\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 134K Nov 14 15:58 Trinity.gmap.bam\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  472 Nov 14 15:58 Trinity.gmap.bam.bai\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:09 abundance-kallisto\n",
    "drwxrwxr-x 5 ubuntu ubuntu 4.0K Nov 14 16:11 trinity_out_dir\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sA_rep1\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sA_rep2\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sA_rep3\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sB_rep1\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sB_rep2\n",
    "drwxrwxr-x 2 ubuntu ubuntu 4.0K Nov 14 16:11 sB_rep3\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  17K Nov 14 16:22 kallisto.isoform.counts.matrix\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  20K Nov 14 16:22 kallisto.isoform.TPM.not_cross_norm\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  11K Nov 14 16:22 kallisto.gene.counts.matrix\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  13K Nov 14 16:22 kallisto.gene.TPM.not_cross_norm\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  335 Nov 14 16:22 kallisto.isoform.TPM.not_cross_norm.TMM_info.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  22K Nov 14 16:22 kallisto.isoform.TMM.EXPR.matrix\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  526 Nov 14 16:22 __tmp_runTMM.R\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  337 Nov 14 16:22 kallisto.gene.TPM.not_cross_norm.TMM_info.txt\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  14K Nov 14 16:22 kallisto.gene.TMM.EXPR.matrix\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.8K Nov 14 16:25 genes_matrix.TPM.not_cross_norm.counts_by_min_TPM\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.9K Nov 14 16:27 isoform_matrix.TPM.not_cross_norm.counts_by_min_TPM\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filtering Transcripts based on expression values"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  $TRINITY_HOME/util/filter_low_expr_transcripts.pl --matrix kallisto.gene.TPM.not_cross_norm  --transcripts trinity_out_dir/Trinity.fasta  --gene_to_trans_map trinity_out_dir/Trinity.fasta.gene_trans_map  --highest_iso_only\n",
    "\n",
    "(...) \n",
    "#stdout is huge\n",
    "#sequences printed to stdout... ?!\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DE analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### edgeR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  $TRINITY_HOME/Analysis/DifferentialExpression/run_DE_analysis.pl  --matrix kallisto.gene.counts.matrix --method edgeR --dispersion 0.1\n",
    "-note, no biological replicates identified, so setting min reps = 2.\n",
    "Got 6 samples, and got: 7 data fields.\n",
    "Header: sA_rep1\tsA_rep2\tsA_rep3\tsB_rep1\tsB_rep2\tsB_rep3\n",
    "Next: TRINITY_DN0_c0_g1\t8.85\t8.45\t12.28\t8.51\t2.89\t5.83\n",
    "\n",
    "$VAR1 = {\n",
    "          'sB_rep3' => 6,\n",
    "          'sA_rep2' => 2,\n",
    "          'sA_rep3' => 3,\n",
    "          'sB_rep2' => 5,\n",
    "          'sA_rep1' => 1,\n",
    "          'sB_rep1' => 4\n",
    "        };\n",
    "$VAR1 = {\n",
    "          'sB_rep1' => [\n",
    "                         'sB_rep1'\n",
    "                       ],\n",
    "          'sA_rep1' => [\n",
    "                         'sA_rep1'\n",
    "                       ],\n",
    "          'sA_rep3' => [\n",
    "                         'sA_rep3'\n",
    "                       ],\n",
    "          'sB_rep2' => [\n",
    "                         'sB_rep2'\n",
    "                       ],\n",
    "          'sA_rep2' => [\n",
    "                         'sA_rep2'\n",
    "                       ],\n",
    "          'sB_rep3' => [\n",
    "                         'sB_rep3'\n",
    "                       ]\n",
    "        };\n",
    "Contrasts to perform are: $VAR1 = [\n",
    "          [\n",
    "            'sA_rep1',\n",
    "            'sA_rep2'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep1',\n",
    "            'sA_rep3'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep1',\n",
    "            'sB_rep1'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep1',\n",
    "            'sB_rep2'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep1',\n",
    "            'sB_rep3'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep2',\n",
    "            'sA_rep3'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep2',\n",
    "            'sB_rep1'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep2',\n",
    "            'sB_rep2'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep2',\n",
    "            'sB_rep3'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep3',\n",
    "            'sB_rep1'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep3',\n",
    "            'sB_rep2'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep3',\n",
    "            'sB_rep3'\n",
    "          ],\n",
    "          [\n",
    "            'sB_rep1',\n",
    "            'sB_rep2'\n",
    "          ],\n",
    "          [\n",
    "            'sB_rep1',\n",
    "            'sB_rep3'\n",
    "          ],\n",
    "          [\n",
    "            'sB_rep2',\n",
    "            'sB_rep3'\n",
    "          ]\n",
    "        ];\n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep2.sA_rep1.vs.sA_rep2.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(1,2)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep1\", 1), rep(\"sA_rep2\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep1\", \"sA_rep2\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep1\", sampleB=\"sA_rep2\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep2.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep2.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep2.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep3.sA_rep1.vs.sA_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(1,3)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep1\", 1), rep(\"sA_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep1\", \"sA_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep1\", sampleB=\"sA_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep1.sA_rep1.vs.sB_rep1.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(1,4)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep1\", 1), rep(\"sB_rep1\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep1\", \"sB_rep1\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep1\", sampleB=\"sB_rep1\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep1.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep1.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep1.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep2.sA_rep1.vs.sB_rep2.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(1,5)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep1\", 1), rep(\"sB_rep2\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep1\", \"sB_rep2\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep1\", sampleB=\"sB_rep2\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep2.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep2.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep2.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep3.sA_rep1.vs.sB_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(1,6)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep1\", 1), rep(\"sB_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep1\", \"sB_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep1\", sampleB=\"sB_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep2_vs_sA_rep3.sA_rep2.vs.sA_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(2,3)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep2\", 1), rep(\"sA_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep2\", \"sA_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep2\", sampleB=\"sA_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep2_vs_sA_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep2_vs_sA_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep2_vs_sA_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep1.sA_rep2.vs.sB_rep1.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(2,4)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep2\", 1), rep(\"sB_rep1\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep2\", \"sB_rep1\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep2\", sampleB=\"sB_rep1\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep1.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep1.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep1.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep2.sA_rep2.vs.sB_rep2.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(2,5)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep2\", 1), rep(\"sB_rep2\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep2\", \"sB_rep2\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep2\", sampleB=\"sB_rep2\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep2.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep2.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep2.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep3.sA_rep2.vs.sB_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(2,6)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep2\", 1), rep(\"sB_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep2\", \"sB_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep2\", sampleB=\"sB_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep1.sA_rep3.vs.sB_rep1.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(3,4)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep3\", 1), rep(\"sB_rep1\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep3\", \"sB_rep1\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep3\", sampleB=\"sB_rep1\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep1.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep1.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep1.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep2.sA_rep3.vs.sB_rep2.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(3,5)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep3\", 1), rep(\"sB_rep2\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep3\", \"sB_rep2\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep3\", sampleB=\"sB_rep2\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep2.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep2.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep2.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep3.sA_rep3.vs.sB_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(3,6)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep3\", 1), rep(\"sB_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep3\", \"sB_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep3\", sampleB=\"sB_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep2.sB_rep1.vs.sB_rep2.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(4,5)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sB_rep1\", 1), rep(\"sB_rep2\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sB_rep1\", \"sB_rep2\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sB_rep1\", sampleB=\"sB_rep2\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep2.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep2.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep2.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep3.sB_rep1.vs.sB_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(4,6)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sB_rep1\", 1), rep(\"sB_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sB_rep1\", \"sB_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sB_rep1\", sampleB=\"sB_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sB_rep2_vs_sB_rep3.sB_rep2.vs.sB_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(5,6)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sB_rep2\", 1), rep(\"sB_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sB_rep2\", \"sB_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sB_rep2\", sampleB=\"sB_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sB_rep2_vs_sB_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sB_rep2_vs_sB_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sB_rep2_vs_sB_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<img src=\"http://ec2-34-221-102-104.us-west-2.compute.amazonaws.com/workspace/transreco/edgeR.22490.dir/kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep2.edgeR.DE_results.MA_n_Volcano.pdf\"/>"
      ],
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from IPython.display import Image\n",
    "from IPython.core.display import HTML \n",
    "Image(url= \"http://ec2-34-221-102-104.us-west-2.compute.amazonaws.com/workspace/transreco/edgeR.22490.dir/kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep2.edgeR.DE_results.MA_n_Volcano.pdf\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## voom\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  $TRINITY_HOME/Analysis/DifferentialExpression/run_DE_analysis.pl  --matrix kallisto.gene.counts.matrix --method voom --samples_file samples.txt \n",
    "Got 6 samples, and got: 7 data fields.\n",
    "Header: sA_rep1\tsA_rep2\tsA_rep3\tsB_rep1\tsB_rep2\tsB_rep3\n",
    "Next: TRINITY_DN0_c0_g1\t8.85\t8.45\t12.28\t8.51\t2.89\t5.83\n",
    "\n",
    "$VAR1 = {\n",
    "          'sB_rep3' => 6,\n",
    "          'sB_rep2' => 5,\n",
    "          'sB_rep1' => 4,\n",
    "          'sA_rep1' => 1,\n",
    "          'sA_rep3' => 3,\n",
    "          'sA_rep2' => 2\n",
    "        };\n",
    "$VAR1 = {\n",
    "          'sA' => [\n",
    "                    'sA_rep1',\n",
    "                    'sA_rep2',\n",
    "                    'sA_rep3'\n",
    "                  ],\n",
    "          'sB' => [\n",
    "                    'sB_rep1',\n",
    "                    'sB_rep2',\n",
    "                    'sB_rep3'\n",
    "                  ]\n",
    "        };\n",
    "Contrasts to perform are: $VAR1 = [\n",
    "          [\n",
    "            'sA',\n",
    "            'sB'\n",
    "          ]\n",
    "        ];\n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_vs_sB.sA.vs.sB.voom.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> library(limma)\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(1,2,3,4,5,6)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 2,]\n",
    "> conditions = factor(c(rep(\"sA\", 3), rep(\"sB\", 3)))\n",
    "> \n",
    "> design = model.matrix(~conditions)\n",
    "> ## TMM normalize data\n",
    "> lib_sizes = colSums(rnaseqMatrix)\n",
    "> tmm_norm_factors = calcNormFactors(rnaseqMatrix, method='TMM')\n",
    "> x = DGEList(counts=rnaseqMatrix)\n",
    "> # voom transformation\n",
    "> y = voom(x, design, lib.size=lib_sizes*tmm_norm_factors, plot=F)\n",
    "> fit = eBayes(lmFit(y,design))\n",
    "> tTags = topTable(fit,coef=2,number=Inf)\n",
    "> # output results, including average expression val for each feature\n",
    "> c = cpm(x)\n",
    "> m = apply(c, 1, mean)\n",
    "> tTags$logFC = -1 * tTags$logFC  # make A/B instead of B/A\n",
    "> tTags2 = cbind(tTags, logCPM=log2(m[rownames(tTags)]))\n",
    "> DE_matrix = data.frame(sampleA=\"sA\", sampleB=\"sB\", logFC=tTags$logFC, logCPM=tTags2$logCPM, PValue=tTags$'P.Value', FDR=tTags$'adj.P.Val')\n",
    "> rownames(DE_matrix) = rownames(tTags)\n",
    "> write.table(DE_matrix, file='kallisto.gene.counts.matrix.sA_vs_sB.voom.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_vs_sB.voom.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> # MA and volcano plots\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_vs_sB.voom.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(tTags2), tTags2$logCPM, tTags$logFC, tTags$'adj.P.Val')\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  $TRINITY_HOME/Analysis/DifferentialExpression/run_DE_analysis.pl  --matrix kallisto.gene.counts.matrix --method edgeR --dispersion 0.1\n",
    "-note, no biological replicates identified, so setting min reps = 2.\n",
    "Got 6 samples, and got: 7 data fields.\n",
    "Header: sA_rep1\tsA_rep2\tsA_rep3\tsB_rep1\tsB_rep2\tsB_rep3\n",
    "Next: TRINITY_DN0_c0_g1\t8.85\t8.45\t12.28\t8.51\t2.89\t5.83\n",
    "\n",
    "$VAR1 = {\n",
    "          'sA_rep1' => 1,\n",
    "          'sB_rep3' => 6,\n",
    "          'sB_rep2' => 5,\n",
    "          'sA_rep2' => 2,\n",
    "          'sB_rep1' => 4,\n",
    "          'sA_rep3' => 3\n",
    "        };\n",
    "$VAR1 = {\n",
    "          'sA_rep2' => [\n",
    "                         'sA_rep2'\n",
    "                       ],\n",
    "          'sA_rep3' => [\n",
    "                         'sA_rep3'\n",
    "                       ],\n",
    "          'sB_rep1' => [\n",
    "                         'sB_rep1'\n",
    "                       ],\n",
    "          'sA_rep1' => [\n",
    "                         'sA_rep1'\n",
    "                       ],\n",
    "          'sB_rep3' => [\n",
    "                         'sB_rep3'\n",
    "                       ],\n",
    "          'sB_rep2' => [\n",
    "                         'sB_rep2'\n",
    "                       ]\n",
    "        };\n",
    "Contrasts to perform are: $VAR1 = [\n",
    "          [\n",
    "            'sA_rep1',\n",
    "            'sA_rep2'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep1',\n",
    "            'sA_rep3'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep1',\n",
    "            'sB_rep1'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep1',\n",
    "            'sB_rep2'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep1',\n",
    "            'sB_rep3'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep2',\n",
    "            'sA_rep3'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep2',\n",
    "            'sB_rep1'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep2',\n",
    "            'sB_rep2'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep2',\n",
    "            'sB_rep3'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep3',\n",
    "            'sB_rep1'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep3',\n",
    "            'sB_rep2'\n",
    "          ],\n",
    "          [\n",
    "            'sA_rep3',\n",
    "            'sB_rep3'\n",
    "          ],\n",
    "          [\n",
    "            'sB_rep1',\n",
    "            'sB_rep2'\n",
    "          ],\n",
    "          [\n",
    "            'sB_rep1',\n",
    "            'sB_rep3'\n",
    "          ],\n",
    "          [\n",
    "            'sB_rep2',\n",
    "            'sB_rep3'\n",
    "          ]\n",
    "        ];\n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep2.sA_rep1.vs.sA_rep2.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(1,2)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep1\", 1), rep(\"sA_rep2\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep1\", \"sA_rep2\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep1\", sampleB=\"sA_rep2\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep2.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep2.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep2.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep3.sA_rep1.vs.sA_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(1,3)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep1\", 1), rep(\"sA_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep1\", \"sA_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep1\", sampleB=\"sA_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep1_vs_sA_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep1.sA_rep1.vs.sB_rep1.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(1,4)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep1\", 1), rep(\"sB_rep1\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep1\", \"sB_rep1\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep1\", sampleB=\"sB_rep1\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep1.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep1.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep1.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep2.sA_rep1.vs.sB_rep2.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(1,5)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep1\", 1), rep(\"sB_rep2\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep1\", \"sB_rep2\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep1\", sampleB=\"sB_rep2\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep2.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep2.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep2.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep3.sA_rep1.vs.sB_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(1,6)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep1\", 1), rep(\"sB_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep1\", \"sB_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep1\", sampleB=\"sB_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep1_vs_sB_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep2_vs_sA_rep3.sA_rep2.vs.sA_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(2,3)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep2\", 1), rep(\"sA_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep2\", \"sA_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep2\", sampleB=\"sA_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep2_vs_sA_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep2_vs_sA_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep2_vs_sA_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep1.sA_rep2.vs.sB_rep1.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(2,4)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep2\", 1), rep(\"sB_rep1\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep2\", \"sB_rep1\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep2\", sampleB=\"sB_rep1\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep1.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep1.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep1.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep2.sA_rep2.vs.sB_rep2.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(2,5)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep2\", 1), rep(\"sB_rep2\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep2\", \"sB_rep2\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep2\", sampleB=\"sB_rep2\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep2.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep2.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep2.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep3.sA_rep2.vs.sB_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(2,6)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep2\", 1), rep(\"sB_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep2\", \"sB_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep2\", sampleB=\"sB_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep2_vs_sB_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep1.sA_rep3.vs.sB_rep1.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(3,4)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep3\", 1), rep(\"sB_rep1\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep3\", \"sB_rep1\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep3\", sampleB=\"sB_rep1\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep1.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep1.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep1.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep2.sA_rep3.vs.sB_rep2.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(3,5)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep3\", 1), rep(\"sB_rep2\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep3\", \"sB_rep2\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep3\", sampleB=\"sB_rep2\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep2.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep2.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep2.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep3.sA_rep3.vs.sB_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(3,6)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sA_rep3\", 1), rep(\"sB_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sA_rep3\", \"sB_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sA_rep3\", sampleB=\"sB_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sA_rep3_vs_sB_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep2.sB_rep1.vs.sB_rep2.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(4,5)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sB_rep1\", 1), rep(\"sB_rep2\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sB_rep1\", \"sB_rep2\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sB_rep1\", sampleB=\"sB_rep2\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep2.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep2.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep2.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep3.sB_rep1.vs.sB_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(4,6)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sB_rep1\", 1), rep(\"sB_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sB_rep1\", \"sB_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sB_rep1\", sampleB=\"sB_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sB_rep1_vs_sB_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "CMD: R --vanilla -q < kallisto.gene.counts.matrix.sB_rep2_vs_sB_rep3.sB_rep2.vs.sB_rep3.EdgeR.Rscript\n",
    "> library(edgeR)\n",
    "Loading required package: limma\n",
    "> \n",
    "> data = read.table(\"/workspace/transreco/kallisto.gene.counts.matrix\", header=T, row.names=1, com='')\n",
    "> col_ordering = c(5,6)\n",
    "> rnaseqMatrix = data[,col_ordering]\n",
    "> rnaseqMatrix = round(rnaseqMatrix)\n",
    "> rnaseqMatrix = rnaseqMatrix[rowSums(cpm(rnaseqMatrix) > 1) >= 1,]\n",
    "> conditions = factor(c(rep(\"sB_rep2\", 1), rep(\"sB_rep3\", 1)))\n",
    "> \n",
    "> exp_study = DGEList(counts=rnaseqMatrix, group=conditions)\n",
    "> exp_study = calcNormFactors(exp_study)\n",
    "> et = exactTest(exp_study, pair=c(\"sB_rep2\", \"sB_rep3\"), dispersion=0.1)\n",
    "> tTags = topTags(et,n=NULL)\n",
    "> result_table = tTags$table\n",
    "> result_table = data.frame(sampleA=\"sB_rep2\", sampleB=\"sB_rep3\", result_table)\n",
    "> result_table$logFC = -1 * result_table$logFC\n",
    "> write.table(result_table, file='kallisto.gene.counts.matrix.sB_rep2_vs_sB_rep3.edgeR.DE_results', sep='\t', quote=F, row.names=T)\n",
    "> write.table(rnaseqMatrix, file='kallisto.gene.counts.matrix.sB_rep2_vs_sB_rep3.edgeR.count_matrix', sep='\t', quote=F, row.names=T)\n",
    "> source(\"/home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/R/rnaseq_plot_funcs.R\")\n",
    "> pdf(\"kallisto.gene.counts.matrix.sB_rep2_vs_sB_rep3.edgeR.DE_results.MA_n_Volcano.pdf\")\n",
    "> plot_MA_and_Volcano(rownames(result_table), result_table$logCPM, result_table$logFC, result_table$FDR)\n",
    "> dev.off()\n",
    "null device \n",
    "          1 \n",
    "> \n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<img src=\"http://ec2-34-221-102-104.us-west-2.compute.amazonaws.com/workspace/transreco/voom.22371.dir/kallisto.gene.counts.matrix.sA_vs_sB.voom.DE_results.MA_n_Volcano.pdf\"/>"
      ],
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from IPython.display import Image\n",
    "from IPython.core.display import HTML \n",
    "Image(url= \"http://ec2-34-221-102-104.us-west-2.compute.amazonaws.com/workspace/transreco/voom.22371.dir/kallisto.gene.counts.matrix.sA_vs_sB.voom.DE_results.MA_n_Volcano.pdf\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Extracting and clustering differentially expressed transcripts"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### edgeR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  $TRINITY_HOME/Analysis/DifferentialExpression/analyze_diff_expr.pl --matrix ../kallisto.gene.TMM.EXPR.matrix -P 1e-3 -C 2\n",
    "\n",
    "\n",
    "** Found 28 features as differentially expressed.\n",
    "\n",
    "CMD: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/PtR -m diffExpr.P1e-3_C2.matrix --log2 --heatmap --min_colSums 0 --min_rowSums 0 --gene_dist euclidean --sample_dist euclidean --sample_cor_matrix --center_rows --save \n",
    "CMD: R --vanilla -q < diffExpr.P1e-3_C2.matrix.R\n",
    "> library(cluster)\n",
    "> library(Biobase)\n",
    "Loading required package: BiocGenerics\n",
    "Loading required package: parallel\n",
    "\n",
    "Attaching package: BiocGenerics\n",
    "\n",
    "The following objects are masked from package:parallel:\n",
    "\n",
    "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
    "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
    "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
    "\n",
    "The following objects are masked from package:stats:\n",
    "\n",
    "    IQR, mad, sd, var, xtabs\n",
    "\n",
    "The following objects are masked from package:base:\n",
    "\n",
    "    anyDuplicated, append, as.data.frame, cbind, colMeans, colnames,\n",
    "    colSums, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
    "    grepl, intersect, is.unsorted, lapply, lengths, Map, mapply, match,\n",
    "    mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
    "    rbind, Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,\n",
    "    table, tapply, union, unique, unsplit, which, which.max, which.min\n",
    "\n",
    "Welcome to Bioconductor\n",
    "\n",
    "    Vignettes contain introductory material; view with\n",
    "    'browseVignettes()'. To cite Bioconductor, see\n",
    "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
    "\n",
    "> library(qvalue)\n",
    "Error in library(qvalue) : there is no package called qvalue\n",
    "Execution halted\n",
    "Error, cmd: R --vanilla -q < diffExpr.P1e-3_C2.matrix.R died with ret 256 at /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/PtR line 1693.\n",
    "Error, Error, cmd: /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/PtR -m diffExpr.P1e-3_C2.matrix --log2 --heatmap --min_colSums 0 --min_rowSums 0 --gene_dist euclidean --sample_dist euclidean --sample_cor_matrix --center_rows --save  died with ret 6400 at /home/ubuntu/trinityrnaseq-Trinity-v2.5.1/Analysis/DifferentialExpression/analyze_diff_expr.pl line 272.\n",
    "[nathalia@aws] $  R\n",
    "\n",
    "R version 3.4.4 (2018-03-15) -- \"Someone to Lean On\"\n",
    "Copyright (C) 2018 The R Foundation for Statistical Computing\n",
    "Platform: x86_64-pc-linux-gnu (64-bit)\n",
    "\n",
    "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
    "You are welcome to redistribute it under certain conditions.\n",
    "Type 'license()' or 'licence()' for distribution details.\n",
    "\n",
    "R is a collaborative project with many contributors.\n",
    "Type 'contributors()' for more information and\n",
    "'citation()' on how to cite R or R packages in publications.\n",
    "\n",
    "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
    "'help.start()' for an HTML browser interface to help.\n",
    "Type 'q()' to quit R.\n",
    "\n",
    "> install.packages(\"qvalue\")\n",
    "Installing package into /home/ubuntu/R/x86_64-pc-linux-gnu-library/3.4\n",
    "(as lib is unspecified)\n",
    "Warning message:\n",
    "package qvalue is not available (for R version 3.4.4) \n",
    "> \n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " no qvalue package bc of R version.... BS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
