{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 11/12/2018\n",
    "\n",
    "GEMINI - Aaron Quilan\n",
    "\n",
    "nathalia\n",
    "\n",
    "https://gemini.readthedocs.io/en/latest/\n",
    "\n",
    "https://gist.github.com/arq5x/9e1928638397ba45da2e#file-gemini-intro-sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/\n",
    "\n",
    "[nathalia@aws] $  mkdir gemini\n",
    "[nathalia@aws] $  cd gemini/\n",
    "[nathalia@aws] $  mkdir mydata\n",
    "[nathalia@aws] $  cd mydata/\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/gemini/mydata\n",
    "\n",
    "#gemini-intro.sh, slide 7\n",
    "\n",
    "[nathalia@aws] $  curl https://s3.amazonaws.com/gemini-tutorials/learnSQL.db > learnSQL.db\n",
    "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
    "                                 Dload  Upload   Total   Spent    Left  Speed\n",
    "  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0\n",
    "  0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0\n",
    "100  2048  100  2048    0     0    351      0  0:00:05  0:00:05 --:--:--   474\n",
    "[nathalia@aws] $  curl https://s3.amazonaws.com/gemini-tutorials/learnSQL2.db > learnSQL2.db\n",
    "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
    "                                 Dload  Upload   Total   Spent    Left  Speed\n",
    "100  3072  100  3072    0     0   8930      0 --:--:-- --:--:-- --:--:--  8930\n",
    "[nathalia@aws] $  curl https://s3.amazonaws.com/gemini-tutorials/chr22.VEP.vcf > chr22.VEP.vcf\n",
    "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
    "                                 Dload  Upload   Total   Spent    Left  Speed\n",
    "100 2503k  100 2503k    0     0  1803k      0  0:00:01  0:00:01 --:--:-- 1802k\n",
    "[nathalia@aws] $  curl https://s3.amazonaws.com/gemini-tutorials/trio.ped > trio.ped\n",
    "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
    "                                 Dload  Upload   Total   Spent    Left  Speed\n",
    "100   154  100   154    0     0    479      0 --:--:-- --:--:-- --:--:--   479\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 2.5M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.0K Nov 12 16:54 learnSQL.db\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 3.0K Nov 12 16:54 learnSQL2.db\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.5M Nov 12 16:54 chr22.VEP.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  154 Nov 12 16:54 trio.ped\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  gemini query -h\n",
    "usage: gemini query [-h] [-q QUERY_STR] [--gt-filter STRING] [--show-samples]\n",
    "                    [--show-families] [--family-wise]\n",
    "                    [--min-kindreds MIN_KINDREDS] [--sample-delim STRING]\n",
    "                    [--header] [--sample-filter SAMPLE_FILTER]\n",
    "                    [--in [{all,none,any,only,not} [{all,none,any,only,not} ...]]]\n",
    "                    [--format FORMAT] [--region REGION]\n",
    "                    [--carrier-summary-by-phenotype CARRIER_SUMMARY] [--dgidb]\n",
    "                    [--use-bcolz]\n",
    "                    db\n",
    "\n",
    "positional arguments:\n",
    "  db                    The name of the database to be queried.\n",
    "\n",
    "optional arguments:\n",
    "  -h, --help            show this help message and exit\n",
    "  -q QUERY_STR          The query to be issued to the database\n",
    "  --gt-filter STRING    Restrictions to apply to genotype values\n",
    "  --show-samples        Add a column of all sample names with a variant to\n",
    "                        each variant.\n",
    "  --show-families       Add a column listing all of the families with a\n",
    "                        variant to each variant.\n",
    "  --family-wise         Perform the sample-filter on a family-wise basis.\n",
    "  --min-kindreds MIN_KINDREDS\n",
    "                        Minimum number of families for a variant passing a\n",
    "                        family-wise filter to be in.\n",
    "  --sample-delim STRING\n",
    "                        The delimiter to be used with the --show-samples\n",
    "                        option.\n",
    "  --header              Add a header of column names to the output.\n",
    "  --sample-filter SAMPLE_FILTER\n",
    "                        SQL filter to use to filter the sample table\n",
    "  --in [{all,none,any,only,not} [{all,none,any,only,not} ...]]\n",
    "                        A variant must be in either all, none or any samples\n",
    "                        passing the --sample-query filter.\n",
    "  --format FORMAT       Format of output (JSON, TPED or default)\n",
    "  --region REGION       Restrict query to this region, e.g. chr1:10-20.\n",
    "  --carrier-summary-by-phenotype CARRIER_SUMMARY\n",
    "                        Output columns of counts of carriers and non-carriers\n",
    "                        stratified by the given sample phenotype column\n",
    "  --dgidb               Request drug-gene interaction info from DGIdb.\n",
    "  --use-bcolz           use a (previously created) bcolz index to speed\n",
    "                        genotype queries\n",
    "                        \n",
    "                        \n",
    "                        \n",
    "[nathalia@aws] $  gemini -h\n",
    "usage: gemini [-h] [-v] [--annotation-dir ANNOTATION_DIR]\n",
    "              {actionable_mutations,amend,annotate,autosomal_dominant,autosomal_recessive,bcolz_index,browser,burden,comp_hets,db_info,de_novo,dump,examples,fusions,gene_wise,interactions,load,load_chunk,lof_interactions,lof_sieve,mendel_errors,merge_chunks,pathways,qc,query,region,roh,set_somatic,stats,update,windower,x_linked_de_novo,x_linked_dominant,x_linked_recessive}\n",
    "              ...\n",
    "\n",
    "optional arguments:\n",
    "  -h, --help            show this help message and exit\n",
    "  -v, --version         Installed gemini version\n",
    "  --annotation-dir ANNOTATION_DIR\n",
    "                        Path to the annotation database. This argument is\n",
    "                        optional and if given will take precedence over the\n",
    "                        default location stored in the gemini config file.\n",
    "                        (default: None)\n",
    "\n",
    "[sub-commands]:\n",
    "  {actionable_mutations,amend,annotate,autosomal_dominant,autosomal_recessive,bcolz_index,browser,burden,comp_hets,db_info,de_novo,dump,examples,fusions,gene_wise,interactions,load,load_chunk,lof_interactions,lof_sieve,mendel_errors,merge_chunks,pathways,qc,query,region,roh,set_somatic,stats,update,windower,x_linked_de_novo,x_linked_dominant,x_linked_recessive}\n",
    "    actionable_mutations\n",
    "                        Retrieve genes with actionable somatic mutations via\n",
    "                        COSMIC and DGIdb.\n",
    "    amend               Amend an already loaded GEMINI database.\n",
    "    annotate            Add new columns for custom annotations\n",
    "    autosomal_dominant  Identify variants meeting an autosomal dominant\n",
    "                        inheritance model\n",
    "    autosomal_recessive\n",
    "                        Identify variants meeting an autosomal recessive\n",
    "                        inheritance model\n",
    "    bcolz_index         index an existing gemini database so it can use bcolze\n",
    "                        for faster genotype queries.\n",
    "    browser             Browser interface to gemini\n",
    "    burden              Gene-level genetic burden tests. By default counts all\n",
    "                        variants with high impact in coding regions as\n",
    "                        contributing to burden.\n",
    "    comp_hets           Identify compound heterozygotes\n",
    "    db_info             Get the names and types of cols. database tables\n",
    "    de_novo             Identify candidate de novo mutations\n",
    "    dump                shortcuts for extracting data from the DB\n",
    "    examples            show usage examples\n",
    "    fusions             Identify somatic fusion genes from a GEMINI database.\n",
    "    interactions        Find interaction partners for a gene in sample\n",
    "                        variants(default mode)\n",
    "    load                load a VCF file in gemini database\n",
    "    load_chunk          load a VCF file in gemini database\n",
    "    lof_interactions    Find interaction partners for a lof gene in sample\n",
    "                        variants(default mode)\n",
    "    lof_sieve           Prioritize LoF mutations\n",
    "    mendel_errors       Identify candidate violations of Mendelian inheritance\n",
    "    merge_chunks        combine intermediate db files into the final gemini\n",
    "    pathways            Map genes and variants to KEGG pathways\n",
    "    qc                  Quality control tools\n",
    "    query               issue ad hoc SQL queries to the DB\n",
    "    region              extract variants from specific genomic loci\n",
    "    roh                 Identify runs of homozygosity\n",
    "    set_somatic         Tag somatic mutations (is_somatic) by comparint\n",
    "                        tumor/normal pairs.\n",
    "    stats               compute useful variant stastics\n",
    "    update              Update gemini software and data files.\n",
    "    windower            Compute statistics across genome \"windows\"\n",
    "    x_linked_de_novo    X-linked de novo variants\n",
    "    x_linked_dominant   X-linked dominant variants\n",
    "    x_linked_recessive  X-linked recessive variants\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Review about SQL\n",
    "\n",
    "```\n",
    "[nathalia@aws] $  gemini query -q \"SELECT name FROM samples\" learnSQL.db\n",
    "John\n",
    "Bob\n",
    "Mary\n",
    "Sue\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT name FROM samples WHERE phenotype == 2\" learnSQL.db\n",
    "John\n",
    "Sue\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT name FROM samples WHERE phenotype <> 2\" learnSQL.db \n",
    "Bob\n",
    "Mary\n",
    "\n",
    "# <>2  represents not equal to 2\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT name FROM samples WHERE sample_id < 3\" learnSQL.db\n",
    "John\n",
    "Bob\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  gemini query --header -q \"SELECT * FROM fakevariants\" learnSQL2.db\n",
    "chrom\tstart\tend\trsid\tin_dbsnp\n",
    "chr1\t1\t2\trs123\t1\n",
    "chr1\t3\t4\t\t0\n",
    "chr2\t1\t2\trs456\t1\n",
    "chr2\t3\t4\t\t0\n",
    "\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT name FROM samples WHERE ethnicity IS NULL\" learnSQL2.db\n",
    "Greg\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT name FROM samples WHERE ethnicity IS NOT NULL\" learnSQL2.db\n",
    "John\n",
    "Bob\n",
    "Mary\n",
    "Sue\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT * FROM fakevariants\" learnSQL2.db\n",
    "chr1\t1\t2\trs123\t1\n",
    "chr1\t3\t4\t\t0\n",
    "chr2\t1\t2\trs456\t1\n",
    "chr2\t3\t4\t\t0\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT chrom,start,end FROM fakevariants WHERE in_dbsnp == 1\" learnSQL2.db\n",
    "chr1\t1\t2\n",
    "chr2\t1\t2\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT chrom,start,end FROM fakevariants WHERE in_dbsnp\" learnSQL2.db\n",
    "chr1\t1\t2\n",
    "chr2\t1\t2\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT COUNT(*) FROM fakevariants WHERE chrom == 'chr1' \" learnSQL2.db\n",
    "2\n",
    "\n",
    "# COUNT() fx that counts the number of rows that have chr1; returns the number of rows that match chr1 in column chrom\n",
    "\n",
    "# 'chr1' in quotes because it's a text\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT COUNT(*) FROM fakevariants WHERE chrom == 'chr1' AND in_dbsnp == 0 \" learnSQL2.db\n",
    "1\n",
    "\n",
    "# how many counts are in chr1 and are novel in in_dbsnp column\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# VEP \n",
    "\n",
    "VCF file was annotated using VEP program:\n",
    "\n",
    "```\n",
    "#perl ~/software/variant_effect_predictor/variant_effect_predictor.pl -i chr22.vcf -o chr22.VEP.vcf --vcf \\\n",
    "#--cache --dir ~/software/variant_effect_predictor/references \\\n",
    "#--compress \"gunzip -c\" \\\n",
    "#--force_overwrite \\\n",
    "#--sift b --polyphen b --symbol --numbers --biotype --total_length \\\n",
    "#--fields Consequence,Codons,Amino_acids,Gene,SYMBOL,Feature,EXON,PolyPhen,SIFT,Protein_position,BIOTYPE \n",
    "```\n",
    "\n",
    "VEP file has a CSQ (consequence) field. Consequence of this new allele on the transcript of a gene. It enters a prediction for every isoform of that gene."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# PED "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  less trio.ped \n",
    "\n",
    "#family_id sample_id paternal_id maternal_id sex phenotype ethnicity\n",
    "family1 1805    -9      -9      2       1       CEU\n",
    "family1 1847    -9      -9      1       1       CEU\n",
    "family1 4805    1847    1805    1       2       CEU\n",
    "\n",
    "```\n",
    "\n",
    "Missing values 0 or -9"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sample names in `.ped` and `.vcf` file should match. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  gemini load -v chr22.VEP.vcf -p trio.ped -t VEP --cores 2 --tempdir . --skip-gene-tables chr22.db\n",
    "\n",
    "CADD scores are not being loaded because the annotation file could not be found.\n",
    "`Run gemini update --dataonly --extra cadd_score` to install the annotation file.\n",
    "\n",
    "\n",
    "GERP per bp is not being loaded because the annotation file could not be found.\n",
    "    Run `gemini update --dataonly --extra gerp_bp` to install the annotation file.\n",
    "\n",
    "[W::hts_idx_load2] The index file is older than the data file: /workspace/lib/gemini/gemini_data/hg19_fitcons_fc-i6-0_V1-01.bed.gz.tbi\n",
    "[W::hts_idx_load2] The index file is older than the data file: /workspace/lib/gemini/gemini_data/geno2mp.variants.tidy.vcf.gz.tbi\n",
    "Bgzipping /workspace/gemini/mydata/chr22.VEP.vcf into /workspace/gemini/mydata/chr22.VEP.vcf.gz.\n",
    "Indexing /workspace/gemini/mydata/chr22.VEP.vcf.gz with grabix.\n",
    "Loading 2445 variants.\n",
    "Breaking /workspace/gemini/mydata/chr22.VEP.vcf.gz into 2 chunks.\n",
    "Loading chunk 0.\n",
    "Loading chunk 1.\n",
    "[W::hts_idx_load2] The index file is older than the data file: /workspace/lib/gemini/gemini_data/hg19_fitcons_fc-i6-0_V1-01.bed.gz.tbi\n",
    "[W::hts_idx_load2] The index file is older than the data file: /workspace/lib/gemini/gemini_data/hg19_fitcons_fc-i6-0_V1-01.bed.gz.tbi\n",
    "[W::hts_idx_load2] The index file is older than the data file: /workspace/lib/gemini/gemini_data/geno2mp.variants.tidy.vcf.gz.tbi\n",
    "[W::hts_idx_load2] The index file is older than the data file: /workspace/lib/gemini/gemini_data/geno2mp.variants.tidy.vcf.gz.tbi\n",
    "pid 9029: 1223 variants processed.\n",
    "pid 9026: 1222 variants processed.\n",
    "Done loading 2445 variants in 2 chunks.\n",
    "2018-11-12 19:18:27 merging 2 chunks.\n",
    "2018-11-12 19:18:28 merged to final database....\n",
    "2018-11-12 19:18:28 moved.\n",
    "storing version, header, etc.\n",
    "building indices\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 18M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.0K Nov 12 16:54 learnSQL.db\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 3.0K Nov 12 16:54 learnSQL2.db\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 2.5M Nov 12 16:54 chr22.VEP.vcf\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  154 Nov 12 16:54 trio.ped\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 374K Nov 12 19:17 chr22.VEP.vcf.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   29 Nov 12 19:17 chr22.VEP.vcf.gz.gbi\n",
    "-rw-r--r-- 1 ubuntu ubuntu  15M Nov 12 19:18 chr22.db\n",
    "\n",
    "\n",
    "[nathalia@aws] $  gemini db_info chr22.db # it's binary\n",
    "table_name          column_name                   type      \n",
    "variants            chrom                         VARCHAR(20)\n",
    "variants            start                         INTEGER   \n",
    "variants            end                           INTEGER   \n",
    "variants            vcf_id                        TEXT      \n",
    "variants            variant_id                    INTEGER   \n",
    "variants            anno_id                       INTEGER   \n",
    "variants            ref                           TEXT      \n",
    "variants            alt                           TEXT      \n",
    "variants            qual                          FLOAT     \n",
    "variants            filter                        TEXT      \n",
    "variants            type                          VARCHAR(20)\n",
    "variants            sub_type                      TEXT      \n",
    "variants            gts                           BLOB      \n",
    "variants            gt_types                      BLOB      \n",
    "variants            gt_phases                     BLOB      \n",
    "variants            gt_depths                     BLOB      \n",
    "variants            gt_ref_depths                 BLOB      \n",
    "variants            gt_alt_depths                 BLOB      \n",
    "variants            gt_alt_freqs                  BLOB      \n",
    "variants            gt_quals                      BLOB      \n",
    "variants            gt_copy_numbers               BLOB      \n",
    "variants            gt_phred_ll_homref            BLOB      \n",
    "variants            gt_phred_ll_het               BLOB      \n",
    "variants            gt_phred_ll_homalt            BLOB      \n",
    "variants            call_rate                     FLOAT     \n",
    "variants            max_aaf_all                   FLOAT     \n",
    "variants            in_dbsnp                      BOOLEAN   \n",
    "variants            rs_ids                        TEXT      \n",
    "variants            sv_cipos_start_left           INTEGER   \n",
    "variants            sv_cipos_end_left             INTEGER   \n",
    "variants            sv_cipos_start_right          INTEGER   \n",
    "variants            sv_cipos_end_right            INTEGER   \n",
    "variants            sv_length                     INTEGER   \n",
    "variants            sv_is_precise                 BOOLEAN   \n",
    "variants            sv_tool                       TEXT      \n",
    "variants            sv_evidence_type              TEXT      \n",
    "variants            sv_event_id                   TEXT      \n",
    "variants            sv_mate_id                    TEXT      \n",
    "variants            sv_strand                     TEXT      \n",
    "variants            in_omim                       BOOLEAN   \n",
    "variants            clinvar_sig                   TEXT      \n",
    "variants            clinvar_disease_name          TEXT      \n",
    "variants            clinvar_dbsource              TEXT      \n",
    "variants            clinvar_dbsource_id           TEXT      \n",
    "variants            clinvar_origin                TEXT      \n",
    "variants            clinvar_dsdb                  TEXT      \n",
    "variants            clinvar_dsdbid                TEXT      \n",
    "variants            clinvar_disease_acc           TEXT      \n",
    "variants            clinvar_in_locus_spec_db      BOOLEAN   \n",
    "variants            clinvar_on_diag_assay         BOOLEAN   \n",
    "variants            clinvar_causal_allele         TEXT      \n",
    "variants            clinvar_gene_phenotype        TEXT      \n",
    "variants            geno2mp_hpo_ct                INTEGER   \n",
    "variants            pfam_domain                   TEXT      \n",
    "variants            cyto_band                     TEXT      \n",
    "variants            rmsk                          TEXT      \n",
    "variants            in_cpg_island                 BOOLEAN   \n",
    "variants            in_segdup                     BOOLEAN   \n",
    "variants            is_conserved                  BOOLEAN   \n",
    "variants            gerp_bp_score                 FLOAT     \n",
    "variants            gerp_element_pval             FLOAT     \n",
    "variants            num_hom_ref                   INTEGER   \n",
    "variants            num_het                       INTEGER   \n",
    "variants            num_hom_alt                   INTEGER   \n",
    "variants            num_unknown                   INTEGER   \n",
    "variants            aaf                           FLOAT     \n",
    "variants            hwe                           FLOAT     \n",
    "variants            inbreeding_coeff              FLOAT     \n",
    "variants            pi                            FLOAT     \n",
    "variants            recomb_rate                   FLOAT     \n",
    "variants            gene                          VARCHAR(60)\n",
    "variants            transcript                    VARCHAR(60)\n",
    "variants            is_exonic                     BOOLEAN   \n",
    "variants            is_coding                     BOOLEAN   \n",
    "variants            is_splicing                   BOOLEAN   \n",
    "variants            is_lof                        BOOLEAN   \n",
    "variants            exon                          TEXT      \n",
    "variants            codon_change                  TEXT      \n",
    "variants            aa_change                     TEXT      \n",
    "variants            aa_length                     TEXT      \n",
    "variants            biotype                       TEXT      \n",
    "variants            impact                        VARCHAR(60)\n",
    "variants            impact_so                     TEXT      \n",
    "variants            impact_severity               VARCHAR(20)\n",
    "variants            polyphen_pred                 TEXT      \n",
    "variants            polyphen_score                FLOAT     \n",
    "variants            sift_pred                     TEXT      \n",
    "variants            sift_score                    FLOAT     \n",
    "variants            anc_allele                    TEXT      \n",
    "variants            rms_bq                        FLOAT     \n",
    "variants            cigar                         TEXT      \n",
    "variants            depth                         INTEGER   \n",
    "variants            strand_bias                   FLOAT     \n",
    "variants            rms_map_qual                  FLOAT     \n",
    "variants            in_hom_run                    INTEGER   \n",
    "variants            num_mapq_zero                 INTEGER   \n",
    "variants            num_alleles                   INTEGER   \n",
    "variants            num_reads_w_dels              FLOAT     \n",
    "variants            haplotype_score               FLOAT     \n",
    "variants            qual_depth                    FLOAT     \n",
    "variants            allele_count                  INTEGER   \n",
    "variants            allele_bal                    FLOAT     \n",
    "variants            in_hm2                        BOOLEAN   \n",
    "variants            in_hm3                        BOOLEAN   \n",
    "variants            is_somatic                    BOOLEAN   \n",
    "variants            somatic_score                 FLOAT     \n",
    "variants            in_esp                        BOOLEAN   \n",
    "variants            aaf_esp_ea                    FLOAT     \n",
    "variants            aaf_esp_aa                    FLOAT     \n",
    "variants            aaf_esp_all                   FLOAT     \n",
    "variants            exome_chip                    BOOLEAN   \n",
    "variants            in_1kg                        BOOLEAN   \n",
    "variants            aaf_1kg_amr                   FLOAT     \n",
    "variants            aaf_1kg_eas                   FLOAT     \n",
    "variants            aaf_1kg_sas                   FLOAT     \n",
    "variants            aaf_1kg_afr                   FLOAT     \n",
    "variants            aaf_1kg_eur                   FLOAT     \n",
    "variants            aaf_1kg_all                   FLOAT     \n",
    "variants            grc                           TEXT      \n",
    "variants            gms_illumina                  FLOAT     \n",
    "variants            gms_solid                     FLOAT     \n",
    "variants            gms_iontorrent                FLOAT     \n",
    "variants            in_cse                        BOOLEAN   \n",
    "variants            encode_tfbs                   TEXT      \n",
    "variants            encode_dnaseI_cell_count      INTEGER   \n",
    "variants            encode_dnaseI_cell_list       TEXT      \n",
    "variants            encode_consensus_gm12878      TEXT      \n",
    "variants            encode_consensus_h1hesc       TEXT      \n",
    "variants            encode_consensus_helas3       TEXT      \n",
    "variants            encode_consensus_hepg2        TEXT      \n",
    "variants            encode_consensus_huvec        TEXT      \n",
    "variants            encode_consensus_k562         TEXT      \n",
    "variants            vista_enhancers               TEXT      \n",
    "variants            cosmic_ids                    TEXT      \n",
    "variants            info                          BLOB      \n",
    "variants            cadd_raw                      FLOAT     \n",
    "variants            cadd_scaled                   FLOAT     \n",
    "variants            fitcons                       FLOAT     \n",
    "variants            in_exac                       BOOLEAN   \n",
    "variants            aaf_exac_all                  FLOAT     \n",
    "variants            aaf_adj_exac_all              FLOAT     \n",
    "variants            aaf_adj_exac_afr              FLOAT     \n",
    "variants            aaf_adj_exac_amr              FLOAT     \n",
    "variants            aaf_adj_exac_eas              FLOAT     \n",
    "variants            aaf_adj_exac_fin              FLOAT     \n",
    "variants            aaf_adj_exac_nfe              FLOAT     \n",
    "variants            aaf_adj_exac_oth              FLOAT     \n",
    "variants            aaf_adj_exac_sas              FLOAT     \n",
    "variants            exac_num_het                  INTEGER   \n",
    "variants            exac_num_hom_alt              INTEGER   \n",
    "variants            exac_num_chroms               INTEGER   \n",
    "variants            aaf_gnomad_all                FLOAT     \n",
    "variants            aaf_gnomad_afr                FLOAT     \n",
    "variants            aaf_gnomad_amr                FLOAT     \n",
    "variants            aaf_gnomad_asj                FLOAT     \n",
    "variants            aaf_gnomad_eas                FLOAT     \n",
    "variants            aaf_gnomad_fin                FLOAT     \n",
    "variants            aaf_gnomad_nfe                FLOAT     \n",
    "variants            aaf_gnomad_oth                FLOAT     \n",
    "variants            aaf_gnomad_sas                FLOAT     \n",
    "variants            gnomad_num_het                INTEGER   \n",
    "variants            gnomad_num_hom_alt            INTEGER   \n",
    "variants            gnomad_num_chroms             INTEGER   \n",
    "variants            vep_allele                    TEXT      \n",
    "variants            vep_feature_type              TEXT      \n",
    "variants            vep_cdna_position             TEXT      \n",
    "variants            vep_cds_position              TEXT      \n",
    "variants            vep_existing_variation        TEXT      \n",
    "variants            vep_intron                    TEXT      \n",
    "variants            vep_distance                  TEXT      \n",
    "variants            vep_strand                    TEXT      \n",
    "variants            vep_symbol_source             TEXT      \n",
    "variant_impacts     variant_id                    INTEGER   \n",
    "variant_impacts     anno_id                       INTEGER   \n",
    "variant_impacts     gene                          VARCHAR(60)\n",
    "variant_impacts     transcript                    VARCHAR(60)\n",
    "variant_impacts     is_exonic                     BOOLEAN   \n",
    "variant_impacts     is_coding                     BOOLEAN   \n",
    "variant_impacts     is_lof                        BOOLEAN   \n",
    "variant_impacts     exon                          TEXT      \n",
    "variant_impacts     codon_change                  TEXT      \n",
    "variant_impacts     aa_change                     TEXT      \n",
    "variant_impacts     aa_length                     TEXT      \n",
    "variant_impacts     biotype                       TEXT      \n",
    "variant_impacts     impact                        VARCHAR(60)\n",
    "variant_impacts     impact_so                     TEXT      \n",
    "variant_impacts     impact_severity               VARCHAR(20)\n",
    "variant_impacts     polyphen_pred                 TEXT      \n",
    "variant_impacts     polyphen_score                FLOAT     \n",
    "variant_impacts     sift_pred                     TEXT      \n",
    "variant_impacts     sift_score                    FLOAT     \n",
    "variant_impacts     vep_allele                    TEXT      \n",
    "variant_impacts     vep_feature_type              TEXT      \n",
    "variant_impacts     vep_cdna_position             TEXT      \n",
    "variant_impacts     vep_cds_position              TEXT      \n",
    "variant_impacts     vep_existing_variation        TEXT      \n",
    "variant_impacts     vep_intron                    TEXT      \n",
    "variant_impacts     vep_distance                  TEXT      \n",
    "variant_impacts     vep_strand                    TEXT      \n",
    "variant_impacts     vep_symbol_source             TEXT      \n",
    "samples             sample_id                     INTEGER   \n",
    "samples             family_id                     TEXT      \n",
    "samples             name                          TEXT      \n",
    "samples             paternal_id                   TEXT      \n",
    "samples             maternal_id                   TEXT      \n",
    "samples             sex                           TEXT      \n",
    "samples             phenotype                     TEXT      \n",
    "samples             ethnicity                     TEXT      \n",
    "gene_detailed       uid                           INTEGER   \n",
    "gene_detailed       chrom                         VARCHAR(60)\n",
    "gene_detailed       gene                          VARCHAR(60)\n",
    "gene_detailed       is_hgnc                       BOOLEAN   \n",
    "gene_detailed       ensembl_gene_id               TEXT      \n",
    "gene_detailed       transcript                    VARCHAR(60)\n",
    "gene_detailed       biotype                       TEXT      \n",
    "gene_detailed       transcript_status             TEXT      \n",
    "gene_detailed       ccds_id                       VARCHAR(60)\n",
    "gene_detailed       hgnc_id                       TEXT      \n",
    "gene_detailed       entrez_id                     TEXT      \n",
    "gene_detailed       cds_length                    TEXT      \n",
    "gene_detailed       protein_length                TEXT      \n",
    "gene_detailed       transcript_start              TEXT      \n",
    "gene_detailed       transcript_end                TEXT      \n",
    "gene_detailed       strand                        TEXT      \n",
    "gene_detailed       synonym                       TEXT      \n",
    "gene_detailed       rvis_pct                      FLOAT     \n",
    "gene_detailed       mam_phenotype_id              TEXT      \n",
    "gene_summary        uid                           INTEGER   \n",
    "gene_summary        chrom                         VARCHAR(60)\n",
    "gene_summary        gene                          VARCHAR(60)\n",
    "gene_summary        is_hgnc                       BOOLEAN   \n",
    "gene_summary        ensembl_gene_id               TEXT      \n",
    "gene_summary        hgnc_id                       TEXT      \n",
    "gene_summary        transcript_min_start          INTEGER   \n",
    "gene_summary        transcript_max_end            INTEGER   \n",
    "gene_summary        strand                        TEXT      \n",
    "gene_summary        synonym                       TEXT      \n",
    "gene_summary        rvis_pct                      FLOAT     \n",
    "gene_summary        mam_phenotype_id              TEXT      \n",
    "gene_summary        in_cosmic_census              BOOLEAN   \n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  gemini query -q \"SELECT name FROM samples\" --header chr22.db\n",
    "name\n",
    "1805\n",
    "1847\n",
    "4805\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT * FROM samples\" --header chr22.db | column -t\n",
    "sample_id  family_id  name  paternal_id  maternal_id  sex  phenotype  ethnicity\n",
    "1          family1    1805  0            0            2    1          CEU\n",
    "2          family1    1847  0            0            1    1          CEU\n",
    "3          family1    4805  1847         1805         1    2          CEU\n",
    "\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  gemini query -q \"SELECT COUNT(*) FROM VARIANTS WHERE filter is NULL\" --header chr22.db\n",
    "COUNT(*)\n",
    "1861\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT * FROM variants WHERE filter is NULL and gene = 'MLC1' \" --header chr22.db\n",
    "\n",
    "# all the variants in the MCL1 gene that filter in null (not flagged as variants)\n",
    "\n",
    "[nathalia@aws] $  gemini query -q \"SELECT * FROM variants WHERE filter is NULL and gene = 'MLC1' \" --header chr22.db | wc -l\n",
    "14\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  gemini query -q \"SELECT gene from variants \\\n",
    "> WHERE clinvar_disease_name is not NULL and aaf_esp_ea <= 0.01\" \\\n",
    "> chr22.db\n",
    "PEX26\n",
    "PEX26\n",
    "PRODH\n",
    "COMT\n",
    "HPS4\n",
    "TCN2\n",
    "TIMP3\n",
    "MYH9\n",
    "MYH9\n",
    "MYH9\n",
    "MYH9\n",
    "MYH9\n",
    "EP300\n",
    "EP300\n",
    "NAGA\n",
    "CYB5R3\n",
    "TRMU\n",
    "ALG12\n",
    "MLC1\n",
    "```\n",
    "\n",
    "Variants that have a prior disease associated with it form this clinvar database\n",
    "\n",
    "`aaf_esp_ea <= 0.01\"` alternate allele frequency, exome sequence project, european-american; 0.01 is pretty rare\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IDENTIFYING HIGH-CONFIDENCE DE NOVO MUTATIONS WITH GEMINI"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://gist.github.com/arq5x/9e1928638397ba45da2e#file-denovo-sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  cd ..\n",
    "[nathalia@aws] $  mkdir denovo_mutations\n",
    "[nathalia@aws] $  cd denovo_mutations/\n",
    "[nathalia@aws] $  pwd\n",
    "/home/ubuntu/workspace/gemini/denovo_mutations\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  curl https://s3.amazonaws.com/gemini-tutorials/trio.trim.vep.vcf.gz > trio.trim.vep.vcf.gz\n",
    "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
    "                                 Dload  Upload   Total   Spent    Left  Speed\n",
    "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n",
    "  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0\n",
    "100 1889k  100 1889k    0     0   299k      0  0:00:06  0:00:06 --:--:--  393k\n",
    "[nathalia@aws] $  curl https://s3.amazonaws.com/gemini-tutorials/denovo.ped > denovo.ped\n",
    "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
    "                                 Dload  Upload   Total   Spent    Left  Speed\n",
    "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n",
    "100   154  100   154    0     0    482      0 --:--:-- --:--:-- --:--:--   481\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 1.9M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.9M Nov 12 19:56 trio.trim.vep.vcf.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  154 Nov 12 19:56 denovo.ped\n",
    "\n",
    "\n",
    "[nathalia@aws] $  gemini load --cores 2 \\\n",
    ">             -v  trio.trim.vep.vcf.gz \\\n",
    ">             -t VEP \\\n",
    ">             --tempdir . \\\n",
    ">             --skip-gene-tables --skip-cadd --skip-gerp-bp \\\n",
    ">             -p denovo.ped \\\n",
    ">        trio.trim.vep.denovo.db\n",
    "[W::hts_idx_load2] The index file is older than the data file: /workspace/lib/gemini/gemini_data/hg19_fitcons_fc-i6-0_V1-01.bed.gz.tbi\n",
    "[W::hts_idx_load2] The index file is older than the data file: /workspace/lib/gemini/gemini_data/geno2mp.variants.tidy.vcf.gz.tbi\n",
    "Indexing /workspace/gemini/denovo_mutations/trio.trim.vep.vcf.gz with grabix.\n",
    "Loading 17237 variants.\n",
    "Breaking /workspace/gemini/denovo_mutations/trio.trim.vep.vcf.gz into 2 chunks.\n",
    "Loading chunk 0.\n",
    "Loading chunk 1.\n",
    "[W::hts_idx_load2] The index file is older than the data file: /workspace/lib/gemini/gemini_data/hg19_fitcons_fc-i6-0_V1-01.bed.gz.tbi\n",
    "[W::hts_idx_load2] The index file is older than the data file: /workspace/lib/gemini/gemini_data/hg19_fitcons_fc-i6-0_V1-01.bed.gz.tbi\n",
    "[W::hts_idx_load2] The index file is older than the data file: /workspace/lib/gemini/gemini_data/geno2mp.variants.tidy.vcf.gz.tbi\n",
    "[W::hts_idx_load2] The index file is older than the data file: /workspace/lib/gemini/gemini_data/geno2mp.variants.tidy.vcf.gz.tbi\n",
    "pid 9271: 8618 variants processed.\n",
    "pid 9274: 8619 variants processed.\n",
    "Done loading 17237 variants in 2 chunks.\n",
    "2018-11-12 19:59:45 merging 2 chunks.\n",
    "2018-11-12 19:59:45 merged to final database....\n",
    "2018-11-12 19:59:45 moved.\n",
    "storing version, header, etc.\n",
    "building indices\n",
    "\n",
    "\n",
    "[nathalia@aws] $  lhtr\n",
    "total 101M\n",
    "-rw-rw-r-- 1 ubuntu ubuntu 1.9M Nov 12 19:56 trio.trim.vep.vcf.gz\n",
    "-rw-rw-r-- 1 ubuntu ubuntu  154 Nov 12 19:56 denovo.ped\n",
    "-rw-rw-r-- 1 ubuntu ubuntu   43 Nov 12 19:57 trio.trim.vep.vcf.gz.gbi\n",
    "-rw-r--r-- 1 ubuntu ubuntu  99M Nov 12 19:59 trio.trim.vep.denovo.db\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "[nathalia@aws] $  gemini de_novo trio.trim.vep.denovo.db | wc -l\n",
    "772\n",
    "\n",
    "# way more than we expected by chance\n",
    "\n",
    "[nathalia@aws] $  gemini de_novo --columns \"chrom, start, end, ref, alt, filter, qual, gene, impact\" trio.trim.vep.denovo.db | wc -l\n",
    "772\n",
    "\n",
    "# more restricted info, but still the same number of rows\n",
    "\n",
    "[nathalia@aws] $  gemini de_novo --columns \"chrom, start, end, ref, alt, filter, qual, gene, impact\" trio.trim.vep.denovo.db --filter \"filter is NULL\" | wc -l\n",
    "73\n",
    "\n",
    "# filtering out the junk var calls on the filter colum\n",
    "\n",
    "[nathalia@aws] $  gemini de_novo --columns \"chrom, start, end, ref, alt, filter, qual, gene, impact\" trio.trim.vep.denovo.db --filter \"filter is NULL and impact_severity != 'LOW'\" | wc -l\n",
    "21\n",
    "\n",
    "#counting the header\n",
    "\n",
    "[nathalia@aws] $  gemini de_novo -d 7 --columns \"chrom, start, end, ref, alt, filter, qual, gene, impact\" trio.trim.vep.denovo.db --filter \"filter is NULL and impact_severity != 'LOW'\" | wc -l\n",
    "15\n",
    "\n",
    "[nathalia@aws] $  gemini de_novo -d 7 --columns \"chrom, start, end, ref, alt, filter, qual, gene, impact\" trio.trim.vep.denovo.db --filter \"filter is NULL and impact_severity != 'LOW' and max_aaf_all < 0.001\" | wc -l\n",
    "3\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
